<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>class Bio::Blast::Report - BioRuby API documentation</title>

<link type="text/css" media="screen" href="../../rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "../../";
</script>

<script type="text/javascript" charset="utf-8" src="../../js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="../../js/darkfish.js"></script>


<body id="top" class="class">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="../../index.html">Home</a>
    <a href="../../table_of_contents.html#classes">Classes</a>
    <a href="../../table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="file-metadata">
    <nav id="file-list-section" class="section">
  <h3 class="section-header">Defined In</h3>
  <ul>
    <li>lib/bio/appl/blast/format8.rb
    <li>lib/bio/appl/blast/report.rb
    <li>lib/bio/appl/blast/rexml.rb
    <li>lib/bio/appl/blast/xmlparser.rb
  </ul>
</nav>

    
  </div>

  <div id="class-metadata">
    
    <nav id="parent-class-section" class="section">
  <h3 class="section-header">Parent</h3>
  
  <p class="link"><a href="../../Object.html">Object</a>
  
</nav>

    
    <!-- Method Quickref -->
<nav id="method-list-section" class="section">
  <h3 class="section-header">Methods</h3>

  <ul class="link-list">
    
    <li><a href="#method-c-new">::new</a>
    
    <li><a href="#method-c-rexml">::rexml</a>
    
    <li><a href="#method-c-tab">::tab</a>
    
    <li><a href="#method-c-xmlparser">::xmlparser</a>
    
    <li><a href="#method-i-db_len">#db_len</a>
    
    <li><a href="#method-i-db_num">#db_num</a>
    
    <li><a href="#method-i-each">#each</a>
    
    <li><a href="#method-i-each_hit">#each_hit</a>
    
    <li><a href="#method-i-each_iteration">#each_iteration</a>
    
    <li><a href="#method-i-eff_space">#eff_space</a>
    
    <li><a href="#method-i-entrez_query">#entrez_query</a>
    
    <li><a href="#method-i-entropy">#entropy</a>
    
    <li><a href="#method-i-expect">#expect</a>
    
    <li><a href="#method-i-filter">#filter</a>
    
    <li><a href="#method-i-gap_extend">#gap_extend</a>
    
    <li><a href="#method-i-gap_open">#gap_open</a>
    
    <li><a href="#method-i-hits">#hits</a>
    
    <li><a href="#method-i-hsp_len">#hsp_len</a>
    
    <li><a href="#method-i-inclusion">#inclusion</a>
    
    <li><a href="#method-i-kappa">#kappa</a>
    
    <li><a href="#method-i-lambda">#lambda</a>
    
    <li><a href="#method-i-matrix">#matrix</a>
    
    <li><a href="#method-i-message">#message</a>
    
    <li><a href="#method-i-pattern">#pattern</a>
    
    <li><a href="#method-i-sc_match">#sc_match</a>
    
    <li><a href="#method-i-sc_mismatch">#sc_mismatch</a>
    
    <li><a href="#method-i-statistics">#statistics</a>
    
  </ul>
</nav>

  </div>

  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="../../ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="../../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="../../LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="../../README_rdoc.html">README</a>
  
    <li class="file"><a href="../../README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="../../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="../../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="../../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="../../Bio.html">Bio</a>
  
    <li><a href="../../Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="../../Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="../../Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="../../Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="../../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="../../Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="../../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="../../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="../../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="../../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="../../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="../../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="../../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="../../Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="../../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="../../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="../../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="../../Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="../../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="../../Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="../../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="../../Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="../../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="../../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="../../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="../../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="../../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="../../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="../../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="../../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="../../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="../../Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="../../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="../../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="../../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="../../Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="../../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="../../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="../../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="../../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="../../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="../../Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="../../Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="../../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="../../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="../../Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="../../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="../../Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="../../Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="../../Bio/Command.html">Bio::Command</a>
  
    <li><a href="../../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="../../Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="../../Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="../../Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="../../Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="../../Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="../../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="../../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="../../Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="../../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="../../Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="../../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="../../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="../../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="../../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="../../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="../../Bio/DB.html">Bio::DB</a>
  
    <li><a href="../../Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="../../Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="../../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="../../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="../../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="../../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="../../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="../../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="../../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="../../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="../../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="../../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="../../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="../../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="../../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="../../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="../../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="../../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="../../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="../../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="../../Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="../../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="../../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="../../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="../../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="../../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="../../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="../../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="../../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="../../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="../../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="../../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="../../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="../../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="../../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="../../Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="../../Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="../../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="../../Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="../../Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="../../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="../../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="../../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="../../Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="../../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="../../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="../../Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="../../Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="../../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="../../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="../../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="../../Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="../../Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="../../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="../../Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="../../Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="../../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="../../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="../../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="../../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="../../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="../../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="../../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="../../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="../../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="../../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="../../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="../../Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="../../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="../../Bio/Features.html">Bio::Features</a>
  
    <li><a href="../../Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="../../Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="../../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="../../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="../../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="../../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="../../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="../../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="../../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="../../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="../../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="../../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="../../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="../../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="../../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="../../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="../../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="../../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="../../Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="../../Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="../../Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="../../Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="../../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="../../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="../../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="../../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="../../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="../../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="../../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="../../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="../../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="../../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="../../Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="../../Bio/GO.html">Bio::GO</a>
  
    <li><a href="../../Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="../../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="../../Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="../../Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="../../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="../../Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="../../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="../../Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="../../Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="../../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="../../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="../../Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="../../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="../../Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="../../Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="../../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="../../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="../../Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="../../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="../../Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="../../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="../../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="../../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="../../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="../../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="../../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="../../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="../../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="../../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="../../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="../../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="../../Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="../../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="../../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="../../Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="../../Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="../../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="../../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="../../Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="../../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="../../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="../../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="../../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="../../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="../../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="../../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="../../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="../../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="../../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="../../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="../../Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="../../Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="../../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="../../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="../../Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="../../Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="../../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="../../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="../../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="../../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="../../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="../../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="../../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="../../Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="../../Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="../../Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="../../Bio/Location.html">Bio::Location</a>
  
    <li><a href="../../Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="../../Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="../../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="../../Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="../../Bio/Map.html">Bio::Map</a>
  
    <li><a href="../../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="../../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="../../Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="../../Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="../../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="../../Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="../../Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="../../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="../../Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="../../Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="../../Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="../../Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="../../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="../../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="../../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="../../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="../../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="../../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="../../Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="../../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="../../Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="../../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="../../Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="../../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="../../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="../../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="../../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="../../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="../../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="../../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="../../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="../../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="../../Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="../../Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="../../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="../../Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="../../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="../../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="../../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="../../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="../../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="../../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="../../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="../../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="../../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="../../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="../../Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="../../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="../../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="../../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="../../Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="../../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="../../Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="../../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="../../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="../../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="../../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="../../Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="../../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="../../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="../../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="../../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="../../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="../../Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="../../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="../../Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="../../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="../../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="../../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="../../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="../../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="../../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="../../Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="../../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="../../Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="../../Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="../../Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="../../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="../../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="../../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="../../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="../../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="../../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="../../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="../../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="../../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="../../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="../../Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="../../Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="../../Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="../../Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="../../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="../../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="../../Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="../../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="../../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="../../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="../../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="../../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="../../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="../../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="../../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="../../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="../../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="../../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="../../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="../../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="../../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="../../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="../../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="../../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="../../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="../../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="../../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="../../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="../../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="../../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="../../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="../../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="../../Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="../../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="../../Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="../../Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="../../Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="../../Bio/References.html">Bio::References</a>
  
    <li><a href="../../Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="../../Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="../../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="../../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="../../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="../../Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="../../Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="../../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="../../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="../../Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="../../Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="../../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="../../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="../../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="../../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="../../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="../../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="../../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="../../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="../../Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="../../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="../../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="../../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="../../Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="../../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="../../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="../../Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="../../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="../../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="../../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="../../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="../../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="../../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="../../Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="../../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="../../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="../../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="../../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="../../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="../../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="../../Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="../../Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="../../Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="../../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="../../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="../../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="../../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="../../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="../../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="../../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="../../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="../../Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="../../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="../../Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="../../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="../../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="../../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="../../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="../../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="../../Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="../../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="../../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="../../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="../../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="../../Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="../../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="../../Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="../../Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="../../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="../../Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="../../Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="../../Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="../../Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="../../Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="../../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="../../Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="../../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="../../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="../../Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="../../Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="../../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="../../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="../../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="../../Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="../../Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="../../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="../../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="../../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="../../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="../../Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="../../Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="../../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="../../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="../../Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="../../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="../../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="../../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="../../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="../../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="../../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="../../Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="../../Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="../../Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="../../Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="../../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="../../Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="../../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="../../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="../../Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="../../Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="../../Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="../../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="../../Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="../../Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="../../BiorubyController.html">BiorubyController</a>
  
    <li><a href="../../BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="../../BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="../../Hash.html">Hash</a>
  
    <li><a href="../../Object.html">Object</a>
  
    <li><a href="../../String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation">
  <h1 class="class">class Bio::Blast::Report</h1>

  <div id="description" class="description">
    
<h1 id="label-Bio%3A%3ABlast%3A%3AReport"><a href="Report.html">Bio::Blast::Report</a></h1>

<p>Parsed results of the blast execution for Tab-delimited and XML output
format.  Tab-delimited reports are consists of</p>

<pre>Query id,
Subject id,
percent of identity,
alignment length,
number of mismatches (not including gaps),
number of gap openings,
start of alignment in query,
end of alignment in query,
start of alignment in subject,
end of alignment in subject,
expected value,
bit score.</pre>

<p>according to the MEGABLAST document (README.mbl).  As for XML output, see
the following DTDs.</p>

<pre>* http://www.ncbi.nlm.nih.gov/dtd/NCBI_BlastOutput.dtd
* http://www.ncbi.nlm.nih.gov/dtd/NCBI_BlastOutput.mod
* http://www.ncbi.nlm.nih.gov/dtd/NCBI_Entity.mod</pre>

  </div><!-- description -->

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    
    <!-- Constants -->
    <section id="constants-list" class="section">
      <h3 class="section-header">Constants</h3>
      <dl>
      
        <dt id="DELIMITER">DELIMITER
        
        <dd class="description"><p>for <a href="../FlatFile.html">Bio::FlatFile</a> support (only for XML
data)</p>
        
      
        <dt id="FLATFILE_SPLITTER">FLATFILE_SPLITTER
        
        <dd class="description"><p>splitter for <a href="../FlatFile.html">Bio::FlatFile</a> support</p>
        
      
      </dl>
    </section>
    

    
    <!-- Attributes -->
    <section id="attribute-method-details" class="method-section section">
      <h3 class="section-header">Attributes</h3>

      
      <div id="attribute-i-db" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">db</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>database name or title (<a href="../../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-iterations" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">iterations</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Returns an Array of <a
href="Report/Iteration.html">Bio::Blast::Report::Iteration</a> objects.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-parameters" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">parameters</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Returns a <a href="../../Hash.html">Hash</a> containing execution
parameters.  Valid keys are: matrix, expect, include, sc-match,
sc-mismatch, gap-open, gap-extend, filter</p>
        
        </div>
      </div>
      
      <div id="attribute-i-program" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">program</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>program name (e.g. blastp) (<a href="../../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-query_def" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">query_def</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>query definition line (<a href="../../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-query_id" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">query_id</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>query ID (<a href="../../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-query_len" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">query_len</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>query length (Integer)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-reference" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">reference</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>reference (<a href="../../String.html">String</a>)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-reports" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">reports</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>When the report contains results for multiple query sequences, returns an
array of <a href="Report.html">Bio::Blast::Report</a> objects corresponding
to the multiple queries. Otherwise, returns nil.</p>

<p>Note for No hits found: When no hits found for a query sequence, the
result for the query is completely void and no information available in the
result XML, including query ID and query definition. The only trace is that
iteration number is skipped. This means that if the no-hit query is the
last query, the query can not be detected, because the result XML is
completely the same as the result XML without the query.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-version" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">version</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>BLAST version (e.g. blastp 2.2.18 [Mar-02-2008]) (<a
href="../../String.html">String</a>)</p>
        
        </div>
      </div>
      
    </section><!-- attribute-method-details -->
    

    <!-- Methods -->
    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Class Methods</h3>

    
      <div id="method-c-new" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(data, parser = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Passing a BLAST output from blastall -m 7 or -m 8 as a <a
href="../../String.html">String</a>. Formats are auto detected.</p>
          

          
          <div class="method-source-code" id="new-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 85</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">data</span>, <span class="ruby-identifier">parser</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-ivar">@iterations</span> = []
  <span class="ruby-ivar">@parameters</span> = {}
  <span class="ruby-keyword">case</span> <span class="ruby-identifier">parser</span>
  <span class="ruby-keyword">when</span> <span class="ruby-value">:xmlparser</span>             <span class="ruby-comment"># format 7</span>
    <span class="ruby-identifier">xmlparser_parse</span>(<span class="ruby-identifier">data</span>)
    <span class="ruby-ivar">@reports</span> = <span class="ruby-identifier">blastxml_split_reports</span>
  <span class="ruby-keyword">when</span> <span class="ruby-value">:rexml</span>         <span class="ruby-comment"># format 7</span>
    <span class="ruby-identifier">rexml_parse</span>(<span class="ruby-identifier">data</span>)
    <span class="ruby-ivar">@reports</span> = <span class="ruby-identifier">blastxml_split_reports</span>
  <span class="ruby-keyword">when</span> <span class="ruby-value">:tab</span>           <span class="ruby-comment"># format 8</span>
    <span class="ruby-identifier">tab_parse</span>(<span class="ruby-identifier">data</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-keyword">false</span>
    <span class="ruby-comment"># do not parse, creates an empty object</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">auto_parse</span>(<span class="ruby-identifier">data</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- new-source -->
          
        </div>

        

        
      </div><!-- new-method -->

    
      <div id="method-c-rexml" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">rexml</span><span
            class="method-args">(data)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Specify to use REXML to parse XML (-m 7) output.</p>
          

          
          <div class="method-source-code" id="rexml-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 59</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">rexml</span>(<span class="ruby-identifier">data</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">data</span>, <span class="ruby-value">:rexml</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- rexml-source -->
          
        </div>

        

        
      </div><!-- rexml-method -->

    
      <div id="method-c-tab" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">tab</span><span
            class="method-args">(data)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Specify to use tab delimited output parser.</p>
          

          
          <div class="method-source-code" id="tab-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 64</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">tab</span>(<span class="ruby-identifier">data</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">data</span>, <span class="ruby-value">:tab</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- tab-source -->
          
        </div>

        

        
      </div><!-- tab-method -->

    
      <div id="method-c-xmlparser" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">xmlparser</span><span
            class="method-args">(data)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Specify to use XMLParser to parse XML (-m 7) output.</p>
          

          
          <div class="method-source-code" id="xmlparser-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 54</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">xmlparser</span>(<span class="ruby-identifier">data</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">data</span>, <span class="ruby-value">:xmlparser</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- xmlparser-source -->
          
        </div>

        

        
      </div><!-- xmlparser-method -->

    
    </section><!-- public-class-method-details -->
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Instance Methods</h3>

    
      <div id="method-i-db_len" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">db_len</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Length of BLAST db</p>
          

          
          <div class="method-source-code" id="db_len-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 191</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">db_len</span>;    <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'db-len'</span>];    <span class="ruby-keyword">end</span></pre>
          </div><!-- db_len-source -->
          
        </div>

        

        
      </div><!-- db_len-method -->

    
      <div id="method-i-db_num" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">db_num</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Number of sequences in BLAST db</p>
          

          
          <div class="method-source-code" id="db_num-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 189</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">db_num</span>;    <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'db-num'</span>];    <span class="ruby-keyword">end</span></pre>
          </div><!-- db_num-source -->
          
        </div>

        

        
      </div><!-- db_num-method -->

    
      <div id="method-i-each" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">each</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Report.html#method-i-each_hit">each_hit</a>
        </div>
        
      </div><!-- each-method -->

    
      <div id="method-i-each_hit" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">each_hit</span><span
            class="method-args">() { |x| ... }</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Iterates on each <a href="Report/Hit.html">Bio::Blast::Report::Hit</a>
object of the the last <a href="Report/Iteration.html">Iteration</a>.
Shortcut for the last iterations hits (for blastall)</p>
          

          
          <div class="method-source-code" id="each_hit-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 167</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">each_hit</span>
  <span class="ruby-ivar">@iterations</span>.<span class="ruby-identifier">last</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">yield</span> <span class="ruby-identifier">x</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- each_hit-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Report.html#method-i-each">each</a>
        </div>
        

        
      </div><!-- each_hit-method -->

    
      <div id="method-i-each_iteration" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">each_iteration</span><span
            class="method-args">() { |x| ... }</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Iterates on each <a
href="Report/Iteration.html">Bio::Blast::Report::Iteration</a> object. (for
blastpgp)</p>
          

          
          <div class="method-source-code" id="each_iteration-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 159</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">each_iteration</span>
  <span class="ruby-ivar">@iterations</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">yield</span> <span class="ruby-identifier">x</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- each_iteration-source -->
          
        </div>

        

        
      </div><!-- each_iteration-method -->

    
      <div id="method-i-eff_space" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">eff_space</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Effective search space</p>
          

          
          <div class="method-source-code" id="eff_space-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 195</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">eff_space</span>; <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'eff-space'</span>]; <span class="ruby-keyword">end</span></pre>
          </div><!-- eff_space-source -->
          
        </div>

        

        
      </div><!-- eff_space-method -->

    
      <div id="method-i-entrez_query" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">entrez_query</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Limit of request to Entrez : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="entrez_query-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 156</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entrez_query</span>; <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'entrez-query'</span>];     <span class="ruby-keyword">end</span></pre>
          </div><!-- entrez_query-source -->
          
        </div>

        

        
      </div><!-- entrez_query-method -->

    
      <div id="method-i-entropy" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">entropy</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Karlin-Altschul parameter H</p>
          

          
          <div class="method-source-code" id="entropy-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 201</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entropy</span>;   <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'entropy'</span>];   <span class="ruby-keyword">end</span></pre>
          </div><!-- entropy-source -->
          
        </div>

        

        
      </div><!-- entropy-method -->

    
      <div id="method-i-expect" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">expect</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Expectation threshold (-e) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="expect-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 140</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">expect</span>;       <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'expect'</span>];           <span class="ruby-keyword">end</span></pre>
          </div><!-- expect-source -->
          
        </div>

        

        
      </div><!-- expect-method -->

    
      <div id="method-i-filter" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">filter</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Filtering options (-F) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="filter-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 152</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">filter</span>;       <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'filter'</span>];           <span class="ruby-keyword">end</span></pre>
          </div><!-- filter-source -->
          
        </div>

        

        
      </div><!-- filter-method -->

    
      <div id="method-i-gap_extend" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">gap_extend</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gap extension cost (-E) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="gap_extend-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 150</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gap_extend</span>;   <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'gap-extend'</span>];       <span class="ruby-keyword">end</span></pre>
          </div><!-- gap_extend-source -->
          
        </div>

        

        
      </div><!-- gap_extend-method -->

    
      <div id="method-i-gap_open" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">gap_open</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gap opening cost (-G) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="gap_open-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 148</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gap_open</span>;     <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'gap-open'</span>];         <span class="ruby-keyword">end</span></pre>
          </div><!-- gap_open-source -->
          
        </div>

        

        
      </div><!-- gap_open-method -->

    
      <div id="method-i-hits" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">hits</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns a Array of Bio::Blast::Report::Hits of the last iteration. Shortcut
for the last iterations hits</p>
          

          
          <div class="method-source-code" id="hits-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 176</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">hits</span>
  <span class="ruby-ivar">@iterations</span>.<span class="ruby-identifier">last</span>.<span class="ruby-identifier">hits</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- hits-source -->
          
        </div>

        

        
      </div><!-- hits-method -->

    
      <div id="method-i-hsp_len" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">hsp_len</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Effective HSP length</p>
          

          
          <div class="method-source-code" id="hsp_len-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 193</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">hsp_len</span>;   <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'hsp-len'</span>];   <span class="ruby-keyword">end</span></pre>
          </div><!-- hsp_len-source -->
          
        </div>

        

        
      </div><!-- hsp_len-method -->

    
      <div id="method-i-inclusion" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">inclusion</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Inclusion threshold (-h) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="inclusion-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 142</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">inclusion</span>;    <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'include'</span>];          <span class="ruby-keyword">end</span></pre>
          </div><!-- inclusion-source -->
          
        </div>

        

        
      </div><!-- inclusion-method -->

    
      <div id="method-i-kappa" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">kappa</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Karlin-Altschul parameter K</p>
          

          
          <div class="method-source-code" id="kappa-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 197</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">kappa</span>;     <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'kappa'</span>];     <span class="ruby-keyword">end</span></pre>
          </div><!-- kappa-source -->
          
        </div>

        

        
      </div><!-- kappa-method -->

    
      <div id="method-i-lambda" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">lambda</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Karlin-Altschul parameter Lamba</p>
          

          
          <div class="method-source-code" id="lambda-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 199</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">lambda</span>;    <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'lambda'</span>];    <span class="ruby-keyword">end</span></pre>
          </div><!-- lambda-source -->
          
        </div>

        

        
      </div><!-- lambda-method -->

    
      <div id="method-i-matrix" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">matrix</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Matrix used (-M) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="matrix-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 138</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">matrix</span>;       <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'matrix'</span>];           <span class="ruby-keyword">end</span></pre>
          </div><!-- matrix-source -->
          
        </div>

        

        
      </div><!-- matrix-method -->

    
      <div id="method-i-message" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">message</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns a <a href="../../String.html">String</a> (or nil) containing
execution message of the last iteration (typically CONVERGED). Shortcut
for the last iterations message (for checking CONVERGED)</p>
          

          
          <div class="method-source-code" id="message-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 206</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">message</span>
  <span class="ruby-ivar">@iterations</span>.<span class="ruby-identifier">last</span>.<span class="ruby-identifier">message</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- message-source -->
          
        </div>

        

        
      </div><!-- message-method -->

    
      <div id="method-i-pattern" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">pattern</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>PHI-BLAST pattern : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="pattern-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 154</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pattern</span>;      <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'pattern'</span>];          <span class="ruby-keyword">end</span></pre>
          </div><!-- pattern-source -->
          
        </div>

        

        
      </div><!-- pattern-method -->

    
      <div id="method-i-sc_match" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">sc_match</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Match score for NT (-r) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="sc_match-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 144</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">sc_match</span>;     <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'sc-match'</span>];         <span class="ruby-keyword">end</span></pre>
          </div><!-- sc_match-source -->
          
        </div>

        

        
      </div><!-- sc_match-method -->

    
      <div id="method-i-sc_mismatch" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">sc_mismatch</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Mismatch score for NT (-q) : shortcuts for @parameters</p>
          

          
          <div class="method-source-code" id="sc_mismatch-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 146</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">sc_mismatch</span>;  <span class="ruby-ivar">@parameters</span>[<span class="ruby-string">'sc-mismatch'</span>];      <span class="ruby-keyword">end</span></pre>
          </div><!-- sc_mismatch-source -->
          
        </div>

        

        
      </div><!-- sc_mismatch-method -->

    
      <div id="method-i-statistics" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">statistics</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns a <a href="../../Hash.html">Hash</a> containing execution
statistics of the last iteration. Valid keys are: db-num, db-len,
hsp-len, eff-space, kappa, lambda, entropy Shortcut for the last
iterations statistics.</p>
          

          
          <div class="method-source-code" id="statistics-source">
            <pre><span class="ruby-comment"># File lib/bio/appl/blast/report.rb, line 184</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">statistics</span>
  <span class="ruby-ivar">@iterations</span>.<span class="ruby-identifier">last</span>.<span class="ruby-identifier">statistics</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- statistics-source -->
          
        </div>

        

        
      </div><!-- statistics-method -->

    
    </section><!-- public-instance-method-details -->
  
  </section><!-- 5Buntitled-5D -->

</div><!-- documentation -->


<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

