<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>module Bio - BioRuby API documentation</title>

<link type="text/css" media="screen" href="./rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "./";
</script>

<script type="text/javascript" charset="utf-8" src="./js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/darkfish.js"></script>


<body id="top" class="module">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="./index.html">Home</a>
    <a href="./table_of_contents.html#classes">Classes</a>
    <a href="./table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="file-metadata">
    <nav id="file-list-section" class="section">
  <h3 class="section-header">Defined In</h3>
  <ul>
    <li>lib/bio.rb
    <li>lib/bio/alignment.rb
    <li>lib/bio/appl/bl2seq/report.rb
    <li>lib/bio/appl/blast.rb
    <li>lib/bio/appl/blast/format0.rb
    <li>lib/bio/appl/blast/format8.rb
    <li>lib/bio/appl/blast/ncbioptions.rb
    <li>lib/bio/appl/blast/remote.rb
    <li>lib/bio/appl/blast/report.rb
    <li>lib/bio/appl/blast/rexml.rb
    <li>lib/bio/appl/blast/rpsblast.rb
    <li>lib/bio/appl/blast/wublast.rb
    <li>lib/bio/appl/blast/xmlparser.rb
    <li>lib/bio/appl/blat/report.rb
    <li>lib/bio/appl/clustalw.rb
    <li>lib/bio/appl/clustalw/report.rb
    <li>lib/bio/appl/emboss.rb
    <li>lib/bio/appl/fasta.rb
    <li>lib/bio/appl/fasta/format10.rb
    <li>lib/bio/appl/gcg/msf.rb
    <li>lib/bio/appl/gcg/seq.rb
    <li>lib/bio/appl/genscan/report.rb
    <li>lib/bio/appl/hmmer.rb
    <li>lib/bio/appl/hmmer/report.rb
    <li>lib/bio/appl/iprscan/report.rb
    <li>lib/bio/appl/mafft.rb
    <li>lib/bio/appl/mafft/report.rb
    <li>lib/bio/appl/meme/mast.rb
    <li>lib/bio/appl/meme/mast/report.rb
    <li>lib/bio/appl/meme/motif.rb
    <li>lib/bio/appl/muscle.rb
    <li>lib/bio/appl/paml/codeml.rb
    <li>lib/bio/appl/paml/common.rb
    <li>lib/bio/appl/phylip/alignment.rb
    <li>lib/bio/appl/phylip/distance_matrix.rb
    <li>lib/bio/appl/probcons.rb
    <li>lib/bio/appl/psort.rb
    <li>lib/bio/appl/psort/report.rb
    <li>lib/bio/appl/pts1.rb
    <li>lib/bio/appl/sim4.rb
    <li>lib/bio/appl/sim4/report.rb
    <li>lib/bio/appl/sosui/report.rb
    <li>lib/bio/appl/spidey/report.rb
    <li>lib/bio/appl/targetp/report.rb
    <li>lib/bio/appl/tcoffee.rb
    <li>lib/bio/appl/tmhmm/report.rb
    <li>lib/bio/command.rb
    <li>lib/bio/compat/features.rb
    <li>lib/bio/compat/references.rb
    <li>lib/bio/data/aa.rb
    <li>lib/bio/data/codontable.rb
    <li>lib/bio/data/na.rb
    <li>lib/bio/db.rb
    <li>lib/bio/db/aaindex.rb
    <li>lib/bio/db/biosql/sequence.rb
    <li>lib/bio/db/embl/common.rb
    <li>lib/bio/db/embl/embl.rb
    <li>lib/bio/db/embl/sptr.rb
    <li>lib/bio/db/embl/swissprot.rb
    <li>lib/bio/db/embl/trembl.rb
    <li>lib/bio/db/embl/uniprot.rb
    <li>lib/bio/db/fantom.rb
    <li>lib/bio/db/fasta.rb
    <li>lib/bio/db/fasta/defline.rb
    <li>lib/bio/db/fasta/qual.rb
    <li>lib/bio/db/fastq.rb
    <li>lib/bio/db/genbank/common.rb
    <li>lib/bio/db/genbank/ddbj.rb
    <li>lib/bio/db/genbank/genbank.rb
    <li>lib/bio/db/genbank/genpept.rb
    <li>lib/bio/db/genbank/refseq.rb
    <li>lib/bio/db/gff.rb
    <li>lib/bio/db/go.rb
    <li>lib/bio/db/kegg/brite.rb
    <li>lib/bio/db/kegg/common.rb
    <li>lib/bio/db/kegg/compound.rb
    <li>lib/bio/db/kegg/drug.rb
    <li>lib/bio/db/kegg/enzyme.rb
    <li>lib/bio/db/kegg/expression.rb
    <li>lib/bio/db/kegg/genes.rb
    <li>lib/bio/db/kegg/genome.rb
    <li>lib/bio/db/kegg/glycan.rb
    <li>lib/bio/db/kegg/keggtab.rb
    <li>lib/bio/db/kegg/kgml.rb
    <li>lib/bio/db/kegg/module.rb
    <li>lib/bio/db/kegg/orthology.rb
    <li>lib/bio/db/kegg/pathway.rb
    <li>lib/bio/db/kegg/reaction.rb
    <li>lib/bio/db/kegg/taxonomy.rb
    <li>lib/bio/db/lasergene.rb
    <li>lib/bio/db/litdb.rb
    <li>lib/bio/db/medline.rb
    <li>lib/bio/db/nbrf.rb
    <li>lib/bio/db/newick.rb
    <li>lib/bio/db/nexus.rb
    <li>lib/bio/db/pdb.rb
    <li>lib/bio/db/pdb/atom.rb
    <li>lib/bio/db/pdb/chain.rb
    <li>lib/bio/db/pdb/chemicalcomponent.rb
    <li>lib/bio/db/pdb/model.rb
    <li>lib/bio/db/pdb/pdb.rb
    <li>lib/bio/db/pdb/residue.rb
    <li>lib/bio/db/pdb/utils.rb
    <li>lib/bio/db/phyloxml/phyloxml_elements.rb
    <li>lib/bio/db/phyloxml/phyloxml_parser.rb
    <li>lib/bio/db/phyloxml/phyloxml_writer.rb
    <li>lib/bio/db/prosite.rb
    <li>lib/bio/db/rebase.rb
    <li>lib/bio/db/sanger_chromatogram/abif.rb
    <li>lib/bio/db/sanger_chromatogram/chromatogram.rb
    <li>lib/bio/db/sanger_chromatogram/scf.rb
    <li>lib/bio/db/soft.rb
    <li>lib/bio/db/transfac.rb
    <li>lib/bio/feature.rb
    <li>lib/bio/io/biosql/ar-biosql.rb
    <li>lib/bio/io/biosql/biosql.rb
    <li>lib/bio/io/das.rb
    <li>lib/bio/io/dbget.rb
    <li>lib/bio/io/ddbjrest.rb
    <li>lib/bio/io/ddbjxml.rb
    <li>lib/bio/io/ebisoap.rb
    <li>lib/bio/io/ensembl.rb
    <li>lib/bio/io/fastacmd.rb
    <li>lib/bio/io/fetch.rb
    <li>lib/bio/io/flatfile.rb
    <li>lib/bio/io/flatfile/autodetection.rb
    <li>lib/bio/io/flatfile/bdb.rb
    <li>lib/bio/io/flatfile/buffer.rb
    <li>lib/bio/io/flatfile/index.rb
    <li>lib/bio/io/flatfile/indexer.rb
    <li>lib/bio/io/flatfile/splitter.rb
    <li>lib/bio/io/higet.rb
    <li>lib/bio/io/hinv.rb
    <li>lib/bio/io/keggapi.rb
    <li>lib/bio/io/ncbirest.rb
    <li>lib/bio/io/ncbisoap.rb
    <li>lib/bio/io/pubmed.rb
    <li>lib/bio/io/registry.rb
    <li>lib/bio/io/soapwsdl.rb
    <li>lib/bio/io/sql.rb
    <li>lib/bio/io/togows.rb
    <li>lib/bio/location.rb
    <li>lib/bio/map.rb
    <li>lib/bio/pathway.rb
    <li>lib/bio/reference.rb
    <li>lib/bio/sequence.rb
    <li>lib/bio/sequence/aa.rb
    <li>lib/bio/sequence/adapter.rb
    <li>lib/bio/sequence/common.rb
    <li>lib/bio/sequence/compat.rb
    <li>lib/bio/sequence/dblink.rb
    <li>lib/bio/sequence/format.rb
    <li>lib/bio/sequence/generic.rb
    <li>lib/bio/sequence/na.rb
    <li>lib/bio/sequence/quality_score.rb
    <li>lib/bio/sequence/sequence_masker.rb
    <li>lib/bio/shell/plugin/das.rb
    <li>lib/bio/tree.rb
    <li>lib/bio/tree/output.rb
    <li>lib/bio/util/color_scheme.rb
    <li>lib/bio/util/contingency_table.rb
    <li>lib/bio/util/restriction_enzyme.rb
    <li>lib/bio/util/restriction_enzyme/analysis.rb
    <li>lib/bio/util/restriction_enzyme/analysis_basic.rb
    <li>lib/bio/util/restriction_enzyme/cut_symbol.rb
    <li>lib/bio/util/restriction_enzyme/dense_int_array.rb
    <li>lib/bio/util/restriction_enzyme/double_stranded.rb
    <li>lib/bio/util/restriction_enzyme/double_stranded/aligned_strands.rb
    <li>lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair.rb
    <li>lib/bio/util/restriction_enzyme/double_stranded/cut_location_pair_in_enzyme_notation.rb
    <li>lib/bio/util/restriction_enzyme/double_stranded/cut_locations.rb
    <li>lib/bio/util/restriction_enzyme/double_stranded/cut_locations_in_enzyme_notation.rb
    <li>lib/bio/util/restriction_enzyme/range/cut_range.rb
    <li>lib/bio/util/restriction_enzyme/range/cut_ranges.rb
    <li>lib/bio/util/restriction_enzyme/range/horizontal_cut_range.rb
    <li>lib/bio/util/restriction_enzyme/range/sequence_range.rb
    <li>lib/bio/util/restriction_enzyme/range/sequence_range/calculated_cuts.rb
    <li>lib/bio/util/restriction_enzyme/range/sequence_range/fragment.rb
    <li>lib/bio/util/restriction_enzyme/range/sequence_range/fragments.rb
    <li>lib/bio/util/restriction_enzyme/range/vertical_cut_range.rb
    <li>lib/bio/util/restriction_enzyme/single_strand.rb
    <li>lib/bio/util/restriction_enzyme/single_strand/cut_locations_in_enzyme_notation.rb
    <li>lib/bio/util/restriction_enzyme/single_strand_complement.rb
    <li>lib/bio/util/restriction_enzyme/sorted_num_array.rb
    <li>lib/bio/util/restriction_enzyme/string_formatting.rb
    <li>lib/bio/util/sirna.rb
    <li>lib/bio/version.rb
  </ul>
</nav>

    
  </div>

  <div id="class-metadata">
    
    
    
    <!-- Method Quickref -->
<nav id="method-list-section" class="section">
  <h3 class="section-header">Methods</h3>

  <ul class="link-list">
    
    <li><a href="#method-c-method_missing">::method_missing</a>
    
  </ul>
</nav>

  </div>

  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="./ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="./KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="./LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="./README_rdoc.html">README</a>
  
    <li class="file"><a href="./README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="./RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="./doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="./Bio.html">Bio</a>
  
    <li><a href="./Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="./Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="./Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="./Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="./Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="./Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="./Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="./Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="./Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="./Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="./Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="./Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="./Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="./Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="./Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="./Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="./Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="./Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="./Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="./Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="./Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="./Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="./Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="./Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="./Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="./Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="./Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="./Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="./Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="./Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="./Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="./Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="./Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="./Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="./Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="./Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="./Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="./Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="./Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="./Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="./Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="./Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="./Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="./Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="./Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="./Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="./Bio/Command.html">Bio::Command</a>
  
    <li><a href="./Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="./Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="./Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="./Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="./Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="./Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="./Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="./Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="./Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="./Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="./Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="./Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="./Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="./Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="./Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="./Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="./Bio/DB.html">Bio::DB</a>
  
    <li><a href="./Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="./Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="./Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="./Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="./Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="./Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="./Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="./Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="./Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="./Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="./Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="./Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="./Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="./Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="./Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="./Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="./Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="./Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="./Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="./Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="./Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="./Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="./Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="./Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="./Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="./Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="./Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="./Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="./Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="./Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="./Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="./Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="./Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="./Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="./Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="./Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="./Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="./Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="./Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="./Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="./Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="./Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="./Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="./Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="./Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="./Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="./Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="./Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="./Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="./Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="./Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="./Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="./Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="./Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="./Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="./Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="./Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="./Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="./Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="./Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="./Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="./Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="./Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="./Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="./Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="./Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="./Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="./Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="./Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="./Bio/Features.html">Bio::Features</a>
  
    <li><a href="./Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="./Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="./Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="./Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="./Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="./Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="./Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="./Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="./Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="./Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="./Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="./Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="./Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="./Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="./Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="./Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="./Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="./Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="./Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="./Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="./Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="./Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="./Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="./Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="./Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="./Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="./Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="./Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="./Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="./Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="./Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="./Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="./Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="./Bio/GO.html">Bio::GO</a>
  
    <li><a href="./Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="./Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="./Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="./Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="./Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="./Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="./Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="./Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="./Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="./Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="./Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="./Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="./Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="./Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="./Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="./Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="./Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="./Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="./Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="./Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="./Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="./Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="./Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="./Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="./Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="./Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="./Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="./Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="./Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="./Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="./Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="./Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="./Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="./Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="./Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="./Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="./Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="./Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="./Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="./Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="./Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="./Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="./Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="./Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="./Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="./Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="./Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="./Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="./Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="./Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="./Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="./Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="./Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="./Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="./Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="./Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="./Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="./Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="./Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="./Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="./Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="./Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="./Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="./Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="./Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="./Bio/Location.html">Bio::Location</a>
  
    <li><a href="./Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="./Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="./Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="./Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="./Bio/Map.html">Bio::Map</a>
  
    <li><a href="./Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="./Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="./Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="./Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="./Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="./Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="./Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="./Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="./Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="./Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="./Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="./Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="./Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="./Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="./Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="./Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="./Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="./Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="./Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="./Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="./Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="./Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="./Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="./Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="./Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="./Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="./Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="./Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="./Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="./Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="./Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="./Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="./Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="./Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="./Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="./Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="./Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="./Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="./Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="./Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="./Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="./Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="./Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="./Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="./Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="./Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="./Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="./Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="./Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="./Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="./Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="./Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="./Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="./Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="./Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="./Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="./Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="./Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="./Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="./Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="./Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="./Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="./Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="./Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="./Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="./Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="./Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="./Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="./Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="./Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="./Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="./Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="./Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="./Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="./Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="./Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="./Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="./Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="./Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="./Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="./Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="./Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="./Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="./Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="./Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="./Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="./Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="./Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="./Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="./Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="./Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="./Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="./Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="./Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="./Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="./Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="./Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="./Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="./Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="./Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="./Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="./Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="./Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="./Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="./Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="./Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="./Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="./Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="./Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="./Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="./Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="./Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="./Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="./Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="./Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="./Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="./Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="./Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="./Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="./Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="./Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="./Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="./Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="./Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="./Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="./Bio/References.html">Bio::References</a>
  
    <li><a href="./Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="./Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="./Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="./Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="./Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="./Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="./Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="./Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="./Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="./Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="./Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="./Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="./Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="./Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="./Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="./Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="./Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="./Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="./Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="./Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="./Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="./Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="./Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="./Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="./Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="./Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="./Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="./Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="./Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="./Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="./Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="./Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="./Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="./Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="./Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="./Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="./Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="./Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="./Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="./Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="./Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="./Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="./Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="./Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="./Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="./Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="./Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="./Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="./Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="./Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="./Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="./Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="./Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="./Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="./Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="./Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="./Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="./Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="./Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="./Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="./Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="./Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="./Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="./Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="./Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="./Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="./Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="./Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="./Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="./Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="./Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="./Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="./Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="./Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="./Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="./Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="./Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="./Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="./Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="./Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="./Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="./Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="./Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="./Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="./Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="./Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="./Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="./Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="./Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="./Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="./Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="./Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="./Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="./Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="./Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="./Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="./Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="./Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="./Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="./Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="./Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="./Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="./Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="./Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="./Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="./Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="./Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="./Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="./Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="./Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="./Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="./Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="./Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="./Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="./Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="./BiorubyController.html">BiorubyController</a>
  
    <li><a href="./BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="./BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="./Hash.html">Hash</a>
  
    <li><a href="./Object.html">Object</a>
  
    <li><a href="./String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation">
  <h1 class="module">module Bio</h1>

  <div id="description" class="description">
    
<h1 id="label-bio.rb+-+Loading+all+BioRuby+modules">bio.rb - Loading all BioRuby modules</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001-2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<p>+</p>

<h1 id="label-bio%2Fappl%2Fbl2seq%2Freport.rb+-+bl2seq+%28BLAST+2+sequences%29+parser">bio/appl/bl2seq/report.rb - bl2seq (BLAST 2 sequences) parser</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2005 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p><a href="Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a> is a
<a href="Bio/NCBI.html">NCBI</a> bl2seq (BLAST 2 sequences) output parser.</p>

<h1 id="label-Acknowledgements">Acknowledgements</h1>

<p>Thanks to Tomoaki NISHIYAMA &lt;tomoakin __at__
kenroku.kanazawa-u.ac.jp&gt;  for providing bl2seq parser patches based on
lib/bio/appl/blast/format0.rb.</p>

<h1 id="label-bio%2Fappl%2Fblast%2Fformat8.rb+-+BLAST+tab-delimited+output+%28-m+8%29+parser">bio/appl/blast/format8.rb - BLAST tab-delimited output (-m 8) parser</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2002, 2003, 2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h2 id="label-Note">Note</h2>

<p>This file is automatically loaded by bio/appl/blast/report.rb</p>

<h1 id="label-bio%2Fappl%2Fblast%2Fremote.rb+-+remote+BLAST+wrapper+basic+module">bio/appl/blast/remote.rb - remote BLAST wrapper basic module</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008  Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fappl%2Fblast%2Fwublast.rb+-+WU-BLAST+default+output+parser">bio/appl/blast/wublast.rb - WU-BLAST default output parser</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003, 2008 Naohisa GOTO &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="label-Description">Description</h2>

<p>WU-BLAST default output parser.</p>

<p>The parser is still incomplete and may contain many bugs, because I didnt
have WU-BLAST license. It was tested under web-based WU-BLAST results and
obsolete version downloaded from <a
href="http://blast.wustl.edu/">blast.wustl.edu/</a> .</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a href="http://blast.wustl.edu/">blast.wustl.edu/</a></p>
</li><li>
<p><a href="http://www.ebi.ac.uk/blast2/">www.ebi.ac.uk/blast2/</a></p>
</li></ul>

<h1 id="label-bio%2Fappl%2Femboss.rb+-+EMBOSS+wrapper">bio/appl/emboss.rb - <a href="Bio/EMBOSS.html">EMBOSS</a> wrapper</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2002, 2005 Toshiaki Katayama&lt;k@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2006       Jan Aerts &lt;bbsrc at jan.aerts.ac.uk&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: emboss.rb,v 1.9 2008/01/10 03:51:06 ngoto Exp $</p>

<p>+</p>

<h1 id="label-bio%2Fappl%2Fgcg%2Fseq.rb+-+GCG+sequence+file+format+class+%28.seq%2F.pep+file%29">bio/appl/gcg/seq.rb - <a href="Bio/GCG.html">GCG</a> sequence file format class (.seq/.pep file)</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003, 2006 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: seq.rb,v 1.3 2007/04/05 23:35:39 trevor Exp $</p>

<h1 id="label-About+Bio%3A%3AGCG%3A%3ASeq">About <a href="Bio/GCG/Seq.html">Bio::GCG::Seq</a></h1>

<p>Please refer document of <a href="Bio/GCG/Seq.html">Bio::GCG::Seq</a>.</p>

<h1 id="label-bio%2Fappl%2Fiprscan%2Freport.rb+-+a+class+for+iprscan+output.">bio/appl/iprscan/report.rb - a class for iprscan output.</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Mitsuteru C. Nakao &lt;mn@kazusa.or.jp&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: report.rb,v 1.9 2007/07/18 11:11:57 nakao Exp $</pre>

<h2 id="label-Report+classes+for+the+iprscan+program.">Report classes for the iprscan program.</h2>

<h1 id="label-bio%2Fappl%2Fmeme%2Fmotif.rb+-+Class+to+represent+a+sequence+motif">bio/appl/meme/motif.rb - Class to represent a sequence motif</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Adam Kraut &lt;adamnkraut@gmail.com&gt;,</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="label-Description">Description</h2>

<p>This file contains a minimal class to represent meme motifs</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a
href="http://meme.sdsc.edu/meme/intro.html">meme.sdsc.edu/meme/intro.html</a></p>
</li></ul>

<h1 id="label-bio%2Fappl%2Fmuscle.rb+-+MUSCLE+application+wrapper+class">bio/appl/muscle.rb - MUSCLE application wrapper class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of
Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: muscle.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Muscle.html">Bio::Muscle</a> is a wrapper class to execute
MUSCLE.</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a href="http://www.drive5.com/muscle/">www.drive5.com/muscle/</a></p>
</li><li>
<p>Edgar R.C. MUSCLE: multiple sequence alignment with high accuracy and high
throughput. Nucleic Acids Res. 32: 1792-1797, 2004.</p>
</li><li>
<p>Edgar, R.C. MUSCLE: a multiple sequence alignment method with reduced time
and space complexity. BMC Bioinformatics 5: 113, 2004.</p>
</li></ul>

<h1 id="label-bio%2Fappl%2Fphylip%2Falignment.rb+-+phylip+multiple+alignment+format+parser">bio/appl/phylip/alignment.rb - phylip multiple alignment format parser</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 GOTO Naohisa &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: alignment.rb,v 1.2 2007/04/05 23:35:40 trevor Exp $</pre>

<h1 id="label-About+Bio%3A%3APhylip%3A%3APhylipFormat">About <a href="Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a></h1>

<p>Please refer document of <a
href="Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a> class.</p>

<h1 id="label-bio%2Fappl%2Fprobcons.rb+-+ProbCons+application+wrapper+class">bio/appl/probcons.rb - ProbCons application wrapper class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of
Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: probcons.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Probcons.html">Bio::Probcons</a> is a wrapper class to execute
ProbCons (Probabilistic Consistency-based Multiple <a
href="Bio/Alignment.html">Alignment</a> of Amino Acid Sequences).</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a href="http://probcons.stanford.edu/">probcons.stanford.edu/</a></p>
</li><li>
<p>Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S. ProbCons:
Probabilistic Consistency-based Multiple <a
href="Bio/Sequence.html">Sequence</a> <a
href="Bio/Alignment.html">Alignment</a>. Genome Research 15: 330-340, 2005.</p>
</li></ul>

<h1 id="label-bio%2Fappl%2Fsim4%2Freport.rb+-+sim4+result+parser">bio/appl/sim4/report.rb - sim4 result parser</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004 GOTO Naohisa &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>The sim4 report parser classes.</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p>Florea, L., et al., A Computer program for aligning a cDNA sequence with a
genomic DNA sequence, Genome Research, 8, 967974, 1998. <a
href="http://www.genome.org/cgi/content/abstract/8/9/967">www.genome.org/cgi/content/abstract/8/9/967</a></p>
</li></ul>

<h1 id="label-bio%2Fappl%2Fsosui%2Freport.rb+-+SOSUI+report+class">bio/appl/sosui/report.rb - <a href="Bio/SOSUI.html">SOSUI</a> report class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003  Mitsuteru C. Nakao &lt;n@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<h2 id="label-Example">Example</h2>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a
href="http://bp.nuap.nagoya-u.ac.jp/sosui/">bp.nuap.nagoya-u.ac.jp/sosui/</a></p>
</li><li>
<p><a
href="http://bp.nuap.nagoya-u.ac.jp/sosui/sosui_submit.html">bp.nuap.nagoya-u.ac.jp/sosui/sosui_submit.html</a></p>
</li></ul>

<h1 id="label-bio%2Fappl%2Ftargetp%2Freport.rb+-+TargetP+report+class">bio/appl/targetp/report.rb - <a href="Bio/TargetP.html">TargetP</a> report class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2003  Mitsuteru C. Nakao &lt;n@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<h2 id="label-Description">Description</h2>

<p><a href="Bio/TargetP.html">TargetP</a> class for <a
href="http://www.cbs.dtu.dk/services/TargetP/">www.cbs.dtu.dk/services/TargetP/</a></p>

<h2 id="label-Example">Example</h2>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>

<h1 id="label-bio%2Fappl%2Ftcoffee.rb+-+T-Coffee+application+wrapper+class">bio/appl/tcoffee.rb - T-Coffee application wrapper class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006-2007 Jeffrey Blakeslee and John Conery University of
Oregon &lt;jeffb@uoregon.edu&gt; Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: tcoffee.rb,v 1.1 2007/07/16 12:25:50 ngoto Exp $</pre>

<p><a href="Bio/Tcoffee.html">Bio::Tcoffee</a> is a wrapper class to execute
T-Coffee.</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a
href="http://www.tcoffee.org/Projects_home_page/t_coffee_home_page.html">www.tcoffee.org/Projects_home_page/t_coffee_home_page.html</a></p>
</li><li>
<p>Notredame, C., Higgins, D.G. and Heringa, J. T-Coffee: A novel method for
fast and accurate multiple sequence alignment. J. Mol. Biol. 302: 205-217,
2000.</p>
</li></ul>

<h1 id="label-bio%2Fcompat%2Freferences.rb+-+Obsoleted+References+class">bio/compat/references.rb - Obsoleted <a href="Bio/References.html">References</a> class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;, Jan Aerts &lt;jandot@bioruby.org&gt;, Naohisa Goto
&lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: references.rb,v 1.1.2.1 2008/03/04 10:07:49 ngoto Exp $</p>

<h2 id="label-Description">Description</h2>

<p>The <a href="Bio/References.html">Bio::References</a> class was obsoleted
after BioRuby 1.2.1. To keep compatibility, some wrapper methods are
provided in this file. As the compatibility methods (and <a
href="Bio/References.html">Bio::References</a>) will soon be removed,
Please change your code not to use <a
href="Bio/References.html">Bio::References</a>.</p>

<p>Note that <a href="Bio/Reference.html">Bio::Reference</a> is different from
<a href="Bio/References.html">Bio::References</a>. <a
href="Bio/Reference.html">Bio::Reference</a> still exists for storing a
reference information in sequence entries.</p>

<h1 id="label-bio%2Fdata%2Faa.rb+-+Amino+Acids">bio/data/aa.rb - Amino Acids</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h1 id="label-bio%2Fdata%2Fcodontable.rb+-+Codon+Table">bio/data/codontable.rb - Codon Table</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2004 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h2 id="label-Data+source">Data source</h2>

<p>Data in this class is converted from the NCBIs genetic codes page.</p>

<pre>* ((&lt;URL:http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=t&gt;))</pre>

<h3 id="label-Examples">Examples</h3>

<p>Obtain a codon table No.1  Standard (Eukaryote)</p>

<pre>table = Bio::CodonTable[1]</pre>

<p>Obtain a copy of the codon table No.1 to modify.  In this example, reassign
a seleno cystein (U) to the tga codon.</p>

<pre>table = Bio::CodonTable.copy(1)
table['tga'] = 'U'</pre>

<p>Create a new codon table by your own from the <a href="Hash.html">Hash</a>
which contains pairs of codon and amino acid.  You can also define the
table name in the second argument.</p>

<pre class="ruby"><span class="ruby-identifier">hash</span> = { <span class="ruby-string">'ttt'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'F'</span>, <span class="ruby-string">'ttc'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'ttc'</span>, <span class="ruby-operator">...</span> }
<span class="ruby-identifier">table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">hash</span>, <span class="ruby-string">&quot;my codon table&quot;</span>)
</pre>

<p>Obtain a translated amino acid by codon.</p>

<pre class="ruby"><span class="ruby-identifier">table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>[<span class="ruby-value">1</span>]
<span class="ruby-identifier">table</span>[<span class="ruby-string">'ttt'</span>]  <span class="ruby-comment"># =&gt; F</span>
</pre>

<p>Reverse translation of a amino acid into a list of relevant codons.</p>

<pre class="ruby"><span class="ruby-identifier">table</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">CodonTable</span>[<span class="ruby-value">1</span>]
<span class="ruby-identifier">table</span>.<span class="ruby-identifier">revtrans</span>(<span class="ruby-string">&quot;A&quot;</span>)  <span class="ruby-comment"># =&gt; [&quot;gcg&quot;, &quot;gct&quot;, &quot;gca&quot;, &quot;gcc&quot;]</span>
</pre>

<h1 id="label-bio%2Fdata%2Fna.rb+-+Nucleic+Acids">bio/data/na.rb - Nucleic Acids</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h2 id="label-Synopsis">Synopsis</h2>

<p><a href="Bio/NucleicAcid.html">Bio::NucleicAcid</a> class contains data
related to nucleic acids.</p>

<h2 id="label-Usage">Usage</h2>

<p>Examples:</p>

<pre class="ruby"><span class="ruby-identifier">require</span> <span class="ruby-string">'bio'</span>

<span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;### na = Bio::NucleicAcid.new&quot;</span>
<span class="ruby-identifier">na</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">new</span>

<span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;# na.to_re('yrwskmbdhvnatgc')&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">na</span>.<span class="ruby-identifier">to_re</span>(<span class="ruby-string">'yrwskmbdhvnatgc'</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;# Bio::NucleicAcid.to_re('yrwskmbdhvnatgc')&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">to_re</span>(<span class="ruby-string">'yrwskmbdhvnatgc'</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;# na.weight('A')&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">na</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">'A'</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;# Bio::NucleicAcid.weight('A')&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">'A'</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;# na.weight('atgc')&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">na</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">'atgc'</span>)

<span class="ruby-identifier">puts</span> <span class="ruby-node">&quot;# Bio::NucleicAcid.weight('atgc')&quot;</span>
<span class="ruby-identifier">p</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">NucleicAcid</span>.<span class="ruby-identifier">weight</span>(<span class="ruby-string">'atgc'</span>)
</pre>

<p>TODO save on db reading from a genbank or embl object</p>

<p>require bio/sequence</p>

<h1 id="label-bio%2Fdb%2Ffasta%2Fdefline.rb+-+FASTA+defline+parser+class">bio/db/fasta/defline.rb - FASTA defline parser class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2002 GOTO Naohisa &lt;ngoto@gen-info.osaka-u.ac.jp&gt;,
Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: defline.rb,v 1.1.2.1 2008/06/20 13:22:32 ngoto Exp $</p>

<h2 id="label-Description">Description</h2>

<p><a href="Bio/FastaDefline.html">Bio::FastaDefline</a> is a parser class for
definition line (defline) of the FASTA format.</p>

<h2 id="label-Examples">Examples</h2>

<pre class="ruby"><span class="ruby-identifier">rub</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">FastaDefline</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">'&gt;gi|671595|emb|CAA85678.1| rubisco large subunit [Perovskia abrotanoides]'</span>)
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">entry_id</span>       <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">'gi|671595'</span>
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">get</span>(<span class="ruby-string">'emb'</span>)     <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">'CAA85678.1'</span>
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">emb</span>            <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">'CAA85678.1'</span>
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">gi</span>             <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">'671595'</span>
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">accession</span>      <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">'CAA85678'</span>
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">accessions</span>     <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> [ <span class="ruby-string">'CAA85678'</span> ]
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">acc_version</span>    <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">'CAA85678.1'</span>
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">locus</span>          <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-keyword">nil</span>
<span class="ruby-identifier">rub</span>.<span class="ruby-identifier">list_ids</span>       <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> [[<span class="ruby-string">&quot;gi&quot;</span>, <span class="ruby-string">&quot;671595&quot;</span>],
                        [<span class="ruby-string">&quot;emb&quot;</span>, <span class="ruby-string">&quot;CAA85678.1&quot;</span>, <span class="ruby-keyword">nil</span>],
                        [<span class="ruby-string">&quot;Perovskia abrotanoides&quot;</span>]]

<span class="ruby-identifier">ckr</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">FastaDefline</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">&quot;&gt;gi|2495000|sp|Q63931|CCKR_CAVPO CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)\0001gi|2147182|pir||I51898 cholecystokinin A receptor - guinea pig\0001gi|544724|gb|AAB29504.1| cholecystokinin A receptor; CCK-A receptor [Cavia]&quot;</span>)
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">entry_id</span>      <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;gi|2495000&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">sp</span>            <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;CCKR_CAVPO&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">pir</span>           <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;I51898&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">gb</span>            <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;AAB29504.1&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">gi</span>            <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;2495000&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">accession</span>     <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;AAB29504&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">accessions</span>    <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> [<span class="ruby-string">&quot;Q63931&quot;</span>, <span class="ruby-string">&quot;AAB29504&quot;</span>]
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">acc_version</span>   <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;AAB29504.1&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">locus</span>         <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> <span class="ruby-keyword">nil</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">description</span>   <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span>
  <span class="ruby-string">&quot;CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)&quot;</span>
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">descriptions</span>  <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span>
  [<span class="ruby-string">&quot;CHOLECYSTOKININ TYPE A RECEPTOR (CCK-A RECEPTOR) (CCK-AR)&quot;</span>,
   <span class="ruby-string">&quot;cholecystokinin A receptor - guinea pig&quot;</span>,
   <span class="ruby-string">&quot;cholecystokinin A receptor; CCK-A receptor [Cavia]&quot;</span>]
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">words</span>         <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span> 
  [<span class="ruby-string">&quot;cavia&quot;</span>, <span class="ruby-string">&quot;cck-a&quot;</span>, <span class="ruby-string">&quot;cck-ar&quot;</span>, <span class="ruby-string">&quot;cholecystokinin&quot;</span>, <span class="ruby-string">&quot;guinea&quot;</span>, <span class="ruby-string">&quot;pig&quot;</span>,
   <span class="ruby-string">&quot;receptor&quot;</span>, <span class="ruby-string">&quot;type&quot;</span>]
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">id_strings</span>    <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span>
  [<span class="ruby-string">&quot;2495000&quot;</span>, <span class="ruby-string">&quot;Q63931&quot;</span>, <span class="ruby-string">&quot;CCKR_CAVPO&quot;</span>, <span class="ruby-string">&quot;2147182&quot;</span>, <span class="ruby-string">&quot;I51898&quot;</span>,
   <span class="ruby-string">&quot;544724&quot;</span>, <span class="ruby-string">&quot;AAB29504.1&quot;</span>, <span class="ruby-string">&quot;Cavia&quot;</span>]
<span class="ruby-identifier">ckr</span>.<span class="ruby-identifier">list_ids</span>      <span class="ruby-operator">==</span><span class="ruby-operator">&gt;</span>
  [[<span class="ruby-string">&quot;gi&quot;</span>, <span class="ruby-string">&quot;2495000&quot;</span>], [<span class="ruby-string">&quot;sp&quot;</span>, <span class="ruby-string">&quot;Q63931&quot;</span>, <span class="ruby-string">&quot;CCKR_CAVPO&quot;</span>],
   [<span class="ruby-string">&quot;gi&quot;</span>, <span class="ruby-string">&quot;2147182&quot;</span>], [<span class="ruby-string">&quot;pir&quot;</span>, <span class="ruby-keyword">nil</span>, <span class="ruby-string">&quot;I51898&quot;</span>], [<span class="ruby-string">&quot;gi&quot;</span>, <span class="ruby-string">&quot;544724&quot;</span>],
   [<span class="ruby-string">&quot;gb&quot;</span>, <span class="ruby-string">&quot;AAB29504.1&quot;</span>, <span class="ruby-keyword">nil</span>], [<span class="ruby-string">&quot;Cavia&quot;</span>]]
</pre>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p>FASTA format (WikiPedia) <a
href="http://en.wikipedia.org/wiki/FASTA_format">en.wikipedia.org/wiki/FASTA_format</a></p>
</li><li>
<p><a href="Bio/Fasta.html">Fasta</a> format description (<a
href="Bio/NCBI.html">NCBI</a>) <a
href="http://www.ncbi.nlm.nih.gov/BLAST/fasta.shtml">www.ncbi.nlm.nih.gov/BLAST/fasta.shtml</a></p>
</li></ul>

<h1 id="label-bio%2Fdb%2Fkegg%2Fcommon.rb+-+Common+methods+for+KEGG+database+classes">bio/db/kegg/common.rb - Common methods for <a href="Bio/KEGG.html">KEGG</a> database classes</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001-2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2003 Masumi Itoh &lt;m@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2009 Kozo Nishida &lt;kozo-ni@is.naist.jp&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="label-Description">Description</h2>

<p>Note that the modules in this file are intended to be Bio::KEGG::* 
internal use only.</p>

<p>This file contains modules that implement methods commonly used from <a
href="Bio/KEGG.html">KEGG</a> database parser classes.</p>

<h1 id="label-bio%2Fdb%2Fkegg%2Fgenes.rb+-+KEGG%2FGENES+database+class">bio/db/kegg/genes.rb - KEGG/GENES database class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2002, 2006, 2010 Toshiaki Katayama
&lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h2 id="label-KEGG+GENES+parser"><a href="Bio/KEGG.html">KEGG</a> GENES parser</h2>

<p>See <a
href="http://www.genome.jp/kegg/genes.html">www.genome.jp/kegg/genes.html</a></p>

<h3 id="label-Examples">Examples</h3>

<pre class="ruby"><span class="ruby-identifier">require</span> <span class="ruby-string">'bio/io/fetch'</span>
<span class="ruby-identifier">entry_string</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Fetch</span>.<span class="ruby-identifier">query</span>(<span class="ruby-string">'genes'</span>, <span class="ruby-string">'b0002'</span>)

<span class="ruby-identifier">entry</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">KEGG</span><span class="ruby-operator">::</span><span class="ruby-constant">GENES</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">entry_string</span>)

<span class="ruby-comment"># ENTRY</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">entry</span>       <span class="ruby-comment"># =&gt; Hash</span>

<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">entry_id</span>    <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">division</span>    <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">organism</span>    <span class="ruby-comment"># =&gt; String</span>

<span class="ruby-comment"># NAME</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">name</span>        <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">names</span>       <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># DEFINITION</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">definition</span>  <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">eclinks</span>     <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># PATHWAY</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">pathway</span>     <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">pathways</span>    <span class="ruby-comment"># =&gt; Hash</span>

<span class="ruby-comment"># POSITION</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">position</span>    <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">chromosome</span>  <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">gbposition</span>  <span class="ruby-comment"># =&gt; String</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">locations</span>   <span class="ruby-comment"># =&gt; Bio::Locations</span>

<span class="ruby-comment"># MOTIF</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">motifs</span>      <span class="ruby-comment"># =&gt; Hash of Array</span>

<span class="ruby-comment"># DBLINKS</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">dblinks</span>     <span class="ruby-comment"># =&gt; Hash of Array</span>

<span class="ruby-comment"># STRUCTURE</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">structure</span>   <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># CODON_USAGE</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">codon_usage</span> <span class="ruby-comment"># =&gt; Hash</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">cu_list</span>     <span class="ruby-comment"># =&gt; Array</span>

<span class="ruby-comment"># AASEQ</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">aaseq</span>       <span class="ruby-comment"># =&gt; Bio::Sequence::AA</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">aalen</span>       <span class="ruby-comment"># =&gt; Fixnum</span>

<span class="ruby-comment"># NTSEQ</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">ntseq</span>       <span class="ruby-comment"># =&gt; Bio::Sequence::NA</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">naseq</span>       <span class="ruby-comment"># =&gt; Bio::Sequence::NA</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">ntlen</span>       <span class="ruby-comment"># =&gt; Fixnum</span>
<span class="ruby-identifier">p</span> <span class="ruby-identifier">entry</span>.<span class="ruby-identifier">nalen</span>       <span class="ruby-comment"># =&gt; Fixnum</span>
</pre>

<h1 id="label-bio%2Fdb%2Fkegg%2Fkeggtab.rb+-+KEGG+keggtab+class">bio/db/kegg/keggtab.rb - <a href="Bio/KEGG.html">KEGG</a> keggtab class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001 Mitsuteru C. Nakao &lt;n@bioruby.org&gt; Copyright (C)
2003, 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id: keggtab.rb,v 1.10 2007/04/05 23:35:41 trevor Exp $</pre>

<h1 id="label-bio%2Fdb%2Fkegg%2Ftaxonomy.rb+-+KEGG+taxonomy+parser+class">bio/db/kegg/taxonomy.rb - <a href="Bio/KEGG.html">KEGG</a> taxonomy parser class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2007 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>bio/db/lasergene.rb - Interface for DNAStar <a
href="Bio/Lasergene.html">Lasergene</a> sequence file format</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2007 Center for Biomedical Research Informatics, University of
Minnesota (<a href="http://cbri.umn.edu">cbri.umn.edu</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>definition of the <a href="Bio/PDB.html">PDB</a> class</p>

<h1 id="label-bio%2Fdb%2Fpdb%2Fchain.rb+-+chain+class+for+PDB">bio/db/pdb/chain.rb - chain class for <a href="Bio/PDB.html">PDB</a></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004, 2006 Alex Gutteridge &lt;alexg@ebi.ac.uk&gt; Naohisa
Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-Bio%3A%3APDB%3A%3AChain"><a href="Bio/PDB/Chain.html">Bio::PDB::Chain</a></h1>

<p>Please refer <a href="Bio/PDB/Chain.html">Bio::PDB::Chain</a>.</p>

<h1 id="label-bio%2Fdb%2Fpdb%2Fchemicalcomponent.rb+-+PDB+Chemical+Component+Dictionary+parser">bio/db/pdb/chemicalcomponent.rb - <a href="Bio/PDB.html">PDB</a> Chemical Component Dictionary parser</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 GOTO Naohisa &lt;ngoto@gen-info.osaka-u.ac.jp&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-About+Bio%3A%3APDB%3A%3AChemicalComponent">About <a href="Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a></h1>

<p>Please refer <a
href="Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>.</p>

<h1 id="label-References"><a href="Bio/References.html">References</a></h1>
<ul><li>
<p>((&lt;URL:<a
href="http://deposit.pdb.org/cc_dict_tut.html">deposit.pdb.org/cc_dict_tut.html</a>&gt;))</p>
</li><li>
<p><a
href="http://deposit.pdb.org/het_dictionary.txt">deposit.pdb.org/het_dictionary.txt</a></p>
</li></ul>

<h1 id="label-bio%2Fdb%2Fpdb%2Fmodel.rb+-+model+class+for+PDB">bio/db/pdb/model.rb - model class for <a href="Bio/PDB.html">PDB</a></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004, 2006 Alex Gutteridge &lt;alexg@ebi.ac.uk&gt; Naohisa
Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-Bio%3A%3APDB%3A%3AModel"><a href="Bio/PDB/Model.html">Bio::PDB::Model</a></h1>

<p>Please refer <a href="Bio/PDB/Model.html">Bio::PDB::Model</a>.</p>

<h1 id="label-bio%2Fdb%2Fpdb%2Fresidue.rb+-+residue+class+for+PDB">bio/db/pdb/residue.rb - residue class for <a href="Bio/PDB.html">PDB</a></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004, 2006 Alex Gutteridge &lt;alexg@ebi.ac.uk&gt; Naohisa
Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-Bio%3A%3APDB%3A%3AResidue"><a href="Bio/PDB/Residue.html">Bio::PDB::Residue</a></h1>

<h1 id="label-Bio%3A%3APDB%3A%3AHeterogen"><a href="Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a></h1>

<p>Autoload definition</p>

<p>bio/db/soft.rb - Interface for <a href="Bio/SOFT.html">SOFT</a> formatted
files</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<h1 id="label-bio%2Ffeature.rb+-+Features%2FFeature+class+%28GenBank+Feature+table%29">bio/feature.rb - Features/Feature class (<a href="Bio/GenBank.html">GenBank</a> <a href="Bio/Feature.html">Feature</a> table)</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright  2002, 2005  Toshiaki Katayama &lt;k@bioruby.org&gt;</p>

<pre>2006        Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;</pre>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fio%2Fflatfile.rb+-+flatfile+access+wrapper+class">bio/io/flatfile.rb - flatfile access wrapper class</h1>

<pre>Copyright (C) 2001-2006 Naohisa Goto &lt;ng@bioruby.org&gt;</pre>
<dl class="rdoc-list note-list"><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p><a href="Bio/FlatFile.html">Bio::FlatFile</a> is a helper and wrapper class
to read a biological data file. It acts like a IO object. It can
automatically detect data format, and users do not need to tell the class
what the data is.</p>

<p>Ok Hilmar gives to me some clarification 1) EMBL/GenBank/SwissProt name
in term table, is only a convention assuming data loaded by genbank embl
ans swissprot formats.</p>

<pre>If your features come from others ways for example blast or alignment ... whatever.. the user as to take care about the source.</pre>

<p>TODO: 1) source_term_id =&gt; surce_term and check before if the source
term is present or not and the level, the root should always be something
EMBL/GenBank/SwissProt or contestualized. 2) Into DummyBase class delete
connection there and use Bio::ArSQL.establish_connection which reads info
from a yml file. 3) Chk <a href="Bio/Locations.html">Locations</a> in
Biofeatures ArSQL</p>

<h1 id="label-bio%2Flocation.rb+-+Locations%2FLocation+class+%28GenBank+location+format%29">bio/location.rb - Locations/Location class (<a href="Bio/GenBank.html">GenBank</a> location format)</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001, 2005 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>

<pre>2006       Jan Aerts &lt;jan.aerts@bbsrc.ac.uk&gt;
2008       Naohisa Goto &lt;ng@bioruby.org&gt;</pre>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id:$</p>

<h1 id="label-bio%2Fsequence.rb+-+biological+sequence+class">bio/sequence.rb - biological sequence class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2000-2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Yoshinori
K. Okuji &lt;okuji@enbug.org&gt;, Naohisa Goto &lt;ng@bioruby.org&gt;, Ryan
Raaum &lt;ryan@raaum.org&gt;, Jan Aerts &lt;bbsrc at jan.aerts.ac.uk&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Faa.rb+-+amino+acid+sequence+class">bio/sequence/aa.rb - amino acid sequence class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Fadapter.rb+-+Bio%3A%3ASequence+adapter+helper+module">bio/sequence/adapter.rb - <a href="Bio/Sequence.html">Bio::Sequence</a> adapter helper module</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Naohisa Goto &lt;ng@bioruby.org&gt;,</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Fcommon.rb+-+common+methods+for+biological+sequence">bio/sequence/common.rb - common methods for biological sequence</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Fcompat.rb+-+methods+for+backward+compatibility">bio/sequence/compat.rb - methods for backward compatibility</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Fdblink.rb+-+sequence+ID+with+database+name">bio/sequence/dblink.rb - sequence ID with database name</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2008 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Fgeneric.rb+-+generic+sequence+class+to+store+an+intact+string">bio/sequence/generic.rb - generic sequence class to store an intact string</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Fna.rb+-+nucleic+acid+sequence+class">bio/sequence/na.rb - nucleic acid sequence class</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;, Ryan Raaum
&lt;ryan@raaum.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fsequence%2Fquality_score.rb+-+Sequence+quality+score+manipulation+modules">bio/sequence/quality_score.rb - <a href="Bio/Sequence.html">Sequence</a> quality score manipulation modules</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2009 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="label-Description">Description</h2>

<p><a href="Bio/Sequence.html">Sequence</a> quality score manipulation
modules, mainly used by <a href="Bio/Fastq.html">Bio::Fastq</a> and related
classes.</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p>FASTQ format specification <a
href="http://maq.sourceforge.net/fastq.shtml">maq.sourceforge.net/fastq.shtml</a></p>
</li></ul>

<h1 id="label-bio%2Fsequence%2Fsequence_masker.rb+-+Sequence+masking+helper+methods">bio/sequence/sequence_masker.rb - <a href="Bio/Sequence.html">Sequence</a> masking helper methods</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2010 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="label-Description">Description</h2>

<p><a
href="Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
is a mix-in module to provide helpful methods for masking a sequence.</p>

<p>For details, see documentation of <a
href="Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>.</p>

<h1 id="label-bio%2Fshell%2Fplugin%2Fkeggdas.rb+-+plugin+for+KEGG+DAS">bio/shell/plugin/keggdas.rb - plugin for <a href="Bio/KEGG.html">KEGG</a> <a href="Bio/DAS.html">DAS</a></h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>$Id: das.rb,v 1.3 2007/04/05 23:35:41 trevor Exp $</p>

<h1 id="label-bio%2Ftree%2Foutput.rb+-+Phylogenetic+tree+formatter">bio/tree/output.rb - Phylogenetic tree formatter</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2004-2006 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h2 id="label-Description">Description</h2>

<p>This file contains formatter of <a href="Bio/Newick.html">Newick</a>, NHX
and <a href="Bio/Phylip.html">Phylip</a> distance matrix.</p>

<h2 id="label-References"><a href="Bio/References.html">References</a></h2>
<ul><li>
<p><a
href="http://evolution.genetics.washington.edu/phylip/newick_doc.html">evolution.genetics.washington.edu/phylip/newick_doc.html</a></p>
</li><li>
<p><a
href="http://www.phylosoft.org/forester/NHX.html">www.phylosoft.org/forester/NHX.html</a></p>
</li></ul>

<p>bio/util/color_scheme.rb - Popular color codings for nucleic and amino
acids</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>bio/util/contingency_table.rb - Statistical contingency table analysis for
aligned sequences</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<pre>$Id:$</pre>

<p>bio/util/restriction_enzyme.rb - Digests DNA based on restriction enzyme
cut patterns</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/analysis.rb - Does the work of fragmenting the
DNA from the enzymes</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/cut_symbol.rb - Defines the symbol used to mark
a cut in an enzyme sequence</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/dense_int_array.rb - Internal data storage for
<a
href="Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></p>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2011 Naohisa Goto &lt;ng@bioruby.org&gt; Tomoaki NISHIYAMA</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded.rb - DoubleStranded restriction
enzyme sequence</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/aligned_strands.rb - Align two
SingleStrand objects</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_location_pair.rb - Stores a
cut location pair in 0-based index notation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_location_pair_in_enzyme_notation.rb
- Inherits from DoubleStranded::CutLocationPair</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_locations.rb - Contains an
Array of CutLocationPair objects</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/double_stranded/cut_locations_in_enzyme_notation.rb
- Inherits from DoubleStrand::CutLocations</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/cut_range.rb - Abstract base class for
HorizontalCutRange and VerticalCutRange</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/cut_ranges.rb - Container for many
CutRange objects or CutRange child objects.</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/horizontal_cut_range.rb -</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/sequence_range.rb - A defined range over
a nucleotide sequence</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/sequence_range/calculated_cuts.rb -</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/sequence_range/fragment.rb -</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/analysis/fragments.rb -</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/range/vertical_cut_range.rb -</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/single_strand/cut_locations_in_enzyme_notation.rb
- The cut locations, in enzyme notation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/single_strand_complement.rb - Single strand
restriction enzyme sequence in complement orientation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/sorted_num_array.rb - Internal data storage for
<a
href="Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></p>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2011 Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<p>bio/util/restriction_enzyme/string_formatting.rb - Useful functions for
string manipulation</p>
<dl class="rdoc-list note-list"><dt>Author
<dd>
<p>Trevor Wennblom  &lt;<a
href="mailto:trevor@corevx.com">trevor@corevx.com</a>&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright  2005-2007 Midwinter Laboratories, LLC (<a
href="http://midwinterlabs.com">midwinterlabs.com</a>)</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

<h1 id="label-bio%2Fversion.rb+-+BioRuby+version+information">bio/version.rb - BioRuby version information</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2001-2012 Toshiaki Katayama &lt;k@bioruby.org&gt;, Naohisa
Goto &lt;ng@bioruby.org&gt;</p>
</dd><dt>License
<dd>
<p>The Ruby License</p>
</dd></dl>

  </div><!-- description -->

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    
    <!-- Constants -->
    <section id="constants-list" class="section">
      <h3 class="section-header">Constants</h3>
      <dl>
      
        <dt id="BIORUBY_EXTRA_VERSION">BIORUBY_EXTRA_VERSION
        
        <dd class="description"><p>Extra version specifier (<a href="String.html">String</a> or nil).
Existance of the value indicates pre-release version or modified version.</p>
<dl class="rdoc-list note-list"><dt>nil                 
<dd>
<p>Release version.</p>
</dd><dt>.0000...4999    
<dd>
<p>Release version with patches.</p>
</dd><dt>.5000             
<dd>
<p>Development unstable version.</p>
</dd><dt>.5001...8999    
<dd>
<p>Pre-alpha version.</p>
</dd><dt>-alphaN (N=0..99) 
<dd>
<p>Alpha version.</p>
</dd><dt>-preN   (N=0..99) 
<dd>
<p>Pre-release test version.</p>
</dd><dt>-rcN    (N=0..99) 
<dd>
<p>Release candidate version.</p>
</dd></dl>
        
      
        <dt id="BIORUBY_VERSION">BIORUBY_VERSION
        
        <dd class="description"><p>BioRuby version (Array containing Integer)</p>
        
      
        <dt id="BIORUBY_VERSION_ID">BIORUBY_VERSION_ID
        
        <dd class="description"><p>Version identifier, including extra version string (<a
href="String.html">String</a>) Unlike <a
href="Bio.html#BIORUBY_VERSION">BIORUBY_VERSION</a>, it is not comparable.</p>
        
      
      </dl>
    </section>
    

    

    <!-- Methods -->
    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Class Methods</h3>

    
      <div id="method-c-method_missing" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">method_missing</span><span
            class="method-args">(*args)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Provide BioRuby shell command also as Bio.command (like ChemRuby)</p>
          

          
          <div class="method-source-code" id="method_missing-source">
            <pre><span class="ruby-comment"># File lib/bio.rb, line 309</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">method_missing</span>(*<span class="ruby-identifier">args</span>)
  <span class="ruby-identifier">require</span> <span class="ruby-string">'bio/shell'</span>
  <span class="ruby-identifier">extend</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Shell</span>
  <span class="ruby-identifier">public_class_method</span>(*<span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Shell</span>.<span class="ruby-identifier">private_instance_methods</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-constant">Bio</span>.<span class="ruby-identifier">respond_to?</span>(<span class="ruby-identifier">args</span>.<span class="ruby-identifier">first</span>)
    <span class="ruby-constant">Bio</span>.<span class="ruby-identifier">send</span>(*<span class="ruby-identifier">args</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">raise</span> <span class="ruby-constant">NameError</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- method_missing-source -->
          
        </div>

        

        
      </div><!-- method_missing-method -->

    
    </section><!-- public-class-method-details -->
  
  </section><!-- 5Buntitled-5D -->

</div><!-- documentation -->


<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

