<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>README_DEV - BioRuby API documentation</title>

<link type="text/css" media="screen" href="./rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "./";
</script>

<script type="text/javascript" charset="utf-8" src="./js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="./js/darkfish.js"></script>


<body class="file">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="./index.html">Home</a>
    <a href="./table_of_contents.html#classes">Classes</a>
    <a href="./table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="./ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="./KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="./LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="./README_rdoc.html">README</a>
  
    <li class="file"><a href="./README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="./RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="./doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="./doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="./Bio.html">Bio</a>
  
    <li><a href="./Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="./Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="./Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="./Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="./Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="./Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="./Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="./Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="./Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="./Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="./Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="./Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="./Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="./Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="./Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="./Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="./Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="./Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="./Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="./Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="./Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="./Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="./Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="./Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="./Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="./Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="./Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="./Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="./Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="./Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="./Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="./Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="./Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="./Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="./Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="./Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="./Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="./Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="./Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="./Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="./Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="./Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="./Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="./Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="./Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="./Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="./Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="./Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="./Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="./Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="./Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="./Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="./Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="./Bio/Command.html">Bio::Command</a>
  
    <li><a href="./Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="./Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="./Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="./Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="./Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="./Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="./Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="./Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="./Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="./Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="./Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="./Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="./Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="./Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="./Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="./Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="./Bio/DB.html">Bio::DB</a>
  
    <li><a href="./Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="./Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="./Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="./Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="./Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="./Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="./Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="./Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="./Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="./Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="./Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="./Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="./Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="./Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="./Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="./Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="./Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="./Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="./Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="./Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="./Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="./Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="./Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="./Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="./Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="./Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="./Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="./Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="./Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="./Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="./Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="./Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="./Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="./Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="./Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="./Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="./Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="./Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="./Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="./Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="./Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="./Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="./Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="./Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="./Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="./Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="./Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="./Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="./Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="./Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="./Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="./Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="./Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="./Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="./Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="./Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="./Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="./Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="./Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="./Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="./Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="./Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="./Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="./Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="./Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="./Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="./Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="./Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="./Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="./Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="./Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="./Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="./Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="./Bio/Features.html">Bio::Features</a>
  
    <li><a href="./Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="./Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="./Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="./Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="./Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="./Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="./Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="./Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="./Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="./Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="./Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="./Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="./Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="./Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="./Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="./Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="./Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="./Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="./Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="./Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="./Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="./Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="./Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="./Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="./Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="./Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="./Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="./Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="./Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="./Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="./Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="./Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="./Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="./Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="./Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="./Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="./Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="./Bio/GO.html">Bio::GO</a>
  
    <li><a href="./Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="./Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="./Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="./Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="./Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="./Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="./Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="./Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="./Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="./Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="./Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="./Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="./Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="./Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="./Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="./Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="./Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="./Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="./Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="./Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="./Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="./Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="./Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="./Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="./Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="./Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="./Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="./Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="./Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="./Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="./Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="./Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="./Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="./Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="./Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="./Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="./Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="./Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="./Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="./Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="./Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="./Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="./Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="./Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="./Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="./Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="./Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="./Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="./Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="./Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="./Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="./Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="./Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="./Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="./Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="./Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="./Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="./Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="./Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="./Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="./Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="./Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="./Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="./Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="./Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="./Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="./Bio/Location.html">Bio::Location</a>
  
    <li><a href="./Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="./Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="./Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="./Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="./Bio/Map.html">Bio::Map</a>
  
    <li><a href="./Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="./Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="./Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="./Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="./Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="./Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="./Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="./Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="./Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="./Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="./Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="./Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="./Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="./Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="./Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="./Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="./Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="./Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="./Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="./Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="./Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="./Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="./Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="./Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="./Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="./Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="./Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="./Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="./Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="./Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="./Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="./Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="./Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="./Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="./Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="./Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="./Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="./Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="./Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="./Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="./Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="./Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="./Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="./Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="./Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="./Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="./Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="./Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="./Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="./Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="./Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="./Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="./Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="./Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="./Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="./Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="./Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="./Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="./Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="./Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="./Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="./Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="./Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="./Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="./Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="./Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="./Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="./Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="./Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="./Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="./Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="./Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="./Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="./Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="./Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="./Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="./Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="./Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="./Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="./Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="./Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="./Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="./Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="./Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="./Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="./Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="./Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="./Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="./Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="./Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="./Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="./Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="./Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="./Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="./Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="./Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="./Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="./Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="./Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="./Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="./Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="./Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="./Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="./Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="./Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="./Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="./Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="./Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="./Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="./Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="./Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="./Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="./Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="./Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="./Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="./Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="./Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="./Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="./Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="./Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="./Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="./Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="./Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="./Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="./Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="./Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="./Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="./Bio/References.html">Bio::References</a>
  
    <li><a href="./Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="./Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="./Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="./Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="./Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="./Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="./Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="./Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="./Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="./Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="./Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="./Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="./Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="./Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="./Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="./Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="./Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="./Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="./Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="./Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="./Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="./Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="./Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="./Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="./Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="./Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="./Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="./Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="./Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="./Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="./Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="./Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="./Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="./Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="./Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="./Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="./Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="./Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="./Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="./Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="./Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="./Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="./Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="./Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="./Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="./Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="./Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="./Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="./Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="./Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="./Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="./Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="./Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="./Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="./Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="./Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="./Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="./Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="./Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="./Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="./Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="./Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="./Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="./Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="./Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="./Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="./Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="./Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="./Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="./Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="./Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="./Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="./Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="./Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="./Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="./Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="./Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="./Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="./Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="./Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="./Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="./Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="./Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="./Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="./Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="./Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="./Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="./Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="./Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="./Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="./Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="./Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="./Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="./Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="./Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="./Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="./Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="./Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="./Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="./Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="./Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="./Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="./Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="./Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="./Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="./Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="./Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="./Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="./Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="./Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="./Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="./Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="./Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="./Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="./Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="./Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="./Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="./Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="./Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="./Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="./Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="./BiorubyController.html">BiorubyController</a>
  
    <li><a href="./BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="./BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="./Hash.html">Hash</a>
  
    <li><a href="./Object.html">Object</a>
  
    <li><a href="./String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation" class="description">
  
<h1 id="label-README.DEV"><a href="README_rdoc.html">README</a>.DEV</h1>
<dl class="rdoc-list note-list"><dt>Copyright
<dd>
<p>Copyright (C) 2005, 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2006, 2008 Jan Aerts &lt;jandot@bioruby.org&gt;</p>
</dd><dt>Copyright
<dd>
<p>Copyright (C) 2011       Naohisa Goto &lt;ng@bioruby.org&gt;</p>
</dd></dl>

<h1 id="label-HOW+TO+CONTRIBUTE+TO+THE+BIORUBY+PROJECT%3F">HOW TO CONTRIBUTE TO THE BIORUBY PROJECT?</h1>

<p>There are many possible ways to contribute to the BioRuby project, such as:</p>
<ul><li>
<p>Join the discussion on the BioRuby mailing list</p>
</li><li>
<p>Send a bug report or write a bug fix patch</p>
</li><li>
<p>Add and correct documentation</p>
</li><li>
<p>Develop code for new features, etc.</p>
</li></ul>

<p>All of these are welcome!  This document mainly focuses on the last option,
how to contribute your code to the BioRuby distribution. This may also be
helpful when you send large patches for existing codes.</p>

<p>We would like to include your contribution as long as the scope of your
module meets the field of bioinformatics.</p>

<h2 id="label-Git">Git</h2>

<p>Bioruby is now under git source control at <a
href="http://github.com/bioruby/bioruby.">github.com/bioruby/bioruby.</a> 
There are two basic ways to contribute: with patches or pull requests. Both
are explained on the bioruby wiki at <a
href="http://bioruby.open-bio.org/wiki.">bioruby.open-bio.org/wiki.</a></p>

<h3 id="label-Preparation+before+sending+patches+or+pull+requests">Preparation before sending patches or pull requests</h3>

<p>Before sending patches or pull requests, rewriting history and reordering
or selecting patches are recommended. See Creating the perfect patch
series in the Git Users Manual. <a
href="http://www.kernel.org/pub/software/scm/git/docs/user-manual.html#patch-series">www.kernel.org/pub/software/scm/git/docs/user-manual.html#patch-series</a></p>

<h3 id="label-Sending+your+contribution">Sending your contribution</h3>

<h4 id="label-With+patches">With patches</h4>

<p>You can send patches with git-format-patch. For a smaller change, unified
diff (diff -u) without using git can also be accepted.</p>

<h4 id="label-With+pull+requests">With pull requests</h4>

<p>We are happy if your commits can be pulled with fast-forward. For the
purpose, using git-rebase before sending pull request is recommended. See
Keeping a patch series up to date using git rebase in the Git Users
Manual. <a
href="http://www.kernel.org/pub/software/scm/git/docs/user-manual.html#using-git-rebase">www.kernel.org/pub/software/scm/git/docs/user-manual.html#using-git-rebase</a></p>

<h3 id="label-Notes+for+the+treatment+of+contributions+in+the+blessed+repository">Notes for the treatment of contributions in the blessed repository</h3>

<h4 id="label-Merging+policy">Merging policy</h4>

<p>We do not always merge your commits as is.  We may edit, rewrite, reorder,
select, and/or mix your commits before and/or after merging to the blessed
repository.</p>

<h4 id="label-Git+commit+management+policy">Git commit management policy</h4>

<p>We want to keep the commit history linear as far as possible, because it is
easy to find problems and regressions in commits.  See Why bisecting merge
commits can be harder than bisecting linear history in the Git Users
Manual. <a
href="http://www.kernel.org/pub/software/scm/git/docs/user-manual.html#bisect-merges">www.kernel.org/pub/software/scm/git/docs/user-manual.html#bisect-merges</a></p>

<p>Note that the above policy is only for the main blessed repository, and
it does not aim to restrict each users fork.</p>

<h1 id="label-LICENSE">LICENSE</h1>

<p>If you would like your module to be included in the BioRuby distribution,
you need to give us right to change the license of your module to make it
compatible with other modules in BioRuby.</p>

<p>BioRuby was previously distributed under the LGPL license, but now is
distributed under the same terms as Ruby.</p>

<h1 id="label-CODING+STYLE">CODING STYLE</h1>

<p>You will need to follow the typical coding styles of the BioRuby modules:</p>

<h2 id="label-Use+the+following+naming+conventions">Use the following naming conventions</h2>
<ul><li>
<p>CamelCase for module and class names</p>
</li><li>
<p>_-separated_lowercase for method names</p>
</li><li>
<p>_-separated_lowercase for variable names</p>
</li><li>
<p>all UPPERCASE for constants</p>
</li></ul>

<h2 id="label-Indentation+must+not+include+tabs">Indentation must not include tabs</h2>
<ul><li>
<p>Use 2 spaces for indentation.</p>
</li><li>
<p>Dont replace spaces to tabs.</p>
</li></ul>

<h2 id="label-Parenthesis+in+the+method+definition+line+should+be+written">Parenthesis in the method definition line should be written</h2>
<ul><li>
<p>Good: <code>def example(str, ary)</code></p>
</li><li>
<p>Discouraged: <code>def example str, ary</code></p>
</li></ul>

<h2 id="label-Comments">Comments</h2>

<p>Dont use <code>=begin</code> and <code>=end</code> blocks for comments. 
If you need to add comments, include it in the RDoc documentation.</p>

<h2 id="label-Documentation+should+be+written+in+the+RDoc+format+in+the+source+code">Documentation should be written in the RDoc format in the source code</h2>

<p>The RDoc format is becoming the popular standard for Ruby documentation. We
are now in transition from the previously used RD format to the RDoc format
in API documentation.</p>

<p>Additional tutorial documentation and working examples are encouraged with
your contribution.  You may use the header part of the file for this
purpose as demonstrated in the previous section.</p>

<h2 id="label-Standard+documentation">Standard documentation</h2>

<h3 id="label-of+files">of files</h3>

<p>Each file should start with a header, which covers the following topics:</p>
<ul><li>
<p>copyright</p>
</li><li>
<p>license</p>
</li><li>
<p>description of the file (<em>not</em> the classes; see below)</p>
</li><li>
<p>any references, if appropriate</p>
</li></ul>

<p>The header should be formatted as follows:</p>

<pre class="ruby"><span class="ruby-comment">#</span>
<span class="ruby-comment"># = bio/db/hoge.rb - Hoge database parser classes</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># Copyright::  Copyright (C) 2001, 2003-2005 Bio R. Hacker &lt;brh@example.org&gt;,</span>
<span class="ruby-comment"># Copyright::  Copyright (C) 2006 Chem R. Hacker &lt;crh@example.org&gt;</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># License::    The Ruby License</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># == Description</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># This file contains classes that implement an interface to the Hoge database.</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># == References</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># * Hoge F. et al., The Hoge database, Nucleic. Acid. Res. 123:100--123 (2030)</span>
<span class="ruby-comment"># * http://hoge.db/</span>
<span class="ruby-comment">#</span>

<span class="ruby-identifier">require</span> <span class="ruby-string">'foo'</span>

<span class="ruby-keyword">module</span> <span class="ruby-constant">Bio</span>

  <span class="ruby-identifier">autoload</span> :<span class="ruby-constant">Bar</span>, <span class="ruby-string">'bio/bar'</span>

<span class="ruby-keyword">class</span> <span class="ruby-constant">Hoge</span>
  <span class="ruby-operator">:</span>
<span class="ruby-keyword">end</span>  <span class="ruby-comment"># Hoge</span>

<span class="ruby-keyword">end</span>  <span class="ruby-comment"># Bio</span>
</pre>

<h3 id="label-of+classes+and+methods+within+those+files">of classes and methods within those files</h3>

<p>Classes and methods should be documented in a standardized format, as in
the following example (from lib/bio/sequence.rb):</p>

<pre class="ruby"><span class="ruby-comment"># == Description</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># Bio::Sequence objects represent annotated sequences in bioruby.</span>
<span class="ruby-comment"># A Bio::Sequence object is a wrapper around the actual sequence, </span>
<span class="ruby-comment"># represented as either a Bio::Sequence::NA or a Bio::Sequence::AA object.</span>
<span class="ruby-comment"># For most users, this encapsulation will be completely transparent.</span>
<span class="ruby-comment"># Bio::Sequence responds to all methods defined for Bio::Sequence::NA/AA</span>
<span class="ruby-comment"># objects using the same arguments and returning the same values (even though </span>
<span class="ruby-comment"># these methods are not documented specifically for Bio::Sequence).</span>
<span class="ruby-comment">#</span>
<span class="ruby-comment"># == Usage</span>
<span class="ruby-comment"># </span>
<span class="ruby-comment">#   require 'bio'</span>
<span class="ruby-comment">#   </span>
<span class="ruby-comment">#   # Create a nucleic or amino acid sequence</span>
<span class="ruby-comment">#   dna = Bio::Sequence.auto('atgcatgcATGCATGCAAAA')</span>
<span class="ruby-comment">#   rna = Bio::Sequence.auto('augcaugcaugcaugcaaaa')</span>
<span class="ruby-comment">#   aa = Bio::Sequence.auto('ACDEFGHIKLMNPQRSTVWYU')</span>
<span class="ruby-comment"># </span>
<span class="ruby-comment">#   # Print in FASTA format</span>
<span class="ruby-comment">#   puts dna.output(:fasta)</span>
<span class="ruby-comment"># </span>
<span class="ruby-comment">#   # Print all codons</span>
<span class="ruby-comment">#   dna.window_search(3,3) do |codon|</span>
<span class="ruby-comment">#     puts codon</span>
<span class="ruby-comment">#   end</span>
<span class="ruby-comment"># </span>
<span class="ruby-keyword">class</span> <span class="ruby-constant">Sequence</span>

  <span class="ruby-comment"># Create a new Bio::Sequence object</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment">#   s = Bio::Sequence.new('atgc')</span>
  <span class="ruby-comment">#   puts s                                  # =&gt; 'atgc'</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># Note that this method does not intialize the contained sequence</span>
  <span class="ruby-comment"># as any kind of bioruby object, only as a simple string</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment">#   puts s.seq.class                        # =&gt; String</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># See Bio::Sequence#na, Bio::Sequence#aa, and Bio::Sequence#auto </span>
  <span class="ruby-comment"># for methods to transform the basic String of a just created </span>
  <span class="ruby-comment"># Bio::Sequence object to a proper bioruby object</span>
  <span class="ruby-comment"># ---</span>
  <span class="ruby-comment"># *Arguments*:</span>
  <span class="ruby-comment"># * (required) _str_: String or Bio::Sequence::NA/AA object</span>
  <span class="ruby-comment"># *Returns*:: Bio::Sequence object</span>
  <span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">str</span>)
    <span class="ruby-ivar">@seq</span> = <span class="ruby-identifier">str</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-comment"># The sequence identifier.  For example, for a sequence</span>
  <span class="ruby-comment"># of Genbank origin, this is the accession number.</span>
  <span class="ruby-identifier">attr_accessor</span> :<span class="ruby-identifier">entry_id</span>

  <span class="ruby-comment"># An Array of Bio::Feature objects</span>
  <span class="ruby-identifier">attr_accessor</span> :<span class="ruby-identifier">features</span>
<span class="ruby-keyword">end</span> <span class="ruby-comment"># Sequence</span>
</pre>

<p>Preceding the class definition (<code>class Sequence</code>), there is at
least a  description and a usage example. Please use the
<code>Description</code> and <code>Usage</code> headings. If appropriate,
refer to other classes that interact with or are related to the class.</p>

<p>The code in the usage example should, if possible, be in a format that a
user can copy-and-paste into a new script to run. It should illustrate the
most important uses of the class. If possible and if it would not clutter
up the example too much, try to provide any input data directly into the
usage example, instead of refering to ARGV or ARGF for input.</p>

<pre>dna = Bio::Sequence.auto('atgcatgcATGCATGCAAAA')</pre>

<p>Otherwise, describe the input shortly, for example:</p>

<pre># input should be string consisting of nucleotides
dna = Bio::Sequence.auto(ARGF.read)</pre>

<p>Methods should be preceded by a comment that describes what the method
does,  including any relevant usage examples. (In contrast to the
documentation for the class itself, headings are not required.) In
addition, any arguments should be listed, as well as the type of thing that
is returned by the method. The format of this information is as follows:</p>

<pre># ---
# *Arguments*:
# * (required) _str_: String or Bio::Sequence::NA
# * (optional) _nr_: a number that means something
# *Returns*:: true or false</pre>

<p>Attribute accessors can be preceded by a short description.</p>

<pre># P-value (Float)
attr_reader :pvalue</pre>

<p>For writing rdoc documentation, putting two or more attributes in a line
(such as <code>attr_reader :evalue, :pvalue</code>) is strongly
discouraged.</p>

<p>Methods looks like attributes can also be preceded by a short description.</p>

<pre class="ruby"><span class="ruby-comment"># Scientific name (String)</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">scientific_name</span>
  <span class="ruby-comment">#...</span>
<span class="ruby-keyword">end</span>

<span class="ruby-comment"># Scientific name (String)</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">scientific_name=</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-comment">#...</span>
<span class="ruby-keyword">end</span>
</pre>

<h2 id="label-Exception+handling">Exception handling</h2>

<p>Dont use</p>

<pre>$stderr.puts &quot;WARNING&quot;</pre>

<p>in your code. Instead, try to avoid printing error messages. For fatal
errors, use <code>raise</code> with an appropriate message.</p>

<p>Kernel#warn can only be used to notice incompatible changes to programmers.
Typically it may be used for deprecated or obsolete usage of a method. For
example,</p>

<pre>warn &quot;The Foo#bar method is obsoleted. Use Foo#baz instead.&quot;</pre>

<h2 id="label-Testing+code+should+use+%27test%2Funit%27">Testing code should use test/unit</h2>

<p>Unit tests should come with your modules by which you can assure what you
meant to do with each method.  The test code is useful to make maintenance
easy and ensure stability. The use of</p>

<pre>if __FILE__ == $0</pre>

<p>is deprecated.</p>

<h2 id="label-Using+autoload">Using autoload</h2>

<p>To quicken the initial load time we have replaced most of require to 
autoload since BioRuby version 0.7.  During this change, we have found
some tips:</p>

<p>You should not separate the same namespace into several files.</p>
<ul><li>
<p>For example, if you have separated definitions of the Bio::Foo class into
two files (e.g. bio/foo.rb and bio/bar.rb), you need to resolve the
dependencies (including the load order) yourself.</p>
</li><li>
<p>If you have a defined Bio::Foo in bio/foo.rb and a defined Bio::Foo::Bar
in bio/foo/bar.rb add the following line in the bio/foo.rb file:</p>

<pre>autoload :Bar, 'bio/foo/bar'</pre>
</li></ul>

<p>You should not put several top level namespaces in one file.</p>
<ul><li>
<p>For example, if you have Bio::A, Bio::B and Bio::C in the file
bio/foo.rb, you need</p>

<pre>autoload :A, 'bio/foo'
autoload :B, 'bio/foo'
autoload :C, 'bio/foo'</pre>

<p>to load the module automatically (instead of require bio/foo). In this
case, you should put them under the new namespace like Bio::Foo::A,
Bio::Foo::B and Bio::Foo::C in the file bio/foo, then use</p>

<pre>autoload :Foo, 'bio/foo'</pre>

<p>so autoload can be written in 1 line.</p>
</li></ul>

<h1 id="label-NAMESPACE">NAMESPACE</h1>

<p>Your module should be located under the top-level module <a
href="Bio.html">Bio</a> and put under the bioruby/lib/bio directory.  The
class/module names and the file names should be short and descriptive.</p>

<p>There are already several sub directories in bioruby/lib:</p>

<pre>bio/*.rb   -- general and widely used basic classes
bio/appl/  -- wrapper and parser for the external applications
bio/data/  -- basic biological data
bio/db/    -- flatfile database entry parsers
bio/io/    -- I/O interfaces for files, RDB, web services etc.
bio/util/  -- utilities and algorithms for bioinformatics</pre>

<p>If your module doesnt match any of the above, please propose an
appropriate directory name when you contribute. Please let the staff
discuss on namespaces (class names), API (method names) before commiting a
new module or making changes on existing modules.</p>

<h1 id="label-MAINTENANCE">MAINTENANCE</h1>

<p>Finally, please maintain the code youve contributed. Please let us know
(on  the bioruby list) before you commit, so that users can discuss on the
change.</p>

<h1 id="label-RUBY+VERSION+and+IMPLEMENTATION">RUBY VERSION and IMPLEMENTATION</h1>

<p>We are mainly using Ruby MRI (Matz Ruby Implementation, or Matz Ruby
Interpreter).  Please confirm that your code is running on current stable
release versions of Ruby MRI.</p>

<p>We are very happy if your code can run on both Ruby 1.8.x and 1.9.x. Note
that Ruby 1.9.0 should be ignored because it was discontinued. Ruby 1.8.5
or earlier versions can also be ignored. See <a
href="README_rdoc.html">README.rdoc</a> and <a
href="RELEASE_NOTES_rdoc.html">RELEASE_NOTES.rdoc</a> for recommended Ruby
versions.</p>

<p>It is welcome to support JRuby, Rubinius, etc, in addition to Ruby MRI.</p>

<p>Of course, it is strongly encouraged to write code that is not affected by
differences between Ruby versions and/or implementations, as far as
possible.</p>

<h1 id="label-OS+and+ARCHITECTURE">OS and ARCHITECTURE</h1>

<p>We hope BioRuby can be run on both UNIX (and UNIX-like OS) and Microsoft
Windows.</p>

</div>



<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

