<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>class Bio::Pathway - BioRuby API documentation</title>

<link type="text/css" media="screen" href="../rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
</script>

<script type="text/javascript" charset="utf-8" src="../js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/darkfish.js"></script>


<body id="top" class="class">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="../index.html">Home</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="file-metadata">
    <nav id="file-list-section" class="section">
  <h3 class="section-header">Defined In</h3>
  <ul>
    <li>lib/bio/pathway.rb
  </ul>
</nav>

    
  </div>

  <div id="class-metadata">
    
    <nav id="parent-class-section" class="section">
  <h3 class="section-header">Parent</h3>
  
  <p class="link"><a href="../Object.html">Object</a>
  
</nav>

    
    <!-- Method Quickref -->
<nav id="method-list-section" class="section">
  <h3 class="section-header">Methods</h3>

  <ul class="link-list">
    
    <li><a href="#method-c-new">::new</a>
    
    <li><a href="#method-i-append">#append</a>
    
    <li><a href="#method-i-bellman_ford">#bellman_ford</a>
    
    <li><a href="#method-i-bfs">#bfs</a>
    
    <li><a href="#method-i-bfs_shortest_path">#bfs_shortest_path</a>
    
    <li><a href="#method-i-breadth_first_search">#breadth_first_search</a>
    
    <li><a href="#method-i-clear_relations-21">#clear_relations!</a>
    
    <li><a href="#method-i-clique">#clique</a>
    
    <li><a href="#method-i-cliquishness">#cliquishness</a>
    
    <li><a href="#method-i-common_subgraph">#common_subgraph</a>
    
    <li><a href="#method-i-delete">#delete</a>
    
    <li><a href="#method-i-depth_first_search">#depth_first_search</a>
    
    <li><a href="#method-i-dfs">#dfs</a>
    
    <li><a href="#method-i-dfs_topological_sort">#dfs_topological_sort</a>
    
    <li><a href="#method-i-dijkstra">#dijkstra</a>
    
    <li><a href="#method-i-directed">#directed</a>
    
    <li><a href="#method-i-directed-3F">#directed?</a>
    
    <li><a href="#method-i-dump_list">#dump_list</a>
    
    <li><a href="#method-i-dump_matrix">#dump_matrix</a>
    
    <li><a href="#method-i-edges">#edges</a>
    
    <li><a href="#method-i-nodes">#nodes</a>
    
    <li><a href="#method-i-small_world">#small_world</a>
    
    <li><a href="#method-i-subgraph">#subgraph</a>
    
    <li><a href="#method-i-to_list">#to_list</a>
    
    <li><a href="#method-i-to_matrix">#to_matrix</a>
    
    <li><a href="#method-i-to_relations">#to_relations</a>
    
    <li><a href="#method-i-undirected">#undirected</a>
    
    <li><a href="#method-i-undirected-3F">#undirected?</a>
    
  </ul>
</nav>

  </div>

  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="../ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="../LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="../README_rdoc.html">README</a>
  
    <li class="file"><a href="../README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="../Bio.html">Bio</a>
  
    <li><a href="../Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="../Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="../Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="../Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="../Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="../Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="../Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="../Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="../Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="../Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="../Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="../Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="../Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="../Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="../Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="../Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="../Bio/Command.html">Bio::Command</a>
  
    <li><a href="../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="../Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="../Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="../Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="../Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="../Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="../Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="../Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="../Bio/DB.html">Bio::DB</a>
  
    <li><a href="../Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="../Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="../Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="../Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="../Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="../Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="../Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="../Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="../Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="../Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="../Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="../Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="../Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="../Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="../Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="../Bio/Features.html">Bio::Features</a>
  
    <li><a href="../Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="../Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="../Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="../Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="../Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="../Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="../Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="../Bio/GO.html">Bio::GO</a>
  
    <li><a href="../Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="../Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="../Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="../Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="../Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="../Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="../Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="../Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="../Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="../Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="../Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="../Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="../Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="../Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="../Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="../Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="../Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="../Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="../Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="../Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="../Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="../Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="../Bio/Location.html">Bio::Location</a>
  
    <li><a href="../Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="../Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="../Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="../Bio/Map.html">Bio::Map</a>
  
    <li><a href="../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="../Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="../Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="../Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="../Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="../Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="../Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="../Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="../Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="../Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="../Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="../Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="../Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="../Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="../Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="../Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="../Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="../Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="../Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="../Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="../Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="../Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="../Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="../Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="../Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="../Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="../Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="../Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="../Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="../Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="../Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="../Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="../Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="../Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="../Bio/References.html">Bio::References</a>
  
    <li><a href="../Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="../Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="../Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="../Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="../Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="../Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="../Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="../Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="../Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="../Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="../Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="../Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="../Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="../Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="../Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="../Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="../Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="../Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="../Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="../Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="../Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="../Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="../Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="../Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="../Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="../Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="../Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="../Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="../Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="../Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="../Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="../Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="../Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="../Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="../Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="../Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="../Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="../Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="../Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="../Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="../Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="../BiorubyController.html">BiorubyController</a>
  
    <li><a href="../BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="../BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="../Hash.html">Hash</a>
  
    <li><a href="../Object.html">Object</a>
  
    <li><a href="../String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation">
  <h1 class="class">class Bio::Pathway</h1>

  <div id="description" class="description">
    
<p><a href="Pathway.html">Bio::Pathway</a> is a general graph object initially
constructed by the list of the ((&lt;Bio::Relation&gt;)) objects.  The
basic concept of the <a href="Pathway.html">Bio::Pathway</a> object is to
store a graph as an adjacency list (in the instance variable @graph), and
converting the list into an adjacency matrix by calling <a
href="Pathway.html#method-i-to_matrix">#to_matrix</a> method on demand. 
However, in some cases, it is convenient to have the original list of the
((&lt;Bio::Relation&gt;))s, <a href="Pathway.html">Bio::Pathway</a> object
also stores the list (as the instance variable @relations) redundantly.</p>

<p>Note: you can clear the @relations list by calling clear_relations! method
to reduce the memory usage, and the content of the @relations can be
re-generated from the @graph by <a
href="Pathway.html#method-i-to_relations">#to_relations</a> method.</p>

  </div><!-- description -->

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    

    
    <!-- Attributes -->
    <section id="attribute-method-details" class="method-section section">
      <h3 class="section-header">Attributes</h3>

      
      <div id="attribute-i-graph" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">graph</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Read-only accessor for the adjacency list of the graph.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-index" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">index</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Read-only accessor for the row/column index (@index) of the adjacency
matrix.  Contents of the hash @index is created by calling <a
href="Pathway.html#method-i-to_matrix">#to_matrix</a> method.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-label" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">label</span><span
            class="attribute-access-type">[RW]</span>
        </div>

        <div class="method-description">
        
        <p>Accessor for the hash of the label assigned to the each node.  You can
label some of the nodes in the graph by passing a hash to the label and
select subgraphs which contain labeled nodes only by subgraph method.</p>

<pre class="ruby"><span class="ruby-identifier">hash</span> = { <span class="ruby-value">1</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'red'</span>, <span class="ruby-value">2</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'green'</span>, <span class="ruby-value">5</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'black'</span> }
<span class="ruby-identifier">g</span>.<span class="ruby-identifier">label</span> = <span class="ruby-identifier">hash</span>
<span class="ruby-identifier">g</span>.<span class="ruby-identifier">label</span>
<span class="ruby-identifier">g</span>.<span class="ruby-identifier">subgraph</span>    <span class="ruby-comment"># =&gt; new graph consists of the node 1, 2, 5 only</span>
</pre>
        
        </div>
      </div>
      
      <div id="attribute-i-relations" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">relations</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Read-only accessor for the internal list of the Bio::Relation objects</p>
        
        </div>
      </div>
      
    </section><!-- attribute-method-details -->
    

    <!-- Methods -->
    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Class Methods</h3>

    
      <div id="method-c-new" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(relations, undirected = false)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Initial graph (adjacency list) generation from the list of Relation.</p>

<p>Generate <a href="Pathway.html">Bio::Pathway</a> object from the list of
Bio::Relation objects. If the second argument is true, undirected graph is
generated.</p>

<pre>r1 = Bio::Relation.new('a', 'b', 1)
r2 = Bio::Relation.new('a', 'c', 5)
r3 = Bio::Relation.new('b', 'c', 3)
list = [ r1, r2, r3 ]
g = Bio::Pathway.new(list, 'undirected')</pre>
          

          
          <div class="method-source-code" id="new-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 41</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">relations</span>, <span class="ruby-identifier">undirected</span> = <span class="ruby-keyword">false</span>)
  <span class="ruby-ivar">@undirected</span> = <span class="ruby-identifier">undirected</span>
  <span class="ruby-ivar">@relations</span> = <span class="ruby-identifier">relations</span>
  <span class="ruby-ivar">@graph</span> = {}         <span class="ruby-comment"># adjacency list expression of the graph</span>
  <span class="ruby-ivar">@index</span> = {}         <span class="ruby-comment"># numbering each node in matrix</span>
  <span class="ruby-ivar">@label</span> = {}         <span class="ruby-comment"># additional information on each node</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">to_list</span>                <span class="ruby-comment"># generate adjacency list</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- new-source -->
          
        </div>

        

        
      </div><!-- new-method -->

    
    </section><!-- public-class-method-details -->
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Instance Methods</h3>

    
      <div id="method-i-append" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">append</span><span
            class="method-args">(rel, add_rel = true)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Add an Bio::Relation object rel to the @graph and @relations. If the
second argument is false, @relations is not modified (only useful when
genarating @graph from @relations internally).</p>
          

          
          <div class="method-source-code" id="append-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 144</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">append</span>(<span class="ruby-identifier">rel</span>, <span class="ruby-identifier">add_rel</span> = <span class="ruby-keyword">true</span>)
  <span class="ruby-ivar">@relations</span>.<span class="ruby-identifier">push</span>(<span class="ruby-identifier">rel</span>) <span class="ruby-keyword">if</span> <span class="ruby-identifier">add_rel</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">from</span>].<span class="ruby-identifier">nil?</span>
    <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">from</span>] = {}
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">to</span>].<span class="ruby-identifier">nil?</span>
    <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">to</span>] = {}
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">from</span>][<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">to</span>] = <span class="ruby-identifier">rel</span>.<span class="ruby-identifier">relation</span>
  <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">to</span>][<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">from</span>] = <span class="ruby-identifier">rel</span>.<span class="ruby-identifier">relation</span> <span class="ruby-keyword">if</span> <span class="ruby-ivar">@undirected</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- append-source -->
          
        </div>

        

        
      </div><!-- append-method -->

    
      <div id="method-i-bellman_ford" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">bellman_ford</span><span
            class="method-args">(root)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Bellman-Ford method for solving the single-source shortest-paths problem in
the graph in which edge weights can be negative.</p>
          

          
          <div class="method-source-code" id="bellman_ford-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 592</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">bellman_ford</span>(<span class="ruby-identifier">root</span>)
  <span class="ruby-identifier">distance</span>, <span class="ruby-identifier">predecessor</span> = <span class="ruby-identifier">initialize_single_source</span>(<span class="ruby-identifier">root</span>)
  <span class="ruby-keyword">for</span> <span class="ruby-identifier">i</span> <span class="ruby-keyword">in</span> <span class="ruby-value">1</span> <span class="ruby-operator">..</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">nodes</span> <span class="ruby-operator">-</span> <span class="ruby-value">1</span>) <span class="ruby-keyword">do</span>
    <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each_key</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">u</span><span class="ruby-operator">|</span>
      <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">u</span>].<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">v</span>, <span class="ruby-identifier">w</span><span class="ruby-operator">|</span>
        <span class="ruby-comment"># relaxing procedure of root -&gt; 'u' -&gt; 'v'</span>
        <span class="ruby-keyword">if</span> <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">v</span>] <span class="ruby-operator">&gt;</span> <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">u</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">w</span>
          <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">v</span>] = <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">u</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">w</span>
          <span class="ruby-identifier">predecessor</span>[<span class="ruby-identifier">v</span>] = <span class="ruby-identifier">u</span>
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span></pre>
          </div><!-- bellman_ford-source -->
          
        </div>

        

        
      </div><!-- bellman_ford-method -->

    
      <div id="method-i-bfs" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">bfs</span><span
            class="method-args">(root)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Alias for the <a
href="Pathway.html#method-i-breadth_first_search">#breadth_first_search</a>
method.</p>
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Pathway.html#method-i-breadth_first_search">breadth_first_search</a>
        </div>
        
      </div><!-- bfs-method -->

    
      <div id="method-i-bfs_shortest_path" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">bfs_shortest_path</span><span
            class="method-args">(node1, node2)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Calculates the shortest path between two nodes by using <a
href="Pathway.html#method-i-breadth_first_search">#breadth_first_search</a>
method and returns steps and the path as Array.</p>
          

          
          <div class="method-source-code" id="bfs_shortest_path-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 450</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">bfs_shortest_path</span>(<span class="ruby-identifier">node1</span>, <span class="ruby-identifier">node2</span>)
  <span class="ruby-identifier">distance</span>, <span class="ruby-identifier">route</span> = <span class="ruby-identifier">breadth_first_search</span>(<span class="ruby-identifier">node1</span>)
  <span class="ruby-identifier">step</span> = <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">node2</span>]
  <span class="ruby-identifier">node</span> = <span class="ruby-identifier">node2</span>
  <span class="ruby-identifier">path</span> = [ <span class="ruby-identifier">node2</span> ]
  <span class="ruby-keyword">while</span> <span class="ruby-identifier">node</span> <span class="ruby-operator">!=</span> <span class="ruby-identifier">node1</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">route</span>[<span class="ruby-identifier">node</span>]
    <span class="ruby-identifier">node</span> = <span class="ruby-identifier">route</span>[<span class="ruby-identifier">node</span>]
    <span class="ruby-identifier">path</span>.<span class="ruby-identifier">unshift</span>(<span class="ruby-identifier">node</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">step</span>, <span class="ruby-identifier">path</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- bfs_shortest_path-source -->
          
        </div>

        

        
      </div><!-- bfs_shortest_path-method -->

    
      <div id="method-i-breadth_first_search" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">breadth_first_search</span><span
            class="method-args">(root)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Breadth first search solves steps and path to the each node and forms a
tree contains all reachable vertices from the root node. This method
returns the result in 2 hashes - 1st one shows the steps from root node and
2nd hash shows the structure of the tree.</p>

<p>The weight of the edges are not considered in this method.</p>
          

          
          <div class="method-source-code" id="breadth_first_search-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 419</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">breadth_first_search</span>(<span class="ruby-identifier">root</span>)
  <span class="ruby-identifier">visited</span> = {}
  <span class="ruby-identifier">distance</span> = {}
  <span class="ruby-identifier">predecessor</span> = {}

  <span class="ruby-identifier">visited</span>[<span class="ruby-identifier">root</span>] = <span class="ruby-keyword">true</span>
  <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">root</span>] = <span class="ruby-value">0</span>
  <span class="ruby-identifier">predecessor</span>[<span class="ruby-identifier">root</span>] = <span class="ruby-keyword">nil</span>

  <span class="ruby-identifier">queue</span> = [ <span class="ruby-identifier">root</span> ]

  <span class="ruby-keyword">while</span> <span class="ruby-identifier">from</span> = <span class="ruby-identifier">queue</span>.<span class="ruby-identifier">shift</span>
    <span class="ruby-keyword">next</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">from</span>]
    <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">from</span>].<span class="ruby-identifier">each_key</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">to</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">unless</span> <span class="ruby-identifier">visited</span>[<span class="ruby-identifier">to</span>]
        <span class="ruby-identifier">visited</span>[<span class="ruby-identifier">to</span>] = <span class="ruby-keyword">true</span>
        <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">to</span>] = <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">from</span>] <span class="ruby-operator">+</span> <span class="ruby-value">1</span>
        <span class="ruby-identifier">predecessor</span>[<span class="ruby-identifier">to</span>] = <span class="ruby-identifier">from</span>
        <span class="ruby-identifier">queue</span>.<span class="ruby-identifier">push</span>(<span class="ruby-identifier">to</span>)
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">distance</span>, <span class="ruby-identifier">predecessor</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- breadth_first_search-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Pathway.html#method-i-bfs">bfs</a>
        </div>
        

        
      </div><!-- breadth_first_search-method -->

    
      <div id="method-i-clear_relations-21" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">clear_relations!</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Clear @relations array to reduce the memory usage.</p>
          

          
          <div class="method-source-code" id="clear_relations-21-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 114</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">clear_relations!</span>
  <span class="ruby-ivar">@relations</span>.<span class="ruby-identifier">clear</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- clear_relations-21-source -->
          
        </div>

        

        
      </div><!-- clear_relations-21-method -->

    
      <div id="method-i-clique" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">clique</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Not implemented yet.</p>
          

          
          <div class="method-source-code" id="clique-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 370</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">clique</span>
  <span class="ruby-identifier">raise</span> <span class="ruby-constant">NotImplementedError</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- clique-source -->
          
        </div>

        

        
      </div><!-- clique-method -->

    
      <div id="method-i-cliquishness" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cliquishness</span><span
            class="method-args">(node)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns completeness of the edge density among the surrounded nodes.</p>

<p>Calculates the value of cliquishness around the node.  This value
indicates completeness of the edge density among the surrounded nodes.</p>

<p>Note: cliquishness (clustering coefficient) for a directed graph is also
calculated. Reference: <a
href="http://en.wikipedia.org/wiki/Clustering_coefficient">en.wikipedia.org/wiki/Clustering_coefficient</a></p>

<p>Note: Cliquishness (clustering coefficient) for a node that has only one
neighbor node is undefined. Currently, it returns NaN, but the behavior may
be changed in the future.</p>
          

          
          <div class="method-source-code" id="cliquishness-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 388</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cliquishness</span>(<span class="ruby-identifier">node</span>)
  <span class="ruby-identifier">neighbors</span> = <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">node</span>].<span class="ruby-identifier">keys</span>
  <span class="ruby-identifier">sg</span> = <span class="ruby-identifier">subgraph</span>(<span class="ruby-identifier">neighbors</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">sg</span>.<span class="ruby-identifier">graph</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">!=</span> <span class="ruby-value">0</span>
    <span class="ruby-identifier">edges</span> = <span class="ruby-identifier">sg</span>.<span class="ruby-identifier">edges</span>
    <span class="ruby-identifier">nodes</span> = <span class="ruby-identifier">neighbors</span>.<span class="ruby-identifier">size</span>
    <span class="ruby-identifier">complete</span> = (<span class="ruby-identifier">nodes</span> * (<span class="ruby-identifier">nodes</span> <span class="ruby-operator">-</span> <span class="ruby-value">1</span>))
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">edges</span>.<span class="ruby-identifier">quo</span>(<span class="ruby-identifier">complete</span>)
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">return</span> <span class="ruby-value">0.0</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- cliquishness-source -->
          
        </div>

        

        
      </div><!-- cliquishness-method -->

    
      <div id="method-i-common_subgraph" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">common_subgraph</span><span
            class="method-args">(graph)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Not implemented yet.</p>
          

          
          <div class="method-source-code" id="common_subgraph-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 364</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">common_subgraph</span>(<span class="ruby-identifier">graph</span>)
  <span class="ruby-identifier">raise</span> <span class="ruby-constant">NotImplementedError</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- common_subgraph-source -->
          
        </div>

        

        
      </div><!-- common_subgraph-method -->

    
      <div id="method-i-delete" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">delete</span><span
            class="method-args">(rel)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Remove an edge indicated by the Bio::Relation object rel from the @graph
and the @relations.</p>
          

          
          <div class="method-source-code" id="delete-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 158</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">delete</span>(<span class="ruby-identifier">rel</span>)
  <span class="ruby-ivar">@relations</span>.<span class="ruby-identifier">delete_if</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">x</span> <span class="ruby-operator">===</span> <span class="ruby-identifier">rel</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">from</span>].<span class="ruby-identifier">delete</span>(<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">to</span>)
  <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">to</span>].<span class="ruby-identifier">delete</span>(<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">from</span>) <span class="ruby-keyword">if</span> <span class="ruby-ivar">@undirected</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- delete-source -->
          
        </div>

        

        
      </div><!-- delete-method -->

    
      <div id="method-i-depth_first_search" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">depth_first_search</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Depth first search yields much information about the structure of the graph
especially on the classification of the edges.  This method returns 5
hashes - 1st one shows the timestamps of each node containing the first
discoverd time and the search finished time in an array.  The 2nd, 3rd,
4th, and 5th hashes contain tree edges, back edges, cross edges,
forward edges respectively.</p>

<p>If $DEBUG is true (e.g. ruby -d), this method prints the progression of the
search.</p>

<p>The weight of the edges are not considered in this method.</p>

<p>Note: The result of this method depends on the order of Hash#each (and
each_key, etc.), which may be variable with Ruby version and Ruby
interpreter variations (JRuby, etc.). For a workaround to remove such
dependency, you can use @index  to set order of <a
href="../Hash.html">Hash</a> keys. Note that this bahavior might be changed
in the future.</p>
          

          
          <div class="method-source-code" id="depth_first_search-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 481</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">depth_first_search</span>
  <span class="ruby-identifier">visited</span> = {}
  <span class="ruby-identifier">timestamp</span> = {}
  <span class="ruby-identifier">tree_edges</span> = {}
  <span class="ruby-identifier">back_edges</span> = {}
  <span class="ruby-identifier">cross_edges</span> = {}
  <span class="ruby-identifier">forward_edges</span> = {}
  <span class="ruby-identifier">count</span> = <span class="ruby-value">0</span>

  <span class="ruby-comment"># begin workaround removing depencency to order of Hash#each</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@index</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-keyword">then</span>
    <span class="ruby-identifier">preference_of_nodes</span> = <span class="ruby-keyword">nil</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">preference_of_nodes</span> = {}.<span class="ruby-identifier">merge</span>(<span class="ruby-ivar">@index</span>)
    <span class="ruby-identifier">i</span> = <span class="ruby-identifier">preference_of_nodes</span>.<span class="ruby-identifier">values</span>.<span class="ruby-identifier">max</span>
    <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each_key</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">node0</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">preference_of_nodes</span>[<span class="ruby-identifier">node0</span>] <span class="ruby-operator">||=</span> (<span class="ruby-identifier">i</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>)
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-comment"># end workaround removing depencency to order of Hash#each</span>

  <span class="ruby-identifier">dfs_visit</span> = <span class="ruby-constant">Proc</span>.<span class="ruby-identifier">new</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">from</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">visited</span>[<span class="ruby-identifier">from</span>] = <span class="ruby-keyword">true</span>
    <span class="ruby-identifier">timestamp</span>[<span class="ruby-identifier">from</span>] = [<span class="ruby-identifier">count</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>]
    <span class="ruby-identifier">ary</span> = <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">from</span>].<span class="ruby-identifier">keys</span>
    <span class="ruby-comment"># begin workaround removing depencency to order of Hash#each</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">preference_of_nodes</span> <span class="ruby-keyword">then</span>
      <span class="ruby-identifier">ary</span> = <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">sort_by</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">node0</span><span class="ruby-operator">|</span> <span class="ruby-identifier">preference_of_nodes</span>[<span class="ruby-identifier">node0</span>] }
    <span class="ruby-keyword">end</span>
    <span class="ruby-comment"># end workaround removing depencency to order of Hash#each</span>
    <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">to</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">visited</span>[<span class="ruby-identifier">to</span>]
        <span class="ruby-keyword">if</span> <span class="ruby-identifier">timestamp</span>[<span class="ruby-identifier">to</span>].<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">1</span>
          <span class="ruby-keyword">if</span> <span class="ruby-identifier">timestamp</span>[<span class="ruby-identifier">from</span>].<span class="ruby-identifier">first</span> <span class="ruby-operator">&lt;</span> <span class="ruby-identifier">timestamp</span>[<span class="ruby-identifier">to</span>].<span class="ruby-identifier">first</span>
      <span class="ruby-comment"># forward edge (black)</span>
      <span class="ruby-identifier">p</span> <span class="ruby-node">&quot;#{from} -&gt; #{to} : forward edge&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
      <span class="ruby-identifier">forward_edges</span>[<span class="ruby-identifier">from</span>] = <span class="ruby-identifier">to</span>
          <span class="ruby-keyword">else</span>
      <span class="ruby-comment"># cross edge (black)</span>
      <span class="ruby-identifier">p</span> <span class="ruby-node">&quot;#{from} -&gt; #{to} : cross edge&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
      <span class="ruby-identifier">cross_edges</span>[<span class="ruby-identifier">from</span>] = <span class="ruby-identifier">to</span>
          <span class="ruby-keyword">end</span>
        <span class="ruby-keyword">else</span>
          <span class="ruby-comment"># back edge (gray)</span>
          <span class="ruby-identifier">p</span> <span class="ruby-node">&quot;#{from} -&gt; #{to} : back edge&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
          <span class="ruby-identifier">back_edges</span>[<span class="ruby-identifier">from</span>] = <span class="ruby-identifier">to</span>
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-comment"># tree edge (white)</span>
        <span class="ruby-identifier">p</span> <span class="ruby-node">&quot;#{from} -&gt; #{to} : tree edge&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">$DEBUG</span>
        <span class="ruby-identifier">tree_edges</span>[<span class="ruby-identifier">to</span>] = <span class="ruby-identifier">from</span>
        <span class="ruby-identifier">dfs_visit</span>.<span class="ruby-identifier">call</span>(<span class="ruby-identifier">to</span>)
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-identifier">timestamp</span>[<span class="ruby-identifier">from</span>].<span class="ruby-identifier">push</span>(<span class="ruby-identifier">count</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>)
  }

  <span class="ruby-identifier">ary</span> = <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">keys</span>
  <span class="ruby-comment"># begin workaround removing depencency to order of Hash#each</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">preference_of_nodes</span> <span class="ruby-keyword">then</span>
    <span class="ruby-identifier">ary</span> = <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">sort_by</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">node0</span><span class="ruby-operator">|</span> <span class="ruby-identifier">preference_of_nodes</span>[<span class="ruby-identifier">node0</span>] }
  <span class="ruby-keyword">end</span>
  <span class="ruby-comment"># end workaround removing depencency to order of Hash#each</span>
  <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">node</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">unless</span> <span class="ruby-identifier">visited</span>[<span class="ruby-identifier">node</span>]
      <span class="ruby-identifier">dfs_visit</span>.<span class="ruby-identifier">call</span>(<span class="ruby-identifier">node</span>)
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">timestamp</span>, <span class="ruby-identifier">tree_edges</span>, <span class="ruby-identifier">back_edges</span>, <span class="ruby-identifier">cross_edges</span>, <span class="ruby-identifier">forward_edges</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- depth_first_search-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="Pathway.html#method-i-dfs">dfs</a>
        </div>
        

        
      </div><!-- depth_first_search-method -->

    
      <div id="method-i-dfs" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">dfs</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Alias for the <a
href="Pathway.html#method-i-depth_first_search">#depth_first_search</a>
method.</p>
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="Pathway.html#method-i-depth_first_search">depth_first_search</a>
        </div>
        
      </div><!-- dfs-method -->

    
      <div id="method-i-dfs_topological_sort" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">dfs_topological_sort</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Topological sort of the directed acyclic graphs (dags) by using
depth_first_search.</p>
          

          
          <div class="method-source-code" id="dfs_topological_sort-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 558</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dfs_topological_sort</span>
  <span class="ruby-comment"># sorted by finished time reversely and collect node names only</span>
  <span class="ruby-identifier">timestamp</span>, = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">depth_first_search</span>
  <span class="ruby-identifier">timestamp</span>.<span class="ruby-identifier">sort</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">a</span>,<span class="ruby-identifier">b</span><span class="ruby-operator">|</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>][<span class="ruby-value">1</span>] <span class="ruby-operator">&lt;=&gt;</span> <span class="ruby-identifier">a</span>[<span class="ruby-value">1</span>][<span class="ruby-value">1</span>]}.<span class="ruby-identifier">collect</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span> <span class="ruby-identifier">x</span>.<span class="ruby-identifier">first</span> }
<span class="ruby-keyword">end</span></pre>
          </div><!-- dfs_topological_sort-source -->
          
        </div>

        

        
      </div><!-- dfs_topological_sort-method -->

    
      <div id="method-i-dijkstra" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">dijkstra</span><span
            class="method-args">(root)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Dijkstra method to solve the shortest path problem in the weighted graph.</p>
          

          
          <div class="method-source-code" id="dijkstra-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 566</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dijkstra</span>(<span class="ruby-identifier">root</span>)
  <span class="ruby-identifier">distance</span>, <span class="ruby-identifier">predecessor</span> = <span class="ruby-identifier">initialize_single_source</span>(<span class="ruby-identifier">root</span>)
  <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">root</span>].<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">v</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">v</span>
    <span class="ruby-identifier">predecessor</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">root</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">queue</span> = <span class="ruby-identifier">distance</span>.<span class="ruby-identifier">dup</span>
  <span class="ruby-identifier">queue</span>.<span class="ruby-identifier">delete</span>(<span class="ruby-identifier">root</span>)

  <span class="ruby-keyword">while</span> <span class="ruby-identifier">queue</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">!=</span> <span class="ruby-value">0</span>
    <span class="ruby-identifier">min</span> = <span class="ruby-identifier">queue</span>.<span class="ruby-identifier">min</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">a</span>, <span class="ruby-identifier">b</span><span class="ruby-operator">|</span> <span class="ruby-identifier">a</span>[<span class="ruby-value">1</span>] <span class="ruby-operator">&lt;=&gt;</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>]}
    <span class="ruby-identifier">u</span> = <span class="ruby-identifier">min</span>[<span class="ruby-value">0</span>]                <span class="ruby-comment"># extranct a node having minimal distance</span>
    <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">u</span>].<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">v</span><span class="ruby-operator">|</span>
      <span class="ruby-comment"># relaxing procedure of root -&gt; 'u' -&gt; 'k'</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">k</span>] <span class="ruby-operator">&gt;</span> <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">u</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">v</span>
        <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">distance</span>[<span class="ruby-identifier">u</span>] <span class="ruby-operator">+</span> <span class="ruby-identifier">v</span>
        <span class="ruby-identifier">predecessor</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">u</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-identifier">queue</span>.<span class="ruby-identifier">delete</span>(<span class="ruby-identifier">u</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">distance</span>, <span class="ruby-identifier">predecessor</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- dijkstra-source -->
          
        </div>

        

        
      </div><!-- dijkstra-method -->

    
      <div id="method-i-directed" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">directed</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Changes the internal state of the graph from undirected to directed and
re-generate adjacency list.  The undirected graph can be converted to
directed graph, however, the edge between two nodes will be simply doubled
to both ends.</p>

<p>Note: this method can not be used without the list of the Bio::Relation
objects (internally stored in @relations variable). Thus if you already
called clear_relations! method, call <a
href="Pathway.html#method-i-to_relations">#to_relations</a> first.</p>
          

          
          <div class="method-source-code" id="directed-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 92</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">directed</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">undirected?</span>
    <span class="ruby-ivar">@undirected</span> = <span class="ruby-keyword">false</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">to_list</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- directed-source -->
          
        </div>

        

        
      </div><!-- directed-method -->

    
      <div id="method-i-directed-3F" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">directed?</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns true or false respond to the internal state of the graph.</p>
          

          
          <div class="method-source-code" id="directed-3F-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 74</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">directed?</span>
  <span class="ruby-ivar">@undirected</span> <span class="ruby-operator">?</span> <span class="ruby-keyword">false</span> <span class="ruby-operator">:</span> <span class="ruby-keyword">true</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- directed-3F-source -->
          
        </div>

        

        
      </div><!-- directed-3F-method -->

    
      <div id="method-i-dump_list" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">dump_list</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Pretty printer of the adjacency list.</p>

<p>Useful when you want to check the internal state of the adjacency list (for
debug purpose etc.) easily.</p>

<p>The result of this method depends on the order of Hash#each (and each_key,
etc.), which may be variable with Ruby version and Ruby interpreter
variations (JRuby, etc.). For a workaround to remove such dependency, you
can use @index  to set order of <a href="../Hash.html">Hash</a> keys. Note
that this behavior might be changed in the future.</p>
          

          
          <div class="method-source-code" id="dump_list-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 293</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dump_list</span>
  <span class="ruby-comment"># begin workaround removing depencency to order of Hash#each</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@index</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-keyword">then</span>
    <span class="ruby-identifier">pref</span> = <span class="ruby-keyword">nil</span>
    <span class="ruby-identifier">enum</span> = <span class="ruby-ivar">@graph</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">pref</span> = {}.<span class="ruby-identifier">merge</span>(<span class="ruby-ivar">@index</span>)
    <span class="ruby-identifier">i</span> = <span class="ruby-identifier">pref</span>.<span class="ruby-identifier">values</span>.<span class="ruby-identifier">max</span>
    <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each_key</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">node</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">pref</span>[<span class="ruby-identifier">node</span>] <span class="ruby-operator">||=</span> (<span class="ruby-identifier">i</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>)
    <span class="ruby-keyword">end</span>
    <span class="ruby-identifier">graph_to_a</span> = <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">to_a</span>
    <span class="ruby-identifier">graph_to_a</span>.<span class="ruby-identifier">sort!</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">x</span>, <span class="ruby-identifier">y</span><span class="ruby-operator">|</span> <span class="ruby-identifier">pref</span>[<span class="ruby-identifier">x</span>[<span class="ruby-value">0</span>]] <span class="ruby-operator">&lt;=&gt;</span> <span class="ruby-identifier">pref</span>[<span class="ruby-identifier">y</span>[<span class="ruby-value">0</span>]] }
    <span class="ruby-identifier">enum</span> = <span class="ruby-identifier">graph_to_a</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-comment"># end workaround removing depencency to order of Hash#each</span>

  <span class="ruby-identifier">list</span> = <span class="ruby-string">&quot;&quot;</span>
  <span class="ruby-identifier">enum</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">from</span>, <span class="ruby-identifier">hash</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">list</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;#{from} =&gt; &quot;</span>
    <span class="ruby-comment"># begin workaround removing depencency to order of Hash#each</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">pref</span> <span class="ruby-keyword">then</span>
      <span class="ruby-identifier">ary</span> = <span class="ruby-identifier">hash</span>.<span class="ruby-identifier">to_a</span>
      <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">sort!</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">x</span>,<span class="ruby-identifier">y</span><span class="ruby-operator">|</span> <span class="ruby-identifier">pref</span>[<span class="ruby-identifier">x</span>[<span class="ruby-value">0</span>]] <span class="ruby-operator">&lt;=&gt;</span> <span class="ruby-identifier">pref</span>[<span class="ruby-identifier">y</span>[<span class="ruby-value">0</span>]] }
      <span class="ruby-identifier">hash</span> = <span class="ruby-identifier">ary</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-comment"># end workaround removing depencency to order of Hash#each</span>
    <span class="ruby-identifier">a</span> = []
    <span class="ruby-identifier">hash</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">to</span>, <span class="ruby-identifier">relation</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">a</span>.<span class="ruby-identifier">push</span>(<span class="ruby-node">&quot;#{to} (#{relation})&quot;</span>)
    <span class="ruby-keyword">end</span>
    <span class="ruby-identifier">list</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">a</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;, &quot;</span>) <span class="ruby-operator">+</span> <span class="ruby-string">&quot;\n&quot;</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">list</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- dump_list-source -->
          
        </div>

        

        
      </div><!-- dump_list-method -->

    
      <div id="method-i-dump_matrix" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">dump_matrix</span><span
            class="method-args">(*arg)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Pretty printer of the adjacency matrix.</p>

<p>The <a href="Pathway.html#method-i-dump_matrix">#dump_matrix</a> method
accepts the same arguments as to_matrix. Useful when you want to check the
internal state of the matrix (for debug purpose etc.) easily.</p>

<p>This method internally calls <a
href="Pathway.html#method-i-to_matrix">#to_matrix</a> method. Read
documents of <a href="Pathway.html#method-i-to_matrix">#to_matrix</a> for
important informations.</p>
          

          
          <div class="method-source-code" id="dump_matrix-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 274</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dump_matrix</span>(*<span class="ruby-identifier">arg</span>)
  <span class="ruby-identifier">matrix</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">to_matrix</span>(*<span class="ruby-identifier">arg</span>)
  <span class="ruby-identifier">sorted</span> = <span class="ruby-ivar">@index</span>.<span class="ruby-identifier">sort</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">a</span>,<span class="ruby-identifier">b</span><span class="ruby-operator">|</span> <span class="ruby-identifier">a</span>[<span class="ruby-value">1</span>] <span class="ruby-operator">&lt;=&gt;</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>]}
  <span class="ruby-node">&quot;[# &quot;</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">sorted</span>.<span class="ruby-identifier">collect</span>{<span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span> <span class="ruby-identifier">x</span>[<span class="ruby-value">0</span>]}.<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;, &quot;</span>) <span class="ruby-operator">+</span> <span class="ruby-string">&quot;\n&quot;</span> <span class="ruby-operator">+</span>
    <span class="ruby-identifier">matrix</span>.<span class="ruby-identifier">to_a</span>.<span class="ruby-identifier">collect</span>{<span class="ruby-operator">|</span><span class="ruby-identifier">row</span><span class="ruby-operator">|</span> <span class="ruby-string">' '</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">row</span>.<span class="ruby-identifier">inspect</span>}.<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;,\n&quot;</span>) <span class="ruby-operator">+</span> <span class="ruby-string">&quot;\n]&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- dump_matrix-source -->
          
        </div>

        

        
      </div><!-- dump_matrix-method -->

    
      <div id="method-i-edges" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">edges</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns the number of the edges in the graph.</p>
          

          
          <div class="method-source-code" id="edges-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 172</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">edges</span>
  <span class="ruby-identifier">edges</span> = <span class="ruby-value">0</span>
  <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each_value</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">v</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">edges</span> <span class="ruby-operator">+=</span> <span class="ruby-identifier">v</span>.<span class="ruby-identifier">size</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">edges</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- edges-source -->
          
        </div>

        

        
      </div><!-- edges-method -->

    
      <div id="method-i-nodes" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">nodes</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns the number of the nodes in the graph.</p>
          

          
          <div class="method-source-code" id="nodes-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 167</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">nodes</span>
  <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">keys</span>.<span class="ruby-identifier">length</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- nodes-source -->
          
        </div>

        

        
      </div><!-- nodes-method -->

    
      <div id="method-i-small_world" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">small_world</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns frequency of the nodes having same number of edges as hash</p>

<p>Calculates the frequency of the nodes having the same number of edges and
returns the value as <a href="../Hash.html">Hash</a>.</p>
          

          
          <div class="method-source-code" id="small_world-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 405</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">small_world</span>
  <span class="ruby-identifier">freq</span> = <span class="ruby-constant">Hash</span>.<span class="ruby-identifier">new</span>(<span class="ruby-value">0</span>)
  <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each_value</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">v</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">freq</span>[<span class="ruby-identifier">v</span>.<span class="ruby-identifier">size</span>] <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">freq</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- small_world-source -->
          
        </div>

        

        
      </div><!-- small_world-method -->

    
      <div id="method-i-subgraph" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">subgraph</span><span
            class="method-args">(list = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Select labeled nodes and generate subgraph</p>

<p>This method select some nodes and returns new <a
href="Pathway.html">Bio::Pathway</a> object consists of selected nodes
only.  If the list of the nodes (as Array) is assigned as the argument, use
the list to select the nodes from the graph.  If no argument is assigned,
internal property of the graph @label is used to select the nodes.</p>

<pre class="ruby"><span class="ruby-identifier">hash</span> = { <span class="ruby-string">'a'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'secret'</span>, <span class="ruby-string">'b'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'important'</span>, <span class="ruby-string">'c'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'important'</span> }
<span class="ruby-identifier">g</span>.<span class="ruby-identifier">label</span> = <span class="ruby-identifier">hash</span>
<span class="ruby-identifier">g</span>.<span class="ruby-identifier">subgraph</span>
<span class="ruby-identifier">list</span> = [ <span class="ruby-string">'a'</span>, <span class="ruby-string">'b'</span>, <span class="ruby-string">'c'</span> ]
 <span class="ruby-identifier">g</span>.<span class="ruby-identifier">subgraph</span>(<span class="ruby-identifier">list</span>)
</pre>
          

          
          <div class="method-source-code" id="subgraph-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 343</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">subgraph</span>(<span class="ruby-identifier">list</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">list</span>
    <span class="ruby-ivar">@label</span>.<span class="ruby-identifier">clear</span>
    <span class="ruby-identifier">list</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">node</span><span class="ruby-operator">|</span>
      <span class="ruby-ivar">@label</span>[<span class="ruby-identifier">node</span>] = <span class="ruby-keyword">true</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">sub_graph</span> = <span class="ruby-constant">Pathway</span>.<span class="ruby-identifier">new</span>([], <span class="ruby-ivar">@undirected</span>)
  <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">from</span>, <span class="ruby-identifier">hash</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">next</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@label</span>[<span class="ruby-identifier">from</span>]
    <span class="ruby-identifier">sub_graph</span>.<span class="ruby-identifier">graph</span>[<span class="ruby-identifier">from</span>] <span class="ruby-operator">||=</span> {}
    <span class="ruby-identifier">hash</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">to</span>, <span class="ruby-identifier">relation</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">next</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@label</span>[<span class="ruby-identifier">to</span>]
      <span class="ruby-identifier">sub_graph</span>.<span class="ruby-identifier">append</span>(<span class="ruby-constant">Relation</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">from</span>, <span class="ruby-identifier">to</span>, <span class="ruby-identifier">relation</span>))
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">sub_graph</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- subgraph-source -->
          
        </div>

        

        
      </div><!-- subgraph-method -->

    
      <div id="method-i-to_list" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_list</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Graph (adjacency list) generation from the Relations</p>

<p>Generate the adjcancecy list @graph from @relations (called by initialize
and in some other cases when @relations has been changed).</p>
          

          
          <div class="method-source-code" id="to_list-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 134</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_list</span>
  <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">clear</span>
  <span class="ruby-ivar">@relations</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">rel</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">append</span>(<span class="ruby-identifier">rel</span>, <span class="ruby-keyword">false</span>)        <span class="ruby-comment"># append to @graph without push to @relations</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_list-source -->
          
        </div>

        

        
      </div><!-- to_list-method -->

    
      <div id="method-i-to_matrix" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_matrix</span><span
            class="method-args">(default_value = nil, diagonal_value = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Convert adjacency list to adjacency matrix</p>

<p>Returns the adjacency matrix expression of the graph as a Matrix object. 
If the first argument was assigned, the matrix will be filled with the
given value.  The second argument indicates the value of the diagonal
constituents of the matrix besides the above.</p>

<p>The result of this method depends on the order of Hash#each (and each_key,
etc.), which may be variable with Ruby version and Ruby interpreter
variations (JRuby, etc.). For a workaround to remove such dependency, you
can use @index  to set order of <a href="../Hash.html">Hash</a> keys. Note
that this behavior might be changed in the future. Be careful that @index
is overwritten by this method.</p>
          

          
          <div class="method-source-code" id="to_matrix-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 196</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_matrix</span>(<span class="ruby-identifier">default_value</span> = <span class="ruby-keyword">nil</span>, <span class="ruby-identifier">diagonal_value</span> = <span class="ruby-keyword">nil</span>)

  <span class="ruby-comment">#--</span>
  <span class="ruby-comment"># Note: following code only fills the outer Array with the reference</span>
  <span class="ruby-comment"># to the same inner Array object.</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment">#   matrix = Array.new(nodes, Array.new(nodes))</span>
  <span class="ruby-comment">#</span>
  <span class="ruby-comment"># so create a new Array object for each row as follows:</span>
  <span class="ruby-comment">#++</span>

  <span class="ruby-identifier">matrix</span> = <span class="ruby-constant">Array</span>.<span class="ruby-identifier">new</span>
  <span class="ruby-identifier">nodes</span>.<span class="ruby-identifier">times</span> <span class="ruby-keyword">do</span>
    <span class="ruby-identifier">matrix</span>.<span class="ruby-identifier">push</span>(<span class="ruby-constant">Array</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">nodes</span>, <span class="ruby-identifier">default_value</span>))
  <span class="ruby-keyword">end</span>

  <span class="ruby-keyword">if</span> <span class="ruby-identifier">diagonal_value</span>
    <span class="ruby-identifier">nodes</span>.<span class="ruby-identifier">times</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">i</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">matrix</span>[<span class="ruby-identifier">i</span>][<span class="ruby-identifier">i</span>] = <span class="ruby-identifier">diagonal_value</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-comment"># assign index number</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@index</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-keyword">then</span>
    <span class="ruby-comment"># assign index number for each node</span>
    <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">keys</span>.<span class="ruby-identifier">each_with_index</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">i</span><span class="ruby-operator">|</span>
      <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">i</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-comment"># begin workaround removing depencency to order of Hash#each</span>
    <span class="ruby-comment"># assign index number from the preset @index</span>
    <span class="ruby-identifier">indices</span> = <span class="ruby-ivar">@index</span>.<span class="ruby-identifier">to_a</span>
    <span class="ruby-identifier">indices</span>.<span class="ruby-identifier">sort!</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">i0</span>, <span class="ruby-identifier">i1</span><span class="ruby-operator">|</span> <span class="ruby-identifier">i0</span>[<span class="ruby-value">1</span>] <span class="ruby-operator">&lt;=&gt;</span> <span class="ruby-identifier">i1</span>[<span class="ruby-value">1</span>] }
    <span class="ruby-identifier">indices</span>.<span class="ruby-identifier">collect!</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">i0</span><span class="ruby-operator">|</span> <span class="ruby-identifier">i0</span>[<span class="ruby-value">0</span>] }
    <span class="ruby-ivar">@index</span>.<span class="ruby-identifier">clear</span>
    <span class="ruby-identifier">v</span> = <span class="ruby-value">0</span>
    <span class="ruby-identifier">indices</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">i</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">k</span>] <span class="ruby-keyword">and</span> <span class="ruby-operator">!</span><span class="ruby-ivar">@index</span>[<span class="ruby-identifier">k</span>] <span class="ruby-keyword">then</span>
        <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">v</span>; <span class="ruby-identifier">v</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each_key</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">k</span>] <span class="ruby-keyword">then</span>
        <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">v</span>; <span class="ruby-identifier">v</span> <span class="ruby-operator">+=</span> <span class="ruby-value">1</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-comment"># end workaround removing depencency to order of Hash#each</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@relations</span>.<span class="ruby-identifier">empty?</span>                <span class="ruby-comment"># only used after clear_relations!</span>
    <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">from</span>, <span class="ruby-identifier">hash</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">hash</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">to</span>, <span class="ruby-identifier">relation</span><span class="ruby-operator">|</span>
        <span class="ruby-identifier">x</span> = <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">from</span>]
        <span class="ruby-identifier">y</span> = <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">to</span>]
        <span class="ruby-identifier">matrix</span>[<span class="ruby-identifier">x</span>][<span class="ruby-identifier">y</span>] = <span class="ruby-identifier">relation</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@relations</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">rel</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">x</span> = <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">from</span>]
      <span class="ruby-identifier">y</span> = <span class="ruby-ivar">@index</span>[<span class="ruby-identifier">rel</span>.<span class="ruby-identifier">to</span>]
      <span class="ruby-identifier">matrix</span>[<span class="ruby-identifier">x</span>][<span class="ruby-identifier">y</span>] = <span class="ruby-identifier">rel</span>.<span class="ruby-identifier">relation</span>
      <span class="ruby-identifier">matrix</span>[<span class="ruby-identifier">y</span>][<span class="ruby-identifier">x</span>] = <span class="ruby-identifier">rel</span>.<span class="ruby-identifier">relation</span> <span class="ruby-keyword">if</span> <span class="ruby-ivar">@undirected</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-constant">Matrix</span>[*<span class="ruby-identifier">matrix</span>]
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_matrix-source -->
          
        </div>

        

        
      </div><!-- to_matrix-method -->

    
      <div id="method-i-to_relations" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_relations</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Reconstruct @relations from the adjacency list @graph.</p>
          

          
          <div class="method-source-code" id="to_relations-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 119</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_relations</span>
  <span class="ruby-ivar">@relations</span>.<span class="ruby-identifier">clear</span>
  <span class="ruby-ivar">@graph</span>.<span class="ruby-identifier">each_key</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">from</span><span class="ruby-operator">|</span>
    <span class="ruby-ivar">@graph</span>[<span class="ruby-identifier">from</span>].<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">to</span>, <span class="ruby-identifier">w</span><span class="ruby-operator">|</span>
      <span class="ruby-ivar">@relations</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-constant">Relation</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">from</span>, <span class="ruby-identifier">to</span>, <span class="ruby-identifier">w</span>)
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-ivar">@relations</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_relations-source -->
          
        </div>

        

        
      </div><!-- to_relations-method -->

    
      <div id="method-i-undirected" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">undirected</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Changes the internal state of the graph from directed to undirected and
re-generate adjacency list.</p>

<p>Note: this method can not be used without the list of the Bio::Relation
objects (internally stored in @relations variable). Thus if you already
called clear_relations! method, call <a
href="Pathway.html#method-i-to_relations">#to_relations</a> first.</p>
          

          
          <div class="method-source-code" id="undirected-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 106</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">undirected</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">directed?</span>
    <span class="ruby-ivar">@undirected</span> = <span class="ruby-keyword">true</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">to_list</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- undirected-source -->
          
        </div>

        

        
      </div><!-- undirected-method -->

    
      <div id="method-i-undirected-3F" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">undirected?</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns true or false respond to the internal state of the graph.</p>
          

          
          <div class="method-source-code" id="undirected-3F-source">
            <pre><span class="ruby-comment"># File lib/bio/pathway.rb, line 79</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">undirected?</span>
  <span class="ruby-ivar">@undirected</span> <span class="ruby-operator">?</span> <span class="ruby-keyword">true</span> <span class="ruby-operator">:</span> <span class="ruby-keyword">false</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- undirected-3F-source -->
          
        </div>

        

        
      </div><!-- undirected-3F-method -->

    
    </section><!-- public-instance-method-details -->
  
  </section><!-- 5Buntitled-5D -->

</div><!-- documentation -->


<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

