<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>class Bio::PDB - BioRuby API documentation</title>

<link type="text/css" media="screen" href="../rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
</script>

<script type="text/javascript" charset="utf-8" src="../js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/darkfish.js"></script>


<body id="top" class="class">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="../index.html">Home</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="file-metadata">
    <nav id="file-list-section" class="section">
  <h3 class="section-header">Defined In</h3>
  <ul>
    <li>lib/bio/db/pdb.rb
    <li>lib/bio/db/pdb/atom.rb
    <li>lib/bio/db/pdb/chain.rb
    <li>lib/bio/db/pdb/chemicalcomponent.rb
    <li>lib/bio/db/pdb/model.rb
    <li>lib/bio/db/pdb/pdb.rb
    <li>lib/bio/db/pdb/residue.rb
    <li>lib/bio/db/pdb/utils.rb
  </ul>
</nav>

    
  </div>

  <div id="class-metadata">
    
    <nav id="parent-class-section" class="section">
  <h3 class="section-header">Parent</h3>
  
  <p class="link"><a href="../Object.html">Object</a>
  
</nav>

    <!-- Included Modules -->
<nav id="includes-section" class="section">
  <h3 class="section-header">Included Modules</h3>

  <ul class="link-list">
  
  
    <li><a class="include" href="PDB/Utils.html">Bio::PDB::Utils</a>
  
  
  
    <li><a class="include" href="PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
  
  
    <li><a class="include" href="PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
  
  
    <li><a class="include" href="PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
  
  
    <li><a class="include" href="PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
  
  
    <li><a class="include" href="PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
  
  
    <li><a class="include" href="PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
  
  
    <li><span class="include">Enumerable</span>
  
  
  </ul>
</nav>

    <!-- Method Quickref -->
<nav id="method-list-section" class="section">
  <h3 class="section-header">Methods</h3>

  <ul class="link-list">
    
    <li><a href="#method-c-new">::new</a>
    
    <li><a href="#method-i-5B-5D">#[]</a>
    
    <li><a href="#method-i-accession">#accession</a>
    
    <li><a href="#method-i-addModel">#addModel</a>
    
    <li><a href="#method-i-authors">#authors</a>
    
    <li><a href="#method-i-classification">#classification</a>
    
    <li><a href="#method-i-dbref">#dbref</a>
    
    <li><a href="#method-i-definition">#definition</a>
    
    <li><a href="#method-i-each">#each</a>
    
    <li><a href="#method-i-each_model">#each_model</a>
    
    <li><a href="#method-i-entry_id">#entry_id</a>
    
    <li><a href="#method-i-helix">#helix</a>
    
    <li><a href="#method-i-inspect">#inspect</a>
    
    <li><a href="#method-i-jrnl">#jrnl</a>
    
    <li><a href="#method-i-keywords">#keywords</a>
    
    <li><a href="#method-i-record">#record</a>
    
    <li><a href="#method-i-remark">#remark</a>
    
    <li><a href="#method-i-seqres">#seqres</a>
    
    <li><a href="#method-i-sheet">#sheet</a>
    
    <li><a href="#method-i-ssbond">#ssbond</a>
    
    <li><a href="#method-i-to_s">#to_s</a>
    
    <li><a href="#method-i-turn">#turn</a>
    
    <li><a href="#method-i-version">#version</a>
    
  </ul>
</nav>

  </div>

  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="../ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="../LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="../README_rdoc.html">README</a>
  
    <li class="file"><a href="../README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="../Bio.html">Bio</a>
  
    <li><a href="../Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="../Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="../Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="../Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="../Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="../Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="../Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="../Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="../Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="../Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="../Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="../Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="../Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="../Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="../Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="../Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="../Bio/Command.html">Bio::Command</a>
  
    <li><a href="../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="../Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="../Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="../Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="../Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="../Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="../Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="../Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="../Bio/DB.html">Bio::DB</a>
  
    <li><a href="../Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="../Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="../Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="../Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="../Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="../Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="../Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="../Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="../Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="../Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="../Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="../Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="../Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="../Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="../Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="../Bio/Features.html">Bio::Features</a>
  
    <li><a href="../Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="../Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="../Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="../Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="../Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="../Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="../Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="../Bio/GO.html">Bio::GO</a>
  
    <li><a href="../Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="../Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="../Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="../Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="../Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="../Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="../Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="../Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="../Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="../Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="../Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="../Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="../Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="../Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="../Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="../Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="../Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="../Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="../Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="../Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="../Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="../Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="../Bio/Location.html">Bio::Location</a>
  
    <li><a href="../Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="../Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="../Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="../Bio/Map.html">Bio::Map</a>
  
    <li><a href="../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="../Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="../Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="../Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="../Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="../Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="../Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="../Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="../Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="../Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="../Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="../Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="../Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="../Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="../Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="../Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="../Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="../Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="../Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="../Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="../Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="../Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="../Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="../Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="../Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="../Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="../Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="../Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="../Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="../Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="../Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="../Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="../Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="../Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="../Bio/References.html">Bio::References</a>
  
    <li><a href="../Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="../Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="../Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="../Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="../Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="../Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="../Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="../Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="../Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="../Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="../Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="../Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="../Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="../Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="../Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="../Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="../Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="../Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="../Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="../Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="../Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="../Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="../Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="../Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="../Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="../Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="../Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="../Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="../Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="../Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="../Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="../Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="../Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="../Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="../Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="../Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="../Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="../Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="../Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="../Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="../Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="../BiorubyController.html">BiorubyController</a>
  
    <li><a href="../BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="../BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="../Hash.html">Hash</a>
  
    <li><a href="../Object.html">Object</a>
  
    <li><a href="../String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation">
  <h1 class="class">class Bio::PDB</h1>

  <div id="description" class="description">
    
<p>This is the main <a href="PDB.html">PDB</a> class which takes care of
parsing, annotations and is the entry way to the co-ordinate data held in
models.</p>

<p>There are many related classes.</p>

<p><a href="PDB/Model.html">Bio::PDB::Model</a> <a
href="PDB/Chain.html">Bio::PDB::Chain</a> <a
href="PDB/Residue.html">Bio::PDB::Residue</a> <a
href="PDB/Heterogen.html">Bio::PDB::Heterogen</a> <a
href="PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a> <a
href="PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
Bio::PDB::Record::* <a href="PDB/Coordinate.html">Bio::PDB::Coordinate</a></p>

  </div><!-- description -->

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    
    <!-- Constants -->
    <section id="constants-list" class="section">
      <h3 class="section-header">Constants</h3>
      <dl>
      
        <dt id="Coordinate_fileds">Coordinate_fileds
        
        <dd class="description">
        
      
        <dt id="DELIMITER">DELIMITER
        
        <dd class="description"><p>delimiter for reading via <a href="FlatFile.html">Bio::FlatFile</a></p>
        
      
      </dl>
    </section>
    

    
    <!-- Attributes -->
    <section id="attribute-method-details" class="method-section section">
      <h3 class="section-header">Attributes</h3>

      
      <div id="attribute-i-data" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">data</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>all records in this entry as an array.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-hash" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">hash</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>all records in this entry as an hash accessed by record names.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-models" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">models</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>models in this entry (array).</p>
        
        </div>
      </div>
      
    </section><!-- attribute-method-details -->
    

    <!-- Methods -->
    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Class Methods</h3>

    
      <div id="method-c-new" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(str)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Creates a new <a href="PDB.html">Bio::PDB</a> object from given
<em>str</em>.</p>
          

          
          <div class="method-source-code" id="new-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1445</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">str</span>)
  <span class="ruby-comment">#Aha! Our entry into the world of PDB parsing, we initialise a PDB</span>
  <span class="ruby-comment">#object with the whole PDB file as a string</span>
  <span class="ruby-comment">#each PDB has an array of the lines of the original file</span>
  <span class="ruby-comment">#a bit memory-tastic! A hash of records and an array of models</span>
  <span class="ruby-comment">#also has an id</span>

  <span class="ruby-ivar">@data</span> = <span class="ruby-identifier">str</span>.<span class="ruby-identifier">split</span>(<span class="ruby-regexp">%r[\r\n]+/</span>)
  <span class="ruby-ivar">@hash</span> = {}
  <span class="ruby-ivar">@models</span> = []
  <span class="ruby-ivar">@id</span> = <span class="ruby-keyword">nil</span>

  <span class="ruby-comment">#Flag to say whether the current line is part of a continuation</span>
  <span class="ruby-identifier">cont</span> = <span class="ruby-keyword">false</span>

  <span class="ruby-comment">#Empty current model</span>
  <span class="ruby-identifier">cModel</span>    = <span class="ruby-constant">Model</span>.<span class="ruby-identifier">new</span>
  <span class="ruby-identifier">cChain</span>    = <span class="ruby-keyword">nil</span> <span class="ruby-comment">#Chain.new</span>
  <span class="ruby-identifier">cResidue</span>  = <span class="ruby-keyword">nil</span> <span class="ruby-comment">#Residue.new</span>
  <span class="ruby-identifier">cLigand</span>   = <span class="ruby-keyword">nil</span> <span class="ruby-comment">#Heterogen.new</span>
  <span class="ruby-identifier">c_atom</span>    = <span class="ruby-keyword">nil</span>

  <span class="ruby-comment">#Goes through each line and replace that line with a PDB::Record</span>
  <span class="ruby-ivar">@data</span>.<span class="ruby-identifier">collect!</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">line</span><span class="ruby-operator">|</span>
    <span class="ruby-comment">#Go to next if the previous line was contiunation able, and</span>
    <span class="ruby-comment">#add_continuation returns true. Line is added by add_continuation</span>
    <span class="ruby-keyword">next</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">cont</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">cont</span> = <span class="ruby-identifier">cont</span>.<span class="ruby-identifier">add_continuation</span>(<span class="ruby-identifier">line</span>)

    <span class="ruby-comment">#Make the new record</span>
    <span class="ruby-identifier">f</span> = <span class="ruby-constant">Record</span>.<span class="ruby-identifier">get_record_class</span>(<span class="ruby-identifier">line</span>).<span class="ruby-identifier">new</span>.<span class="ruby-identifier">initialize_from_string</span>(<span class="ruby-identifier">line</span>)
    <span class="ruby-comment">#p f</span>
    <span class="ruby-comment">#Set cont</span>
    <span class="ruby-identifier">cont</span> = <span class="ruby-identifier">f</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">continue?</span>
    <span class="ruby-comment">#Set the hash to point to this record either by adding to an</span>
    <span class="ruby-comment">#array, or on it's own</span>
    <span class="ruby-identifier">key</span> = <span class="ruby-identifier">f</span>.<span class="ruby-identifier">record_name</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">a</span> = <span class="ruby-ivar">@hash</span>[<span class="ruby-identifier">key</span>] <span class="ruby-keyword">then</span>
      <span class="ruby-identifier">a</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">f</span>
    <span class="ruby-keyword">else</span>
      <span class="ruby-ivar">@hash</span>[<span class="ruby-identifier">key</span>] = [ <span class="ruby-identifier">f</span> ]
    <span class="ruby-keyword">end</span>

    <span class="ruby-comment"># Do something for ATOM and HETATM</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">key</span> <span class="ruby-operator">==</span> <span class="ruby-string">'ATOM'</span> <span class="ruby-keyword">or</span> <span class="ruby-identifier">key</span> <span class="ruby-operator">==</span> <span class="ruby-string">'HETATM'</span> <span class="ruby-keyword">then</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">cChain</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">chainID</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">cChain</span>.<span class="ruby-identifier">id</span>
        <span class="ruby-identifier">chain</span> = <span class="ruby-identifier">cChain</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-keyword">if</span> <span class="ruby-identifier">chain</span> = <span class="ruby-identifier">cModel</span>[<span class="ruby-identifier">f</span>.<span class="ruby-identifier">chainID</span>]
          <span class="ruby-identifier">cChain</span> = <span class="ruby-identifier">chain</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">cChain</span>
        <span class="ruby-keyword">else</span>
          <span class="ruby-comment"># If we don't have chain, add a new chain</span>
          <span class="ruby-identifier">newChain</span> = <span class="ruby-constant">Chain</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">f</span>.<span class="ruby-identifier">chainID</span>, <span class="ruby-identifier">cModel</span>)
          <span class="ruby-identifier">cModel</span>.<span class="ruby-identifier">addChain</span>(<span class="ruby-identifier">newChain</span>)
          <span class="ruby-identifier">cChain</span> = <span class="ruby-identifier">newChain</span>
          <span class="ruby-identifier">chain</span> = <span class="ruby-identifier">newChain</span>
        <span class="ruby-keyword">end</span>
        <span class="ruby-comment"># chain might be changed, clearing cResidue and cLigand</span>
        <span class="ruby-identifier">cResidue</span> = <span class="ruby-keyword">nil</span>
        <span class="ruby-identifier">cLigand</span> = <span class="ruby-keyword">nil</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>

    <span class="ruby-keyword">case</span> <span class="ruby-identifier">key</span>
    <span class="ruby-keyword">when</span> <span class="ruby-string">'ATOM'</span>
      <span class="ruby-identifier">c_atom</span> = <span class="ruby-identifier">f</span>
      <span class="ruby-identifier">residueID</span> = <span class="ruby-constant">Residue</span>.<span class="ruby-identifier">get_residue_id_from_atom</span>(<span class="ruby-identifier">f</span>)

      <span class="ruby-keyword">if</span> <span class="ruby-identifier">cResidue</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">residueID</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">cResidue</span>.<span class="ruby-identifier">id</span>
        <span class="ruby-identifier">residue</span> = <span class="ruby-identifier">cResidue</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-keyword">if</span> <span class="ruby-identifier">residue</span> = <span class="ruby-identifier">chain</span>.<span class="ruby-identifier">get_residue_by_id</span>(<span class="ruby-identifier">residueID</span>)
          <span class="ruby-identifier">cResidue</span> = <span class="ruby-identifier">residue</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">cResidue</span>
        <span class="ruby-keyword">else</span>
          <span class="ruby-comment"># add a new residue</span>
          <span class="ruby-identifier">newResidue</span> = <span class="ruby-constant">Residue</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">f</span>.<span class="ruby-identifier">resName</span>, <span class="ruby-identifier">f</span>.<span class="ruby-identifier">resSeq</span>, <span class="ruby-identifier">f</span>.<span class="ruby-identifier">iCode</span>, <span class="ruby-identifier">chain</span>)
          <span class="ruby-identifier">chain</span>.<span class="ruby-identifier">addResidue</span>(<span class="ruby-identifier">newResidue</span>)
          <span class="ruby-identifier">cResidue</span> = <span class="ruby-identifier">newResidue</span>
          <span class="ruby-identifier">residue</span> = <span class="ruby-identifier">newResidue</span>
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span>
      
      <span class="ruby-identifier">f</span>.<span class="ruby-identifier">residue</span> = <span class="ruby-identifier">residue</span>
      <span class="ruby-identifier">residue</span>.<span class="ruby-identifier">addAtom</span>(<span class="ruby-identifier">f</span>)

    <span class="ruby-keyword">when</span> <span class="ruby-string">'HETATM'</span>
      <span class="ruby-identifier">c_atom</span> = <span class="ruby-identifier">f</span>
      <span class="ruby-identifier">residueID</span> = <span class="ruby-constant">Heterogen</span>.<span class="ruby-identifier">get_residue_id_from_atom</span>(<span class="ruby-identifier">f</span>)

      <span class="ruby-keyword">if</span> <span class="ruby-identifier">cLigand</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">residueID</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">cLigand</span>.<span class="ruby-identifier">id</span>
        <span class="ruby-identifier">ligand</span> = <span class="ruby-identifier">cLigand</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-keyword">if</span> <span class="ruby-identifier">ligand</span> = <span class="ruby-identifier">chain</span>.<span class="ruby-identifier">get_heterogen_by_id</span>(<span class="ruby-identifier">residueID</span>)
          <span class="ruby-identifier">cLigand</span> = <span class="ruby-identifier">ligand</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">cLigand</span>
        <span class="ruby-keyword">else</span>
          <span class="ruby-comment"># add a new heterogen</span>
          <span class="ruby-identifier">newLigand</span> = <span class="ruby-constant">Heterogen</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">f</span>.<span class="ruby-identifier">resName</span>, <span class="ruby-identifier">f</span>.<span class="ruby-identifier">resSeq</span>, <span class="ruby-identifier">f</span>.<span class="ruby-identifier">iCode</span>, <span class="ruby-identifier">chain</span>)
          <span class="ruby-identifier">chain</span>.<span class="ruby-identifier">addLigand</span>(<span class="ruby-identifier">newLigand</span>)
          <span class="ruby-identifier">cLigand</span> = <span class="ruby-identifier">newLigand</span>
          <span class="ruby-identifier">ligand</span> = <span class="ruby-identifier">newLigand</span>
          <span class="ruby-comment">#Each model has a special solvent chain. (for compatibility)</span>
          <span class="ruby-keyword">if</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">resName</span> <span class="ruby-operator">==</span> <span class="ruby-string">'HOH'</span>
            <span class="ruby-identifier">cModel</span>.<span class="ruby-identifier">addSolvent</span>(<span class="ruby-identifier">newLigand</span>)
          <span class="ruby-keyword">end</span>
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span>

      <span class="ruby-identifier">f</span>.<span class="ruby-identifier">residue</span> = <span class="ruby-identifier">ligand</span>
      <span class="ruby-identifier">ligand</span>.<span class="ruby-identifier">addAtom</span>(<span class="ruby-identifier">f</span>)

    <span class="ruby-keyword">when</span> <span class="ruby-string">'MODEL'</span>
      <span class="ruby-identifier">c_atom</span> = <span class="ruby-keyword">nil</span>
      <span class="ruby-identifier">cChain</span> = <span class="ruby-keyword">nil</span>
      <span class="ruby-identifier">cResidue</span> = <span class="ruby-keyword">nil</span>
      <span class="ruby-identifier">cLigand</span> = <span class="ruby-keyword">nil</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">cModel</span>.<span class="ruby-identifier">model_serial</span> <span class="ruby-keyword">or</span> <span class="ruby-identifier">cModel</span>.<span class="ruby-identifier">chains</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">0</span> <span class="ruby-keyword">then</span>
        <span class="ruby-keyword">self</span>.<span class="ruby-identifier">addModel</span>(<span class="ruby-identifier">cModel</span>)
      <span class="ruby-keyword">end</span>
      <span class="ruby-identifier">cModel</span> = <span class="ruby-constant">Model</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">f</span>.<span class="ruby-identifier">serial</span>)

    <span class="ruby-keyword">when</span> <span class="ruby-string">'TER'</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">c_atom</span>
        <span class="ruby-identifier">c_atom</span>.<span class="ruby-identifier">ter</span> = <span class="ruby-identifier">f</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-comment">#$stderr.puts &quot;Warning: stray TER?&quot;</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">when</span> <span class="ruby-string">'SIGATM'</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">c_atom</span>
        <span class="ruby-comment">#$stderr.puts &quot;Warning: duplicated SIGATM?&quot; if c_atom.sigatm</span>
        <span class="ruby-identifier">c_atom</span>.<span class="ruby-identifier">sigatm</span> = <span class="ruby-identifier">f</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-comment">#$stderr.puts &quot;Warning: stray SIGATM?&quot;</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">when</span> <span class="ruby-string">'ANISOU'</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">c_atom</span>
        <span class="ruby-comment">#$stderr.puts &quot;Warning: duplicated ANISOU?&quot; if c_atom.anisou</span>
        <span class="ruby-identifier">c_atom</span>.<span class="ruby-identifier">anisou</span> = <span class="ruby-identifier">f</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-comment">#$stderr.puts &quot;Warning: stray ANISOU?&quot;</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">when</span> <span class="ruby-string">'SIGUIJ'</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">c_atom</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">c_atom</span>.<span class="ruby-identifier">anisou</span>
        <span class="ruby-comment">#$stderr.puts &quot;Warning: duplicated SIGUIJ?&quot; if c_atom.anisou.siguij</span>
        <span class="ruby-identifier">c_atom</span>.<span class="ruby-identifier">anisou</span>.<span class="ruby-identifier">siguij</span> = <span class="ruby-identifier">f</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-comment">#$stderr.puts &quot;Warning: stray SIGUIJ?&quot;</span>
      <span class="ruby-keyword">end</span>

    <span class="ruby-keyword">else</span>
      <span class="ruby-identifier">c_atom</span> = <span class="ruby-keyword">nil</span>

    <span class="ruby-keyword">end</span>
    <span class="ruby-identifier">f</span>
  <span class="ruby-keyword">end</span> <span class="ruby-comment">#each</span>
  <span class="ruby-comment">#At the end we need to add the final model</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">addModel</span>(<span class="ruby-identifier">cModel</span>)
  <span class="ruby-ivar">@data</span>.<span class="ruby-identifier">compact!</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- new-source -->
          
        </div>

        

        
      </div><!-- new-method -->

    
    </section><!-- public-class-method-details -->
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Instance Methods</h3>

    
      <div id="method-i-5B-5D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">[]</span><span
            class="method-args">(key)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Provides keyed access to the models based on serial number returns nil if
its not there</p>
          

          
          <div class="method-source-code" id="5B-5D-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1633</span>
<span class="ruby-keyword">def</span> <span class="ruby-operator">[]</span>(<span class="ruby-identifier">key</span>)
  <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">find</span>{ <span class="ruby-operator">|</span><span class="ruby-identifier">model</span><span class="ruby-operator">|</span> <span class="ruby-identifier">key</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">model</span>.<span class="ruby-identifier">model_serial</span> }
<span class="ruby-keyword">end</span></pre>
          </div><!-- 5B-5D-source -->
          
        </div>

        

        
      </div><!-- 5B-5D-method -->

    
      <div id="method-i-accession" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">accession</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Same as <code>Bio::PDB#entry_id</code>.</p>
          

          
          <div class="method-source-code" id="accession-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1887</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">accession</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">entry_id</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- accession-source -->
          
        </div>

        

        
      </div><!-- accession-method -->

    
      <div id="method-i-addModel" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">addModel</span><span
            class="method-args">(model)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Adds a <code>Bio::Model</code> object to the current strucutre. Adds a
model to the current structure. Returns self.</p>
          

          
          <div class="method-source-code" id="addModel-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1615</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">addModel</span>(<span class="ruby-identifier">model</span>)
  <span class="ruby-identifier">raise</span> <span class="ruby-string">&quot;Expecting a Bio::PDB::Model&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-keyword">not</span> <span class="ruby-identifier">model</span>.<span class="ruby-identifier">is_a?</span> <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">PDB</span><span class="ruby-operator">::</span><span class="ruby-constant">Model</span>
  <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">push</span>(<span class="ruby-identifier">model</span>)
  <span class="ruby-keyword">self</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- addModel-source -->
          
        </div>

        

        
      </div><!-- addModel-method -->

    
      <div id="method-i-authors" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">authors</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Get authors in AUTHOR.</p>
          

          
          <div class="method-source-code" id="authors-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1869</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">authors</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'AUTHOR'</span>).<span class="ruby-identifier">collect</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">authorList</span> }.<span class="ruby-identifier">flatten</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- authors-source -->
          
        </div>

        

        
      </div><!-- authors-method -->

    
      <div id="method-i-classification" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">classification</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Classification in HEADER.</p>
          

          
          <div class="method-source-code" id="classification-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1863</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">classification</span>
  <span class="ruby-identifier">f</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'HEADER'</span>).<span class="ruby-identifier">first</span>
  <span class="ruby-identifier">f</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">classification</span> <span class="ruby-operator">:</span> <span class="ruby-keyword">nil</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- classification-source -->
          
        </div>

        

        
      </div><!-- classification-method -->

    
      <div id="method-i-dbref" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">dbref</span><span
            class="method-args">(chainID = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets DBREF records. Returns an array of Bio::PDB::Record::DBREF objects.</p>

<p>If <em>chainID</em> is given, it returns corresponding DBREF records.</p>
          

          
          <div class="method-source-code" id="dbref-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1848</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dbref</span>(<span class="ruby-identifier">chainID</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">chainID</span> <span class="ruby-keyword">then</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'DBREF'</span>).<span class="ruby-identifier">find_all</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">chainID</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">chainID</span> }
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'DBREF'</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- dbref-source -->
          
        </div>

        

        
      </div><!-- dbref-method -->

    
      <div id="method-i-definition" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">definition</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Title of this entry in TITLE.</p>
          

          
          <div class="method-source-code" id="definition-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1892</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">definition</span>
  <span class="ruby-identifier">f</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'TITLE'</span>).<span class="ruby-identifier">first</span>
  <span class="ruby-identifier">f</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">title</span> <span class="ruby-operator">:</span> <span class="ruby-keyword">nil</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- definition-source -->
          
        </div>

        

        
      </div><!-- definition-method -->

    
      <div id="method-i-each" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">each</span><span
            class="method-args">() { |model| ... }</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Iterates over each model. Iterates over each of the models in the
structure. Returns <code>self</code>.</p>
          

          
          <div class="method-source-code" id="each-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1624</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">each</span>
  <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">each</span>{ <span class="ruby-operator">|</span><span class="ruby-identifier">model</span><span class="ruby-operator">|</span> <span class="ruby-keyword">yield</span> <span class="ruby-identifier">model</span> }
  <span class="ruby-keyword">self</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- each-source -->
          
        </div>

        
        <div class="aliases">
          Also aliased as: <a href="PDB.html#method-i-each_model">each_model</a>
        </div>
        

        
      </div><!-- each-method -->

    
      <div id="method-i-each_model" class="method-detail method-alias">
        
        <div class="method-heading">
          <span class="method-name">each_model</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Alias needed for <a href="PDB/ModelFinder.html">Bio::PDB::ModelFinder</a></p>
          

          
        </div>

        

        
        <div class="aliases">
          Alias for: <a href="PDB.html#method-i-each">each</a>
        </div>
        
      </div><!-- each_model-method -->

    
      <div id="method-i-entry_id" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">entry_id</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p><a href="PDB.html">PDB</a> identifier written in "HEADER". (e.g. 1A00)</p>
          

          
          <div class="method-source-code" id="entry_id-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1878</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entry_id</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@id</span>
    <span class="ruby-identifier">f</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'HEADER'</span>).<span class="ruby-identifier">first</span>
    <span class="ruby-ivar">@id</span> = <span class="ruby-identifier">f</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">idCode</span> <span class="ruby-operator">:</span> <span class="ruby-keyword">nil</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@id</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- entry_id-source -->
          
        </div>

        

        
      </div><!-- entry_id-method -->

    
      <div id="method-i-helix" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">helix</span><span
            class="method-args">(helixID = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets HELIX records. If no arguments are given, it returns all HELIX
records. (Returns an array of <code>Bio::PDB::Record::HELIX</code>
instances.) If <em>helixID</em> is given, it only returns records
corresponding to given <em>helixID</em>. (Returns an
<code>Bio::PDB::Record::HELIX</code> instance.)</p>
          

          
          <div class="method-source-code" id="helix-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1749</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">helix</span>(<span class="ruby-identifier">helixID</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">helixID</span> <span class="ruby-keyword">then</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'HELIX'</span>).<span class="ruby-identifier">find</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">helixID</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">helixID</span> }
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'HELIX'</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- helix-source -->
          
        </div>

        

        
      </div><!-- helix-method -->

    
      <div id="method-i-inspect" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">inspect</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>returns a string containing human-readable representation of this object.</p>
          

          
          <div class="method-source-code" id="inspect-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1905</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">inspect</span>
  <span class="ruby-node">&quot;#&lt;#{self.class.to_s} entry_id=#{entry_id.inspect}&gt;&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- inspect-source -->
          
        </div>

        

        
      </div><!-- inspect-method -->

    
      <div id="method-i-jrnl" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">jrnl</span><span
            class="method-args">(sub_record = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets JRNL records. If no arguments, it returns all JRNL records as a hash.
If sub record name is specified, it returns only corresponding records as
an array of <a href="PDB/Record.html">Bio::PDB::Record</a> instances.</p>
          

          
          <div class="method-source-code" id="jrnl-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1730</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">jrnl</span>(<span class="ruby-identifier">sub_record</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-keyword">defined?</span>(<span class="ruby-ivar">@jrnl</span>)
    <span class="ruby-ivar">@jrnl</span> = <span class="ruby-identifier">make_hash</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'JRNL'</span>), <span class="ruby-value">:sub_record</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">sub_record</span> <span class="ruby-operator">?</span> <span class="ruby-ivar">@jrnl</span>[<span class="ruby-identifier">sub_record</span>] <span class="ruby-operator">:</span> <span class="ruby-ivar">@jrnl</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- jrnl-source -->
          
        </div>

        

        
      </div><!-- jrnl-method -->

    
      <div id="method-i-keywords" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">keywords</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Keywords in KEYWDS. Returns an array of string.</p>
          

          
          <div class="method-source-code" id="keywords-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1858</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">keywords</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'KEYWDS'</span>).<span class="ruby-identifier">collect</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">keywds</span> }.<span class="ruby-identifier">flatten</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- keywords-source -->
          
        </div>

        

        
      </div><!-- keywords-method -->

    
      <div id="method-i-record" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">record</span><span
            class="method-args">(name = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets all records whose record type is <em>name</em>. Returns an array of
<code>Bio::PDB::Record::*</code> objects.</p>

<p>if <em>name</em> is nil, returns hash storing all record data.</p>

<p>Example: p pdb.record(HETATM) p <a href="http://'HETATM'">pdb.record</a></p>
          

          
          <div class="method-source-code" id="record-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1696</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">record</span>(<span class="ruby-identifier">name</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">name</span> <span class="ruby-operator">?</span> (<span class="ruby-ivar">@hash</span>[<span class="ruby-identifier">name</span>] <span class="ruby-operator">||</span> []) <span class="ruby-operator">:</span> <span class="ruby-ivar">@hash</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- record-source -->
          
        </div>

        

        
      </div><!-- record-method -->

    
      <div id="method-i-remark" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">remark</span><span
            class="method-args">(nn = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets REMARK records. If no arguments, it returns all REMARK records as a
hash. If remark number is specified, returns only corresponding REMARK
records. If number == 1 or 2 (REMARK   1 or REMARK   2), returns an
array of <a href="PDB/Record.html">Bio::PDB::Record</a> instances.
Otherwise, returns an array of strings.</p>
          

          
          <div class="method-source-code" id="remark-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1711</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">remark</span>(<span class="ruby-identifier">nn</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-keyword">defined?</span>(<span class="ruby-ivar">@remark</span>)
    <span class="ruby-identifier">h</span> = <span class="ruby-identifier">make_hash</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'REMARK'</span>), <span class="ruby-value">:remarkNum</span>)
    <span class="ruby-identifier">h</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">i</span>, <span class="ruby-identifier">a</span><span class="ruby-operator">|</span>
        <span class="ruby-identifier">a</span>.<span class="ruby-identifier">shift</span> <span class="ruby-comment"># remove first record (= space only)</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">i</span> <span class="ruby-operator">!=</span> <span class="ruby-value">1</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">i</span> <span class="ruby-operator">!=</span> <span class="ruby-value">2</span> <span class="ruby-keyword">then</span>
        <span class="ruby-identifier">a</span>.<span class="ruby-identifier">collect!</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">text</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">%r\s+\z/</span>, <span class="ruby-string">''</span>) }
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@remark</span> = <span class="ruby-identifier">h</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">nn</span> <span class="ruby-operator">?</span> <span class="ruby-ivar">@remark</span>[<span class="ruby-identifier">nn</span>] <span class="ruby-operator">:</span> <span class="ruby-ivar">@remark</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- remark-source -->
          
        </div>

        

        
      </div><!-- remark-method -->

    
      <div id="method-i-seqres" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">seqres</span><span
            class="method-args">(chainID = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Amino acid or nucleic acid sequence of backbone residues in SEQRES. If
<em>chainID</em> is given, it returns corresponding sequence as an array of
string. Otherwise, returns a hash which contains all sequences.</p>
          

          
          <div class="method-source-code" id="seqres-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1802</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">seqres</span>(<span class="ruby-identifier">chainID</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-keyword">defined?</span>(<span class="ruby-ivar">@seqres</span>)
    <span class="ruby-identifier">h</span> = <span class="ruby-identifier">make_hash</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'SEQRES'</span>), <span class="ruby-value">:chainID</span>)
    <span class="ruby-identifier">newHash</span> = {}
    <span class="ruby-identifier">h</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">a</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">a</span>.<span class="ruby-identifier">collect!</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">resName</span> }
      <span class="ruby-identifier">a</span>.<span class="ruby-identifier">flatten!</span>
      <span class="ruby-comment"># determine nuc or aa?</span>
      <span class="ruby-identifier">tmp</span> = <span class="ruby-constant">Hash</span>.<span class="ruby-identifier">new</span>(<span class="ruby-value">0</span>)
      <span class="ruby-identifier">a</span>[<span class="ruby-value">0</span>,<span class="ruby-value">13</span>].<span class="ruby-identifier">each</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span> <span class="ruby-identifier">tmp</span>[<span class="ruby-identifier">x</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">strip</span>.<span class="ruby-identifier">size</span>] <span class="ruby-operator">+=</span> <span class="ruby-value">1</span> }
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">tmp</span>[<span class="ruby-value">3</span>] <span class="ruby-operator">&gt;=</span> <span class="ruby-identifier">tmp</span>[<span class="ruby-value">1</span>] <span class="ruby-keyword">then</span>
        <span class="ruby-comment"># amino acid sequence</span>
        <span class="ruby-identifier">a</span>.<span class="ruby-identifier">collect!</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">aa</span><span class="ruby-operator">|</span>
          <span class="ruby-comment">#aa is three letter code: i.e. ALA</span>
          <span class="ruby-comment">#need to look up with Ala</span>
          <span class="ruby-identifier">aa</span> = <span class="ruby-identifier">aa</span>.<span class="ruby-identifier">capitalize</span>
          (<span class="ruby-keyword">begin</span>
             <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">AminoAcid</span>.<span class="ruby-identifier">three2one</span>(<span class="ruby-identifier">aa</span>)
           <span class="ruby-keyword">rescue</span> <span class="ruby-constant">ArgumentError</span>
             <span class="ruby-keyword">nil</span>
           <span class="ruby-keyword">end</span> <span class="ruby-operator">||</span> <span class="ruby-string">'X'</span>)
        <span class="ruby-keyword">end</span>
        <span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">AA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">a</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">''</span>))
      <span class="ruby-keyword">else</span>
        <span class="ruby-comment"># nucleic acid sequence</span>
        <span class="ruby-identifier">a</span>.<span class="ruby-identifier">collect!</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">na</span><span class="ruby-operator">|</span>
          <span class="ruby-identifier">na</span> = <span class="ruby-identifier">na</span>.<span class="ruby-identifier">delete</span>(<span class="ruby-string">'^a-zA-Z'</span>)
          <span class="ruby-identifier">na</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">==</span> <span class="ruby-value">1</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">na</span> <span class="ruby-operator">:</span> <span class="ruby-string">'n'</span>
        <span class="ruby-keyword">end</span>
        <span class="ruby-identifier">seq</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">NA</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">a</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">''</span>))
      <span class="ruby-keyword">end</span>
      <span class="ruby-identifier">newHash</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">seq</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@seqres</span> = <span class="ruby-identifier">newHash</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">chainID</span> <span class="ruby-keyword">then</span>
    <span class="ruby-ivar">@seqres</span>[<span class="ruby-identifier">chainID</span>]
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@seqres</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- seqres-source -->
          
        </div>

        

        
      </div><!-- seqres-method -->

    
      <div id="method-i-sheet" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">sheet</span><span
            class="method-args">(sheetID = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets SHEET records. If no arguments are given, it returns all SHEET records
as an array of arrays of <code>Bio::PDB::Record::SHEET</code> instances. If
<em>sheetID</em> is given, it returns an array of
<code>Bio::PDB::Record::SHEET</code> instances.</p>
          

          
          <div class="method-source-code" id="sheet-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1777</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">sheet</span>(<span class="ruby-identifier">sheetID</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-keyword">defined?</span>(<span class="ruby-ivar">@sheet</span>)
    <span class="ruby-ivar">@sheet</span> = <span class="ruby-identifier">make_grouping</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'SHEET'</span>), <span class="ruby-value">:sheetID</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">sheetID</span> <span class="ruby-keyword">then</span>
    <span class="ruby-ivar">@sheet</span>.<span class="ruby-identifier">find_all</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">first</span>.<span class="ruby-identifier">sheetID</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">sheetID</span> }
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@sheet</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- sheet-source -->
          
        </div>

        

        
      </div><!-- sheet-method -->

    
      <div id="method-i-ssbond" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">ssbond</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets SSBOND records.</p>
          

          
          <div class="method-source-code" id="ssbond-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1789</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ssbond</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'SSBOND'</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- ssbond-source -->
          
        </div>

        

        
      </div><!-- ssbond-method -->

    
      <div id="method-i-to_s" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">to_s</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns a string of Bio::PDB::Models. This propogates down the heirarchy
till you get to <a href="PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
which are outputed in <a href="PDB.html">PDB</a> format</p>
          

          
          <div class="method-source-code" id="to_s-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1647</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">to_s</span>
  <span class="ruby-identifier">string</span> = <span class="ruby-string">&quot;&quot;</span>
  <span class="ruby-ivar">@models</span>.<span class="ruby-identifier">each</span>{ <span class="ruby-operator">|</span><span class="ruby-identifier">model</span><span class="ruby-operator">|</span> <span class="ruby-identifier">string</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">model</span>.<span class="ruby-identifier">to_s</span> }
  <span class="ruby-identifier">string</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-string">&quot;END\n&quot;</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">string</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- to_s-source -->
          
        </div>

        

        
      </div><!-- to_s-method -->

    
      <div id="method-i-turn" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">turn</span><span
            class="method-args">(turnId = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Gets TURN records. If no arguments are given, it returns all TURN records. 
(Returns an array of <code>Bio::PDB::Record::TURN</code> instances.) If
<em>turnId</em> is given, it only returns a record corresponding to given
<em>turnId</em>. (Returns an <code>Bio::PDB::Record::TURN</code> instance.)</p>
          

          
          <div class="method-source-code" id="turn-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1764</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">turn</span>(<span class="ruby-identifier">turnId</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">turnId</span> <span class="ruby-keyword">then</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'TURN'</span>).<span class="ruby-identifier">find</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">f</span><span class="ruby-operator">|</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">turnId</span> <span class="ruby-operator">==</span> <span class="ruby-identifier">turnId</span> }
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'TURN'</span>)
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- turn-source -->
          
        </div>

        

        
      </div><!-- turn-method -->

    
      <div id="method-i-version" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">version</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Current modification number in REVDAT.</p>
          

          
          <div class="method-source-code" id="version-source">
            <pre><span class="ruby-comment"># File lib/bio/db/pdb/pdb.rb, line 1898</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">version</span>
  <span class="ruby-identifier">f</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">record</span>(<span class="ruby-string">'REVDAT'</span>).<span class="ruby-identifier">first</span>
  <span class="ruby-identifier">f</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">f</span>.<span class="ruby-identifier">modNum</span> <span class="ruby-operator">:</span> <span class="ruby-keyword">nil</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- version-source -->
          
        </div>

        

        
      </div><!-- version-method -->

    
    </section><!-- public-instance-method-details -->
  
  </section><!-- 5Buntitled-5D -->

</div><!-- documentation -->


<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

