<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />

  <title>Class: Bio::PROSITE</title>

  <link rel="stylesheet" href="../rdoc.css" type="text/css" media="screen" />

  <script src="../js/jquery.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/thickbox-compressed.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/quicksearch.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/darkfish.js" type="text/javascript" charset="utf-8"></script>

</head>
<body id="top" class="class">

  <div id="metadata">
    <div id="home-metadata">
      <div id="home-section" class="section">
        <h3 class="section-header">
          <a href="../index.html">Home</a>
          <a href="../index.html#classes">Classes</a>
          <a href="../index.html#methods">Methods</a>
        </h3>
      </div>
    </div>

    <div id="file-metadata">
      <div id="file-list-section" class="section">
        <h3 class="section-header">In Files</h3>
        <div class="section-body">
          <ul>
          
            <li><a href="../lib/bio/db/prosite_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/prosite.rb">lib/bio/db/prosite.rb</a></li>
          
          </ul>
        </div>
      </div>

      
    </div>

    <div id="class-metadata">
      
      <!-- Parent Class -->
      <div id="parent-class-section" class="section">
        <h3 class="section-header">Parent</h3>
        
        <p class="link"><a href="EMBLDB.html">Bio::EMBLDB</a></p>
        
      </div>
      

      

      

      
      <!-- Method Quickref -->
      <div id="method-list-section" class="section">
        <h3 class="section-header">Methods</h3>
        <ul class="link-list">
          
          <li><a href="#method-c-new">::new</a></li>
          
          <li><a href="#method-c-pa2re">::pa2re</a></li>
          
          <li><a href="#method-i-ac">#ac</a></li>
          
          <li><a href="#method-i-cc">#cc</a></li>
          
          <li><a href="#method-i-comment">#comment</a></li>
          
          <li><a href="#method-i-date">#date</a></li>
          
          <li><a href="#method-i-de">#de</a></li>
          
          <li><a href="#method-i-definition">#definition</a></li>
          
          <li><a href="#method-i-division">#division</a></li>
          
          <li><a href="#method-i-dr">#dr</a></li>
          
          <li><a href="#method-i-dt">#dt</a></li>
          
          <li><a href="#method-i-entry_id">#entry_id</a></li>
          
          <li><a href="#method-i-false_neg">#false_neg</a></li>
          
          <li><a href="#method-i-false_negative_hits">#false_negative_hits</a></li>
          
          <li><a href="#method-i-false_pos">#false_pos</a></li>
          
          <li><a href="#method-i-false_positive_hits">#false_positive_hits</a></li>
          
          <li><a href="#method-i-false_positive_sequences">#false_positive_sequences</a></li>
          
          <li><a href="#method-i-list_falsenegative">#list_falsenegative</a></li>
          
          <li><a href="#method-i-list_falsepositive">#list_falsepositive</a></li>
          
          <li><a href="#method-i-list_potentialhit">#list_potentialhit</a></li>
          
          <li><a href="#method-i-list_truepositive">#list_truepositive</a></li>
          
          <li><a href="#method-i-list_unknown">#list_unknown</a></li>
          
          <li><a href="#method-i-list_xref">#list_xref</a></li>
          
          <li><a href="#method-i-ma">#ma</a></li>
          
          <li><a href="#method-i-ma2re">#ma2re</a></li>
          
          <li><a href="#method-i-max_repeat">#max_repeat</a></li>
          
          <li><a href="#method-i-name">#name</a></li>
          
          <li><a href="#method-i-nr">#nr</a></li>
          
          <li><a href="#method-i-pa">#pa</a></li>
          
          <li><a href="#method-i-pa2re">#pa2re</a></li>
          
          <li><a href="#method-i-partial">#partial</a></li>
          
          <li><a href="#method-i-pattern">#pattern</a></li>
          
          <li><a href="#method-i-pdb_xref">#pdb_xref</a></li>
          
          <li><a href="#method-i-pdoc_xref">#pdoc_xref</a></li>
          
          <li><a href="#method-i-positive">#positive</a></li>
          
          <li><a href="#method-i-positive_hits">#positive_hits</a></li>
          
          <li><a href="#method-i-positive_sequences">#positive_sequences</a></li>
          
          <li><a href="#method-i-profile">#profile</a></li>
          
          <li><a href="#method-i-re">#re</a></li>
          
          <li><a href="#method-i-release">#release</a></li>
          
          <li><a href="#method-i-ru">#ru</a></li>
          
          <li><a href="#method-i-rule">#rule</a></li>
          
          <li><a href="#method-i-site">#site</a></li>
          
          <li><a href="#method-i-skip_flag">#skip_flag</a></li>
          
          <li><a href="#method-i-sp_xref">#sp_xref</a></li>
          
          <li><a href="#method-i-statistics">#statistics</a></li>
          
          <li><a href="#method-i-swissprot_release_number">#swissprot_release_number</a></li>
          
          <li><a href="#method-i-swissprot_release_sequences">#swissprot_release_sequences</a></li>
          
          <li><a href="#method-i-taxon_range">#taxon_range</a></li>
          
          <li><a href="#method-i-total">#total</a></li>
          
          <li><a href="#method-i-total_hits">#total_hits</a></li>
          
          <li><a href="#method-i-total_sequences">#total_sequences</a></li>
          
          <li><a href="#method-i-unknown">#unknown</a></li>
          
          <li><a href="#method-i-unknown_hits">#unknown_hits</a></li>
          
          <li><a href="#method-i-unknown_sequences">#unknown_sequences</a></li>
          
        </ul>
      </div>
      

      
    </div>

    <div id="project-metadata">
      
      
      <div id="fileindex-section" class="section project-section">
        <h3 class="section-header">Files</h3>
        <ul>
        
          <li class="file"><a href="../ChangeLog.html">ChangeLog</a></li>
        
          <li class="file"><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a></li>
        
          <li class="file"><a href="../README_rdoc.html">README.rdoc</a></li>
        
          <li class="file"><a href="../README_DEV_rdoc.html">README_DEV.rdoc</a></li>
        
          <li class="file"><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES.rdoc</a></li>
        
          <li class="file"><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3.rdoc</a></li>
        
          <li class="file"><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0.rdoc</a></li>
        
          <li class="file"><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1.rdoc</a></li>
        
        </ul>
      </div>
      

      <div id="classindex-section" class="section project-section">
        <h3 class="section-header">Class/Module Index
          <span class="search-toggle"><img src="../images/find.png"
            height="16" width="16" alt="[+]"
            title="show/hide quicksearch" /></span></h3>
        <form action="#" method="get" accept-charset="utf-8" class="initially-hidden">
        <fieldset>
          <legend>Quicksearch</legend>
          <input type="text" name="quicksearch" value=""
            class="quicksearch-field" />
        </fieldset>
        </form>

        <ul class="link-list">
        
          <li><a href="../Bio.html">Bio</a></li>
        
          <li><a href="../Bio/AAindex.html">Bio::AAindex</a></li>
        
          <li><a href="../Bio/AAindex1.html">Bio::AAindex1</a></li>
        
          <li><a href="../Bio/AAindex2.html">Bio::AAindex2</a></li>
        
          <li><a href="../Bio/Abif.html">Bio::Abif</a></li>
        
          <li><a href="../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a></li>
        
          <li><a href="../Bio/Alignment.html">Bio::Alignment</a></li>
        
          <li><a href="../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a></li>
        
          <li><a href="../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a></li>
        
          <li><a href="../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a></li>
        
          <li><a href="../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a></li>
        
          <li><a href="../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a></li>
        
          <li><a href="../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a></li>
        
          <li><a href="../Bio/Alignment/Output.html">Bio::Alignment::Output</a></li>
        
          <li><a href="../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a></li>
        
          <li><a href="../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a></li>
        
          <li><a href="../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a></li>
        
          <li><a href="../Bio/Alignment/Site.html">Bio::Alignment::Site</a></li>
        
          <li><a href="../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a></li>
        
          <li><a href="../Bio/AminoAcid.html">Bio::AminoAcid</a></li>
        
          <li><a href="../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a></li>
        
          <li><a href="../Bio/Blast.html">Bio::Blast</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Default.html">Bio::Blast::Default</a></li>
        
          <li><a href="../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a></li>
        
          <li><a href="../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a></li>
        
          <li><a href="../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Remote.html">Bio::Blast::Remote</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a></li>
        
          <li><a href="../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a></li>
        
          <li><a href="../Bio/Blast/Report.html">Bio::Blast::Report</a></li>
        
          <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a></li>
        
          <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a></li>
        
          <li><a href="../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a></li>
        
          <li><a href="../Bio/Blast/WU.html">Bio::Blast::WU</a></li>
        
          <li><a href="../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a></li>
        
          <li><a href="../Bio/Blat.html">Bio::Blat</a></li>
        
          <li><a href="../Bio/Blat/Report.html">Bio::Blat::Report</a></li>
        
          <li><a href="../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a></li>
        
          <li><a href="../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/ClustalW.html">Bio::ClustalW</a></li>
        
          <li><a href="../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a></li>
        
          <li><a href="../Bio/CodonTable.html">Bio::CodonTable</a></li>
        
          <li><a href="../Bio/ColorScheme.html">Bio::ColorScheme</a></li>
        
          <li><a href="../Bio/Command.html">Bio::Command</a></li>
        
          <li><a href="../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a></li>
        
          <li><a href="../Bio/ContingencyTable.html">Bio::ContingencyTable</a></li>
        
          <li><a href="../Bio/DAS.html">Bio::DAS</a></li>
        
          <li><a href="../Bio/DAS/DNA.html">Bio::DAS::DNA</a></li>
        
          <li><a href="../Bio/DAS/DSN.html">Bio::DAS::DSN</a></li>
        
          <li><a href="../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a></li>
        
          <li><a href="../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a></li>
        
          <li><a href="../Bio/DAS/GFF.html">Bio::DAS::GFF</a></li>
        
          <li><a href="../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a></li>
        
          <li><a href="../Bio/DAS/LINK.html">Bio::DAS::LINK</a></li>
        
          <li><a href="../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a></li>
        
          <li><a href="../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a></li>
        
          <li><a href="../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a></li>
        
          <li><a href="../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a></li>
        
          <li><a href="../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a></li>
        
          <li><a href="../Bio/DB.html">Bio::DB</a></li>
        
          <li><a href="../Bio/DBGET.html">Bio::DBGET</a></li>
        
          <li><a href="../Bio/DDBJ.html">Bio::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a></li>
        
          <li><a href="../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a></li>
        
          <li><a href="../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a></li>
        
          <li><a href="../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a></li>
        
          <li><a href="../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a></li>
        
          <li><a href="../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a></li>
        
          <li><a href="../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a></li>
        
          <li><a href="../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a></li>
        
          <li><a href="../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a></li>
        
          <li><a href="../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a></li>
        
          <li><a href="../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a></li>
        
          <li><a href="../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a></li>
        
          <li><a href="../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a></li>
        
          <li><a href="../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a></li>
        
          <li><a href="../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a></li>
        
          <li><a href="../Bio/EBI.html">Bio::EBI</a></li>
        
          <li><a href="../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a></li>
        
          <li><a href="../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a></li>
        
          <li><a href="../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a></li>
        
          <li><a href="../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a></li>
        
          <li><a href="../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a></li>
        
          <li><a href="../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a></li>
        
          <li><a href="../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a></li>
        
          <li><a href="../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a></li>
        
          <li><a href="../Bio/EMBL.html">Bio::EMBL</a></li>
        
          <li><a href="../Bio/EMBLDB.html">Bio::EMBLDB</a></li>
        
          <li><a href="../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a></li>
        
          <li><a href="../Bio/EMBOSS.html">Bio::EMBOSS</a></li>
        
          <li><a href="../Bio/Ensembl.html">Bio::Ensembl</a></li>
        
          <li><a href="../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a></li>
        
          <li><a href="../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a></li>
        
          <li><a href="../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a></li>
        
          <li><a href="../Bio/FANTOM.html">Bio::FANTOM</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a></li>
        
          <li><a href="../Bio/Fasta.html">Bio::Fasta</a></li>
        
          <li><a href="../Bio/Fasta/Report.html">Bio::Fasta::Report</a></li>
        
          <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a></li>
        
          <li><a href="../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a></li>
        
          <li><a href="../Bio/FastaDefline.html">Bio::FastaDefline</a></li>
        
          <li><a href="../Bio/FastaFormat.html">Bio::FastaFormat</a></li>
        
          <li><a href="../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a></li>
        
          <li><a href="../Bio/Fastq.html">Bio::Fastq</a></li>
        
          <li><a href="../Bio/Fastq/Error.html">Bio::Fastq::Error</a></li>
        
          <li><a href="../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a></li>
        
          <li><a href="../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a></li>
        
          <li><a href="../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a></li>
        
          <li><a href="../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a></li>
        
          <li><a href="../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a></li>
        
          <li><a href="../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a></li>
        
          <li><a href="../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a></li>
        
          <li><a href="../Bio/Feature.html">Bio::Feature</a></li>
        
          <li><a href="../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a></li>
        
          <li><a href="../Bio/Features.html">Bio::Features</a></li>
        
          <li><a href="../Bio/Fetch.html">Bio::Fetch</a></li>
        
          <li><a href="../Bio/FlatFile.html">Bio::FlatFile</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a></li>
        
          <li><a href="../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a></li>
        
          <li><a href="../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a></li>
        
          <li><a href="../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a></li>
        
          <li><a href="../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a></li>
        
          <li><a href="../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a></li>
        
          <li><a href="../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a></li>
        
          <li><a href="../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a></li>
        
          <li><a href="../Bio/GCG.html">Bio::GCG</a></li>
        
          <li><a href="../Bio/GCG/Msf.html">Bio::GCG::Msf</a></li>
        
          <li><a href="../Bio/GCG/Seq.html">Bio::GCG::Seq</a></li>
        
          <li><a href="../Bio/GFF.html">Bio::GFF</a></li>
        
          <li><a href="../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a></li>
        
          <li><a href="../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a></li>
        
          <li><a href="../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a></li>
        
          <li><a href="../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a></li>
        
          <li><a href="../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a></li>
        
          <li><a href="../Bio/GFF/Record.html">Bio::GFF::Record</a></li>
        
          <li><a href="../Bio/GO.html">Bio::GO</a></li>
        
          <li><a href="../Bio/GO/External2go.html">Bio::GO::External2go</a></li>
        
          <li><a href="../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a></li>
        
          <li><a href="../Bio/GO/Ontology.html">Bio::GO::Ontology</a></li>
        
          <li><a href="../Bio/GenBank.html">Bio::GenBank</a></li>
        
          <li><a href="../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a></li>
        
          <li><a href="../Bio/GenPept.html">Bio::GenPept</a></li>
        
          <li><a href="../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a></li>
        
          <li><a href="../Bio/Genscan.html">Bio::Genscan</a></li>
        
          <li><a href="../Bio/Genscan/Report.html">Bio::Genscan::Report</a></li>
        
          <li><a href="../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a></li>
        
          <li><a href="../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a></li>
        
          <li><a href="../Bio/HGC.html">Bio::HGC</a></li>
        
          <li><a href="../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a></li>
        
          <li><a href="../Bio/HMMER.html">Bio::HMMER</a></li>
        
          <li><a href="../Bio/HMMER/Report.html">Bio::HMMER::Report</a></li>
        
          <li><a href="../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a></li>
        
          <li><a href="../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a></li>
        
          <li><a href="../Bio/Hinv.html">Bio::Hinv</a></li>
        
          <li><a href="../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a></li>
        
          <li><a href="../Bio/Hinv/Common.html">Bio::Hinv::Common</a></li>
        
          <li><a href="../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a></li>
        
          <li><a href="../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a></li>
        
          <li><a href="../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a></li>
        
          <li><a href="../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a></li>
        
          <li><a href="../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a></li>
        
          <li><a href="../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a></li>
        
          <li><a href="../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a></li>
        
          <li><a href="../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a></li>
        
          <li><a href="../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a></li>
        
          <li><a href="../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a></li>
        
          <li><a href="../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a></li>
        
          <li><a href="../Bio/Iprscan.html">Bio::Iprscan</a></li>
        
          <li><a href="../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a></li>
        
          <li><a href="../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a></li>
        
          <li><a href="../Bio/KEGG.html">Bio::KEGG</a></li>
        
          <li><a href="../Bio/KEGG/API.html">Bio::KEGG::API</a></li>
        
          <li><a href="../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a></li>
        
          <li><a href="../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a></li>
        
          <li><a href="../Bio/KEGG/Common.html">Bio::KEGG::Common</a></li>
        
          <li><a href="../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a></li>
        
          <li><a href="../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a></li>
        
          <li><a href="../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a></li>
        
          <li><a href="../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a></li>
        
          <li><a href="../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a></li>
        
          <li><a href="../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a></li>
        
          <li><a href="../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a></li>
        
          <li><a href="../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a></li>
        
          <li><a href="../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a></li>
        
          <li><a href="../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a></li>
        
          <li><a href="../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a></li>
        
          <li><a href="../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a></li>
        
          <li><a href="../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a></li>
        
          <li><a href="../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a></li>
        
          <li><a href="../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a></li>
        
          <li><a href="../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a></li>
        
          <li><a href="../Bio/KEGGDB.html">Bio::KEGGDB</a></li>
        
          <li><a href="../Bio/LITDB.html">Bio::LITDB</a></li>
        
          <li><a href="../Bio/Lasergene.html">Bio::Lasergene</a></li>
        
          <li><a href="../Bio/Location.html">Bio::Location</a></li>
        
          <li><a href="../Bio/Locations.html">Bio::Locations</a></li>
        
          <li><a href="../Bio/MAFFT.html">Bio::MAFFT</a></li>
        
          <li><a href="../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a></li>
        
          <li><a href="../Bio/MEDLINE.html">Bio::MEDLINE</a></li>
        
          <li><a href="../Bio/Map.html">Bio::Map</a></li>
        
          <li><a href="../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a></li>
        
          <li><a href="../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a></li>
        
          <li><a href="../Bio/Map/Mapping.html">Bio::Map::Mapping</a></li>
        
          <li><a href="../Bio/Map/Marker.html">Bio::Map::Marker</a></li>
        
          <li><a href="../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a></li>
        
          <li><a href="../Bio/Meme.html">Bio::Meme</a></li>
        
          <li><a href="../Bio/Meme/Mast.html">Bio::Meme::Mast</a></li>
        
          <li><a href="../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a></li>
        
          <li><a href="../Bio/Meme/Motif.html">Bio::Meme::Motif</a></li>
        
          <li><a href="../Bio/Muscle.html">Bio::Muscle</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/NBRF.html">Bio::NBRF</a></li>
        
          <li><a href="../Bio/NCBI.html">Bio::NCBI</a></li>
        
          <li><a href="../Bio/NCBI/REST.html">Bio::NCBI::REST</a></li>
        
          <li><a href="../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a></li>
        
          <li><a href="../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a></li>
        
          <li><a href="../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a></li>
        
          <li><a href="../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a></li>
        
          <li><a href="../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a></li>
        
          <li><a href="../Bio/NCBIDB.html">Bio::NCBIDB</a></li>
        
          <li><a href="../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a></li>
        
          <li><a href="../Bio/Newick.html">Bio::Newick</a></li>
        
          <li><a href="../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a></li>
        
          <li><a href="../Bio/Nexus.html">Bio::Nexus</a></li>
        
          <li><a href="../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a></li>
        
          <li><a href="../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a></li>
        
          <li><a href="../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a></li>
        
          <li><a href="../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a></li>
        
          <li><a href="../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a></li>
        
          <li><a href="../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a></li>
        
          <li><a href="../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a></li>
        
          <li><a href="../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a></li>
        
          <li><a href="../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a></li>
        
          <li><a href="../Bio/Nexus/Util.html">Bio::Nexus::Util</a></li>
        
          <li><a href="../Bio/NucleicAcid.html">Bio::NucleicAcid</a></li>
        
          <li><a href="../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a></li>
        
          <li><a href="../Bio/PAML.html">Bio::PAML</a></li>
        
          <li><a href="../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a></li>
        
          <li><a href="../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a></li>
        
          <li><a href="../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a></li>
        
          <li><a href="../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a></li>
        
          <li><a href="../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a></li>
        
          <li><a href="../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a></li>
        
          <li><a href="../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a></li>
        
          <li><a href="../Bio/PAML/Common.html">Bio::PAML::Common</a></li>
        
          <li><a href="../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a></li>
        
          <li><a href="../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a></li>
        
          <li><a href="../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a></li>
        
          <li><a href="../Bio/PDB.html">Bio::PDB</a></li>
        
          <li><a href="../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a></li>
        
          <li><a href="../Bio/PDB/Chain.html">Bio::PDB::Chain</a></li>
        
          <li><a href="../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a></li>
        
          <li><a href="../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a></li>
        
          <li><a href="../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a></li>
        
          <li><a href="../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a></li>
        
          <li><a href="../Bio/PDB/DataType.html">Bio::PDB::DataType</a></li>
        
          <li><a href="../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a></li>
        
          <li><a href="../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a></li>
        
          <li><a href="../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a></li>
        
          <li><a href="../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a></li>
        
          <li><a href="../Bio/PDB/Model.html">Bio::PDB::Model</a></li>
        
          <li><a href="../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a></li>
        
          <li><a href="../Bio/PDB/Record.html">Bio::PDB::Record</a></li>
        
          <li><a href="../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a></li>
        
          <li><a href="../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a></li>
        
          <li><a href="../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a></li>
        
          <li><a href="../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a></li>
        
          <li><a href="../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a></li>
        
          <li><a href="../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a></li>
        
          <li><a href="../Bio/PDB/Residue.html">Bio::PDB::Residue</a></li>
        
          <li><a href="../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a></li>
        
          <li><a href="../Bio/PDB/Utils.html">Bio::PDB::Utils</a></li>
        
          <li><a href="../Bio/PROSITE.html">Bio::PROSITE</a></li>
        
          <li><a href="../Bio/PSORT.html">Bio::PSORT</a></li>
        
          <li><a href="../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a></li>
        
          <li><a href="../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a></li>
        
          <li><a href="../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a></li>
        
          <li><a href="../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a></li>
        
          <li><a href="../Bio/PTS1.html">Bio::PTS1</a></li>
        
          <li><a href="../Bio/PTS1/Report.html">Bio::PTS1::Report</a></li>
        
          <li><a href="../Bio/Pathway.html">Bio::Pathway</a></li>
        
          <li><a href="../Bio/Phylip.html">Bio::Phylip</a></li>
        
          <li><a href="../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a></li>
        
          <li><a href="../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a></li>
        
          <li><a href="../Bio/PhyloXML.html">Bio::PhyloXML</a></li>
        
          <li><a href="../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a></li>
        
          <li><a href="../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a></li>
        
          <li><a href="../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a></li>
        
          <li><a href="../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a></li>
        
          <li><a href="../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a></li>
        
          <li><a href="../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a></li>
        
          <li><a href="../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a></li>
        
          <li><a href="../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a></li>
        
          <li><a href="../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a></li>
        
          <li><a href="../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a></li>
        
          <li><a href="../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a></li>
        
          <li><a href="../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a></li>
        
          <li><a href="../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a></li>
        
          <li><a href="../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a></li>
        
          <li><a href="../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a></li>
        
          <li><a href="../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a></li>
        
          <li><a href="../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a></li>
        
          <li><a href="../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a></li>
        
          <li><a href="../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a></li>
        
          <li><a href="../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a></li>
        
          <li><a href="../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a></li>
        
          <li><a href="../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a></li>
        
          <li><a href="../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a></li>
        
          <li><a href="../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a></li>
        
          <li><a href="../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a></li>
        
          <li><a href="../Bio/Probcons.html">Bio::Probcons</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/PubMed.html">Bio::PubMed</a></li>
        
          <li><a href="../Bio/REBASE.html">Bio::REBASE</a></li>
        
          <li><a href="../Bio/RefSeq.html">Bio::RefSeq</a></li>
        
          <li><a href="../Bio/Reference.html">Bio::Reference</a></li>
        
          <li><a href="../Bio/References.html">Bio::References</a></li>
        
          <li><a href="../Bio/Registry.html">Bio::Registry</a></li>
        
          <li><a href="../Bio/Registry/DB.html">Bio::Registry::DB</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a></li>
        
          <li><a href="../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a></li>
        
          <li><a href="../Bio/SOFT.html">Bio::SOFT</a></li>
        
          <li><a href="../Bio/SOSUI.html">Bio::SOSUI</a></li>
        
          <li><a href="../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a></li>
        
          <li><a href="../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a></li>
        
          <li><a href="../Bio/SPTR.html">Bio::SPTR</a></li>
        
          <li><a href="../Bio/SQL.html">Bio::SQL</a></li>
        
          <li><a href="../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a></li>
        
          <li><a href="../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a></li>
        
          <li><a href="../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a></li>
        
          <li><a href="../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a></li>
        
          <li><a href="../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a></li>
        
          <li><a href="../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a></li>
        
          <li><a href="../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a></li>
        
          <li><a href="../Bio/SQL/Comment.html">Bio::SQL::Comment</a></li>
        
          <li><a href="../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a></li>
        
          <li><a href="../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a></li>
        
          <li><a href="../Bio/SQL/Location.html">Bio::SQL::Location</a></li>
        
          <li><a href="../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a></li>
        
          <li><a href="../Bio/SQL/Reference.html">Bio::SQL::Reference</a></li>
        
          <li><a href="../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a></li>
        
          <li><a href="../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a></li>
        
          <li><a href="../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a></li>
        
          <li><a href="../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a></li>
        
          <li><a href="../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a></li>
        
          <li><a href="../Bio/SQL/Term.html">Bio::SQL::Term</a></li>
        
          <li><a href="../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a></li>
        
          <li><a href="../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a></li>
        
          <li><a href="../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a></li>
        
          <li><a href="../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a></li>
        
          <li><a href="../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a></li>
        
          <li><a href="../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a></li>
        
          <li><a href="../Bio/Scf.html">Bio::Scf</a></li>
        
          <li><a href="../Bio/Sequence.html">Bio::Sequence</a></li>
        
          <li><a href="../Bio/Sequence/AA.html">Bio::Sequence::AA</a></li>
        
          <li><a href="../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a></li>
        
          <li><a href="../Bio/Sequence/Common.html">Bio::Sequence::Common</a></li>
        
          <li><a href="../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a></li>
        
          <li><a href="../Bio/Sequence/Format.html">Bio::Sequence::Format</a></li>
        
          <li><a href="../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a></li>
        
          <li><a href="../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a></li>
        
          <li><a href="../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a></li>
        
          <li><a href="../Bio/Sequence/NA.html">Bio::Sequence::NA</a></li>
        
          <li><a href="../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a></li>
        
          <li><a href="../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a></li>
        
          <li><a href="../Bio/Shell.html">Bio::Shell</a></li>
        
          <li><a href="../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a></li>
        
          <li><a href="../Bio/Shell/Core.html">Bio::Shell::Core</a></li>
        
          <li><a href="../Bio/Shell/Demo.html">Bio::Shell::Demo</a></li>
        
          <li><a href="../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a></li>
        
          <li><a href="../Bio/Shell/Irb.html">Bio::Shell::Irb</a></li>
        
          <li><a href="../Bio/Shell/Private.html">Bio::Shell::Private</a></li>
        
          <li><a href="../Bio/Shell/Script.html">Bio::Shell::Script</a></li>
        
          <li><a href="../Bio/Shell/Setup.html">Bio::Shell::Setup</a></li>
        
          <li><a href="../Bio/Shell/Web.html">Bio::Shell::Web</a></li>
        
          <li><a href="../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a></li>
        
          <li><a href="../Bio/SiRNA.html">Bio::SiRNA</a></li>
        
          <li><a href="../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a></li>
        
          <li><a href="../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a></li>
        
          <li><a href="../Bio/Sim4.html">Bio::Sim4</a></li>
        
          <li><a href="../Bio/Sim4/Report.html">Bio::Sim4::Report</a></li>
        
          <li><a href="../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a></li>
        
          <li><a href="../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a></li>
        
          <li><a href="../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/Spidey.html">Bio::Spidey</a></li>
        
          <li><a href="../Bio/Spidey/Report.html">Bio::Spidey::Report</a></li>
        
          <li><a href="../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a></li>
        
          <li><a href="../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a></li>
        
          <li><a href="../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/SwissProt.html">Bio::SwissProt</a></li>
        
          <li><a href="../Bio/TMHMM.html">Bio::TMHMM</a></li>
        
          <li><a href="../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a></li>
        
          <li><a href="../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a></li>
        
          <li><a href="../Bio/TRANSFAC.html">Bio::TRANSFAC</a></li>
        
          <li><a href="../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a></li>
        
          <li><a href="../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a></li>
        
          <li><a href="../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a></li>
        
          <li><a href="../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a></li>
        
          <li><a href="../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a></li>
        
          <li><a href="../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a></li>
        
          <li><a href="../Bio/TargetP.html">Bio::TargetP</a></li>
        
          <li><a href="../Bio/TargetP/Report.html">Bio::TargetP::Report</a></li>
        
          <li><a href="../Bio/Taxonomy.html">Bio::Taxonomy</a></li>
        
          <li><a href="../Bio/Tcoffee.html">Bio::Tcoffee</a></li>
        
          <li><a href="../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/TogoWS.html">Bio::TogoWS</a></li>
        
          <li><a href="../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a></li>
        
          <li><a href="../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a></li>
        
          <li><a href="../Bio/TrEMBL.html">Bio::TrEMBL</a></li>
        
          <li><a href="../Bio/Tree.html">Bio::Tree</a></li>
        
          <li><a href="../Bio/Tree/Edge.html">Bio::Tree::Edge</a></li>
        
          <li><a href="../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a></li>
        
          <li><a href="../Bio/Tree/Node.html">Bio::Tree::Node</a></li>
        
          <li><a href="../Bio/UniProt.html">Bio::UniProt</a></li>
        
          <li><a href="../BiorubyController.html">BiorubyController</a></li>
        
          <li><a href="../BiorubyGenerator.html">BiorubyGenerator</a></li>
        
          <li><a href="../BiorubyHelper.html">BiorubyHelper</a></li>
        
          <li><a href="../Hash.html">Hash</a></li>
        
          <li><a href="../Object.html">Object</a></li>
        
          <li><a href="../String.html">String</a></li>
        
        </ul>
        <div id="no-class-search-results" style="display: none;">No matching classes.</div>
      </div>

      
    </div>
  </div>

  <div id="documentation">
    <h1 class="class">Bio::PROSITE</h1>

    <div id="description" class="description">
      
    </div><!-- description -->

    
    
    
    <div id="5Buntitled-5D" class="documentation-section">
      

      

      
      <!-- Constants -->
      <div id="constants-list" class="section">
        <h3 class="section-header">Constants</h3>
        <dl>
        
          <dt><a name="DELIMITER">DELIMITER</a></dt>
          
          <dd class="description"><p>Delimiter</p></dd>
          
        
          <dt><a name="RS">RS</a></dt>
          
          <dd class="description"><p>Delimiter</p></dd>
          
        
          <dt><a name="TAGSIZE">TAGSIZE</a></dt>
          
          <dd class="description"><p><a href="DB.html">Bio::DB</a> API</p></dd>
          
        
        </dl>
      </div>
      

      

      <!-- Methods -->
      
      <div id="public-class-method-details" class="method-section section">
        <h3 class="section-header">Public Class Methods</h3>

      
        <div id="new-method" class="method-detail ">
          <a name="method-c-new"></a>

          
          <div class="method-heading">
            <span class="method-name">new</span><span
              class="method-args">(entry)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
            <div class="method-source-code" id="new-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 26</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">entry</span>)
  <span class="ruby-keyword">super</span>(<span class="ruby-identifier">entry</span>, <span class="ruby-constant">TAGSIZE</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- new-source -->
            
          </div>

          

          
        </div><!-- new-method -->

      
        <div id="pa2re-method" class="method-detail ">
          <a name="method-c-pa2re"></a>

          
          <div class="method-heading">
            <span class="method-name">pa2re</span><span
              class="method-args">(pattern)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <pre>prosite pattern to regular expression</pre>

<p>prosite/prosuser.txt:</p>

<p>The PA (PAttern) lines contains the definition of a <a
href="PROSITE.html">PROSITE</a> pattern. The patterns are described using
the following conventions:</p>

<p>0) The standard IUPAC one-letter codes for the amino acids are used. 0)
Ambiguities are indicated by listing the acceptable amino acids for a</p>

<pre>given position, between square parentheses `[ ]'. For example: [ALT]
stands for Ala or Leu or Thr.</pre>

<p>1) A period ends the pattern. 2) When a pattern is restricted to either the
N- or C-terminal of a</p>

<pre>sequence, that pattern either starts with a `&lt;' symbol or respectively
ends with a `&gt;' symbol.</pre>

<p>3) Ambiguities are also indicated by listing between a pair of curly</p>

<pre>brackets `{ }' the amino acids that are not accepted at a given
position. For example: {AM} stands for any amino acid except Ala and
Met.</pre>

<p>4) Repetition of an element of the pattern can be indicated by following</p>

<pre>that element with a numerical value or a numerical range between
parenthesis. Examples: x(3) corresponds to x-x-x, x(2,4) corresponds to
x-x or x-x-x or x-x-x-x.</pre>

<p>5) The symbol `x is used for a position where any amino acid is accepted.
6) Each element in a pattern is separated from its neighbor by a `-.</p>

<p>Examples:</p>

<p>PA  [AC]-x-V-x(4)-{ED}.</p>

<p>This pattern is translated as: [Ala or Cys]-any-Val-any-any-any-any-{any
but Glu or Asp}</p>

<p>PA  &lt;<a href="http://ST">A-x-</a>(2)-x(0,1)-V.</p>

<p>This pattern, which must be in the N-terminal of the sequence (`&lt;), is
translated as: Ala-any-[Ser or Thr]-[Ser or Thr]-(any or none)-Val</p>
            

            
            <div class="method-source-code" id="pa2re-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 467</span>
<span class="ruby-keyword">def</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">pa2re</span>(<span class="ruby-identifier">pattern</span>)
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">gsub!</span>(<span class="ruby-regexp">/\s/</span>, <span class="ruby-string">''</span>)     <span class="ruby-comment"># remove white spaces</span>
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">sub!</span>(<span class="ruby-regexp">/\.$/</span>, <span class="ruby-string">''</span>)     <span class="ruby-comment"># (1) remove trailing '.'</span>
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">sub!</span>(<span class="ruby-regexp">/^&lt;/</span>, <span class="ruby-string">'^'</span>)     <span class="ruby-comment"># (2) restricted to the N-terminal : `&lt;'</span>
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">sub!</span>(<span class="ruby-regexp">/&gt;$/</span>, <span class="ruby-string">'$'</span>)     <span class="ruby-comment"># (2) restricted to the C-terminal : `&gt;'</span>
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">gsub!</span>(<span class="ruby-regexp">/\{(\w+)\}/</span>) { <span class="ruby-operator">|</span><span class="ruby-identifier">m</span><span class="ruby-operator">|</span>
    <span class="ruby-string">'[^'</span> <span class="ruby-operator">+</span> <span class="ruby-node">$1</span> <span class="ruby-operator">+</span> <span class="ruby-string">']'</span>           <span class="ruby-comment"># (3) not accepted at a given position : '{}'</span>
  }
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">gsub!</span>(<span class="ruby-regexp">/\(([\d,]+)\)/</span>) { <span class="ruby-operator">|</span><span class="ruby-identifier">m</span><span class="ruby-operator">|</span>
    <span class="ruby-string">'{'</span> <span class="ruby-operator">+</span> <span class="ruby-node">$1</span> <span class="ruby-operator">+</span> <span class="ruby-string">'}'</span>            <span class="ruby-comment"># (4) repetition of an element : (n), (n,m)</span>
  }
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">tr!</span>(<span class="ruby-string">'x'</span>, <span class="ruby-string">'.'</span>)       <span class="ruby-comment"># (5) any amino acid is accepted : 'x'</span>
  <span class="ruby-identifier">pattern</span>.<span class="ruby-identifier">tr!</span>(<span class="ruby-string">'-'</span>, <span class="ruby-string">''</span>)        <span class="ruby-comment"># (6) each element is separated by a '-'</span>
  <span class="ruby-constant">Regexp</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">pattern</span>, <span class="ruby-constant">Regexp</span><span class="ruby-operator">::</span><span class="ruby-constant">IGNORECASE</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- pa2re-source -->
            
          </div>

          

          
        </div><!-- pa2re-method -->

      
      </div><!-- public-class-method-details -->
    
      <div id="public-instance-method-details" class="method-section section">
        <h3 class="section-header">Public Instance Methods</h3>

      
        <div id="ac-method" class="method-detail ">
          <a name="method-i-ac"></a>

          
          <div class="method-heading">
            <span class="method-name">ac</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>AC  Accession number                   (1 per entry)</p>

<pre>AC   PSnnnnn;</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="ac-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 57</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ac</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'AC'</span>]
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'AC'</span>] = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'AC'</span>).<span class="ruby-identifier">chomp</span>(<span class="ruby-string">';'</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'AC'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- ac-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-entry_id">entry_id</a>
          </div>
          

          
        </div><!-- ac-method -->

      
        <div id="cc-method" class="method-detail ">
          <a name="method-i-cc"></a>

          
          <div class="method-heading">
            <span class="method-name">cc</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>CC  Comments                           (&gt;=0 per entry)</p>

<pre>CC   /QUALIFIER=data; /QUALIFIER=data; .......</pre>

<p>/TAXO-RANGE  Taxonomic range. /MAX-REPEAT  Maximum known  number of 
repetitions of  the pattern  in a</p>

<pre>single protein.</pre>

<p>/SITE        Indication of an `interesting site in the pattern. /SKIP-FLAG
Indication of  an entry that can be, in some cases, ignored</p>

<pre>by a program (because it is too unspecific).</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="cc-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 273</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cc</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>]
    <span class="ruby-identifier">hash</span> = {}                 <span class="ruby-comment"># temporal hash</span>
    <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'CC'</span>).<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">%{/(\S+)=([^;]+);}</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">v</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">v</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>] = <span class="ruby-identifier">hash</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- cc-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-comment">comment</a>
          </div>
          

          
        </div><!-- cc-method -->

      
        <div id="comment-method" class="method-detail method-alias">
          <a name="method-i-comment"></a>

          
          <div class="method-heading">
            <span class="method-name">comment</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-cc">cc</a>
          </div>
          
        </div><!-- comment-method -->

      
        <div id="date-method" class="method-detail method-alias">
          <a name="method-i-date"></a>

          
          <div class="method-heading">
            <span class="method-name">date</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-dt">dt</a>
          </div>
          
        </div><!-- date-method -->

      
        <div id="de-method" class="method-detail ">
          <a name="method-i-de"></a>

          
          <div class="method-heading">
            <span class="method-name">de</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>DE  Short description                  (1 per entry)</p>

<pre>DE   Description.</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="de-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 84</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">de</span>
  <span class="ruby-identifier">field_fetch</span>(<span class="ruby-string">'DE'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- de-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-definition">definition</a>
          </div>
          

          
        </div><!-- de-method -->

      
        <div id="definition-method" class="method-detail method-alias">
          <a name="method-i-definition"></a>

          
          <div class="method-heading">
            <span class="method-name">definition</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-de">de</a>
          </div>
          
        </div><!-- definition-method -->

      
        <div id="division-method" class="method-detail ">
          <a name="method-i-division"></a>

          
          <div class="method-heading">
            <span class="method-name">division</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="division-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 44</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">division</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'TYPE'</span>]
    <span class="ruby-identifier">name</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'TYPE'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- division-source -->
            
          </div>

          

          
        </div><!-- division-method -->

      
        <div id="dr-method" class="method-detail ">
          <a name="method-i-dr"></a>

          
          <div class="method-heading">
            <span class="method-name">dr</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>DR  Cross-references to SWISS-PROT     (&gt;=0 per entry)</p>

<pre>DR   AC_NB, ENTRY_NAME, C; AC_NB, ENTRY_NAME, C; AC_NB, ENTRY_NAME, C;</pre>
<ul><li>
<p>`AC_NB is the SWISS-PROT primary accession number of the entry to which
reference is being made.</p>
</li><li>
<p>`ENTRY_NAME is the SWISS-PROT entry name.</p>
</li><li>
<p>`C is a one character flag that can be one of the following:</p>
</li></ul>

<p>T For a true positive. N For a false negative; a sequence which belongs to
the set under</p>

<pre>consideration, but which has not been picked up by the pattern or
profile.</pre>

<p>P For a `potential hit; a sequence that belongs to the set under</p>

<pre>consideration, but which was not picked up because the region(s) that
are used as a 'fingerprint' (pattern or profile) is not yet available
in the data bank (partial sequence).</pre>

<p>? For an unknown; a sequence which possibly could belong to the set under</p>

<pre>consideration.</pre>

<p>F For a false positive; a sequence which does not belong to the set in</p>

<pre>consideration.</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="dr-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 349</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dr</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DR'</span>]
    <span class="ruby-identifier">hash</span> = {}                 <span class="ruby-comment"># temporal hash</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'DR'</span>)
      <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'DR'</span>).<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">/(\w+)\s*, (\w+)\s*, (.);/</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">a</span>, <span class="ruby-identifier">e</span>, <span class="ruby-identifier">c</span><span class="ruby-operator">|</span>
        <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">a</span>] = [<span class="ruby-identifier">e</span>, <span class="ruby-identifier">c</span>]      <span class="ruby-comment"># SWISS-PROT : accession, entry, true/false</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DR'</span>] = <span class="ruby-identifier">hash</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DR'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- dr-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-sp_xref">sp_xref</a>
          </div>
          

          
        </div><!-- dr-method -->

      
        <div id="dt-method" class="method-detail ">
          <a name="method-i-dt"></a>

          
          <div class="method-heading">
            <span class="method-name">dt</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>DT  Date                               (1 per entry)</p>

<pre>DT   MMM-YYYY (CREATED); MMM-YYYY (DATA UPDATE); MMM-YYYY (INFO UPDATE).</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="dt-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 72</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dt</span>
  <span class="ruby-identifier">field_fetch</span>(<span class="ruby-string">'DT'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- dt-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-date">date</a>
          </div>
          

          
        </div><!-- dt-method -->

      
        <div id="entry_id-method" class="method-detail method-alias">
          <a name="method-i-entry_id"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_id</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-ac">ac</a>
          </div>
          
        </div><!-- entry_id-method -->

      
        <div id="false_neg-method" class="method-detail ">
          <a name="method-i-false_neg"></a>

          
          <div class="method-heading">
            <span class="method-name">false_neg</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="false_neg-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 250</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">false_neg</span>
  <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'FALSE_NEG'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- false_neg-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-false_negative_hits">false_negative_hits</a>
          </div>
          

          
        </div><!-- false_neg-method -->

      
        <div id="false_negative_hits-method" class="method-detail method-alias">
          <a name="method-i-false_negative_hits"></a>

          
          <div class="method-heading">
            <span class="method-name">false_negative_hits</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-false_neg">false_neg</a>
          </div>
          
        </div><!-- false_negative_hits-method -->

      
        <div id="false_pos-method" class="method-detail ">
          <a name="method-i-false_pos"></a>

          
          <div class="method-heading">
            <span class="method-name">false_pos</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="false_pos-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 235</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">false_pos</span>
  <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'FALSE_POS'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- false_pos-source -->
            
          </div>

          

          
        </div><!-- false_pos-method -->

      
        <div id="false_positive_hits-method" class="method-detail ">
          <a name="method-i-false_positive_hits"></a>

          
          <div class="method-heading">
            <span class="method-name">false_positive_hits</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="false_positive_hits-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 240</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">false_positive_hits</span>
  <span class="ruby-identifier">false_pos</span>.<span class="ruby-identifier">first</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- false_positive_hits-source -->
            
          </div>

          

          
        </div><!-- false_positive_hits-method -->

      
        <div id="false_positive_sequences-method" class="method-detail ">
          <a name="method-i-false_positive_sequences"></a>

          
          <div class="method-heading">
            <span class="method-name">false_positive_sequences</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="false_positive_sequences-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 245</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">false_positive_sequences</span>
  <span class="ruby-identifier">false_pos</span>.<span class="ruby-identifier">last</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- false_positive_sequences-source -->
            
          </div>

          

          
        </div><!-- false_positive_sequences-method -->

      
        <div id="list_falsenegative-method" class="method-detail ">
          <a name="method-i-list_falsenegative"></a>

          
          <div class="method-heading">
            <span class="method-name">list_falsenegative</span><span
              class="method-args">(by_name = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="list_falsenegative-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 386</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">list_falsenegative</span>(<span class="ruby-identifier">by_name</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">list_xref</span>(<span class="ruby-string">'F'</span>, <span class="ruby-identifier">by_name</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- list_falsenegative-source -->
            
          </div>

          

          
        </div><!-- list_falsenegative-method -->

      
        <div id="list_falsepositive-method" class="method-detail ">
          <a name="method-i-list_falsepositive"></a>

          
          <div class="method-heading">
            <span class="method-name">list_falsepositive</span><span
              class="method-args">(by_name = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="list_falsepositive-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 391</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">list_falsepositive</span>(<span class="ruby-identifier">by_name</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">list_xref</span>(<span class="ruby-string">'P'</span>, <span class="ruby-identifier">by_name</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- list_falsepositive-source -->
            
          </div>

          

          
        </div><!-- list_falsepositive-method -->

      
        <div id="list_potentialhit-method" class="method-detail ">
          <a name="method-i-list_potentialhit"></a>

          
          <div class="method-heading">
            <span class="method-name">list_potentialhit</span><span
              class="method-args">(by_name = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="list_potentialhit-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 396</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">list_potentialhit</span>(<span class="ruby-identifier">by_name</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">list_xref</span>(<span class="ruby-string">'P'</span>, <span class="ruby-identifier">by_name</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- list_potentialhit-source -->
            
          </div>

          

          
        </div><!-- list_potentialhit-method -->

      
        <div id="list_truepositive-method" class="method-detail ">
          <a name="method-i-list_truepositive"></a>

          
          <div class="method-heading">
            <span class="method-name">list_truepositive</span><span
              class="method-args">(by_name = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="list_truepositive-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 381</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">list_truepositive</span>(<span class="ruby-identifier">by_name</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">list_xref</span>(<span class="ruby-string">'T'</span>, <span class="ruby-identifier">by_name</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- list_truepositive-source -->
            
          </div>

          

          
        </div><!-- list_truepositive-method -->

      
        <div id="list_unknown-method" class="method-detail ">
          <a name="method-i-list_unknown"></a>

          
          <div class="method-heading">
            <span class="method-name">list_unknown</span><span
              class="method-args">(by_name = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="list_unknown-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 401</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">list_unknown</span>(<span class="ruby-identifier">by_name</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">list_xref</span>(<span class="ruby-string">'?'</span>, <span class="ruby-identifier">by_name</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- list_unknown-source -->
            
          </div>

          

          
        </div><!-- list_unknown-method -->

      
        <div id="list_xref-method" class="method-detail ">
          <a name="method-i-list_xref"></a>

          
          <div class="method-heading">
            <span class="method-name">list_xref</span><span
              class="method-args">(flag, by_name = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="list_xref-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 365</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">list_xref</span>(<span class="ruby-identifier">flag</span>, <span class="ruby-identifier">by_name</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">ary</span> = []
  <span class="ruby-identifier">sp_xref</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">sp_acc</span>, <span class="ruby-identifier">value</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">value</span>[<span class="ruby-value">1</span>] <span class="ruby-operator">==</span> <span class="ruby-identifier">flag</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">by_name</span>
        <span class="ruby-identifier">sp_name</span> = <span class="ruby-identifier">value</span>[<span class="ruby-value">0</span>]
        <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">push</span>(<span class="ruby-identifier">sp_name</span>)
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">push</span>(<span class="ruby-identifier">sp_acc</span>)
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">ary</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- list_xref-source -->
            
          </div>

          

          
        </div><!-- list_xref-method -->

      
        <div id="ma-method" class="method-detail ">
          <a name="method-i-ma"></a>

          
          <div class="method-heading">
            <span class="method-name">ma</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>MA  Matrix/profile                     (&gt;=0 per entry)</p>

<pre>see - ma2re method</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="ma-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 111</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ma</span>
  <span class="ruby-identifier">field_fetch</span>(<span class="ruby-string">'MA'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- ma-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-profile">profile</a>
          </div>
          

          
        </div><!-- ma-method -->

      
        <div id="ma2re-method" class="method-detail ">
          <a name="method-i-ma2re"></a>

          
          <div class="method-heading">
            <span class="method-name">ma2re</span><span
              class="method-args">(matrix)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <pre>prosite profile to regular expression</pre>

<p>prosite/profile.txt:</p>

<p>Returns</p>
            

            
            <div class="method-source-code" id="ma2re-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 497</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ma2re</span>(<span class="ruby-identifier">matrix</span>)
  <span class="ruby-identifier">raise</span> <span class="ruby-constant">NotImplementedError</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- ma2re-source -->
            
          </div>

          

          
        </div><!-- ma2re-method -->

      
        <div id="max_repeat-method" class="method-detail ">
          <a name="method-i-max_repeat"></a>

          
          <div class="method-heading">
            <span class="method-name">max_repeat</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="max_repeat-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 306</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">max_repeat</span>
  <span class="ruby-identifier">comment</span>[<span class="ruby-string">'MAX-REPEAT'</span>].<span class="ruby-identifier">to_i</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- max_repeat-source -->
            
          </div>

          

          
        </div><!-- max_repeat-method -->

      
        <div id="name-method" class="method-detail ">
          <a name="method-i-name"></a>

          
          <div class="method-heading">
            <span class="method-name">name</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>ID  Identification                     (Begins each entry; 1 per entry)</p>

<pre>ID   ENTRY_NAME; ENTRY_TYPE.  (ENTRY_TYPE : PATTERN, MATRIX, RULE)</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="name-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 36</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">name</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'ID'</span>]
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'ID'</span>], <span class="ruby-ivar">@data</span>[<span class="ruby-string">'TYPE'</span>] = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'ID'</span>).<span class="ruby-identifier">chomp</span>(<span class="ruby-string">'.'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-string">'; '</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'ID'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- name-source -->
            
          </div>

          

          
        </div><!-- name-method -->

      
        <div id="nr-method" class="method-detail ">
          <a name="method-i-nr"></a>

          
          <div class="method-heading">
            <span class="method-name">nr</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>NR  Numerical results                  (&gt;=0 per entry)</p>

<pre>- SWISS-PROT scan statistics of true and false positives/negatives</pre>

<p>/RELEASE     SWISS-PROT release  number and  total  number  of  sequence</p>

<pre>entries in that release.</pre>

<p>/TOTAL       Total number of hits in SWISS-PROT. /POSITIVE    Number of 
hits on proteins that are known to belong to the</p>

<pre>set in consideration.</pre>

<p>/UNKNOWN     Number of  hits on  proteins that  could possibly belong to</p>

<pre>the set in consideration.</pre>

<p>/FALSE_POS   Number of false hits (on unrelated proteins). /FALSE_NEG  
Number of known missed hits. /PARTIAL     Number of  partial sequences 
which belong  to the  set  in</p>

<pre>consideration, but  which  are  not  hit  by the pattern or
profile because they are partial (fragment) sequences.</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="nr-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 150</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">nr</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'NR'</span>]
    <span class="ruby-identifier">hash</span> = {}                 <span class="ruby-comment"># temporal hash</span>
    <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'NR'</span>).<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">%{/(\S+)=([^;]+);}</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">v</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">v</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/^(\d+)\((\d+)\)$/</span>
        <span class="ruby-identifier">hits</span> = <span class="ruby-node">$1</span>.<span class="ruby-identifier">to_i</span>                <span class="ruby-comment"># the number of hits</span>
        <span class="ruby-identifier">seqs</span> = <span class="ruby-node">$2</span>.<span class="ruby-identifier">to_i</span>                <span class="ruby-comment"># the number of sequences</span>
        <span class="ruby-identifier">v</span> = [<span class="ruby-identifier">hits</span>, <span class="ruby-identifier">seqs</span>]
      <span class="ruby-keyword">elsif</span> <span class="ruby-identifier">v</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/([\d\.]+),(\d+)/</span>
        <span class="ruby-identifier">sprel</span> = <span class="ruby-node">$1</span>                    <span class="ruby-comment"># the number of SWISS-PROT release</span>
        <span class="ruby-identifier">spseq</span> = <span class="ruby-node">$2</span>.<span class="ruby-identifier">to_i</span>               <span class="ruby-comment"># the number of SWISS-PROT sequences</span>
        <span class="ruby-identifier">v</span> = [<span class="ruby-identifier">sprel</span>, <span class="ruby-identifier">spseq</span>]
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">v</span> = <span class="ruby-identifier">v</span>.<span class="ruby-identifier">to_i</span>
      <span class="ruby-keyword">end</span>
      <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-identifier">v</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'NR'</span>] = <span class="ruby-identifier">hash</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'NR'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- nr-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-statistics">statistics</a>
          </div>
          

          
        </div><!-- nr-method -->

      
        <div id="pa-method" class="method-detail ">
          <a name="method-i-pa"></a>

          
          <div class="method-heading">
            <span class="method-name">pa</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>PA  Pattern                            (&gt;=0 per entry)</p>

<pre>see - pa2re method</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="pa-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 96</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pa</span>
  <span class="ruby-identifier">field_fetch</span>(<span class="ruby-string">'PA'</span>)
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'PA'</span>] = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'PA'</span>) <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'PA'</span>]
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'PA'</span>].<span class="ruby-identifier">gsub!</span>(<span class="ruby-regexp">/\s+/</span>, <span class="ruby-string">''</span>) <span class="ruby-keyword">if</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'PA'</span>]
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'PA'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- pa-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-pattern">pattern</a>
          </div>
          

          
        </div><!-- pa-method -->

      
        <div id="pa2re-method" class="method-detail ">
          <a name="method-i-pa2re"></a>

          
          <div class="method-heading">
            <span class="method-name">pa2re</span><span
              class="method-args">(pattern)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
            <div class="method-source-code" id="pa2re-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 483</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pa2re</span>(<span class="ruby-identifier">pattern</span>)
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">class</span>.<span class="ruby-identifier">pa2re</span>(<span class="ruby-identifier">pattern</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- pa2re-source -->
            
          </div>

          

          
        </div><!-- pa2re-method -->

      
        <div id="partial-method" class="method-detail ">
          <a name="method-i-partial"></a>

          
          <div class="method-heading">
            <span class="method-name">partial</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="partial-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 256</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">partial</span>
  <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'PARTIAL'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- partial-source -->
            
          </div>

          

          
        </div><!-- partial-method -->

      
        <div id="pattern-method" class="method-detail method-alias">
          <a name="method-i-pattern"></a>

          
          <div class="method-heading">
            <span class="method-name">pattern</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-pa">pa</a>
          </div>
          
        </div><!-- pattern-method -->

      
        <div id="pdb_xref-method" class="method-detail ">
          <a name="method-i-pdb_xref"></a>

          
          <div class="method-heading">
            <span class="method-name">pdb_xref</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>3D  Cross-references to <a href="PDB.html">PDB</a>            (&gt;=0 per
entry)</p>

<pre>3D   name; [name2;...]</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="pdb_xref-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 411</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pdb_xref</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'3D'</span>]
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'3D'</span>] = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'3D'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/; */</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'3D'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- pdb_xref-source -->
            
          </div>

          

          
        </div><!-- pdb_xref-method -->

      
        <div id="pdoc_xref-method" class="method-detail ">
          <a name="method-i-pdoc_xref"></a>

          
          <div class="method-heading">
            <span class="method-name">pdoc_xref</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>DO  Pointer to the documentation file  (1 per entry)</p>

<pre>DO   PDOCnnnnn;</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="pdoc_xref-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 424</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pdoc_xref</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DO'</span>] = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'DO'</span>).<span class="ruby-identifier">chomp</span>(<span class="ruby-string">';'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- pdoc_xref-source -->
            
          </div>

          

          
        </div><!-- pdoc_xref-method -->

      
        <div id="positive-method" class="method-detail ">
          <a name="method-i-positive"></a>

          
          <div class="method-heading">
            <span class="method-name">positive</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="positive-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 205</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">positive</span>
  <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'POSITIVE'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- positive-source -->
            
          </div>

          

          
        </div><!-- positive-method -->

      
        <div id="positive_hits-method" class="method-detail ">
          <a name="method-i-positive_hits"></a>

          
          <div class="method-heading">
            <span class="method-name">positive_hits</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="positive_hits-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 210</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">positive_hits</span>
  <span class="ruby-identifier">positive</span>.<span class="ruby-identifier">first</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- positive_hits-source -->
            
          </div>

          

          
        </div><!-- positive_hits-method -->

      
        <div id="positive_sequences-method" class="method-detail ">
          <a name="method-i-positive_sequences"></a>

          
          <div class="method-heading">
            <span class="method-name">positive_sequences</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="positive_sequences-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 215</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">positive_sequences</span>
  <span class="ruby-identifier">positive</span>.<span class="ruby-identifier">last</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- positive_sequences-source -->
            
          </div>

          

          
        </div><!-- positive_sequences-method -->

      
        <div id="profile-method" class="method-detail method-alias">
          <a name="method-i-profile"></a>

          
          <div class="method-heading">
            <span class="method-name">profile</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-ma">ma</a>
          </div>
          
        </div><!-- profile-method -->

      
        <div id="re-method" class="method-detail ">
          <a name="method-i-re"></a>

          
          <div class="method-heading">
            <span class="method-name">re</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
            <div class="method-source-code" id="re-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 487</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">re</span>
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">class</span>.<span class="ruby-identifier">pa2re</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">pa</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- re-source -->
            
          </div>

          

          
        </div><!-- re-method -->

      
        <div id="release-method" class="method-detail ">
          <a name="method-i-release"></a>

          
          <div class="method-heading">
            <span class="method-name">release</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="release-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 175</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">release</span>
  <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'RELEASE'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- release-source -->
            
          </div>

          

          
        </div><!-- release-method -->

      
        <div id="ru-method" class="method-detail ">
          <a name="method-i-ru"></a>

          
          <div class="method-heading">
            <span class="method-name">ru</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>RU  Rule                               (&gt;=0 per entry)</p>

<pre>RU   Rule_Description.

The rule is described in ordinary English and is free-format.</pre>

<p>Returns</p>
            

            
            <div class="method-source-code" id="ru-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 125</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ru</span>
  <span class="ruby-identifier">field_fetch</span>(<span class="ruby-string">'RU'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- ru-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="PROSITE.html#method-i-rule">rule</a>
          </div>
          

          
        </div><!-- ru-method -->

      
        <div id="rule-method" class="method-detail method-alias">
          <a name="method-i-rule"></a>

          
          <div class="method-heading">
            <span class="method-name">rule</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-ru">ru</a>
          </div>
          
        </div><!-- rule-method -->

      
        <div id="site-method" class="method-detail ">
          <a name="method-i-site"></a>

          
          <div class="method-heading">
            <span class="method-name">site</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="site-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 311</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">site</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">comment</span>[<span class="ruby-string">'SITE'</span>]
    <span class="ruby-identifier">num</span>, <span class="ruby-identifier">desc</span> = <span class="ruby-identifier">comment</span>[<span class="ruby-string">'SITE'</span>].<span class="ruby-identifier">split</span>(<span class="ruby-string">','</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> [<span class="ruby-identifier">num</span>.<span class="ruby-identifier">to_i</span>, <span class="ruby-identifier">desc</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- site-source -->
            
          </div>

          

          
        </div><!-- site-method -->

      
        <div id="skip_flag-method" class="method-detail ">
          <a name="method-i-skip_flag"></a>

          
          <div class="method-heading">
            <span class="method-name">skip_flag</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="skip_flag-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 319</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">skip_flag</span>
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">comment</span>[<span class="ruby-string">'SKIP-FLAG'</span>] <span class="ruby-operator">==</span> <span class="ruby-string">'TRUE'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-keyword">true</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- skip_flag-source -->
            
          </div>

          

          
        </div><!-- skip_flag-method -->

      
        <div id="sp_xref-method" class="method-detail method-alias">
          <a name="method-i-sp_xref"></a>

          
          <div class="method-heading">
            <span class="method-name">sp_xref</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-dr">dr</a>
          </div>
          
        </div><!-- sp_xref-method -->

      
        <div id="statistics-method" class="method-detail method-alias">
          <a name="method-i-statistics"></a>

          
          <div class="method-heading">
            <span class="method-name">statistics</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="PROSITE.html#method-i-nr">nr</a>
          </div>
          
        </div><!-- statistics-method -->

      
        <div id="swissprot_release_number-method" class="method-detail ">
          <a name="method-i-swissprot_release_number"></a>

          
          <div class="method-heading">
            <span class="method-name">swissprot_release_number</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="swissprot_release_number-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 180</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">swissprot_release_number</span>
  <span class="ruby-identifier">release</span>.<span class="ruby-identifier">first</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- swissprot_release_number-source -->
            
          </div>

          

          
        </div><!-- swissprot_release_number-method -->

      
        <div id="swissprot_release_sequences-method" class="method-detail ">
          <a name="method-i-swissprot_release_sequences"></a>

          
          <div class="method-heading">
            <span class="method-name">swissprot_release_sequences</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="swissprot_release_sequences-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 185</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">swissprot_release_sequences</span>
  <span class="ruby-identifier">release</span>.<span class="ruby-identifier">last</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- swissprot_release_sequences-source -->
            
          </div>

          

          
        </div><!-- swissprot_release_sequences-method -->

      
        <div id="taxon_range-method" class="method-detail ">
          <a name="method-i-taxon_range"></a>

          
          <div class="method-heading">
            <span class="method-name">taxon_range</span><span
              class="method-args">(expand = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="taxon_range-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 287</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">taxon_range</span>(<span class="ruby-identifier">expand</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-identifier">range</span> = <span class="ruby-identifier">comment</span>[<span class="ruby-string">'TAXO-RANGE'</span>]
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">range</span> <span class="ruby-keyword">and</span> <span class="ruby-identifier">expand</span>
    <span class="ruby-identifier">expand</span> = []
    <span class="ruby-identifier">range</span>.<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">/./</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">case</span> <span class="ruby-identifier">x</span>
      <span class="ruby-keyword">when</span> <span class="ruby-string">'A'</span>; <span class="ruby-identifier">expand</span>.<span class="ruby-identifier">push</span>(<span class="ruby-string">'archaebacteria'</span>)
      <span class="ruby-keyword">when</span> <span class="ruby-string">'B'</span>; <span class="ruby-identifier">expand</span>.<span class="ruby-identifier">push</span>(<span class="ruby-string">'bacteriophages'</span>)
      <span class="ruby-keyword">when</span> <span class="ruby-string">'E'</span>; <span class="ruby-identifier">expand</span>.<span class="ruby-identifier">push</span>(<span class="ruby-string">'eukaryotes'</span>)
      <span class="ruby-keyword">when</span> <span class="ruby-string">'P'</span>; <span class="ruby-identifier">expand</span>.<span class="ruby-identifier">push</span>(<span class="ruby-string">'prokaryotes'</span>)
      <span class="ruby-keyword">when</span> <span class="ruby-string">'V'</span>; <span class="ruby-identifier">expand</span>.<span class="ruby-identifier">push</span>(<span class="ruby-string">'eukaryotic viruses'</span>)
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-identifier">range</span> = <span class="ruby-identifier">expand</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">range</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- taxon_range-source -->
            
          </div>

          

          
        </div><!-- taxon_range-method -->

      
        <div id="total-method" class="method-detail ">
          <a name="method-i-total"></a>

          
          <div class="method-heading">
            <span class="method-name">total</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="total-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 190</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">total</span>
  <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'TOTAL'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- total-source -->
            
          </div>

          

          
        </div><!-- total-method -->

      
        <div id="total_hits-method" class="method-detail ">
          <a name="method-i-total_hits"></a>

          
          <div class="method-heading">
            <span class="method-name">total_hits</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="total_hits-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 195</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">total_hits</span>
  <span class="ruby-identifier">total</span>.<span class="ruby-identifier">first</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- total_hits-source -->
            
          </div>

          

          
        </div><!-- total_hits-method -->

      
        <div id="total_sequences-method" class="method-detail ">
          <a name="method-i-total_sequences"></a>

          
          <div class="method-heading">
            <span class="method-name">total_sequences</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="total_sequences-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 200</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">total_sequences</span>
  <span class="ruby-identifier">total</span>.<span class="ruby-identifier">last</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- total_sequences-source -->
            
          </div>

          

          
        </div><!-- total_sequences-method -->

      
        <div id="unknown-method" class="method-detail ">
          <a name="method-i-unknown"></a>

          
          <div class="method-heading">
            <span class="method-name">unknown</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="unknown-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 220</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">unknown</span>
  <span class="ruby-identifier">statistics</span>[<span class="ruby-string">'UNKNOWN'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- unknown-source -->
            
          </div>

          

          
        </div><!-- unknown-method -->

      
        <div id="unknown_hits-method" class="method-detail ">
          <a name="method-i-unknown_hits"></a>

          
          <div class="method-heading">
            <span class="method-name">unknown_hits</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="unknown_hits-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 225</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">unknown_hits</span>
  <span class="ruby-identifier">unknown</span>.<span class="ruby-identifier">first</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- unknown_hits-source -->
            
          </div>

          

          
        </div><!-- unknown_hits-method -->

      
        <div id="unknown_sequences-method" class="method-detail ">
          <a name="method-i-unknown_sequences"></a>

          
          <div class="method-heading">
            <span class="method-name">unknown_sequences</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Returns</p>
            

            
            <div class="method-source-code" id="unknown_sequences-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/prosite.rb, line 230</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">unknown_sequences</span>
  <span class="ruby-identifier">unknown</span>.<span class="ruby-identifier">last</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- unknown_sequences-source -->
            
          </div>

          

          
        </div><!-- unknown_sequences-method -->

      
      </div><!-- public-instance-method-details -->
    
    </div><!-- 5Buntitled-5D -->
  

  </div><!-- documentation -->

  <div id="validator-badges">
    <p><small><a href="http://validator.w3.org/check/referer">[Validate]</a></small></p>
    <p><small>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish
      Rdoc Generator</a> 2</small>.</p>
  </div>

</body>
</html>

