<!DOCTYPE html>

<html>
<head>
<meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

<title>class Bio::Reference - BioRuby API documentation</title>

<link type="text/css" media="screen" href="../rdoc.css" rel="stylesheet">

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
</script>

<script type="text/javascript" charset="utf-8" src="../js/jquery.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/navigation.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/search_index.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/search.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/searcher.js"></script>
<script type="text/javascript" charset="utf-8" src="../js/darkfish.js"></script>


<body id="top" class="class">
<nav id="metadata">
  <nav id="home-section" class="section">
  <h3 class="section-header">
    <a href="../index.html">Home</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </h3>
</nav>


  <nav id="search-section" class="section project-section" class="initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <h3 class="section-header">
      <input type="text" name="search" placeholder="Search" id="search-field"
             title="Type to search, Up and Down to navigate, Enter to load">
    </h3>
  </form>

  <ul id="search-results" class="initially-hidden"></ul>
</nav>


  <div id="file-metadata">
    <nav id="file-list-section" class="section">
  <h3 class="section-header">Defined In</h3>
  <ul>
    <li>lib/bio/reference.rb
  </ul>
</nav>

    
  </div>

  <div id="class-metadata">
    
    <nav id="parent-class-section" class="section">
  <h3 class="section-header">Parent</h3>
  
  <p class="link"><a href="../Object.html">Object</a>
  
</nav>

    
    <!-- Method Quickref -->
<nav id="method-list-section" class="section">
  <h3 class="section-header">Methods</h3>

  <ul class="link-list">
    
    <li><a href="#method-c-new">::new</a>
    
    <li><a href="#method-i-3D-3D">#==</a>
    
    <li><a href="#method-i-bibitem">#bibitem</a>
    
    <li><a href="#method-i-bibtex">#bibtex</a>
    
    <li><a href="#method-i-cell">#cell</a>
    
    <li><a href="#method-i-current">#current</a>
    
    <li><a href="#method-i-embl">#embl</a>
    
    <li><a href="#method-i-endnote">#endnote</a>
    
    <li><a href="#method-i-format">#format</a>
    
    <li><a href="#method-i-general">#general</a>
    
    <li><a href="#method-i-genome_biol">#genome_biol</a>
    
    <li><a href="#method-i-genome_res">#genome_res</a>
    
    <li><a href="#method-i-nar">#nar</a>
    
    <li><a href="#method-i-nature">#nature</a>
    
    <li><a href="#method-i-pubmed_url">#pubmed_url</a>
    
    <li><a href="#method-i-rd">#rd</a>
    
    <li><a href="#method-i-science">#science</a>
    
    <li><a href="#method-i-trends">#trends</a>
    
  </ul>
</nav>

  </div>

  <div id="project-metadata">
    <nav id="fileindex-section" class="section project-section">
  <h3 class="section-header">Pages</h3>

  <ul>
  
    <li class="file"><a href="../ChangeLog.html">ChangeLog</a>
  
    <li class="file"><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES</a>
  
    <li class="file"><a href="../LEGAL.html">LEGAL</a>
  
    <li class="file"><a href="../README_rdoc.html">README</a>
  
    <li class="file"><a href="../README_DEV_rdoc.html">README_DEV</a>
  
    <li class="file"><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES</a>
  
    <li class="file"><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1</a>
  
    <li class="file"><a href="../doc/RELEASE_NOTES-1_4_2_rdoc.html">RELEASE_NOTES-1.4.2</a>
  
  </ul>
</nav>

    <nav id="classindex-section" class="section project-section">
  <h3 class="section-header">Class and Module Index</h3>

  <ul class="link-list">
  
    <li><a href="../Bio.html">Bio</a>
  
    <li><a href="../Bio/AAindex.html">Bio::AAindex</a>
  
    <li><a href="../Bio/AAindex1.html">Bio::AAindex1</a>
  
    <li><a href="../Bio/AAindex2.html">Bio::AAindex2</a>
  
    <li><a href="../Bio/Abif.html">Bio::Abif</a>
  
    <li><a href="../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a>
  
    <li><a href="../Bio/Alignment.html">Bio::Alignment</a>
  
    <li><a href="../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a>
  
    <li><a href="../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a>
  
    <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a>
  
    <li><a href="../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a>
  
    <li><a href="../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a>
  
    <li><a href="../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a>
  
    <li><a href="../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a>
  
    <li><a href="../Bio/Alignment/Output.html">Bio::Alignment::Output</a>
  
    <li><a href="../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a>
  
    <li><a href="../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a>
  
    <li><a href="../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a>
  
    <li><a href="../Bio/Alignment/Site.html">Bio::Alignment::Site</a>
  
    <li><a href="../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a>
  
    <li><a href="../Bio/AminoAcid.html">Bio::AminoAcid</a>
  
    <li><a href="../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a>
  
    <li><a href="../Bio/Blast.html">Bio::Blast</a>
  
    <li><a href="../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Default.html">Bio::Blast::Default</a>
  
    <li><a href="../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a>
  
    <li><a href="../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a>
  
    <li><a href="../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a>
  
    <li><a href="../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a>
  
    <li><a href="../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a>
  
    <li><a href="../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a>
  
    <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a>
  
    <li><a href="../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a>
  
    <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Remote.html">Bio::Blast::Remote</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a>
  
    <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a>
  
    <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a>
  
    <li><a href="../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a>
  
    <li><a href="../Bio/Blast/Report.html">Bio::Blast::Report</a>
  
    <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a>
  
    <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a>
  
    <li><a href="../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a>
  
    <li><a href="../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a>
  
    <li><a href="../Bio/Blast/WU.html">Bio::Blast::WU</a>
  
    <li><a href="../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a>
  
    <li><a href="../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a>
  
    <li><a href="../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a>
  
    <li><a href="../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a>
  
    <li><a href="../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a>
  
    <li><a href="../Bio/Blat.html">Bio::Blat</a>
  
    <li><a href="../Bio/Blat/Report.html">Bio::Blat::Report</a>
  
    <li><a href="../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a>
  
    <li><a href="../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a>
  
    <li><a href="../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a>
  
    <li><a href="../Bio/ClustalW.html">Bio::ClustalW</a>
  
    <li><a href="../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a>
  
    <li><a href="../Bio/CodonTable.html">Bio::CodonTable</a>
  
    <li><a href="../Bio/ColorScheme.html">Bio::ColorScheme</a>
  
    <li><a href="../Bio/Command.html">Bio::Command</a>
  
    <li><a href="../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a>
  
    <li><a href="../Bio/Command/Tmpdir/Remover.html">Bio::Command::Tmpdir::Remover</a>
  
    <li><a href="../Bio/ContingencyTable.html">Bio::ContingencyTable</a>
  
    <li><a href="../Bio/DAS.html">Bio::DAS</a>
  
    <li><a href="../Bio/DAS/DNA.html">Bio::DAS::DNA</a>
  
    <li><a href="../Bio/DAS/DSN.html">Bio::DAS::DSN</a>
  
    <li><a href="../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a>
  
    <li><a href="../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a>
  
    <li><a href="../Bio/DAS/GFF.html">Bio::DAS::GFF</a>
  
    <li><a href="../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a>
  
    <li><a href="../Bio/DAS/LINK.html">Bio::DAS::LINK</a>
  
    <li><a href="../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a>
  
    <li><a href="../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a>
  
    <li><a href="../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a>
  
    <li><a href="../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a>
  
    <li><a href="../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a>
  
    <li><a href="../Bio/DB.html">Bio::DB</a>
  
    <li><a href="../Bio/DBGET.html">Bio::DBGET</a>
  
    <li><a href="../Bio/DDBJ.html">Bio::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a>
  
    <li><a href="../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a>
  
    <li><a href="../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a>
  
    <li><a href="../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a>
  
    <li><a href="../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a>
  
    <li><a href="../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a>
  
    <li><a href="../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a>
  
    <li><a href="../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a>
  
    <li><a href="../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a>
  
    <li><a href="../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a>
  
    <li><a href="../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a>
  
    <li><a href="../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a>
  
    <li><a href="../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a>
  
    <li><a href="../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a>
  
    <li><a href="../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a>
  
    <li><a href="../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a>
  
    <li><a href="../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a>
  
    <li><a href="../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a>
  
    <li><a href="../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a>
  
    <li><a href="../Bio/EBI.html">Bio::EBI</a>
  
    <li><a href="../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a>
  
    <li><a href="../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a>
  
    <li><a href="../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a>
  
    <li><a href="../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a>
  
    <li><a href="../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a>
  
    <li><a href="../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a>
  
    <li><a href="../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a>
  
    <li><a href="../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a>
  
    <li><a href="../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a>
  
    <li><a href="../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a>
  
    <li><a href="../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a>
  
    <li><a href="../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a>
  
    <li><a href="../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a>
  
    <li><a href="../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a>
  
    <li><a href="../Bio/EMBL.html">Bio::EMBL</a>
  
    <li><a href="../Bio/EMBLDB.html">Bio::EMBLDB</a>
  
    <li><a href="../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a>
  
    <li><a href="../Bio/EMBOSS.html">Bio::EMBOSS</a>
  
    <li><a href="../Bio/Ensembl.html">Bio::Ensembl</a>
  
    <li><a href="../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a>
  
    <li><a href="../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a>
  
    <li><a href="../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a>
  
    <li><a href="../Bio/FANTOM.html">Bio::FANTOM</a>
  
    <li><a href="../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a>
  
    <li><a href="../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a>
  
    <li><a href="../Bio/Fasta.html">Bio::Fasta</a>
  
    <li><a href="../Bio/Fasta/Report.html">Bio::Fasta::Report</a>
  
    <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a>
  
    <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a>
  
    <li><a href="../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a>
  
    <li><a href="../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a>
  
    <li><a href="../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a>
  
    <li><a href="../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a>
  
    <li><a href="../Bio/FastaDefline.html">Bio::FastaDefline</a>
  
    <li><a href="../Bio/FastaFormat.html">Bio::FastaFormat</a>
  
    <li><a href="../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a>
  
    <li><a href="../Bio/Fastq.html">Bio::Fastq</a>
  
    <li><a href="../Bio/Fastq/Error.html">Bio::Fastq::Error</a>
  
    <li><a href="../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a>
  
    <li><a href="../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a>
  
    <li><a href="../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a>
  
    <li><a href="../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a>
  
    <li><a href="../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a>
  
    <li><a href="../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a>
  
    <li><a href="../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a>
  
    <li><a href="../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a>
  
    <li><a href="../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a>
  
    <li><a href="../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a>
  
    <li><a href="../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a>
  
    <li><a href="../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a>
  
    <li><a href="../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a>
  
    <li><a href="../Bio/Feature.html">Bio::Feature</a>
  
    <li><a href="../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a>
  
    <li><a href="../Bio/Features.html">Bio::Features</a>
  
    <li><a href="../Bio/Fetch.html">Bio::Fetch</a>
  
    <li><a href="../Bio/FlatFile.html">Bio::FlatFile</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a>
  
    <li><a href="../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a>
  
    <li><a href="../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a>
  
    <li><a href="../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a>
  
    <li><a href="../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a>
  
    <li><a href="../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a>
  
    <li><a href="../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a>
  
    <li><a href="../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a>
  
    <li><a href="../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a>
  
    <li><a href="../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a>
  
    <li><a href="../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a>
  
    <li><a href="../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a>
  
    <li><a href="../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a>
  
    <li><a href="../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a>
  
    <li><a href="../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a>
  
    <li><a href="../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a>
  
    <li><a href="../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a>
  
    <li><a href="../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a>
  
    <li><a href="../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a>
  
    <li><a href="../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a>
  
    <li><a href="../Bio/GCG.html">Bio::GCG</a>
  
    <li><a href="../Bio/GCG/Msf.html">Bio::GCG::Msf</a>
  
    <li><a href="../Bio/GCG/Seq.html">Bio::GCG::Seq</a>
  
    <li><a href="../Bio/GFF.html">Bio::GFF</a>
  
    <li><a href="../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a>
  
    <li><a href="../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a>
  
    <li><a href="../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a>
  
    <li><a href="../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a>
  
    <li><a href="../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a>
  
    <li><a href="../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a>
  
    <li><a href="../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a>
  
    <li><a href="../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a>
  
    <li><a href="../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a>
  
    <li><a href="../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a>
  
    <li><a href="../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a>
  
    <li><a href="../Bio/GFF/Record.html">Bio::GFF::Record</a>
  
    <li><a href="../Bio/GO.html">Bio::GO</a>
  
    <li><a href="../Bio/GO/External2go.html">Bio::GO::External2go</a>
  
    <li><a href="../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a>
  
    <li><a href="../Bio/GO/Ontology.html">Bio::GO::Ontology</a>
  
    <li><a href="../Bio/GenBank.html">Bio::GenBank</a>
  
    <li><a href="../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a>
  
    <li><a href="../Bio/GenPept.html">Bio::GenPept</a>
  
    <li><a href="../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a>
  
    <li><a href="../Bio/Genscan.html">Bio::Genscan</a>
  
    <li><a href="../Bio/Genscan/Report.html">Bio::Genscan::Report</a>
  
    <li><a href="../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a>
  
    <li><a href="../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a>
  
    <li><a href="../Bio/HGC.html">Bio::HGC</a>
  
    <li><a href="../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a>
  
    <li><a href="../Bio/HMMER.html">Bio::HMMER</a>
  
    <li><a href="../Bio/HMMER/Report.html">Bio::HMMER::Report</a>
  
    <li><a href="../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a>
  
    <li><a href="../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a>
  
    <li><a href="../Bio/Hinv.html">Bio::Hinv</a>
  
    <li><a href="../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a>
  
    <li><a href="../Bio/Hinv/Common.html">Bio::Hinv::Common</a>
  
    <li><a href="../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a>
  
    <li><a href="../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a>
  
    <li><a href="../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a>
  
    <li><a href="../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a>
  
    <li><a href="../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a>
  
    <li><a href="../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a>
  
    <li><a href="../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a>
  
    <li><a href="../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a>
  
    <li><a href="../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a>
  
    <li><a href="../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a>
  
    <li><a href="../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a>
  
    <li><a href="../Bio/Iprscan.html">Bio::Iprscan</a>
  
    <li><a href="../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a>
  
    <li><a href="../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a>
  
    <li><a href="../Bio/KEGG.html">Bio::KEGG</a>
  
    <li><a href="../Bio/KEGG/API.html">Bio::KEGG::API</a>
  
    <li><a href="../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a>
  
    <li><a href="../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a>
  
    <li><a href="../Bio/KEGG/Common.html">Bio::KEGG::Common</a>
  
    <li><a href="../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a>
  
    <li><a href="../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a>
  
    <li><a href="../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a>
  
    <li><a href="../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a>
  
    <li><a href="../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a>
  
    <li><a href="../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a>
  
    <li><a href="../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a>
  
    <li><a href="../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a>
  
    <li><a href="../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a>
  
    <li><a href="../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a>
  
    <li><a href="../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a>
  
    <li><a href="../Bio/KEGG/KGML/Graphics.html">Bio::KEGG::KGML::Graphics</a>
  
    <li><a href="../Bio/KEGG/KGML/Product.html">Bio::KEGG::KGML::Product</a>
  
    <li><a href="../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a>
  
    <li><a href="../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a>
  
    <li><a href="../Bio/KEGG/KGML/Substrate.html">Bio::KEGG::KGML::Substrate</a>
  
    <li><a href="../Bio/KEGG/KGML/SubstrateProduct.html">Bio::KEGG::KGML::SubstrateProduct</a>
  
    <li><a href="../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a>
  
    <li><a href="../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a>
  
    <li><a href="../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a>
  
    <li><a href="../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a>
  
    <li><a href="../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a>
  
    <li><a href="../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a>
  
    <li><a href="../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a>
  
    <li><a href="../Bio/KEGGDB.html">Bio::KEGGDB</a>
  
    <li><a href="../Bio/LITDB.html">Bio::LITDB</a>
  
    <li><a href="../Bio/Lasergene.html">Bio::Lasergene</a>
  
    <li><a href="../Bio/Location.html">Bio::Location</a>
  
    <li><a href="../Bio/Locations.html">Bio::Locations</a>
  
    <li><a href="../Bio/MAFFT.html">Bio::MAFFT</a>
  
    <li><a href="../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a>
  
    <li><a href="../Bio/MEDLINE.html">Bio::MEDLINE</a>
  
    <li><a href="../Bio/Map.html">Bio::Map</a>
  
    <li><a href="../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a>
  
    <li><a href="../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a>
  
    <li><a href="../Bio/Map/Mapping.html">Bio::Map::Mapping</a>
  
    <li><a href="../Bio/Map/Marker.html">Bio::Map::Marker</a>
  
    <li><a href="../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a>
  
    <li><a href="../Bio/Meme.html">Bio::Meme</a>
  
    <li><a href="../Bio/Meme/Mast.html">Bio::Meme::Mast</a>
  
    <li><a href="../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a>
  
    <li><a href="../Bio/Meme/Motif.html">Bio::Meme::Motif</a>
  
    <li><a href="../Bio/Muscle.html">Bio::Muscle</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/NBRF.html">Bio::NBRF</a>
  
    <li><a href="../Bio/NCBI.html">Bio::NCBI</a>
  
    <li><a href="../Bio/NCBI/REST.html">Bio::NCBI::REST</a>
  
    <li><a href="../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a>
  
    <li><a href="../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a>
  
    <li><a href="../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a>
  
    <li><a href="../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a>
  
    <li><a href="../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a>
  
    <li><a href="../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a>
  
    <li><a href="../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a>
  
    <li><a href="../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a>
  
    <li><a href="../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a>
  
    <li><a href="../Bio/NCBIDB.html">Bio::NCBIDB</a>
  
    <li><a href="../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a>
  
    <li><a href="../Bio/Newick.html">Bio::Newick</a>
  
    <li><a href="../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a>
  
    <li><a href="../Bio/Nexus.html">Bio::Nexus</a>
  
    <li><a href="../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a>
  
    <li><a href="../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a>
  
    <li><a href="../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a>
  
    <li><a href="../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a>
  
    <li><a href="../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a>
  
    <li><a href="../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a>
  
    <li><a href="../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a>
  
    <li><a href="../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a>
  
    <li><a href="../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a>
  
    <li><a href="../Bio/Nexus/Util.html">Bio::Nexus::Util</a>
  
    <li><a href="../Bio/NucleicAcid.html">Bio::NucleicAcid</a>
  
    <li><a href="../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a>
  
    <li><a href="../Bio/PAML.html">Bio::PAML</a>
  
    <li><a href="../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a>
  
    <li><a href="../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a>
  
    <li><a href="../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a>
  
    <li><a href="../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a>
  
    <li><a href="../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a>
  
    <li><a href="../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a>
  
    <li><a href="../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a>
  
    <li><a href="../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a>
  
    <li><a href="../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a>
  
    <li><a href="../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a>
  
    <li><a href="../Bio/PAML/Common.html">Bio::PAML::Common</a>
  
    <li><a href="../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a>
  
    <li><a href="../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a>
  
    <li><a href="../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a>
  
    <li><a href="../Bio/PDB.html">Bio::PDB</a>
  
    <li><a href="../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a>
  
    <li><a href="../Bio/PDB/Chain.html">Bio::PDB::Chain</a>
  
    <li><a href="../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a>
  
    <li><a href="../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a>
  
    <li><a href="../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a>
  
    <li><a href="../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a>
  
    <li><a href="../Bio/PDB/DataType.html">Bio::PDB::DataType</a>
  
    <li><a href="../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a>
  
    <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a>
  
    <li><a href="../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a>
  
    <li><a href="../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a>
  
    <li><a href="../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a>
  
    <li><a href="../Bio/PDB/Model.html">Bio::PDB::Model</a>
  
    <li><a href="../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a>
  
    <li><a href="../Bio/PDB/Record.html">Bio::PDB::Record</a>
  
    <li><a href="../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a>
  
    <li><a href="../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a>
  
    <li><a href="../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a>
  
    <li><a href="../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a>
  
    <li><a href="../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a>
  
    <li><a href="../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a>
  
    <li><a href="../Bio/PDB/Residue.html">Bio::PDB::Residue</a>
  
    <li><a href="../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a>
  
    <li><a href="../Bio/PDB/Utils.html">Bio::PDB::Utils</a>
  
    <li><a href="../Bio/PROSITE.html">Bio::PROSITE</a>
  
    <li><a href="../Bio/PSORT.html">Bio::PSORT</a>
  
    <li><a href="../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a>
  
    <li><a href="../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a>
  
    <li><a href="../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a>
  
    <li><a href="../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a>
  
    <li><a href="../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a>
  
    <li><a href="../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a>
  
    <li><a href="../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a>
  
    <li><a href="../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a>
  
    <li><a href="../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a>
  
    <li><a href="../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a>
  
    <li><a href="../Bio/PTS1.html">Bio::PTS1</a>
  
    <li><a href="../Bio/PTS1/Report.html">Bio::PTS1::Report</a>
  
    <li><a href="../Bio/Pathway.html">Bio::Pathway</a>
  
    <li><a href="../Bio/Phylip.html">Bio::Phylip</a>
  
    <li><a href="../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a>
  
    <li><a href="../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a>
  
    <li><a href="../Bio/PhyloXML.html">Bio::PhyloXML</a>
  
    <li><a href="../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a>
  
    <li><a href="../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a>
  
    <li><a href="../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a>
  
    <li><a href="../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a>
  
    <li><a href="../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a>
  
    <li><a href="../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a>
  
    <li><a href="../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a>
  
    <li><a href="../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a>
  
    <li><a href="../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a>
  
    <li><a href="../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a>
  
    <li><a href="../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a>
  
    <li><a href="../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a>
  
    <li><a href="../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a>
  
    <li><a href="../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a>
  
    <li><a href="../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a>
  
    <li><a href="../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a>
  
    <li><a href="../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a>
  
    <li><a href="../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a>
  
    <li><a href="../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a>
  
    <li><a href="../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a>
  
    <li><a href="../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a>
  
    <li><a href="../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a>
  
    <li><a href="../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a>
  
    <li><a href="../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a>
  
    <li><a href="../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a>
  
    <li><a href="../Bio/Probcons.html">Bio::Probcons</a>
  
    <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/PubMed.html">Bio::PubMed</a>
  
    <li><a href="../Bio/REBASE.html">Bio::REBASE</a>
  
    <li><a href="../Bio/RefSeq.html">Bio::RefSeq</a>
  
    <li><a href="../Bio/Reference.html">Bio::Reference</a>
  
    <li><a href="../Bio/References.html">Bio::References</a>
  
    <li><a href="../Bio/Registry.html">Bio::Registry</a>
  
    <li><a href="../Bio/Registry/DB.html">Bio::Registry::DB</a>
  
    <li><a href="../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a>
  
    <li><a href="../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a>
  
    <li><a href="../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a>
  
    <li><a href="../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a>
  
    <li><a href="../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a>
  
    <li><a href="../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a>
  
    <li><a href="../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a>
  
    <li><a href="../Bio/SOFT.html">Bio::SOFT</a>
  
    <li><a href="../Bio/SOSUI.html">Bio::SOSUI</a>
  
    <li><a href="../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a>
  
    <li><a href="../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a>
  
    <li><a href="../Bio/SPTR.html">Bio::SPTR</a>
  
    <li><a href="../Bio/SQL.html">Bio::SQL</a>
  
    <li><a href="../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a>
  
    <li><a href="../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a>
  
    <li><a href="../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a>
  
    <li><a href="../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a>
  
    <li><a href="../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a>
  
    <li><a href="../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a>
  
    <li><a href="../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a>
  
    <li><a href="../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a>
  
    <li><a href="../Bio/SQL/Comment.html">Bio::SQL::Comment</a>
  
    <li><a href="../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a>
  
    <li><a href="../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a>
  
    <li><a href="../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a>
  
    <li><a href="../Bio/SQL/Location.html">Bio::SQL::Location</a>
  
    <li><a href="../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a>
  
    <li><a href="../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a>
  
    <li><a href="../Bio/SQL/Reference.html">Bio::SQL::Reference</a>
  
    <li><a href="../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a>
  
    <li><a href="../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a>
  
    <li><a href="../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a>
  
    <li><a href="../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a>
  
    <li><a href="../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a>
  
    <li><a href="../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a>
  
    <li><a href="../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a>
  
    <li><a href="../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a>
  
    <li><a href="../Bio/SQL/Term.html">Bio::SQL::Term</a>
  
    <li><a href="../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a>
  
    <li><a href="../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a>
  
    <li><a href="../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a>
  
    <li><a href="../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a>
  
    <li><a href="../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a>
  
    <li><a href="../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a>
  
    <li><a href="../Bio/Scf.html">Bio::Scf</a>
  
    <li><a href="../Bio/Sequence.html">Bio::Sequence</a>
  
    <li><a href="../Bio/Sequence/AA.html">Bio::Sequence::AA</a>
  
    <li><a href="../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a>
  
    <li><a href="../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a>
  
    <li><a href="../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a>
  
    <li><a href="../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a>
  
    <li><a href="../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a>
  
    <li><a href="../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a>
  
    <li><a href="../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a>
  
    <li><a href="../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a>
  
    <li><a href="../Bio/Sequence/Common.html">Bio::Sequence::Common</a>
  
    <li><a href="../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a>
  
    <li><a href="../Bio/Sequence/Format.html">Bio::Sequence::Format</a>
  
    <li><a href="../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a>
  
    <li><a href="../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a>
  
    <li><a href="../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a>
  
    <li><a href="../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a>
  
    <li><a href="../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a>
  
    <li><a href="../Bio/Sequence/NA.html">Bio::Sequence::NA</a>
  
    <li><a href="../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a>
  
    <li><a href="../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a>
  
    <li><a href="../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a>
  
    <li><a href="../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a>
  
    <li><a href="../Bio/Shell.html">Bio::Shell</a>
  
    <li><a href="../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a>
  
    <li><a href="../Bio/Shell/Core.html">Bio::Shell::Core</a>
  
    <li><a href="../Bio/Shell/Demo.html">Bio::Shell::Demo</a>
  
    <li><a href="../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a>
  
    <li><a href="../Bio/Shell/Irb.html">Bio::Shell::Irb</a>
  
    <li><a href="../Bio/Shell/Private.html">Bio::Shell::Private</a>
  
    <li><a href="../Bio/Shell/Script.html">Bio::Shell::Script</a>
  
    <li><a href="../Bio/Shell/Setup.html">Bio::Shell::Setup</a>
  
    <li><a href="../Bio/Shell/Web.html">Bio::Shell::Web</a>
  
    <li><a href="../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a>
  
    <li><a href="../Bio/SiRNA.html">Bio::SiRNA</a>
  
    <li><a href="../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a>
  
    <li><a href="../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a>
  
    <li><a href="../Bio/Sim4.html">Bio::Sim4</a>
  
    <li><a href="../Bio/Sim4/Report.html">Bio::Sim4::Report</a>
  
    <li><a href="../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a>
  
    <li><a href="../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a>
  
    <li><a href="../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a>
  
    <li><a href="../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a>
  
    <li><a href="../Bio/Spidey.html">Bio::Spidey</a>
  
    <li><a href="../Bio/Spidey/Report.html">Bio::Spidey::Report</a>
  
    <li><a href="../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a>
  
    <li><a href="../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a>
  
    <li><a href="../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a>
  
    <li><a href="../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a>
  
    <li><a href="../Bio/SwissProt.html">Bio::SwissProt</a>
  
    <li><a href="../Bio/TMHMM.html">Bio::TMHMM</a>
  
    <li><a href="../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a>
  
    <li><a href="../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a>
  
    <li><a href="../Bio/TRANSFAC.html">Bio::TRANSFAC</a>
  
    <li><a href="../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a>
  
    <li><a href="../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a>
  
    <li><a href="../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a>
  
    <li><a href="../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a>
  
    <li><a href="../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a>
  
    <li><a href="../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a>
  
    <li><a href="../Bio/TargetP.html">Bio::TargetP</a>
  
    <li><a href="../Bio/TargetP/Report.html">Bio::TargetP::Report</a>
  
    <li><a href="../Bio/Taxonomy.html">Bio::Taxonomy</a>
  
    <li><a href="../Bio/Tcoffee.html">Bio::Tcoffee</a>
  
    <li><a href="../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a>
  
    <li><a href="../Bio/TogoWS.html">Bio::TogoWS</a>
  
    <li><a href="../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a>
  
    <li><a href="../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a>
  
    <li><a href="../Bio/TrEMBL.html">Bio::TrEMBL</a>
  
    <li><a href="../Bio/Tree.html">Bio::Tree</a>
  
    <li><a href="../Bio/Tree/Edge.html">Bio::Tree::Edge</a>
  
    <li><a href="../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a>
  
    <li><a href="../Bio/Tree/Node.html">Bio::Tree::Node</a>
  
    <li><a href="../Bio/UniProt.html">Bio::UniProt</a>
  
    <li><a href="../BiorubyController.html">BiorubyController</a>
  
    <li><a href="../BiorubyGenerator.html">BiorubyGenerator</a>
  
    <li><a href="../BiorubyHelper.html">BiorubyHelper</a>
  
    <li><a href="../Hash.html">Hash</a>
  
    <li><a href="../Object.html">Object</a>
  
    <li><a href="../String.html">String</a>
  
  </ul>
</nav>

  </div>
</nav>

<div id="documentation">
  <h1 class="class">class Bio::Reference</h1>

  <div id="description" class="description">
    
<h1 id="label-DESCRIPTION">DESCRIPTION</h1>

<p>A class for journal reference information.</p>

<h1 id="label-USAGE">USAGE</h1>

<pre class="ruby"><span class="ruby-identifier">hash</span> = {<span class="ruby-string">'authors'</span> =<span class="ruby-operator">&gt;</span> [ <span class="ruby-string">&quot;Hoge, J.P.&quot;</span>, <span class="ruby-string">&quot;Fuga, F.B.&quot;</span> ], 
        <span class="ruby-string">'title'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;Title of the study.&quot;</span>,
        <span class="ruby-string">'journal'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;Theor. J. Hoge&quot;</span>, 
        <span class="ruby-string">'volume'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">12</span>, 
        <span class="ruby-string">'issue'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">3</span>, 
        <span class="ruby-string">'pages'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;123-145&quot;</span>,
        <span class="ruby-string">'year'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">2001</span>, 
        <span class="ruby-string">'pubmed'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">12345678</span>, 
        <span class="ruby-string">'medline'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">98765432</span>, 
        <span class="ruby-string">'abstract'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;Hoge fuga. ...&quot;</span>,
        <span class="ruby-string">'url'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;http://example.com&quot;</span>, 
        <span class="ruby-string">'mesh'</span> =<span class="ruby-operator">&gt;</span> [], 
        <span class="ruby-string">'affiliations'</span> =<span class="ruby-operator">&gt;</span> []}
<span class="ruby-identifier">ref</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Reference</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">hash</span>)

<span class="ruby-comment"># Formats in the BiBTeX style.</span>
<span class="ruby-identifier">ref</span>.<span class="ruby-identifier">format</span>(<span class="ruby-string">&quot;bibtex&quot;</span>)

<span class="ruby-comment"># Short-cut for Bio::Reference#format(&quot;bibtex&quot;)</span>
<span class="ruby-identifier">ref</span>.<span class="ruby-identifier">bibtex</span>
</pre>

  </div><!-- description -->

  
  
  
  <section id="5Buntitled-5D" class="documentation-section">
    

    

    

    
    <!-- Attributes -->
    <section id="attribute-method-details" class="method-section section">
      <h3 class="section-header">Attributes</h3>

      
      <div id="attribute-i-abstract" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">abstract</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Abstract text in <a href="../String.html">String</a>.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-affiliations" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">affiliations</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Affiliations in an Array.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-authors" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">authors</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Author names in an Array, [ Hoge, J.P., Fuga, F.B. ].</p>
        
        </div>
      </div>
      
      <div id="attribute-i-comments" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">comments</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Comments for the reference (typically Array of <a
href="../String.html">String</a>, or nil)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-doi" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">doi</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>DOI identifier (typically <a href="../String.html">String</a>, e.g.
10.1126/science.1110418)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-embl_gb_record_number" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">embl_gb_record_number</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p><a href="Sequence.html">Sequence</a> number in EMBL/GenBank records</p>
        
        </div>
      </div>
      
      <div id="attribute-i-issue" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">issue</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>issue number (typically Fixnum)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-journal" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">journal</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p><a href="../String.html">String</a> with journal name</p>
        
        </div>
      </div>
      
      <div id="attribute-i-medline" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">medline</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>medline identifier (typically Fixnum)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-mesh" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">mesh</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>MeSH terms in an Array.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-pages" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">pages</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>page range (typically <a href="../String.html">String</a>, e.g. 123-145)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-pubmed" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">pubmed</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>pubmed identifier (typically Fixnum)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-sequence_position" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">sequence_position</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>Position in a sequence that this reference refers to</p>
        
        </div>
      </div>
      
      <div id="attribute-i-title" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">title</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p><a href="../String.html">String</a> with title of the study</p>
        
        </div>
      </div>
      
      <div id="attribute-i-url" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">url</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>An URL <a href="../String.html">String</a>.</p>
        
        </div>
      </div>
      
      <div id="attribute-i-volume" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">volume</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>volume number (typically Fixnum)</p>
        
        </div>
      </div>
      
      <div id="attribute-i-year" class="method-detail">
        <div class="method-heading attribute-method-heading">
          <span class="method-name">year</span><span
            class="attribute-access-type">[R]</span>
        </div>

        <div class="method-description">
        
        <p>year of publication (typically Fixnum)</p>
        
        </div>
      </div>
      
    </section><!-- attribute-method-details -->
    

    <!-- Methods -->
    
     <section id="public-class-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Class Methods</h3>

    
      <div id="method-c-new" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">new</span><span
            class="method-args">(hash)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Create a new <a href="Reference.html">Bio::Reference</a> object from a <a
href="../Hash.html">Hash</a> of values.  Data is extracted from the values
for keys:</p>
<ul><li>
<p>authors - expected value: Array of Strings</p>
</li><li>
<p>title - expected value: <a href="../String.html">String</a></p>
</li><li>
<p>journal - expected value: <a href="../String.html">String</a></p>
</li><li>
<p>volume - expected value: Fixnum or <a href="../String.html">String</a></p>
</li><li>
<p>issue - expected value: Fixnum or <a href="../String.html">String</a></p>
</li><li>
<p>pages - expected value: <a href="../String.html">String</a></p>
</li><li>
<p>year - expected value: Fixnum or <a href="../String.html">String</a></p>
</li><li>
<p>pubmed - expected value: Fixnum or <a href="../String.html">String</a></p>
</li><li>
<p>medline - expected value: Fixnum or <a href="../String.html">String</a></p>
</li><li>
<p>abstract - expected value: <a href="../String.html">String</a></p>
</li><li>
<p>url - expected value: <a href="../String.html">String</a></p>
</li><li>
<p>mesh - expected value: Array of Strings</p>
</li><li>
<p>affiliations - expected value: Array of Strings</p>

<pre class="ruby"><span class="ruby-identifier">hash</span> = {<span class="ruby-string">'authors'</span> =<span class="ruby-operator">&gt;</span> [ <span class="ruby-string">&quot;Hoge, J.P.&quot;</span>, <span class="ruby-string">&quot;Fuga, F.B.&quot;</span> ], 
        <span class="ruby-string">'title'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;Title of the study.&quot;</span>,
        <span class="ruby-string">'journal'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;Theor. J. Hoge&quot;</span>, 
        <span class="ruby-string">'volume'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">12</span>, 
        <span class="ruby-string">'issue'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">3</span>, 
        <span class="ruby-string">'pages'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;123-145&quot;</span>,
        <span class="ruby-string">'year'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">2001</span>, 
        <span class="ruby-string">'pubmed'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">12345678</span>, 
        <span class="ruby-string">'medline'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-value">98765432</span>, 
        <span class="ruby-string">'abstract'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;Hoge fuga. ...&quot;</span>,
        <span class="ruby-string">'url'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">&quot;http://example.com&quot;</span>, 
        <span class="ruby-string">'mesh'</span> =<span class="ruby-operator">&gt;</span> [], 
        <span class="ruby-string">'affiliations'</span> =<span class="ruby-operator">&gt;</span> []}
<span class="ruby-identifier">ref</span> = <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Reference</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">hash</span>)
</pre>
</li></ul>
<hr style="height: 1px">

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(required) <em>hash</em>: <a href="../Hash.html">Hash</a></p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="Reference.html">Bio::Reference</a> object</p>
</dd></dl>
          

          
          <div class="method-source-code" id="new-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 133</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">initialize</span>(<span class="ruby-identifier">hash</span>)
  <span class="ruby-ivar">@authors</span>  = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'authors'</span>] <span class="ruby-operator">||</span> [] <span class="ruby-comment"># [ &quot;Hoge, J.P.&quot;, &quot;Fuga, F.B.&quot; ]</span>
  <span class="ruby-ivar">@title</span>    = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'title'</span>]   <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># &quot;Title of the study.&quot;</span>
  <span class="ruby-ivar">@journal</span>  = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'journal'</span>] <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># &quot;Theor. J. Hoge&quot;</span>
  <span class="ruby-ivar">@volume</span>   = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'volume'</span>]  <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># 12</span>
  <span class="ruby-ivar">@issue</span>    = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'issue'</span>]   <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># 3</span>
  <span class="ruby-ivar">@pages</span>    = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'pages'</span>]   <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># 123-145</span>
  <span class="ruby-ivar">@year</span>     = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'year'</span>]    <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># 2001</span>
  <span class="ruby-ivar">@pubmed</span>   = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'pubmed'</span>]  <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># 12345678</span>
  <span class="ruby-ivar">@medline</span>  = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'medline'</span>] <span class="ruby-operator">||</span> <span class="ruby-string">''</span> <span class="ruby-comment"># 98765432</span>
  <span class="ruby-ivar">@doi</span>      = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'doi'</span>]
  <span class="ruby-ivar">@abstract</span> = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'abstract'</span>] <span class="ruby-operator">||</span> <span class="ruby-string">''</span> 
  <span class="ruby-ivar">@url</span>      = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'url'</span>]
  <span class="ruby-ivar">@mesh</span>     = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'mesh'</span>] <span class="ruby-operator">||</span> []
  <span class="ruby-ivar">@embl_gb_record_number</span> = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'embl_gb_record_number'</span>] <span class="ruby-operator">||</span> <span class="ruby-keyword">nil</span>
  <span class="ruby-ivar">@sequence_position</span> = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'sequence_position'</span>] <span class="ruby-operator">||</span> <span class="ruby-keyword">nil</span>
  <span class="ruby-ivar">@comments</span>  = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'comments'</span>]
  <span class="ruby-ivar">@affiliations</span> = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'affiliations'</span>] <span class="ruby-operator">||</span> []
<span class="ruby-keyword">end</span></pre>
          </div><!-- new-source -->
          
        </div>

        

        
      </div><!-- new-method -->

    
    </section><!-- public-class-method-details -->
  
     <section id="public-instance-5Buntitled-5D-method-details" class="method-section section">
      <h3 class="section-header">Public Instance Methods</h3>

    
      <div id="method-i-3D-3D" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">==</span><span
            class="method-args">(other)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>If <em>other</em> is equal with the self, returns true. Otherwise, returns
false.</p>
<hr style="height: 1px">

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(required) <em>other</em>: any object</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p>true or false</p>
</dd></dl>
          

          
          <div class="method-source-code" id="3D-3D-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 159</span>
<span class="ruby-keyword">def</span> <span class="ruby-operator">==</span>(<span class="ruby-identifier">other</span>)
  <span class="ruby-keyword">return</span> <span class="ruby-keyword">true</span> <span class="ruby-keyword">if</span> <span class="ruby-keyword">super</span>(<span class="ruby-identifier">other</span>)
  <span class="ruby-keyword">return</span> <span class="ruby-keyword">false</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">other</span>.<span class="ruby-identifier">instance_of?</span>(<span class="ruby-keyword">self</span>.<span class="ruby-identifier">class</span>)
  <span class="ruby-identifier">flag</span> = <span class="ruby-keyword">false</span>
  [ <span class="ruby-value">:authors</span>, <span class="ruby-value">:title</span>, <span class="ruby-value">:journal</span>, <span class="ruby-value">:volume</span>, <span class="ruby-value">:issue</span>, <span class="ruby-value">:pages</span>,
    <span class="ruby-value">:year</span>, <span class="ruby-value">:pubmed</span>, <span class="ruby-value">:medline</span>, <span class="ruby-value">:doi</span>, <span class="ruby-value">:abstract</span>,
    <span class="ruby-value">:url</span>, <span class="ruby-value">:mesh</span>, <span class="ruby-value">:embl_gb_record_number</span>,
    <span class="ruby-value">:sequence_position</span>, <span class="ruby-value">:comments</span>, <span class="ruby-value">:affiliations</span> ].<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">m</span><span class="ruby-operator">|</span>
    <span class="ruby-keyword">begin</span>
      <span class="ruby-identifier">flag</span> = (<span class="ruby-keyword">self</span>.<span class="ruby-identifier">__send__</span>(<span class="ruby-identifier">m</span>) <span class="ruby-operator">==</span> <span class="ruby-identifier">other</span>.<span class="ruby-identifier">__send__</span>(<span class="ruby-identifier">m</span>))
    <span class="ruby-keyword">rescue</span> <span class="ruby-constant">NoMethodError</span>, <span class="ruby-constant">ArgumentError</span>, <span class="ruby-constant">NameError</span>
      <span class="ruby-identifier">flag</span> = <span class="ruby-keyword">false</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">break</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">flag</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">flag</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- 3D-3D-source -->
          
        </div>

        

        
      </div><!-- 3D-3D-method -->

    
      <div id="method-i-bibitem" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">bibitem</span><span
            class="method-args">(item = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the bibitem style</p>

<pre># ref is a Bio::Reference object
puts ref.bibitem

  \bibitem{PMID:12345678}
  Hoge, J.P., Fuga, F.B.
  Title of the study.,
  {\em Theor. J. Hoge}, 12(3):123--145, 2001.</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) <em>item</em>: label string (default:
<code>&quot;PMID:#{pubmed}&quot;</code>).</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="bibitem-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 316</span>
    <span class="ruby-keyword">def</span> <span class="ruby-identifier">bibitem</span>(<span class="ruby-identifier">item</span> = <span class="ruby-keyword">nil</span>)
      <span class="ruby-identifier">item</span>  = <span class="ruby-node">&quot;PMID:#{@pubmed}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">item</span>
      <span class="ruby-identifier">pages</span> = <span class="ruby-ivar">@pages</span>.<span class="ruby-identifier">sub</span>(<span class="ruby-string">'-'</span>, <span class="ruby-string">'--'</span>)
      <span class="ruby-keyword">return</span> <span class="ruby-string">&quot;        \\bibitem{#{item}}
        #{@authors.join(', ')}
        #{@title},
        {\\em #{@journal}}, #{@volume}(#{@issue}):#{pages}, #{@year}.
&quot;</span>.<span class="ruby-identifier">enum_for</span>(<span class="ruby-value">:each_line</span>).<span class="ruby-identifier">collect</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">line</span><span class="ruby-operator">|</span> <span class="ruby-identifier">line</span>.<span class="ruby-identifier">strip</span>}.<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;\n&quot;</span>)
    <span class="ruby-keyword">end</span></pre>
          </div><!-- bibitem-source -->
          
        </div>

        

        
      </div><!-- bibitem-method -->

    
      <div id="method-i-bibtex" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">bibtex</span><span
            class="method-args">(section = nil, label = nil, keywords = {})</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the BiBTeX style.</p>

<pre># ref is a Bio::Reference object
puts ref.bibtex

  @article{PMID:12345678,
    author  = {Hoge, J.P. and Fuga, F.B.},
    title   = {Title of the study.},
    journal = {Theor. J. Hoge},
    year    = {2001},
    volume  = {12},
    number  = {3},
    pages   = {123--145},
  }

# using a different section (e.g. &quot;book&quot;)
# (but not really configured for anything other than articles)
puts ref.bibtex(&quot;book&quot;)

  @book{PMID:12345678,
    author  = {Hoge, J.P. and Fuga, F.B.},
    title   = {Title of the study.},
    journal = {Theor. J. Hoge},
    year    = {2001},
    volume  = {12},
    number  = {3},
    pages   = {123--145},
  }</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) <em>section</em>: BiBTeX section as <a
href="../String.html">String</a></p>
</li><li>
<p>(optional) <em>label</em>: Label string cited by LaTeX documents.</p>

<pre>Default is &lt;tt&gt;&quot;PMID:#{pubmed}&quot;&lt;/tt&gt;.</pre>
</li><li>
<p>(optional) <em>keywords</em>: <a href="../Hash.html">Hash</a> of additional
keywords,</p>

<pre class="ruby"><span class="ruby-identifier">e</span>.<span class="ruby-identifier">g</span>. { <span class="ruby-string">'abstract'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">'This is abstract.'</span> }.
<span class="ruby-constant">You</span> <span class="ruby-identifier">can</span> <span class="ruby-identifier">also</span> <span class="ruby-identifier">override</span> <span class="ruby-identifier">default</span> <span class="ruby-identifier">keywords</span>.
<span class="ruby-constant">To</span> <span class="ruby-identifier">disable</span> <span class="ruby-identifier">default</span> <span class="ruby-identifier">keywords</span>, <span class="ruby-identifier">specify</span> <span class="ruby-keyword">false</span> <span class="ruby-identifier">as</span>
<span class="ruby-identifier">value</span>, <span class="ruby-identifier">e</span>.<span class="ruby-identifier">g</span>. { <span class="ruby-string">'url'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-keyword">false</span>, <span class="ruby-string">'year'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-keyword">false</span> }.
</pre>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="bibtex-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 366</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">bibtex</span>(<span class="ruby-identifier">section</span> = <span class="ruby-keyword">nil</span>, <span class="ruby-identifier">label</span> = <span class="ruby-keyword">nil</span>, <span class="ruby-identifier">keywords</span> = {})
  <span class="ruby-identifier">section</span> = <span class="ruby-string">&quot;article&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">section</span>
  <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' and '</span>, <span class="ruby-string">' and '</span>)
  <span class="ruby-identifier">thepages</span> = <span class="ruby-identifier">pages</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-operator">?</span> <span class="ruby-keyword">nil</span> <span class="ruby-operator">:</span> <span class="ruby-identifier">pages</span>.<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">%r\-/</span>, <span class="ruby-string">'--'</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-identifier">label</span> <span class="ruby-keyword">then</span>
    <span class="ruby-identifier">label</span> = <span class="ruby-node">&quot;PMID:#{pubmed}&quot;</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">theurl</span> = <span class="ruby-keyword">if</span> <span class="ruby-operator">!</span>(<span class="ruby-identifier">url</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span>) <span class="ruby-keyword">then</span>
             <span class="ruby-identifier">url</span>
           <span class="ruby-keyword">elsif</span> <span class="ruby-identifier">pmurl</span> = <span class="ruby-identifier">pubmed_url</span> <span class="ruby-keyword">and</span> <span class="ruby-operator">!</span>(<span class="ruby-identifier">pmurl</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span>) <span class="ruby-keyword">then</span>
             <span class="ruby-identifier">pmurl</span>
           <span class="ruby-keyword">else</span>
             <span class="ruby-keyword">nil</span>
           <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">hash</span> = {
    <span class="ruby-string">'author'</span>  =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">authors</span>.<span class="ruby-identifier">empty?</span>    <span class="ruby-operator">?</span> <span class="ruby-keyword">nil</span> <span class="ruby-operator">:</span> <span class="ruby-identifier">authors</span>,
    <span class="ruby-string">'title'</span>   =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">title</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-operator">?</span> <span class="ruby-keyword">nil</span> <span class="ruby-operator">:</span> <span class="ruby-identifier">title</span>,
    <span class="ruby-string">'number'</span>  =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">issue</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-operator">?</span> <span class="ruby-keyword">nil</span> <span class="ruby-operator">:</span> <span class="ruby-identifier">issue</span>,
    <span class="ruby-string">'pages'</span>   =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">thepages</span>,
    <span class="ruby-string">'url'</span>     =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">theurl</span>
  }
  <span class="ruby-identifier">keys</span> = <span class="ruby-node">%w( author title journal year volume number pages url )</span>
  <span class="ruby-identifier">keys</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">k</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">k</span>] = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">__send__</span>(<span class="ruby-identifier">k</span>.<span class="ruby-identifier">intern</span>) <span class="ruby-keyword">unless</span> <span class="ruby-identifier">hash</span>.<span class="ruby-identifier">has_key?</span>(<span class="ruby-identifier">k</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">hash</span>.<span class="ruby-identifier">merge!</span>(<span class="ruby-identifier">keywords</span>) { <span class="ruby-operator">|</span><span class="ruby-identifier">k</span>, <span class="ruby-identifier">v1</span>, <span class="ruby-identifier">v2</span><span class="ruby-operator">|</span> <span class="ruby-identifier">v2</span>.<span class="ruby-identifier">nil?</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">v1</span> <span class="ruby-operator">:</span> <span class="ruby-identifier">v2</span> }
  <span class="ruby-identifier">bib</span> = [ <span class="ruby-node">&quot;@#{section}{#{label},&quot;</span> ]
  <span class="ruby-identifier">keys</span>.<span class="ruby-identifier">concat</span>((<span class="ruby-identifier">hash</span>.<span class="ruby-identifier">keys</span> <span class="ruby-operator">-</span> <span class="ruby-identifier">keys</span>).<span class="ruby-identifier">sort</span>)
  <span class="ruby-identifier">keys</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">kw</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">ref</span> = <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">kw</span>]
    <span class="ruby-identifier">bib</span>.<span class="ruby-identifier">push</span> <span class="ruby-node">&quot;  #{kw.ljust(12)} = {#{ref}},&quot;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">ref</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">bib</span>.<span class="ruby-identifier">push</span> <span class="ruby-string">&quot;}\n&quot;</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">bib</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;\n&quot;</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- bibtex-source -->
          
        </div>

        

        
      </div><!-- bibtex-method -->

    
      <div id="method-i-cell" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">cell</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the  <a href="http://www.cell.com">CELL</a>
Press style.</p>

<pre># ref is a Bio::Reference object
puts ref.cell

  Hoge, J.P. and Fuga, F.B. (2001). Title of the study. Theor. J. Hoge 12, 123-145.</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="cell-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 560</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cell</span>
  <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' and '</span>)
  <span class="ruby-node">&quot;#{authors} (#{@year}). #{@title} #{@journal} #{@volume}, #{pages}.&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- cell-source -->
          
        </div>

        

        
      </div><!-- cell-method -->

    
      <div id="method-i-current" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">current</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the Current Biology  (<a
href="http://current-biology.com">current-biology.com</a>) style. (Same as
the Genome Biology style)</p>

<pre># ref is a Bio::Reference object
puts ref.current

  Hoge JP, Fuga FB: Title of the study. Theor J Hoge 2001, 12:123-145.</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="current-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 518</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">current</span> 
  <span class="ruby-keyword">self</span>.<span class="ruby-identifier">genome_biol</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- current-source -->
          
        </div>

        

        
      </div><!-- current-method -->

    
      <div id="method-i-embl" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">embl</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the <a href="EMBL.html">EMBL</a> style.</p>

<pre># ref is a Bio::Reference object
puts ref.embl

  RP   1-1859
  RX   PUBMED; 1907511.
  RA   Oxtoby E., Dunn M.A., Pancoro A., Hughes M.A.;
  RT   &quot;Nucleotide and derived amino acid sequence of the cyanogenic
  RT   beta-glucosidase (linamarase) from white clover (Trifolium repens L.)&quot;;
  RL   Plant Mol. Biol. 17(2):209-219(1991).</pre>
          

          
          <div class="method-source-code" id="embl-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 296</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">embl</span>
  <span class="ruby-identifier">r</span> = <span class="ruby-keyword">self</span>
  <span class="ruby-constant">Bio</span><span class="ruby-operator">::</span><span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">Format</span><span class="ruby-operator">::</span><span class="ruby-constant">NucFormatter</span><span class="ruby-operator">::</span><span class="ruby-constant">Embl</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">''</span>).<span class="ruby-identifier">instance_eval</span> {
    <span class="ruby-identifier">reference_format_embl</span>(<span class="ruby-identifier">r</span>)
  }
<span class="ruby-keyword">end</span></pre>
          </div><!-- embl-source -->
          
        </div>

        

        
      </div><!-- embl-method -->

    
      <div id="method-i-endnote" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">endnote</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the Endnote style.</p>

<pre># ref is a Bio::Reference object
puts ref.endnote

  %0 Journal Article
  %A Hoge, J.P.
  %A Fuga, F.B.
  %D 2001
  %T Title of the study.
  %J Theor. J. Hoge
  %V 12
  %N 3
  %P 123-145
  %M 12345678
  %U http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Citation&amp;list_uids=12345678
  %X Hoge fuga. ...</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="endnote-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 262</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">endnote</span>
  <span class="ruby-identifier">lines</span> = []
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-string">&quot;%0 Journal Article&quot;</span>
  <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">author</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%A #{author}&quot;</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%D #{@year}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@year</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%T #{@title}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@title</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%J #{@journal}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@journal</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%V #{@volume}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@volume</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%N #{@issue}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@issue</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%P #{@pages}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@pages</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%M #{@pubmed}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@pubmed</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">u</span> = <span class="ruby-ivar">@url</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span> <span class="ruby-operator">?</span> <span class="ruby-identifier">pubmed_url</span> <span class="ruby-operator">:</span> <span class="ruby-ivar">@url</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%U #{u}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">u</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%X #{@abstract}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@abstract</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-ivar">@mesh</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">term</span><span class="ruby-operator">|</span>
    <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%K #{term}&quot;</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;%+ #{@affiliations.join(' ')}&quot;</span> <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@affiliations</span>.<span class="ruby-identifier">empty?</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">lines</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;\n&quot;</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- endnote-source -->
          
        </div>

        

        
      </div><!-- endnote-method -->

    
      <div id="method-i-format" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">format</span><span
            class="method-args">(style = nil, *options)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Formats the reference in a given style.</p>

<p>Styles:</p>
<ol><li>
<p>nil - general</p>
</li><li>
<p>endnote - Endnote</p>
</li><li>
<p>bibitem - Bibitem (option available)</p>
</li><li>
<p>bibtex - BiBTeX (option available)</p>
</li><li>
<p>rd - rd (option available)</p>
</li><li>
<p>nature - Nature (option available)</p>
</li><li>
<p>science - Science</p>
</li><li>
<p><a href="Reference.html#method-i-genome_biol">#genome_biol</a> - Genome
Biology</p>
</li><li>
<p><a href="Reference.html#method-i-genome_res">#genome_res</a> - Genome
Research</p>
</li><li>
<p>nar - Nucleic Acids Research</p>
</li><li>
<p>current - Current Biology</p>
</li><li>
<p>trends - Trends in *</p>
</li><li>
<p>cell - Cell Press</p>
</li></ol>

<p>See individual methods for details. Basic usage is:</p>

<pre># ref is Bio::Reference object
# using simplest possible call (for general style)
puts ref.format

# output in Nature style
puts ref.format(&quot;nature&quot;)      # alternatively, puts ref.nature

# output in Nature short style (see Bio::Reference#nature)
puts ref.format(&quot;nature&quot;,true) # alternatively, puts ref.nature(true)</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) <em>style</em>: <a href="../String.html">String</a> with style
identifier</p>
</li><li>
<p>(optional) <em>options</em>: Options for styles accepting one</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="format-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 210</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">format</span>(<span class="ruby-identifier">style</span> = <span class="ruby-keyword">nil</span>, *<span class="ruby-identifier">options</span>)
  <span class="ruby-keyword">case</span> <span class="ruby-identifier">style</span>
  <span class="ruby-keyword">when</span> <span class="ruby-string">'embl'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">embl</span>
  <span class="ruby-keyword">when</span> <span class="ruby-string">'endnote'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">endnote</span>
  <span class="ruby-keyword">when</span> <span class="ruby-string">'bibitem'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">bibitem</span>(*<span class="ruby-identifier">options</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-string">'bibtex'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">bibtex</span>(*<span class="ruby-identifier">options</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-string">'rd'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">rd</span>(*<span class="ruby-identifier">options</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^nature$/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">nature</span>(*<span class="ruby-identifier">options</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^science$/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">science</span>
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^genome\s*_*biol/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">genome_biol</span>
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^genome\s*_*res/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">genome_res</span>
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^nar$/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">nar</span>
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^current/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">current</span>
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^trends/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">trends</span>
  <span class="ruby-keyword">when</span> <span class="ruby-regexp">%r^cell$/</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cell</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">general</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- format-source -->
          
        </div>

        

        
      </div><!-- format-method -->

    
      <div id="method-i-general" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">general</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in a general/generic style.</p>

<pre># ref is a Bio::Reference object
puts ref.general

  Hoge, J.P., Fuga, F.B. (2001). &quot;Title of the study.&quot; Theor. J. Hoge 12:123-145.</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="general-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 410</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">general</span>
  <span class="ruby-identifier">authors</span> = <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">', '</span>)
  <span class="ruby-node">&quot;#{authors} (#{@year}). \&quot;#{@title}\&quot; #{@journal} #{@volume}:#{@pages}.&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- general-source -->
          
        </div>

        

        
      </div><!-- general-method -->

    
      <div id="method-i-genome_biol" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">genome_biol</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the Genome Biology  (<a
href="http://genomebiology.com">genomebiology.com</a>) style.</p>

<pre># ref is a Bio::Reference object
puts ref.genome_biol

  Hoge JP, Fuga FB: Title of the study. Theor J Hoge 2001, 12:123-145.</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="genome_biol-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 503</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">genome_biol</span>
  <span class="ruby-identifier">authors</span> = <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">collect</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">name</span><span class="ruby-operator">|</span> <span class="ruby-identifier">strip_dots</span>(<span class="ruby-identifier">name</span>)}.<span class="ruby-identifier">join</span>(<span class="ruby-string">', '</span>)
  <span class="ruby-identifier">journal</span> = <span class="ruby-identifier">strip_dots</span>(<span class="ruby-ivar">@journal</span>)
  <span class="ruby-node">&quot;#{authors}: #{@title} #{journal} #{@year}, #{@volume}:#{@pages}.&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- genome_biol-source -->
          
        </div>

        

        
      </div><!-- genome_biol-method -->

    
      <div id="method-i-genome_res" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">genome_res</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the Genome Research  (<a
href="http://genome.org">genome.org</a>) style.</p>

<pre># ref is a Bio::Reference object
puts ref.genome_res

  Hoge, J.P. and Fuga, F.B. 2001.
    Title of the study. Theor. J. Hoge 12: 123-145.</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="genome_res-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 532</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">genome_res</span>
  <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' and '</span>)
  <span class="ruby-node">&quot;#{authors} #{@year}.\n  #{@title} #{@journal} #{@volume}: #{@pages}.&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- genome_res-source -->
          
        </div>

        

        
      </div><!-- genome_res-method -->

    
      <div id="method-i-nar" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">nar</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the Nucleic Acids Reseach  (<a
href="http://nar.oxfordjournals.org">nar.oxfordjournals.org</a>) style.</p>

<pre># ref is a Bio::Reference object
puts ref.nar

  Hoge, J.P. and Fuga, F.B. (2001) Title of the study. Theor. J. Hoge, 12, 123-145.</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="nar-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 546</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">nar</span>
  <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' and '</span>)
  <span class="ruby-node">&quot;#{authors} (#{@year}) #{@title} #{@journal}, #{@volume}, #{@pages}.&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- nar-source -->
          
        </div>

        

        
      </div><!-- nar-method -->

    
      <div id="method-i-nature" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">nature</span><span
            class="method-args">(short = false)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Formats in the Nature Publishing Group  (<a
href="http://www.nature.com">www.nature.com</a>) style.</p>

<pre># ref is a Bio::Reference object
puts ref.nature

  Hoge, J.P. &amp; Fuga, F.B. Title of the study. Theor. J. Hoge 12, 123-145 (2001).

# optionally, output short version
puts ref.nature(true)  # or puts ref.nature(short=true)

  Hoge, J.P. &amp; Fuga, F.B. Theor. J. Hoge 12, 123-145 (2001).</pre>
<hr style="height: 1px">

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) <em>short</em>: Boolean (default false)</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="nature-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 459</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">nature</span>(<span class="ruby-identifier">short</span> = <span class="ruby-keyword">false</span>)
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">short</span>
    <span class="ruby-keyword">if</span> <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">4</span>
      <span class="ruby-identifier">authors</span> = <span class="ruby-node">&quot;#{@authors[0]} et al.&quot;</span>
    <span class="ruby-keyword">elsif</span> <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">==</span> <span class="ruby-value">1</span>
      <span class="ruby-identifier">authors</span> = <span class="ruby-node">&quot;#{@authors[0]}&quot;</span>
    <span class="ruby-keyword">else</span>
      <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' &amp; '</span>)
    <span class="ruby-keyword">end</span>
    <span class="ruby-node">&quot;#{authors} #{@journal} #{@volume}, #{@pages} (#{@year}).&quot;</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' &amp; '</span>)
    <span class="ruby-node">&quot;#{authors} #{@title} #{@journal} #{@volume}, #{@pages} (#{@year}).&quot;</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- nature-source -->
          
        </div>

        

        
      </div><!-- nature-method -->

    
      <div id="method-i-pubmed_url" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">pubmed_url</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns a valid URL for pubmed records</p>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="pubmed_url-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 588</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">pubmed_url</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@pubmed</span>.<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">empty?</span>
    <span class="ruby-identifier">head</span> = <span class="ruby-string">&quot;http://www.ncbi.nlm.nih.gov/pubmed&quot;</span>
    <span class="ruby-keyword">return</span> <span class="ruby-node">&quot;#{head}/#{@pubmed}&quot;</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-string">''</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- pubmed_url-source -->
          
        </div>

        

        
      </div><!-- pubmed_url-method -->

    
      <div id="method-i-rd" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">rd</span><span
            class="method-args">(str = nil)</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Return reference formatted in the RD style.</p>

<pre># ref is a Bio::Reference object
puts ref.rd

  == Title of the study.

  * Hoge, J.P. and Fuga, F.B.

  * Theor. J. Hoge 2001 12:123-145 [PMID:12345678]

  Hoge fuga. ...</pre>

<p>An optional string argument can be supplied, but does nothing.</p>
<hr style="height: 1px">

<p><strong>Arguments</strong>:</p>
<ul><li>
<p>(optional) str: <a href="../String.html">String</a> (default nil)</p>
</li></ul>
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="rd-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 433</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">rd</span>(<span class="ruby-identifier">str</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-ivar">@abstract</span> <span class="ruby-operator">||=</span> <span class="ruby-identifier">str</span>
  <span class="ruby-identifier">lines</span> = []
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-string">&quot;== &quot;</span> <span class="ruby-operator">+</span> <span class="ruby-ivar">@title</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-string">&quot;* &quot;</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' and '</span>)
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-node">&quot;* #{@journal} #{@year} #{@volume}:#{@pages} [PMID:#{@pubmed}]&quot;</span>
  <span class="ruby-identifier">lines</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-ivar">@abstract</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">lines</span>.<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;\n\n&quot;</span>)
<span class="ruby-keyword">end</span></pre>
          </div><!-- rd-source -->
          
        </div>

        

        
      </div><!-- rd-method -->

    
      <div id="method-i-science" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">science</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the  <a
href="http://www.sciencemag.org">Science</a> style.</p>

<pre># ref is a Bio::Reference object
puts ref.science

  J.P. Hoge, F.B. Fuga, Theor. J. Hoge 12 123 (2001).</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="science-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 484</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">science</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">4</span>
    <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">rev_name</span>(<span class="ruby-ivar">@authors</span>[<span class="ruby-value">0</span>]) <span class="ruby-operator">+</span> <span class="ruby-string">&quot; et al.&quot;</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">authors</span> = <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">collect</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">name</span><span class="ruby-operator">|</span> <span class="ruby-identifier">rev_name</span>(<span class="ruby-identifier">name</span>)}.<span class="ruby-identifier">join</span>(<span class="ruby-string">', '</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-identifier">page_from</span>, = <span class="ruby-ivar">@pages</span>.<span class="ruby-identifier">split</span>(<span class="ruby-string">'-'</span>)
  <span class="ruby-node">&quot;#{authors}, #{@journal} #{@volume} #{page_from} (#{@year}).&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- science-source -->
          
        </div>

        

        
      </div><!-- science-method -->

    
      <div id="method-i-trends" class="method-detail ">
        
        <div class="method-heading">
          <span class="method-name">trends</span><span
            class="method-args">()</span>
          <span class="method-click-advice">click to toggle source</span>
        </div>
        

        <div class="method-description">
          
          <p>Returns reference formatted in the  <a
href="http://www.trends.com">TRENDS</a> style.</p>

<pre># ref is a Bio::Reference object
puts ref.trends

  Hoge, J.P. and Fuga, F.B. (2001) Title of the study. Theor. J. Hoge 12, 123-145</pre>
<hr style="height: 1px">
<dl class="rdoc-list note-list"><dt><strong>Returns</strong>
<dd>
<p><a href="../String.html">String</a></p>
</dd></dl>
          

          
          <div class="method-source-code" id="trends-source">
            <pre><span class="ruby-comment"># File lib/bio/reference.rb, line 574</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">trends</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">2</span>
    <span class="ruby-identifier">authors</span> = <span class="ruby-node">&quot;#{@authors[0]} et al.&quot;</span>
  <span class="ruby-keyword">elsif</span> <span class="ruby-ivar">@authors</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">==</span> <span class="ruby-value">1</span>
    <span class="ruby-identifier">authors</span> = <span class="ruby-node">&quot;#{@authors[0]}&quot;</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">authors</span> = <span class="ruby-identifier">authors_join</span>(<span class="ruby-string">' and '</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-node">&quot;#{authors} (#{@year}) #{@title} #{@journal} #{@volume}, #{@pages}&quot;</span>
<span class="ruby-keyword">end</span></pre>
          </div><!-- trends-source -->
          
        </div>

        

        
      </div><!-- trends-method -->

    
    </section><!-- public-instance-method-details -->
  
  </section><!-- 5Buntitled-5D -->

</div><!-- documentation -->


<footer id="validator-badges">
  <p><a href="http://validator.w3.org/check/referer">[Validate]</a>
  <p>Generated by <a href="https://github.com/rdoc/rdoc">RDoc</a> 3.12.
  <p>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish Rdoc Generator</a> 3.
</footer>

