<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />

  <title>Class: Bio::SPTR</title>

  <link rel="stylesheet" href="../rdoc.css" type="text/css" media="screen" />

  <script src="../js/jquery.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/thickbox-compressed.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/quicksearch.js" type="text/javascript" charset="utf-8"></script>
  <script src="../js/darkfish.js" type="text/javascript" charset="utf-8"></script>

</head>
<body id="top" class="class">

  <div id="metadata">
    <div id="home-metadata">
      <div id="home-section" class="section">
        <h3 class="section-header">
          <a href="../index.html">Home</a>
          <a href="../index.html#classes">Classes</a>
          <a href="../index.html#methods">Methods</a>
        </h3>
      </div>
    </div>

    <div id="file-metadata">
      <div id="file-list-section" class="section">
        <h3 class="section-header">In Files</h3>
        <div class="section-body">
          <ul>
          
            <li><a href="../lib/bio/db/embl/sptr_rb.html?TB_iframe=true&amp;height=550&amp;width=785"
              class="thickbox" title="lib/bio/db/embl/sptr.rb">lib/bio/db/embl/sptr.rb</a></li>
          
          </ul>
        </div>
      </div>

      
    </div>

    <div id="class-metadata">
      
      <!-- Parent Class -->
      <div id="parent-class-section" class="section">
        <h3 class="section-header">Parent</h3>
        
        <p class="link"><a href="EMBLDB.html">Bio::EMBLDB</a></p>
        
      </div>
      

      

      

      
      <!-- Method Quickref -->
      <div id="method-list-section" class="section">
        <h3 class="section-header">Methods</h3>
        <ul class="link-list">
          
          <li><a href="#method-i-aalen">#aalen</a></li>
          
          <li><a href="#method-i-aaseq">#aaseq</a></li>
          
          <li><a href="#method-i-cc">#cc</a></li>
          
          <li><a href="#method-i-dr">#dr</a></li>
          
          <li><a href="#method-i-dt">#dt</a></li>
          
          <li><a href="#method-i-embl_dr">#embl_dr</a></li>
          
          <li><a href="#method-i-entry">#entry</a></li>
          
          <li><a href="#method-i-entry_id">#entry_id</a></li>
          
          <li><a href="#method-i-entry_name">#entry_name</a></li>
          
          <li><a href="#method-i-ft">#ft</a></li>
          
          <li><a href="#method-i-gene_name">#gene_name</a></li>
          
          <li><a href="#method-i-gene_names">#gene_names</a></li>
          
          <li><a href="#method-i-gn">#gn</a></li>
          
          <li><a href="#method-i-hi">#hi</a></li>
          
          <li><a href="#method-i-id_line">#id_line</a></li>
          
          <li><a href="#method-i-molecule">#molecule</a></li>
          
          <li><a href="#method-i-molecule_type">#molecule_type</a></li>
          
          <li><a href="#method-i-oh">#oh</a></li>
          
          <li><a href="#method-i-os">#os</a></li>
          
          <li><a href="#method-i-ox">#ox</a></li>
          
          <li><a href="#method-i-protein_name">#protein_name</a></li>
          
          <li><a href="#method-i-ref">#ref</a></li>
          
          <li><a href="#method-i-references">#references</a></li>
          
          <li><a href="#method-i-seq">#seq</a></li>
          
          <li><a href="#method-i-sequence_length">#sequence_length</a></li>
          
          <li><a href="#method-i-set_RN">#set_RN</a></li>
          
          <li><a href="#method-i-sq">#sq</a></li>
          
          <li><a href="#method-i-synonyms">#synonyms</a></li>
          
        </ul>
      </div>
      

      
      <!-- Included Modules -->
      <div id="includes-section" class="section">
        <h3 class="section-header">Included Modules</h3>
        <ul class="link-list">
        
        
          <li><a class="include" href="EMBLDB/Common.html">Bio::EMBLDB::Common</a></li>
        
        
        </ul>
      </div>
      
    </div>

    <div id="project-metadata">
      
      
      <div id="fileindex-section" class="section project-section">
        <h3 class="section-header">Files</h3>
        <ul>
        
          <li class="file"><a href="../ChangeLog.html">ChangeLog</a></li>
        
          <li class="file"><a href="../KNOWN_ISSUES_rdoc.html">KNOWN_ISSUES.rdoc</a></li>
        
          <li class="file"><a href="../README_rdoc.html">README.rdoc</a></li>
        
          <li class="file"><a href="../README_DEV_rdoc.html">README_DEV.rdoc</a></li>
        
          <li class="file"><a href="../RELEASE_NOTES_rdoc.html">RELEASE_NOTES.rdoc</a></li>
        
          <li class="file"><a href="../doc/Changes-1_3_rdoc.html">Changes-1.3.rdoc</a></li>
        
          <li class="file"><a href="../doc/RELEASE_NOTES-1_4_0_rdoc.html">RELEASE_NOTES-1.4.0.rdoc</a></li>
        
          <li class="file"><a href="../doc/RELEASE_NOTES-1_4_1_rdoc.html">RELEASE_NOTES-1.4.1.rdoc</a></li>
        
        </ul>
      </div>
      

      <div id="classindex-section" class="section project-section">
        <h3 class="section-header">Class/Module Index
          <span class="search-toggle"><img src="../images/find.png"
            height="16" width="16" alt="[+]"
            title="show/hide quicksearch" /></span></h3>
        <form action="#" method="get" accept-charset="utf-8" class="initially-hidden">
        <fieldset>
          <legend>Quicksearch</legend>
          <input type="text" name="quicksearch" value=""
            class="quicksearch-field" />
        </fieldset>
        </form>

        <ul class="link-list">
        
          <li><a href="../Bio.html">Bio</a></li>
        
          <li><a href="../Bio/AAindex.html">Bio::AAindex</a></li>
        
          <li><a href="../Bio/AAindex1.html">Bio::AAindex1</a></li>
        
          <li><a href="../Bio/AAindex2.html">Bio::AAindex2</a></li>
        
          <li><a href="../Bio/Abif.html">Bio::Abif</a></li>
        
          <li><a href="../Bio/Abif/DirectoryEntry.html">Bio::Abif::DirectoryEntry</a></li>
        
          <li><a href="../Bio/Alignment.html">Bio::Alignment</a></li>
        
          <li><a href="../Bio/Alignment/ArrayExtension.html">Bio::Alignment::ArrayExtension</a></li>
        
          <li><a href="../Bio/Alignment/EnumerableExtension.html">Bio::Alignment::EnumerableExtension</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate.html">Bio::Alignment::FactoryTemplate</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOut.html">Bio::Alignment::FactoryTemplate::FileInFileOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInFileOutWithTree.html">Bio::Alignment::FactoryTemplate::FileInFileOutWithTree</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/FileInStdoutOut.html">Bio::Alignment::FactoryTemplate::FileInStdoutOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/Simple.html">Bio::Alignment::FactoryTemplate::Simple</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/StdinInFileOut.html">Bio::Alignment::FactoryTemplate::StdinInFileOut</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputStdin.html">Bio::Alignment::FactoryTemplate::WrapInputStdin</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapInputTempfile.html">Bio::Alignment::FactoryTemplate::WrapInputTempfile</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputStdout.html">Bio::Alignment::FactoryTemplate::WrapOutputStdout</a></li>
        
          <li><a href="../Bio/Alignment/FactoryTemplate/WrapOutputTempfile.html">Bio::Alignment::FactoryTemplate::WrapOutputTempfile</a></li>
        
          <li><a href="../Bio/Alignment/GAP.html">Bio::Alignment::GAP</a></li>
        
          <li><a href="../Bio/Alignment/HashExtension.html">Bio::Alignment::HashExtension</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Alignment::MultiFastaFormat</a></li>
        
          <li><a href="../Bio/Alignment/OriginalAlignment.html">Bio::Alignment::OriginalAlignment</a></li>
        
          <li><a href="../Bio/Alignment/OriginalPrivate.html">Bio::Alignment::OriginalPrivate</a></li>
        
          <li><a href="../Bio/Alignment/Output.html">Bio::Alignment::Output</a></li>
        
          <li><a href="../Bio/Alignment/PropertyMethods.html">Bio::Alignment::PropertyMethods</a></li>
        
          <li><a href="../Bio/Alignment/SequenceArray.html">Bio::Alignment::SequenceArray</a></li>
        
          <li><a href="../Bio/Alignment/SequenceHash.html">Bio::Alignment::SequenceHash</a></li>
        
          <li><a href="../Bio/Alignment/Site.html">Bio::Alignment::Site</a></li>
        
          <li><a href="../Bio/Alignment/SiteMethods.html">Bio::Alignment::SiteMethods</a></li>
        
          <li><a href="../Bio/AminoAcid.html">Bio::AminoAcid</a></li>
        
          <li><a href="../Bio/AminoAcid/Data.html">Bio::AminoAcid::Data</a></li>
        
          <li><a href="../Bio/Blast.html">Bio::Blast</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq.html">Bio::Blast::Bl2seq</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report.html">Bio::Blast::Bl2seq::Report</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/HSP.html">Bio::Blast::Bl2seq::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/Hit.html">Bio::Blast::Bl2seq::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Bl2seq/Report/Iteration.html">Bio::Blast::Bl2seq::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Default.html">Bio::Blast::Default</a></li>
        
          <li><a href="../Bio/Blast/Default/Report.html">Bio::Blast::Default::Report</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/HSP.html">Bio::Blast::Default::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/Hit.html">Bio::Blast::Default::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Default/Report/Iteration.html">Bio::Blast::Default::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Default/Report_TBlast.html">Bio::Blast::Default::Report_TBlast</a></li>
        
          <li><a href="../Bio/Blast/Fastacmd.html">Bio::Blast::Fastacmd</a></li>
        
          <li><a href="../Bio/Blast/NCBIOptions.html">Bio::Blast::NCBIOptions</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast.html">Bio::Blast::RPSBlast</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::RPSBlastSplitter</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/Report.html">Bio::Blast::RPSBlast::Report</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/RPSBlastSplitter.html">Bio::Blast::RPSBlast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Blast/RPSBlast/Report/Iteration.html">Bio::Blast::RPSBlast::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Remote.html">Bio::Blast::Remote</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::DDBJ</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ/Information.html">Bio::Blast::Remote::DDBJ::Information</a></li>
        
          <li><a href="../Bio/Blast/Remote/DDBJ.html">Bio::Blast::Remote::Ddbj</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::GenomeNet</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet/Information.html">Bio::Blast::Remote::GenomeNet::Information</a></li>
        
          <li><a href="../Bio/Blast/Remote/GenomeNet.html">Bio::Blast::Remote::Genomenet</a></li>
        
          <li><a href="../Bio/Blast/Remote/Information.html">Bio::Blast::Remote::Information</a></li>
        
          <li><a href="../Bio/Blast/Report.html">Bio::Blast::Report</a></li>
        
          <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::BlastXmlSplitter</a></li>
        
          <li><a href="../Bio/Blast/Report/BlastXmlSplitter.html">Bio::Blast::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Blast/Report/Hit.html">Bio::Blast::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/Report/Hsp.html">Bio::Blast::Report::Hsp</a></li>
        
          <li><a href="../Bio/Blast/Report/Iteration.html">Bio::Blast::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/Report_tab.html">Bio::Blast::Report_tab</a></li>
        
          <li><a href="../Bio/Blast/WU.html">Bio::Blast::WU</a></li>
        
          <li><a href="../Bio/Blast/WU/Report.html">Bio::Blast::WU::Report</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/HSP.html">Bio::Blast::WU::Report::HSP</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/Hit.html">Bio::Blast::WU::Report::Hit</a></li>
        
          <li><a href="../Bio/Blast/WU/Report/Iteration.html">Bio::Blast::WU::Report::Iteration</a></li>
        
          <li><a href="../Bio/Blast/WU/Report_TBlast.html">Bio::Blast::WU::Report_TBlast</a></li>
        
          <li><a href="../Bio/Blat.html">Bio::Blat</a></li>
        
          <li><a href="../Bio/Blat/Report.html">Bio::Blat::Report</a></li>
        
          <li><a href="../Bio/Blat/Report/Hit.html">Bio::Blat::Report::Hit</a></li>
        
          <li><a href="../Bio/Blat/Report/SegmentPair.html">Bio::Blat::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Blat/Report/SeqDesc.html">Bio::Blat::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/ClustalW.html">Bio::ClustalW</a></li>
        
          <li><a href="../Bio/ClustalW/Report.html">Bio::ClustalW::Report</a></li>
        
          <li><a href="../Bio/CodonTable.html">Bio::CodonTable</a></li>
        
          <li><a href="../Bio/ColorScheme.html">Bio::ColorScheme</a></li>
        
          <li><a href="../Bio/Command.html">Bio::Command</a></li>
        
          <li><a href="../Bio/Command/Tmpdir.html">Bio::Command::Tmpdir</a></li>
        
          <li><a href="../Bio/ContingencyTable.html">Bio::ContingencyTable</a></li>
        
          <li><a href="../Bio/DAS.html">Bio::DAS</a></li>
        
          <li><a href="../Bio/DAS/DNA.html">Bio::DAS::DNA</a></li>
        
          <li><a href="../Bio/DAS/DSN.html">Bio::DAS::DSN</a></li>
        
          <li><a href="../Bio/DAS/ENTRY_POINT.html">Bio::DAS::ENTRY_POINT</a></li>
        
          <li><a href="../Bio/DAS/FEATURE.html">Bio::DAS::FEATURE</a></li>
        
          <li><a href="../Bio/DAS/GFF.html">Bio::DAS::GFF</a></li>
        
          <li><a href="../Bio/DAS/GROUP.html">Bio::DAS::GROUP</a></li>
        
          <li><a href="../Bio/DAS/LINK.html">Bio::DAS::LINK</a></li>
        
          <li><a href="../Bio/DAS/SEGMENT.html">Bio::DAS::SEGMENT</a></li>
        
          <li><a href="../Bio/DAS/SEQUENCE.html">Bio::DAS::SEQUENCE</a></li>
        
          <li><a href="../Bio/DAS/TARGET.html">Bio::DAS::TARGET</a></li>
        
          <li><a href="../Bio/DAS/TYPE.html">Bio::DAS::TYPE</a></li>
        
          <li><a href="../Bio/DAS/TYPES.html">Bio::DAS::TYPES</a></li>
        
          <li><a href="../Bio/DB.html">Bio::DB</a></li>
        
          <li><a href="../Bio/DBGET.html">Bio::DBGET</a></li>
        
          <li><a href="../Bio/DDBJ.html">Bio::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/REST.html">Bio::DDBJ::REST</a></li>
        
          <li><a href="../Bio/DDBJ/REST/Blast.html">Bio::DDBJ::REST::Blast</a></li>
        
          <li><a href="../Bio/DDBJ/REST/ClustalW.html">Bio::DDBJ::REST::ClustalW</a></li>
        
          <li><a href="../Bio/DDBJ/REST/DDBJ.html">Bio::DDBJ::REST::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/REST/Mafft.html">Bio::DDBJ::REST::Mafft</a></li>
        
          <li><a href="../Bio/DDBJ/REST/RequestManager.html">Bio::DDBJ::REST::RequestManager</a></li>
        
          <li><a href="../Bio/DDBJ/REST/WABItemplate.html">Bio::DDBJ::REST::WABItemplate</a></li>
        
          <li><a href="../Bio/DDBJ/XML.html">Bio::DDBJ::XML</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Blast.html">Bio::DDBJ::XML::Blast</a></li>
        
          <li><a href="../Bio/DDBJ/XML/ClustalW.html">Bio::DDBJ::XML::ClustalW</a></li>
        
          <li><a href="../Bio/DDBJ/XML/DDBJ.html">Bio::DDBJ::XML::DDBJ</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Fasta.html">Bio::DDBJ::XML::Fasta</a></li>
        
          <li><a href="../Bio/DDBJ/XML/GetEntry.html">Bio::DDBJ::XML::GetEntry</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Gib.html">Bio::DDBJ::XML::Gib</a></li>
        
          <li><a href="../Bio/DDBJ/XML/Gtop.html">Bio::DDBJ::XML::Gtop</a></li>
        
          <li><a href="../Bio/DDBJ/XML/PML.html">Bio::DDBJ::XML::PML</a></li>
        
          <li><a href="../Bio/DDBJ/XML/RequestManager.html">Bio::DDBJ::XML::RequestManager</a></li>
        
          <li><a href="../Bio/DDBJ/XML/RequestManager/REST.html">Bio::DDBJ::XML::RequestManager::REST</a></li>
        
          <li><a href="../Bio/DDBJ/XML/SRS.html">Bio::DDBJ::XML::SRS</a></li>
        
          <li><a href="../Bio/DDBJ/XML/TxSearch.html">Bio::DDBJ::XML::TxSearch</a></li>
        
          <li><a href="../Bio/EBI.html">Bio::EBI</a></li>
        
          <li><a href="../Bio/EBI/SOAP.html">Bio::EBI::SOAP</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Citation.html">Bio::EBI::SOAP::Citation</a></li>
        
          <li><a href="../Bio/EBI/SOAP/ClustalW.html">Bio::EBI::SOAP::ClustalW</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Dbfetch.html">Bio::EBI::SOAP::Dbfetch</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Emboss.html">Bio::EBI::SOAP::Emboss</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Fasta.html">Bio::EBI::SOAP::Fasta</a></li>
        
          <li><a href="../Bio/EBI/SOAP/InterProScan.html">Bio::EBI::SOAP::InterProScan</a></li>
        
          <li><a href="../Bio/EBI/SOAP/MPsrch.html">Bio::EBI::SOAP::MPsrch</a></li>
        
          <li><a href="../Bio/EBI/SOAP/MSD.html">Bio::EBI::SOAP::MSD</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Muscle.html">Bio::EBI::SOAP::Muscle</a></li>
        
          <li><a href="../Bio/EBI/SOAP/Ontology.html">Bio::EBI::SOAP::Ontology</a></li>
        
          <li><a href="../Bio/EBI/SOAP/ScanPS.html">Bio::EBI::SOAP::ScanPS</a></li>
        
          <li><a href="../Bio/EBI/SOAP/TCoffee.html">Bio::EBI::SOAP::TCoffee</a></li>
        
          <li><a href="../Bio/EBI/SOAP/WUBlast.html">Bio::EBI::SOAP::WUBlast</a></li>
        
          <li><a href="../Bio/EMBL.html">Bio::EMBL</a></li>
        
          <li><a href="../Bio/EMBLDB.html">Bio::EMBLDB</a></li>
        
          <li><a href="../Bio/EMBLDB/Common.html">Bio::EMBLDB::Common</a></li>
        
          <li><a href="../Bio/EMBOSS.html">Bio::EMBOSS</a></li>
        
          <li><a href="../Bio/Ensembl.html">Bio::Ensembl</a></li>
        
          <li><a href="../Bio/Ensembl/Base.html">Bio::Ensembl::Base</a></li>
        
          <li><a href="../Bio/Ensembl/Human.html">Bio::Ensembl::Human</a></li>
        
          <li><a href="../Bio/Ensembl/Mouse.html">Bio::Ensembl::Mouse</a></li>
        
          <li><a href="../Bio/FANTOM.html">Bio::FANTOM</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML.html">Bio::FANTOM::MaXML</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotation.html">Bio::FANTOM::MaXML::Annotation</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotation/DataSrc.html">Bio::FANTOM::MaXML::Annotation::DataSrc</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Annotations.html">Bio::FANTOM::MaXML::Annotations</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Cluster.html">Bio::FANTOM::MaXML::Cluster</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Sequence.html">Bio::FANTOM::MaXML::Sequence</a></li>
        
          <li><a href="../Bio/FANTOM/MaXML/Sequences.html">Bio::FANTOM::MaXML::Sequences</a></li>
        
          <li><a href="../Bio/Fasta.html">Bio::Fasta</a></li>
        
          <li><a href="../Bio/Fasta/Report.html">Bio::Fasta::Report</a></li>
        
          <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FLATFILE_SPLITTER</a></li>
        
          <li><a href="../Bio/Fasta/Report/FastaFormat10Splitter.html">Bio::Fasta::Report::FastaFormat10Splitter</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit.html">Bio::Fasta::Report::Hit</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit/Query.html">Bio::Fasta::Report::Hit::Query</a></li>
        
          <li><a href="../Bio/Fasta/Report/Hit/Target.html">Bio::Fasta::Report::Hit::Target</a></li>
        
          <li><a href="../Bio/Fasta/Report/Program.html">Bio::Fasta::Report::Program</a></li>
        
          <li><a href="../Bio/FastaDefline.html">Bio::FastaDefline</a></li>
        
          <li><a href="../Bio/FastaFormat.html">Bio::FastaFormat</a></li>
        
          <li><a href="../Bio/FastaNumericFormat.html">Bio::FastaNumericFormat</a></li>
        
          <li><a href="../Bio/Fastq.html">Bio::Fastq</a></li>
        
          <li><a href="../Bio/Fastq/Error.html">Bio::Fastq::Error</a></li>
        
          <li><a href="../Bio/Fastq/Error/Diff_ids.html">Bio::Fastq::Error::Diff_ids</a></li>
        
          <li><a href="../Bio/Fastq/Error/Long_qual.html">Bio::Fastq::Error::Long_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_atmark.html">Bio::Fastq::Error::No_atmark</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_ids.html">Bio::Fastq::Error::No_ids</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_qual.html">Bio::Fastq::Error::No_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/No_seq.html">Bio::Fastq::Error::No_seq</a></li>
        
          <li><a href="../Bio/Fastq/Error/Qual_char.html">Bio::Fastq::Error::Qual_char</a></li>
        
          <li><a href="../Bio/Fastq/Error/Qual_range.html">Bio::Fastq::Error::Qual_range</a></li>
        
          <li><a href="../Bio/Fastq/Error/Seq_char.html">Bio::Fastq::Error::Seq_char</a></li>
        
          <li><a href="../Bio/Fastq/Error/Short_qual.html">Bio::Fastq::Error::Short_qual</a></li>
        
          <li><a href="../Bio/Fastq/Error/Skipped_unformatted_lines.html">Bio::Fastq::Error::Skipped_unformatted_lines</a></li>
        
          <li><a href="../Bio/Fastq/FormatData.html">Bio::Fastq::FormatData</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_ILLUMINA.html">Bio::Fastq::FormatData::FASTQ_ILLUMINA</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_SANGER.html">Bio::Fastq::FormatData::FASTQ_SANGER</a></li>
        
          <li><a href="../Bio/Fastq/FormatData/FASTQ_SOLEXA.html">Bio::Fastq::FormatData::FASTQ_SOLEXA</a></li>
        
          <li><a href="../Bio/Feature.html">Bio::Feature</a></li>
        
          <li><a href="../Bio/Feature/Qualifier.html">Bio::Feature::Qualifier</a></li>
        
          <li><a href="../Bio/Features.html">Bio::Features</a></li>
        
          <li><a href="../Bio/Fetch.html">Bio::Fetch</a></li>
        
          <li><a href="../Bio/FlatFile.html">Bio::FlatFile</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect.html">Bio::FlatFile::AutoDetect</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleDebug.html">Bio::FlatFile::AutoDetect::RuleDebug</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleProc.html">Bio::FlatFile::AutoDetect::RuleProc</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp.html">Bio::FlatFile::AutoDetect::RuleRegexp</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleRegexp2.html">Bio::FlatFile::AutoDetect::RuleRegexp2</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleSpecial.html">Bio::FlatFile::AutoDetect::RuleSpecial</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RuleTemplate.html">Bio::FlatFile::AutoDetect::RuleTemplate</a></li>
        
          <li><a href="../Bio/FlatFile/AutoDetect/RulesArray.html">Bio::FlatFile::AutoDetect::RulesArray</a></li>
        
          <li><a href="../Bio/FlatFile/BufferedInputStream.html">Bio::FlatFile::BufferedInputStream</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter.html">Bio::FlatFile::Splitter</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/Default.html">Bio::FlatFile::Splitter::Default</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/LineOriented.html">Bio::FlatFile::Splitter::LineOriented</a></li>
        
          <li><a href="../Bio/FlatFile/Splitter/Template.html">Bio::FlatFile::Splitter::Template</a></li>
        
          <li><a href="../Bio/FlatFile/UnknownDataFormatError.html">Bio::FlatFile::UnknownDataFormatError</a></li>
        
          <li><a href="../Bio/FlatFileIndex.html">Bio::FlatFileIndex</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1.html">Bio::FlatFileIndex::BDB_1</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/BDBMappingFile.html">Bio::FlatFileIndex::BDB_1::BDBMappingFile</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/PrimaryNameSpace.html">Bio::FlatFileIndex::BDB_1::PrimaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDB_1/SecondaryNameSpace.html">Bio::FlatFileIndex::BDB_1::SecondaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDBdefault.html">Bio::FlatFileIndex::BDBdefault</a></li>
        
          <li><a href="../Bio/FlatFileIndex/BDBwrapper.html">Bio::FlatFileIndex::BDBwrapper</a></li>
        
          <li><a href="../Bio/FlatFileIndex/DEBUG.html">Bio::FlatFileIndex::DEBUG</a></li>
        
          <li><a href="../Bio/FlatFileIndex/DataBank.html">Bio::FlatFileIndex::DataBank</a></li>
        
          <li><a href="../Bio/FlatFileIndex/FileID.html">Bio::FlatFileIndex::FileID</a></li>
        
          <li><a href="../Bio/FlatFileIndex/FileIDs.html">Bio::FlatFileIndex::FileIDs</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1.html">Bio::FlatFileIndex::Flat_1</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/FlatMappingFile.html">Bio::FlatFileIndex::Flat_1::FlatMappingFile</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/PrimaryNameSpace.html">Bio::FlatFileIndex::Flat_1::PrimaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/Record.html">Bio::FlatFileIndex::Flat_1::Record</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Flat_1/SecondaryNameSpace.html">Bio::FlatFileIndex::Flat_1::SecondaryNameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer.html">Bio::FlatFileIndex::Indexer</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/NameSpace.html">Bio::FlatFileIndex::Indexer::NameSpace</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/NameSpaces.html">Bio::FlatFileIndex::Indexer::NameSpaces</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser.html">Bio::FlatFileIndex::Indexer::Parser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/BlastDefaultParser.html">Bio::FlatFileIndex::Indexer::Parser::BlastDefaultParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/EMBLParser.html">Bio::FlatFileIndex::Indexer::Parser::EMBLParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/FastaFormatParser.html">Bio::FlatFileIndex::Indexer::Parser::FastaFormatParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenBankParser.html">Bio::FlatFileIndex::Indexer::Parser::GenBankParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/GenPeptParser.html">Bio::FlatFileIndex::Indexer::Parser::GenPeptParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLClusterParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLClusterParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/MaXMLSequenceParser.html">Bio::FlatFileIndex::Indexer::Parser::MaXMLSequenceParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/PDBChemicalComponentParser.html">Bio::FlatFileIndex::Indexer::Parser::PDBChemicalComponentParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/SPTRParser.html">Bio::FlatFileIndex::Indexer::Parser::SPTRParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Indexer/Parser/TemplateParser.html">Bio::FlatFileIndex::Indexer::Parser::TemplateParser</a></li>
        
          <li><a href="../Bio/FlatFileIndex/NameSpaces.html">Bio::FlatFileIndex::NameSpaces</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Results.html">Bio::FlatFileIndex::Results</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Template.html">Bio::FlatFileIndex::Template</a></li>
        
          <li><a href="../Bio/FlatFileIndex/Template/NameSpace.html">Bio::FlatFileIndex::Template::NameSpace</a></li>
        
          <li><a href="../Bio/GCG.html">Bio::GCG</a></li>
        
          <li><a href="../Bio/GCG/Msf.html">Bio::GCG::Msf</a></li>
        
          <li><a href="../Bio/GCG/Seq.html">Bio::GCG::Seq</a></li>
        
          <li><a href="../Bio/GFF.html">Bio::GFF</a></li>
        
          <li><a href="../Bio/GFF/GFF2.html">Bio::GFF::GFF2</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Escape.html">Bio::GFF::GFF2::Escape</a></li>
        
          <li><a href="../Bio/GFF/GFF2/MetaData.html">Bio::GFF::GFF2::MetaData</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Record.html">Bio::GFF::GFF2::Record</a></li>
        
          <li><a href="../Bio/GFF/GFF2/Record/Value.html">Bio::GFF::GFF2::Record::Value</a></li>
        
          <li><a href="../Bio/GFF/GFF3.html">Bio::GFF::GFF3</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Escape.html">Bio::GFF::GFF3::Escape</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record.html">Bio::GFF::GFF3::Record</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Gap.html">Bio::GFF::GFF3::Record::Gap</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Gap/Code.html">Bio::GFF::GFF3::Record::Gap::Code</a></li>
        
          <li><a href="../Bio/GFF/GFF3/Record/Target.html">Bio::GFF::GFF3::Record::Target</a></li>
        
          <li><a href="../Bio/GFF/GFF3/RecordBoundary.html">Bio::GFF::GFF3::RecordBoundary</a></li>
        
          <li><a href="../Bio/GFF/GFF3/SequenceRegion.html">Bio::GFF::GFF3::SequenceRegion</a></li>
        
          <li><a href="../Bio/GFF/Record.html">Bio::GFF::Record</a></li>
        
          <li><a href="../Bio/GO.html">Bio::GO</a></li>
        
          <li><a href="../Bio/GO/External2go.html">Bio::GO::External2go</a></li>
        
          <li><a href="../Bio/GO/GeneAssociation.html">Bio::GO::GeneAssociation</a></li>
        
          <li><a href="../Bio/GO/Ontology.html">Bio::GO::Ontology</a></li>
        
          <li><a href="../Bio/GenBank.html">Bio::GenBank</a></li>
        
          <li><a href="../Bio/GenBank/Locus.html">Bio::GenBank::Locus</a></li>
        
          <li><a href="../Bio/GenPept.html">Bio::GenPept</a></li>
        
          <li><a href="../Bio/GenPept/Locus.html">Bio::GenPept::Locus</a></li>
        
          <li><a href="../Bio/Genscan.html">Bio::Genscan</a></li>
        
          <li><a href="../Bio/Genscan/Report.html">Bio::Genscan::Report</a></li>
        
          <li><a href="../Bio/Genscan/Report/Exon.html">Bio::Genscan::Report::Exon</a></li>
        
          <li><a href="../Bio/Genscan/Report/Gene.html">Bio::Genscan::Report::Gene</a></li>
        
          <li><a href="../Bio/HGC.html">Bio::HGC</a></li>
        
          <li><a href="../Bio/HGC/HiGet.html">Bio::HGC::HiGet</a></li>
        
          <li><a href="../Bio/HMMER.html">Bio::HMMER</a></li>
        
          <li><a href="../Bio/HMMER/Report.html">Bio::HMMER::Report</a></li>
        
          <li><a href="../Bio/HMMER/Report/Hit.html">Bio::HMMER::Report::Hit</a></li>
        
          <li><a href="../Bio/HMMER/Report/Hsp.html">Bio::HMMER::Report::Hsp</a></li>
        
          <li><a href="../Bio/Hinv.html">Bio::Hinv</a></li>
        
          <li><a href="../Bio/Hinv/Acc2hit.html">Bio::Hinv::Acc2hit</a></li>
        
          <li><a href="../Bio/Hinv/Common.html">Bio::Hinv::Common</a></li>
        
          <li><a href="../Bio/Hinv/Hit2acc.html">Bio::Hinv::Hit2acc</a></li>
        
          <li><a href="../Bio/Hinv/HitCnt.html">Bio::Hinv::HitCnt</a></li>
        
          <li><a href="../Bio/Hinv/HitDefinition.html">Bio::Hinv::HitDefinition</a></li>
        
          <li><a href="../Bio/Hinv/HitPubmedId.html">Bio::Hinv::HitPubmedId</a></li>
        
          <li><a href="../Bio/Hinv/HitXML.html">Bio::Hinv::HitXML</a></li>
        
          <li><a href="../Bio/Hinv/Hix2hit.html">Bio::Hinv::Hix2hit</a></li>
        
          <li><a href="../Bio/Hinv/HixCnt.html">Bio::Hinv::HixCnt</a></li>
        
          <li><a href="../Bio/Hinv/HixRepresent.html">Bio::Hinv::HixRepresent</a></li>
        
          <li><a href="../Bio/Hinv/HixXML.html">Bio::Hinv::HixXML</a></li>
        
          <li><a href="../Bio/Hinv/IdSearch.html">Bio::Hinv::IdSearch</a></li>
        
          <li><a href="../Bio/Hinv/KeywordSearch.html">Bio::Hinv::KeywordSearch</a></li>
        
          <li><a href="../Bio/Iprscan.html">Bio::Iprscan</a></li>
        
          <li><a href="../Bio/Iprscan/Report.html">Bio::Iprscan::Report</a></li>
        
          <li><a href="../Bio/Iprscan/Report/Match.html">Bio::Iprscan::Report::Match</a></li>
        
          <li><a href="../Bio/KEGG.html">Bio::KEGG</a></li>
        
          <li><a href="../Bio/KEGG/API.html">Bio::KEGG::API</a></li>
        
          <li><a href="../Bio/KEGG/BRITE.html">Bio::KEGG::BRITE</a></li>
        
          <li><a href="../Bio/KEGG/COMPOUND.html">Bio::KEGG::COMPOUND</a></li>
        
          <li><a href="../Bio/KEGG/Common.html">Bio::KEGG::Common</a></li>
        
          <li><a href="../Bio/KEGG/Common/DblinksAsHash.html">Bio::KEGG::Common::DblinksAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/GenesAsHash.html">Bio::KEGG::Common::GenesAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/ModulesAsHash.html">Bio::KEGG::Common::ModulesAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/OrthologsAsHash.html">Bio::KEGG::Common::OrthologsAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/PathwaysAsHash.html">Bio::KEGG::Common::PathwaysAsHash</a></li>
        
          <li><a href="../Bio/KEGG/Common/References.html">Bio::KEGG::Common::References</a></li>
        
          <li><a href="../Bio/KEGG/Common/StringsAsHash.html">Bio::KEGG::Common::StringsAsHash</a></li>
        
          <li><a href="../Bio/KEGG/DRUG.html">Bio::KEGG::DRUG</a></li>
        
          <li><a href="../Bio/KEGG/ENZYME.html">Bio::KEGG::ENZYME</a></li>
        
          <li><a href="../Bio/KEGG/EXPRESSION.html">Bio::KEGG::EXPRESSION</a></li>
        
          <li><a href="../Bio/KEGG/GENES.html">Bio::KEGG::GENES</a></li>
        
          <li><a href="../Bio/KEGG/GENOME.html">Bio::KEGG::GENOME</a></li>
        
          <li><a href="../Bio/KEGG/GLYCAN.html">Bio::KEGG::GLYCAN</a></li>
        
          <li><a href="../Bio/KEGG/KGML.html">Bio::KEGG::KGML</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Entry.html">Bio::KEGG::KGML::Entry</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Reaction.html">Bio::KEGG::KGML::Reaction</a></li>
        
          <li><a href="../Bio/KEGG/KGML/Relation.html">Bio::KEGG::KGML::Relation</a></li>
        
          <li><a href="../Bio/KEGG/Keggtab.html">Bio::KEGG::Keggtab</a></li>
        
          <li><a href="../Bio/KEGG/Keggtab/DB.html">Bio::KEGG::Keggtab::DB</a></li>
        
          <li><a href="../Bio/KEGG/MODULE.html">Bio::KEGG::MODULE</a></li>
        
          <li><a href="../Bio/KEGG/ORTHOLOGY.html">Bio::KEGG::ORTHOLOGY</a></li>
        
          <li><a href="../Bio/KEGG/PATHWAY.html">Bio::KEGG::PATHWAY</a></li>
        
          <li><a href="../Bio/KEGG/REACTION.html">Bio::KEGG::REACTION</a></li>
        
          <li><a href="../Bio/KEGG/Taxonomy.html">Bio::KEGG::Taxonomy</a></li>
        
          <li><a href="../Bio/KEGGDB.html">Bio::KEGGDB</a></li>
        
          <li><a href="../Bio/LITDB.html">Bio::LITDB</a></li>
        
          <li><a href="../Bio/Lasergene.html">Bio::Lasergene</a></li>
        
          <li><a href="../Bio/Location.html">Bio::Location</a></li>
        
          <li><a href="../Bio/Locations.html">Bio::Locations</a></li>
        
          <li><a href="../Bio/MAFFT.html">Bio::MAFFT</a></li>
        
          <li><a href="../Bio/MAFFT/Report.html">Bio::MAFFT::Report</a></li>
        
          <li><a href="../Bio/MEDLINE.html">Bio::MEDLINE</a></li>
        
          <li><a href="../Bio/Map.html">Bio::Map</a></li>
        
          <li><a href="../Bio/Map/ActsLikeMap.html">Bio::Map::ActsLikeMap</a></li>
        
          <li><a href="../Bio/Map/ActsLikeMarker.html">Bio::Map::ActsLikeMarker</a></li>
        
          <li><a href="../Bio/Map/Mapping.html">Bio::Map::Mapping</a></li>
        
          <li><a href="../Bio/Map/Marker.html">Bio::Map::Marker</a></li>
        
          <li><a href="../Bio/Map/SimpleMap.html">Bio::Map::SimpleMap</a></li>
        
          <li><a href="../Bio/Meme.html">Bio::Meme</a></li>
        
          <li><a href="../Bio/Meme/Mast.html">Bio::Meme::Mast</a></li>
        
          <li><a href="../Bio/Meme/Mast/Report.html">Bio::Meme::Mast::Report</a></li>
        
          <li><a href="../Bio/Meme/Motif.html">Bio::Meme::Motif</a></li>
        
          <li><a href="../Bio/Muscle.html">Bio::Muscle</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Muscle::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/NBRF.html">Bio::NBRF</a></li>
        
          <li><a href="../Bio/NCBI.html">Bio::NCBI</a></li>
        
          <li><a href="../Bio/NCBI/REST.html">Bio::NCBI::REST</a></li>
        
          <li><a href="../Bio/NCBI/REST/EFetch.html">Bio::NCBI::REST::EFetch</a></li>
        
          <li><a href="../Bio/NCBI/REST/EFetch/Methods.html">Bio::NCBI::REST::EFetch::Methods</a></li>
        
          <li><a href="../Bio/NCBI/REST/ESearch.html">Bio::NCBI::REST::ESearch</a></li>
        
          <li><a href="../Bio/NCBI/REST/ESearch/Methods.html">Bio::NCBI::REST::ESearch::Methods</a></li>
        
          <li><a href="../Bio/NCBI/SOAP.html">Bio::NCBI::SOAP</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EFetch.html">Bio::NCBI::SOAP::EFetch</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EFetchLite.html">Bio::NCBI::SOAP::EFetchLite</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EUtils.html">Bio::NCBI::SOAP::EUtils</a></li>
        
          <li><a href="../Bio/NCBI/SOAP/EUtilsLite.html">Bio::NCBI::SOAP::EUtilsLite</a></li>
        
          <li><a href="../Bio/NCBIDB.html">Bio::NCBIDB</a></li>
        
          <li><a href="../Bio/NCBIDB/Common.html">Bio::NCBIDB::Common</a></li>
        
          <li><a href="../Bio/Newick.html">Bio::Newick</a></li>
        
          <li><a href="../Bio/Newick/ParseError.html">Bio::Newick::ParseError</a></li>
        
          <li><a href="../Bio/Nexus.html">Bio::Nexus</a></li>
        
          <li><a href="../Bio/Nexus/CharactersBlock.html">Bio::Nexus::CharactersBlock</a></li>
        
          <li><a href="../Bio/Nexus/DataBlock.html">Bio::Nexus::DataBlock</a></li>
        
          <li><a href="../Bio/Nexus/DistancesBlock.html">Bio::Nexus::DistancesBlock</a></li>
        
          <li><a href="../Bio/Nexus/GenericBlock.html">Bio::Nexus::GenericBlock</a></li>
        
          <li><a href="../Bio/Nexus/NexusMatrix.html">Bio::Nexus::NexusMatrix</a></li>
        
          <li><a href="../Bio/Nexus/NexusMatrix/NexusMatrixError.html">Bio::Nexus::NexusMatrix::NexusMatrixError</a></li>
        
          <li><a href="../Bio/Nexus/NexusParseError.html">Bio::Nexus::NexusParseError</a></li>
        
          <li><a href="../Bio/Nexus/TaxaBlock.html">Bio::Nexus::TaxaBlock</a></li>
        
          <li><a href="../Bio/Nexus/TreesBlock.html">Bio::Nexus::TreesBlock</a></li>
        
          <li><a href="../Bio/Nexus/Util.html">Bio::Nexus::Util</a></li>
        
          <li><a href="../Bio/NucleicAcid.html">Bio::NucleicAcid</a></li>
        
          <li><a href="../Bio/NucleicAcid/Data.html">Bio::NucleicAcid::Data</a></li>
        
          <li><a href="../Bio/PAML.html">Bio::PAML</a></li>
        
          <li><a href="../Bio/PAML/Baseml.html">Bio::PAML::Baseml</a></li>
        
          <li><a href="../Bio/PAML/Baseml/Report.html">Bio::PAML::Baseml::Report</a></li>
        
          <li><a href="../Bio/PAML/Codeml.html">Bio::PAML::Codeml</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Model.html">Bio::PAML::Codeml::Model</a></li>
        
          <li><a href="../Bio/PAML/Codeml/PositiveSite.html">Bio::PAML::Codeml::PositiveSite</a></li>
        
          <li><a href="../Bio/PAML/Codeml/PositiveSites.html">Bio::PAML::Codeml::PositiveSites</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Rates.html">Bio::PAML::Codeml::Rates</a></li>
        
          <li><a href="../Bio/PAML/Codeml/Report.html">Bio::PAML::Codeml::Report</a></li>
        
          <li><a href="../Bio/PAML/Codeml/ReportError.html">Bio::PAML::Codeml::ReportError</a></li>
        
          <li><a href="../Bio/PAML/Codeml/ReportSingle.html">Bio::PAML::Codeml::ReportSingle</a></li>
        
          <li><a href="../Bio/PAML/Common.html">Bio::PAML::Common</a></li>
        
          <li><a href="../Bio/PAML/Common/Report.html">Bio::PAML::Common::Report</a></li>
        
          <li><a href="../Bio/PAML/Yn00.html">Bio::PAML::Yn00</a></li>
        
          <li><a href="../Bio/PAML/Yn00/Report.html">Bio::PAML::Yn00::Report</a></li>
        
          <li><a href="../Bio/PDB.html">Bio::PDB</a></li>
        
          <li><a href="../Bio/PDB/AtomFinder.html">Bio::PDB::AtomFinder</a></li>
        
          <li><a href="../Bio/PDB/Chain.html">Bio::PDB::Chain</a></li>
        
          <li><a href="../Bio/PDB/ChainFinder.html">Bio::PDB::ChainFinder</a></li>
        
          <li><a href="../Bio/PDB/ChemicalComponent.html">Bio::PDB::ChemicalComponent</a></li>
        
          <li><a href="../Bio/PDB/ChemicalComponent/Record.html">Bio::PDB::ChemicalComponent::Record</a></li>
        
          <li><a href="../Bio/PDB/Coordinate.html">Bio::PDB::Coordinate</a></li>
        
          <li><a href="../Bio/PDB/DataType.html">Bio::PDB::DataType</a></li>
        
          <li><a href="../Bio/PDB/DataType/ConstLikeMethod.html">Bio::PDB::DataType::ConstLikeMethod</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_AChar</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Atom</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_Character</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Date</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_IDcode</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Integer.html">Bio::PDB::DataType::Pdb_Integer</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_LString.html">Bio::PDB::DataType::Pdb_LString</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_List.html">Bio::PDB::DataType::Pdb_List</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Real.html">Bio::PDB::DataType::Pdb_Real</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_Residue_name</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_SList.html">Bio::PDB::DataType::Pdb_SList</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_Specification_list.html">Bio::PDB::DataType::Pdb_Specification_list</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_String</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_StringRJ.html">Bio::PDB::DataType::Pdb_StringRJ</a></li>
        
          <li><a href="../Bio/PDB/DataType/Pdb_String.html">Bio::PDB::DataType::Pdb_SymOP</a></li>
        
          <li><a href="../Bio/PDB/HetatmFinder.html">Bio::PDB::HetatmFinder</a></li>
        
          <li><a href="../Bio/PDB/Heterogen.html">Bio::PDB::Heterogen</a></li>
        
          <li><a href="../Bio/PDB/HeterogenFinder.html">Bio::PDB::HeterogenFinder</a></li>
        
          <li><a href="../Bio/PDB/Model.html">Bio::PDB::Model</a></li>
        
          <li><a href="../Bio/PDB/ModelFinder.html">Bio::PDB::ModelFinder</a></li>
        
          <li><a href="../Bio/PDB/Record.html">Bio::PDB::Record</a></li>
        
          <li><a href="../Bio/PDB/Record/ANISOU.html">Bio::PDB::Record::ANISOU</a></li>
        
          <li><a href="../Bio/PDB/Record/ATOM.html">Bio::PDB::Record::ATOM</a></li>
        
          <li><a href="../Bio/PDB/Record/HETATM.html">Bio::PDB::Record::HETATM</a></li>
        
          <li><a href="../Bio/PDB/Record/Jrnl.html">Bio::PDB::Record::Jrnl</a></li>
        
          <li><a href="../Bio/PDB/Record/Remark1.html">Bio::PDB::Record::Remark1</a></li>
        
          <li><a href="../Bio/PDB/Record/Remark2.html">Bio::PDB::Record::Remark2</a></li>
        
          <li><a href="../Bio/PDB/Residue.html">Bio::PDB::Residue</a></li>
        
          <li><a href="../Bio/PDB/ResidueFinder.html">Bio::PDB::ResidueFinder</a></li>
        
          <li><a href="../Bio/PDB/Utils.html">Bio::PDB::Utils</a></li>
        
          <li><a href="../Bio/PROSITE.html">Bio::PROSITE</a></li>
        
          <li><a href="../Bio/PSORT.html">Bio::PSORT</a></li>
        
          <li><a href="../Bio/PSORT/CGIDriver.html">Bio::PSORT::CGIDriver</a></li>
        
          <li><a href="../Bio/PSORT/IPSORT.html">Bio::PSORT::IPSORT</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1.html">Bio::PSORT::PSORT1</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1/Remote.html">Bio::PSORT::PSORT1::Remote</a></li>
        
          <li><a href="../Bio/PSORT/PSORT1/Report.html">Bio::PSORT::PSORT1::Report</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2.html">Bio::PSORT::PSORT2</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2/Remote.html">Bio::PSORT::PSORT2::Remote</a></li>
        
          <li><a href="../Bio/PSORT/PSORT2/Report.html">Bio::PSORT::PSORT2::Report</a></li>
        
          <li><a href="../Bio/PSORT/PSORTB.html">Bio::PSORT::PSORTB</a></li>
        
          <li><a href="../Bio/PSORT/WoLF_PSORT.html">Bio::PSORT::WoLF_PSORT</a></li>
        
          <li><a href="../Bio/PTS1.html">Bio::PTS1</a></li>
        
          <li><a href="../Bio/PTS1/Report.html">Bio::PTS1::Report</a></li>
        
          <li><a href="../Bio/Pathway.html">Bio::Pathway</a></li>
        
          <li><a href="../Bio/Phylip.html">Bio::Phylip</a></li>
        
          <li><a href="../Bio/Phylip/DistanceMatrix.html">Bio::Phylip::DistanceMatrix</a></li>
        
          <li><a href="../Bio/Phylip/PhylipFormat.html">Bio::Phylip::PhylipFormat</a></li>
        
          <li><a href="../Bio/PhyloXML.html">Bio::PhyloXML</a></li>
        
          <li><a href="../Bio/PhyloXML/Accession.html">Bio::PhyloXML::Accession</a></li>
        
          <li><a href="../Bio/PhyloXML/Annotation.html">Bio::PhyloXML::Annotation</a></li>
        
          <li><a href="../Bio/PhyloXML/BinaryCharacters.html">Bio::PhyloXML::BinaryCharacters</a></li>
        
          <li><a href="../Bio/PhyloXML/BranchColor.html">Bio::PhyloXML::BranchColor</a></li>
        
          <li><a href="../Bio/PhyloXML/CladeRelation.html">Bio::PhyloXML::CladeRelation</a></li>
        
          <li><a href="../Bio/PhyloXML/Confidence.html">Bio::PhyloXML::Confidence</a></li>
        
          <li><a href="../Bio/PhyloXML/Date.html">Bio::PhyloXML::Date</a></li>
        
          <li><a href="../Bio/PhyloXML/Distribution.html">Bio::PhyloXML::Distribution</a></li>
        
          <li><a href="../Bio/PhyloXML/DomainArchitecture.html">Bio::PhyloXML::DomainArchitecture</a></li>
        
          <li><a href="../Bio/PhyloXML/Events.html">Bio::PhyloXML::Events</a></li>
        
          <li><a href="../Bio/PhyloXML/Id.html">Bio::PhyloXML::Id</a></li>
        
          <li><a href="../Bio/PhyloXML/Node.html">Bio::PhyloXML::Node</a></li>
        
          <li><a href="../Bio/PhyloXML/Other.html">Bio::PhyloXML::Other</a></li>
        
          <li><a href="../Bio/PhyloXML/Parser.html">Bio::PhyloXML::Parser</a></li>
        
          <li><a href="../Bio/PhyloXML/Point.html">Bio::PhyloXML::Point</a></li>
        
          <li><a href="../Bio/PhyloXML/Polygon.html">Bio::PhyloXML::Polygon</a></li>
        
          <li><a href="../Bio/PhyloXML/Property.html">Bio::PhyloXML::Property</a></li>
        
          <li><a href="../Bio/PhyloXML/ProteinDomain.html">Bio::PhyloXML::ProteinDomain</a></li>
        
          <li><a href="../Bio/PhyloXML/Reference.html">Bio::PhyloXML::Reference</a></li>
        
          <li><a href="../Bio/PhyloXML/Sequence.html">Bio::PhyloXML::Sequence</a></li>
        
          <li><a href="../Bio/PhyloXML/SequenceRelation.html">Bio::PhyloXML::SequenceRelation</a></li>
        
          <li><a href="../Bio/PhyloXML/Taxonomy.html">Bio::PhyloXML::Taxonomy</a></li>
        
          <li><a href="../Bio/PhyloXML/Tree.html">Bio::PhyloXML::Tree</a></li>
        
          <li><a href="../Bio/PhyloXML/Uri.html">Bio::PhyloXML::Uri</a></li>
        
          <li><a href="../Bio/PhyloXML/Writer.html">Bio::PhyloXML::Writer</a></li>
        
          <li><a href="../Bio/Probcons.html">Bio::Probcons</a></li>
        
          <li><a href="../Bio/Alignment/MultiFastaFormat.html">Bio::Probcons::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/PubMed.html">Bio::PubMed</a></li>
        
          <li><a href="../Bio/REBASE.html">Bio::REBASE</a></li>
        
          <li><a href="../Bio/RefSeq.html">Bio::RefSeq</a></li>
        
          <li><a href="../Bio/Reference.html">Bio::Reference</a></li>
        
          <li><a href="../Bio/References.html">Bio::References</a></li>
        
          <li><a href="../Bio/Registry.html">Bio::Registry</a></li>
        
          <li><a href="../Bio/Registry/DB.html">Bio::Registry::DB</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme.html">Bio::RestrictionEnzyme</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Analysis.html">Bio::RestrictionEnzyme::Analysis</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/CutSymbol.html">Bio::RestrictionEnzyme::CutSymbol</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/CutSymbol/CutSymbol__.html">Bio::RestrictionEnzyme::CutSymbol::CutSymbol__</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DenseIntArray.html">Bio::RestrictionEnzyme::DenseIntArray</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded.html">Bio::RestrictionEnzyme::DoubleStranded</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/AlignedStrands.html">Bio::RestrictionEnzyme::DoubleStranded::AlignedStrands</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPair.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPair</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationPairInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationPairInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocations.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocations</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::DoubleStranded::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/DoubleStranded/EnzymeAction.html">Bio::RestrictionEnzyme::DoubleStranded::EnzymeAction</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Fragments.html">Bio::RestrictionEnzyme::Fragments</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range.html">Bio::RestrictionEnzyme::Range</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/CutRange.html">Bio::RestrictionEnzyme::Range::CutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/CutRanges.html">Bio::RestrictionEnzyme::Range::CutRanges</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/HorizontalCutRange.html">Bio::RestrictionEnzyme::Range::HorizontalCutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange.html">Bio::RestrictionEnzyme::Range::SequenceRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/CalculatedCuts.html">Bio::RestrictionEnzyme::Range::SequenceRange::CalculatedCuts</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragment.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragment</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/SequenceRange/Fragments.html">Bio::RestrictionEnzyme::Range::SequenceRange::Fragments</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/Range/VerticalCutRange.html">Bio::RestrictionEnzyme::Range::VerticalCutRange</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrand.html">Bio::RestrictionEnzyme::SingleStrand</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrand/CutLocationsInEnzymeNotation.html">Bio::RestrictionEnzyme::SingleStrand::CutLocationsInEnzymeNotation</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SingleStrandComplement.html">Bio::RestrictionEnzyme::SingleStrandComplement</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/SortedNumArray.html">Bio::RestrictionEnzyme::SortedNumArray</a></li>
        
          <li><a href="../Bio/RestrictionEnzyme/StringFormatting.html">Bio::RestrictionEnzyme::StringFormatting</a></li>
        
          <li><a href="../Bio/SOAPWSDL.html">Bio::SOAPWSDL</a></li>
        
          <li><a href="../Bio/SOFT.html">Bio::SOFT</a></li>
        
          <li><a href="../Bio/SOSUI.html">Bio::SOSUI</a></li>
        
          <li><a href="../Bio/SOSUI/Report.html">Bio::SOSUI::Report</a></li>
        
          <li><a href="../Bio/SOSUI/Report/TMH.html">Bio::SOSUI::Report::TMH</a></li>
        
          <li><a href="../Bio/SPTR.html">Bio::SPTR</a></li>
        
          <li><a href="../Bio/SQL.html">Bio::SQL</a></li>
        
          <li><a href="../Bio/SQL/Biodatabase.html">Bio::SQL::Biodatabase</a></li>
        
          <li><a href="../Bio/SQL/Bioentry.html">Bio::SQL::Bioentry</a></li>
        
          <li><a href="../Bio/SQL/BioentryDbxref.html">Bio::SQL::BioentryDbxref</a></li>
        
          <li><a href="../Bio/SQL/BioentryPath.html">Bio::SQL::BioentryPath</a></li>
        
          <li><a href="../Bio/SQL/BioentryQualifierValue.html">Bio::SQL::BioentryQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/BioentryReference.html">Bio::SQL::BioentryReference</a></li>
        
          <li><a href="../Bio/SQL/BioentryRelationship.html">Bio::SQL::BioentryRelationship</a></li>
        
          <li><a href="../Bio/SQL/Biosequence.html">Bio::SQL::Biosequence</a></li>
        
          <li><a href="../Bio/SQL/Comment.html">Bio::SQL::Comment</a></li>
        
          <li><a href="../Bio/SQL/Dbxref.html">Bio::SQL::Dbxref</a></li>
        
          <li><a href="../Bio/SQL/DbxrefQualifierValue.html">Bio::SQL::DbxrefQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/DummyBase.html">Bio::SQL::DummyBase</a></li>
        
          <li><a href="../Bio/SQL/Location.html">Bio::SQL::Location</a></li>
        
          <li><a href="../Bio/SQL/LocationQualifierValue.html">Bio::SQL::LocationQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/Ontology.html">Bio::SQL::Ontology</a></li>
        
          <li><a href="../Bio/SQL/Reference.html">Bio::SQL::Reference</a></li>
        
          <li><a href="../Bio/SQL/Seqfeature.html">Bio::SQL::Seqfeature</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureDbxref.html">Bio::SQL::SeqfeatureDbxref</a></li>
        
          <li><a href="../Bio/SQL/SeqfeaturePath.html">Bio::SQL::SeqfeaturePath</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureQualifierValue.html">Bio::SQL::SeqfeatureQualifierValue</a></li>
        
          <li><a href="../Bio/SQL/SeqfeatureRelationship.html">Bio::SQL::SeqfeatureRelationship</a></li>
        
          <li><a href="../Bio/SQL/Sequence.html">Bio::SQL::Sequence</a></li>
        
          <li><a href="../Bio/SQL/Taxon.html">Bio::SQL::Taxon</a></li>
        
          <li><a href="../Bio/SQL/TaxonName.html">Bio::SQL::TaxonName</a></li>
        
          <li><a href="../Bio/SQL/Term.html">Bio::SQL::Term</a></li>
        
          <li><a href="../Bio/SQL/TermDbxref.html">Bio::SQL::TermDbxref</a></li>
        
          <li><a href="../Bio/SQL/TermPath.html">Bio::SQL::TermPath</a></li>
        
          <li><a href="../Bio/SQL/TermRelationship.html">Bio::SQL::TermRelationship</a></li>
        
          <li><a href="../Bio/SQL/TermRelationshipTerm.html">Bio::SQL::TermRelationshipTerm</a></li>
        
          <li><a href="../Bio/SQL/TermSynonym.html">Bio::SQL::TermSynonym</a></li>
        
          <li><a href="../Bio/SangerChromatogram.html">Bio::SangerChromatogram</a></li>
        
          <li><a href="../Bio/Scf.html">Bio::Scf</a></li>
        
          <li><a href="../Bio/Sequence.html">Bio::Sequence</a></li>
        
          <li><a href="../Bio/Sequence/AA.html">Bio::Sequence::AA</a></li>
        
          <li><a href="../Bio/Sequence/Adapter.html">Bio::Sequence::Adapter</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/BioSQL.html">Bio::Sequence::Adapter::BioSQL</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/EMBL.html">Bio::Sequence::Adapter::EMBL</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/FastaFormat.html">Bio::Sequence::Adapter::FastaFormat</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/FastaNumericFormat.html">Bio::Sequence::Adapter::FastaNumericFormat</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/Fastq.html">Bio::Sequence::Adapter::Fastq</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/GenBank.html">Bio::Sequence::Adapter::GenBank</a></li>
        
          <li><a href="../Bio/Sequence/Adapter/SangerChromatogram.html">Bio::Sequence::Adapter::SangerChromatogram</a></li>
        
          <li><a href="../Bio/Sequence/Common.html">Bio::Sequence::Common</a></li>
        
          <li><a href="../Bio/Sequence/DBLink.html">Bio::Sequence::DBLink</a></li>
        
          <li><a href="../Bio/Sequence/Format.html">Bio::Sequence::Format</a></li>
        
          <li><a href="../Bio/Sequence/Format/AminoFormatter.html">Bio::Sequence::Format::AminoFormatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter.html">Bio::Sequence::Format::Formatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta.html">Bio::Sequence::Format::Formatter::Fasta</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta_ncbi.html">Bio::Sequence::Format::Formatter::Fasta_ncbi</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fasta_numeric.html">Bio::Sequence::Format::Formatter::Fasta_numeric</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq_illumina.html">Bio::Sequence::Format::Formatter::Fastq_illumina</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq.html">Bio::Sequence::Format::Formatter::Fastq_sanger</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Fastq_solexa.html">Bio::Sequence::Format::Formatter::Fastq_solexa</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Qual.html">Bio::Sequence::Format::Formatter::Qual</a></li>
        
          <li><a href="../Bio/Sequence/Format/Formatter/Raw.html">Bio::Sequence::Format::Formatter::Raw</a></li>
        
          <li><a href="../Bio/Sequence/Format/FormatterBase.html">Bio::Sequence::Format::FormatterBase</a></li>
        
          <li><a href="../Bio/Sequence/Format/INSDFeatureHelper.html">Bio::Sequence::Format::INSDFeatureHelper</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter.html">Bio::Sequence::Format::NucFormatter</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter/Embl.html">Bio::Sequence::Format::NucFormatter::Embl</a></li>
        
          <li><a href="../Bio/Sequence/Format/NucFormatter/Genbank.html">Bio::Sequence::Format::NucFormatter::Genbank</a></li>
        
          <li><a href="../Bio/Sequence/NA.html">Bio::Sequence::NA</a></li>
        
          <li><a href="../Bio/Sequence/NA/MidiTrack.html">Bio::Sequence::NA::MidiTrack</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore.html">Bio::Sequence::QualityScore</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Converter.html">Bio::Sequence::QualityScore::Converter</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Phred.html">Bio::Sequence::QualityScore::Phred</a></li>
        
          <li><a href="../Bio/Sequence/QualityScore/Solexa.html">Bio::Sequence::QualityScore::Solexa</a></li>
        
          <li><a href="../Bio/Sequence/SequenceMasker.html">Bio::Sequence::SequenceMasker</a></li>
        
          <li><a href="../Bio/Shell.html">Bio::Shell</a></li>
        
          <li><a href="../Bio/Shell/ColoredCodonTable.html">Bio::Shell::ColoredCodonTable</a></li>
        
          <li><a href="../Bio/Shell/Core.html">Bio::Shell::Core</a></li>
        
          <li><a href="../Bio/Shell/Demo.html">Bio::Shell::Demo</a></li>
        
          <li><a href="../Bio/Shell/Ghost.html">Bio::Shell::Ghost</a></li>
        
          <li><a href="../Bio/Shell/Irb.html">Bio::Shell::Irb</a></li>
        
          <li><a href="../Bio/Shell/Private.html">Bio::Shell::Private</a></li>
        
          <li><a href="../Bio/Shell/Script.html">Bio::Shell::Script</a></li>
        
          <li><a href="../Bio/Shell/Setup.html">Bio::Shell::Setup</a></li>
        
          <li><a href="../Bio/Shell/Web.html">Bio::Shell::Web</a></li>
        
          <li><a href="../Bio/Shell/Web/Results.html">Bio::Shell::Web::Results</a></li>
        
          <li><a href="../Bio/SiRNA.html">Bio::SiRNA</a></li>
        
          <li><a href="../Bio/SiRNA/Pair.html">Bio::SiRNA::Pair</a></li>
        
          <li><a href="../Bio/SiRNA/ShRNA.html">Bio::SiRNA::ShRNA</a></li>
        
          <li><a href="../Bio/Sim4.html">Bio::Sim4</a></li>
        
          <li><a href="../Bio/Sim4/Report.html">Bio::Sim4::Report</a></li>
        
          <li><a href="../Bio/Sim4/Report/Hit.html">Bio::Sim4::Report::Hit</a></li>
        
          <li><a href="../Bio/Sim4/Report/Segment.html">Bio::Sim4::Report::Segment</a></li>
        
          <li><a href="../Bio/Sim4/Report/SegmentPair.html">Bio::Sim4::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Sim4/Report/SeqDesc.html">Bio::Sim4::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/Spidey.html">Bio::Spidey</a></li>
        
          <li><a href="../Bio/Spidey/Report.html">Bio::Spidey::Report</a></li>
        
          <li><a href="../Bio/Spidey/Report/Hit.html">Bio::Spidey::Report::Hit</a></li>
        
          <li><a href="../Bio/Spidey/Report/Segment.html">Bio::Spidey::Report::Segment</a></li>
        
          <li><a href="../Bio/Spidey/Report/SegmentPair.html">Bio::Spidey::Report::SegmentPair</a></li>
        
          <li><a href="../Bio/Spidey/Report/SeqDesc.html">Bio::Spidey::Report::SeqDesc</a></li>
        
          <li><a href="../Bio/SwissProt.html">Bio::SwissProt</a></li>
        
          <li><a href="../Bio/TMHMM.html">Bio::TMHMM</a></li>
        
          <li><a href="../Bio/TMHMM/Report.html">Bio::TMHMM::Report</a></li>
        
          <li><a href="../Bio/TMHMM/TMH.html">Bio::TMHMM::TMH</a></li>
        
          <li><a href="../Bio/TRANSFAC.html">Bio::TRANSFAC</a></li>
        
          <li><a href="../Bio/TRANSFAC/CELL.html">Bio::TRANSFAC::CELL</a></li>
        
          <li><a href="../Bio/TRANSFAC/CLASS.html">Bio::TRANSFAC::CLASS</a></li>
        
          <li><a href="../Bio/TRANSFAC/FACTOR.html">Bio::TRANSFAC::FACTOR</a></li>
        
          <li><a href="../Bio/TRANSFAC/GENE.html">Bio::TRANSFAC::GENE</a></li>
        
          <li><a href="../Bio/TRANSFAC/MATRIX.html">Bio::TRANSFAC::MATRIX</a></li>
        
          <li><a href="../Bio/TRANSFAC/SITE.html">Bio::TRANSFAC::SITE</a></li>
        
          <li><a href="../Bio/TargetP.html">Bio::TargetP</a></li>
        
          <li><a href="../Bio/TargetP/Report.html">Bio::TargetP::Report</a></li>
        
          <li><a href="../Bio/Taxonomy.html">Bio::Taxonomy</a></li>
        
          <li><a href="../Bio/Tcoffee.html">Bio::Tcoffee</a></li>
        
          <li><a href="../Bio/ClustalW/Report.html">Bio::Tcoffee::DEFAULT_PARSER</a></li>
        
          <li><a href="../Bio/TogoWS.html">Bio::TogoWS</a></li>
        
          <li><a href="../Bio/TogoWS/AccessWait.html">Bio::TogoWS::AccessWait</a></li>
        
          <li><a href="../Bio/TogoWS/REST.html">Bio::TogoWS::REST</a></li>
        
          <li><a href="../Bio/TrEMBL.html">Bio::TrEMBL</a></li>
        
          <li><a href="../Bio/Tree.html">Bio::Tree</a></li>
        
          <li><a href="../Bio/Tree/Edge.html">Bio::Tree::Edge</a></li>
        
          <li><a href="../Bio/Tree/NoPathError.html">Bio::Tree::NoPathError</a></li>
        
          <li><a href="../Bio/Tree/Node.html">Bio::Tree::Node</a></li>
        
          <li><a href="../Bio/UniProt.html">Bio::UniProt</a></li>
        
          <li><a href="../BiorubyController.html">BiorubyController</a></li>
        
          <li><a href="../BiorubyGenerator.html">BiorubyGenerator</a></li>
        
          <li><a href="../BiorubyHelper.html">BiorubyHelper</a></li>
        
          <li><a href="../Hash.html">Hash</a></li>
        
          <li><a href="../Object.html">Object</a></li>
        
          <li><a href="../String.html">String</a></li>
        
        </ul>
        <div id="no-class-search-results" style="display: none;">No matching classes.</div>
      </div>

      
    </div>
  </div>

  <div id="documentation">
    <h1 class="class">Bio::SPTR</h1>

    <div id="description" class="description">
      
<p>Parser class for UniProtKB/SwissProt and <a href="TrEMBL.html">TrEMBL</a>
database entry.</p>

    </div><!-- description -->

    
    
    
    <div id="5Buntitled-5D" class="documentation-section">
      

      

      

      

      <!-- Methods -->
      
      <div id="public-instance-method-details" class="method-section section">
        <h3 class="section-header">Public Instance Methods</h3>

      
        <div id="aalen-method" class="method-detail method-alias">
          <a name="method-i-aalen"></a>

          
          <div class="method-heading">
            <span class="method-name">aalen</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="SPTR.html#method-i-sequence_length">sequence_length</a>
          </div>
          
        </div><!-- aalen-method -->

      
        <div id="aaseq-method" class="method-detail method-alias">
          <a name="method-i-aaseq"></a>

          
          <div class="method-heading">
            <span class="method-name">aaseq</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="SPTR.html#method-i-seq">seq</a>
          </div>
          
        </div><!-- aaseq-method -->

      
        <div id="cc-method" class="method-detail ">
          <a name="method-i-cc"></a>

          
          <div class="method-heading">
            <span class="method-name">cc</span><span
              class="method-args">(topic = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns contents in the CC lines.</p>
<ul><li>
<p><a href="SPTR.html#method-i-cc">Bio::SPTR#cc</a> -&gt; <a
href="../Hash.html">Hash</a></p>
</li></ul>

<p>returns an object of contents in the TOPIC.</p>
<ul><li>
<p><a href="SPTR.html#method-i-cc">Bio::SPTR#cc(TOPIC)</a> -&gt; Array w/in <a
href="../Hash.html">Hash</a>, <a href="../Hash.html">Hash</a></p>
</li></ul>

<p>returns contents of the ALTERNATIVE PRODUCTS.</p>
<ul><li>
<p><a href="SPTR.html#method-i-cc">Bio::SPTR#cc</a>('ALTERNATIVE PRODUCTS')
-&gt; <a href="../Hash.html">Hash</a></p>

<pre>{'Event' =&gt; str, 
 'Named isoforms' =&gt; int,  
 'Comment' =&gt; str,
 'Variants'=&gt;[{'Name' =&gt; str, 'Synonyms' =&gt; str, 'IsoId' =&gt; str, 'Sequence' =&gt; []}]}

CC   -!- ALTERNATIVE PRODUCTS:
CC       Event=Alternative splicing; Named isoforms=15;
...
CC         placentae isoforms. All tissues differentially splice exon 13;
CC       Name=A; Synonyms=no del;
CC         IsoId=P15529-1; Sequence=Displayed;</pre>
</li></ul>

<p>returns contents of the DATABASE.</p>
<ul><li>
<p><a href="SPTR.html#method-i-cc">Bio::SPTR#cc</a>('DATABASE') -&gt; Array</p>

<pre>[{'NAME'=&gt;str,'NOTE'=&gt;str, 'WWW'=&gt;URI,'FTP'=&gt;URI}, ...]

CC   -!- DATABASE: NAME=Text[; NOTE=Text][; WWW=&quot;Address&quot;][; FTP=&quot;Address&quot;].</pre>
</li></ul>

<p>returns contents of the MASS SPECTROMETRY.</p>
<ul><li>
<p><a href="SPTR.html#method-i-cc">Bio::SPTR#cc</a>('MASS SPECTROMETRY') -&gt;
Array</p>

<pre>[{'MW&quot;=&gt;float,'MW_ERR'=&gt;float, 'METHOD'=&gt;str,'RANGE'=&gt;str}, ...]

CC   -!- MASS SPECTROMETRY: MW=XXX[; MW_ERR=XX][; METHOD=XX][;RANGE=XX-XX].</pre>
</li></ul>

<h3>CC lines (&gt;=0, optional)</h3>

<pre>CC   -!- TISSUE SPECIFICITY: HIGHEST LEVELS FOUND IN TESTIS. ALSO PRESENT
CC       IN LIVER, KIDNEY, LUNG AND BRAIN.

CC   -!- TOPIC: FIRST LINE OF A COMMENT BLOCK;
CC       SECOND AND SUBSEQUENT LINES OF A COMMENT BLOCK.</pre>

<p>See also <a
href="http://www.expasy.org/sprot/userman.html#CC_line">www.expasy.org/sprot/userman.html#CC_line</a></p>
            

            
            <div class="method-source-code" id="cc-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 775</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">cc</span>(<span class="ruby-identifier">topic</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>]
    <span class="ruby-identifier">cc</span>  = <span class="ruby-constant">Hash</span>.<span class="ruby-identifier">new</span>
    <span class="ruby-identifier">comment_border</span>= <span class="ruby-string">'-'</span> * (<span class="ruby-value">77</span> <span class="ruby-operator">-</span> <span class="ruby-value">4</span> <span class="ruby-operator">+</span> <span class="ruby-value">1</span>)
    <span class="ruby-identifier">dlm</span> = <span class="ruby-regexp">/-!- /</span>

    <span class="ruby-comment"># 12KD_MYCSM has no CC lines.</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">get</span>(<span class="ruby-string">'CC'</span>).<span class="ruby-identifier">size</span> <span class="ruby-operator">==</span> <span class="ruby-value">0</span>
    
    <span class="ruby-identifier">cc_raw</span> = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'CC'</span>)

    <span class="ruby-comment"># Removing the copyright statement.</span>
    <span class="ruby-identifier">cc_raw</span>.<span class="ruby-identifier">sub!</span>(<span class="ruby-regexp">/ *---.+---/</span>, <span class="ruby-string">''</span>)

    <span class="ruby-comment"># Not any CC Lines without the copyright statement.</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">cc_raw</span> <span class="ruby-operator">==</span> <span class="ruby-string">''</span>

    <span class="ruby-keyword">begin</span>
      <span class="ruby-identifier">cc_raw</span>, <span class="ruby-identifier">copyright</span> = <span class="ruby-identifier">cc_raw</span>.<span class="ruby-identifier">split</span>(<span class="ruby-node">/#{comment_border}/</span>)[<span class="ruby-value">0</span>]
      <span class="ruby-identifier">cc_raw</span> = <span class="ruby-identifier">cc_raw</span>.<span class="ruby-identifier">sub</span>(<span class="ruby-identifier">dlm</span>,<span class="ruby-string">''</span>)
      <span class="ruby-identifier">cc_raw</span>.<span class="ruby-identifier">split</span>(<span class="ruby-identifier">dlm</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">tmp</span><span class="ruby-operator">|</span>
        <span class="ruby-identifier">tmp</span> = <span class="ruby-identifier">tmp</span>.<span class="ruby-identifier">strip</span>

        <span class="ruby-keyword">if</span> <span class="ruby-regexp">/(^[A-Z ]+[A-Z]): (.+)/</span> <span class="ruby-operator">=~</span> <span class="ruby-identifier">tmp</span>
          <span class="ruby-identifier">key</span>  = <span class="ruby-node">$1</span>
          <span class="ruby-identifier">body</span> = <span class="ruby-node">$2</span>
          <span class="ruby-identifier">body</span>.<span class="ruby-identifier">gsub!</span>(<span class="ruby-regexp">/- (?!AND)/</span>,<span class="ruby-string">'-'</span>)
          <span class="ruby-identifier">body</span>.<span class="ruby-identifier">strip!</span>
          <span class="ruby-keyword">unless</span> <span class="ruby-identifier">cc</span>[<span class="ruby-identifier">key</span>]
            <span class="ruby-identifier">cc</span>[<span class="ruby-identifier">key</span>] = [<span class="ruby-identifier">body</span>]
          <span class="ruby-keyword">else</span>
            <span class="ruby-identifier">cc</span>[<span class="ruby-identifier">key</span>].<span class="ruby-identifier">push</span>(<span class="ruby-identifier">body</span>)
          <span class="ruby-keyword">end</span>
        <span class="ruby-keyword">else</span>
          <span class="ruby-identifier">raise</span> [<span class="ruby-node">&quot;Error: [#{entry_id}]: CC Lines&quot;</span>, <span class="ruby-string">'&quot;'</span>, <span class="ruby-identifier">tmp</span>, <span class="ruby-string">'&quot;'</span>,
                 <span class="ruby-string">''</span>, <span class="ruby-identifier">get</span>(<span class="ruby-string">'CC'</span>),<span class="ruby-string">''</span>].<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot;\n&quot;</span>)
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">rescue</span> <span class="ruby-constant">NameError</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'CC'</span>) <span class="ruby-operator">==</span> <span class="ruby-string">''</span>
        <span class="ruby-keyword">return</span> {}
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">raise</span> [<span class="ruby-node">&quot;Error: Invalid CC Lines: [#{entry_id}]: &quot;</span>,
               <span class="ruby-node">&quot;\n'#{self.get('CC')}'\n&quot;</span>, <span class="ruby-node">&quot;(#{$!})&quot;</span>].<span class="ruby-identifier">join</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">rescue</span> <span class="ruby-constant">NoMethodError</span>
    <span class="ruby-keyword">end</span>
    
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>] = <span class="ruby-identifier">cc</span>
  <span class="ruby-keyword">end</span>


  <span class="ruby-keyword">case</span> <span class="ruby-identifier">topic</span>
  <span class="ruby-keyword">when</span> <span class="ruby-string">'ALLERGEN'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'ALTERNATIVE PRODUCTS'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_alternative_products</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'BIOPHYSICOCHEMICAL PROPERTIES'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_biophysiochemical_properties</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'BIOTECHNOLOGY'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'CATALITIC ACTIVITY'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_catalytic_activity</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'CAUTION'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_caution</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'COFACTOR'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'DEVELOPMENTAL STAGE'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>].<span class="ruby-identifier">join</span>(<span class="ruby-string">''</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-string">'DISEASE'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>].<span class="ruby-identifier">join</span>(<span class="ruby-string">''</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-string">'DOMAIN'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'ENZYME REGULATION'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>].<span class="ruby-identifier">join</span>(<span class="ruby-string">''</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-string">'FUNCTION'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>].<span class="ruby-identifier">join</span>(<span class="ruby-string">''</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-string">'INDUCTION'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>].<span class="ruby-identifier">join</span>(<span class="ruby-string">''</span>)
  <span class="ruby-keyword">when</span> <span class="ruby-string">'INTERACTION'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_interaction</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'MASS SPECTROMETRY'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_mass_spectrometry</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'MISCELLANEOUS'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'PATHWAY'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_pathway</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'PHARMACEUTICAL'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'POLYMORPHISM'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'PTM'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'RNA EDITING'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_rna_editing</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'SIMILARITY'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'SUBCELLULAR LOCATION'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_subcellular_location</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'SUBUNIT'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'TISSUE SPECIFICITY'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'TOXIC DOSE'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">when</span> <span class="ruby-string">'WEB RESOURCE'</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">cc_web_resource</span>(<span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>])
  <span class="ruby-keyword">when</span> <span class="ruby-string">'DATABASE'</span>
    <span class="ruby-comment"># DATABASE: NAME=Text[; NOTE=Text][; WWW=&quot;Address&quot;][; FTP=&quot;Address&quot;].</span>
    <span class="ruby-identifier">tmp</span> = <span class="ruby-constant">Array</span>.<span class="ruby-identifier">new</span>
    <span class="ruby-identifier">db</span> = <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-string">'DATABASE'</span>]
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">db</span> <span class="ruby-keyword">unless</span> <span class="ruby-identifier">db</span>

    <span class="ruby-identifier">db</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">e</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">db</span> = {<span class="ruby-string">'NAME'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-keyword">nil</span>, <span class="ruby-string">'NOTE'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-keyword">nil</span>, <span class="ruby-string">'WWW'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-keyword">nil</span>, <span class="ruby-string">'FTP'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-keyword">nil</span>}
      <span class="ruby-identifier">e</span>.<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/.$/</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/;/</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">line</span><span class="ruby-operator">|</span>
        <span class="ruby-keyword">case</span> <span class="ruby-identifier">line</span>
        <span class="ruby-keyword">when</span> <span class="ruby-regexp">/NAME=(.+)/</span>
          <span class="ruby-identifier">db</span>[<span class="ruby-string">'NAME'</span>] = <span class="ruby-node">$1</span>
        <span class="ruby-keyword">when</span> <span class="ruby-regexp">/NOTE=(.+)/</span>
          <span class="ruby-identifier">db</span>[<span class="ruby-string">'NOTE'</span>] = <span class="ruby-node">$1</span>
        <span class="ruby-keyword">when</span> <span class="ruby-regexp">/WWW=&quot;(.+)&quot;/</span>
          <span class="ruby-identifier">db</span>[<span class="ruby-string">'WWW'</span>] = <span class="ruby-node">$1</span>
        <span class="ruby-keyword">when</span> <span class="ruby-regexp">/FTP=&quot;(.+)&quot;/</span>
          <span class="ruby-identifier">db</span>[<span class="ruby-string">'FTP'</span>] = <span class="ruby-node">$1</span>
        <span class="ruby-keyword">end</span> 
      <span class="ruby-keyword">end</span>
      <span class="ruby-identifier">tmp</span>.<span class="ruby-identifier">push</span>(<span class="ruby-identifier">db</span>)
    <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">return</span> <span class="ruby-identifier">tmp</span>
  <span class="ruby-keyword">when</span> <span class="ruby-keyword">nil</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>]
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'CC'</span>][<span class="ruby-identifier">topic</span>]
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- cc-source -->
            
          </div>

          

          
        </div><!-- cc-method -->

      
        <div id="dr-method" class="method-detail ">
          <a name="method-i-dr"></a>

          
          <div class="method-heading">
            <span class="method-name">dr</span><span
              class="method-args">(key = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p><a href="SPTR.html#method-i-dr">Bio::SPTR#dr</a></p>
            

            
            <div class="method-source-code" id="dr-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 1131</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dr</span>(<span class="ruby-identifier">key</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-identifier">key</span>
    <span class="ruby-identifier">embl_dr</span>
  <span class="ruby-keyword">else</span>
    (<span class="ruby-identifier">embl_dr</span>[<span class="ruby-identifier">key</span>] <span class="ruby-keyword">or</span> []).<span class="ruby-identifier">map</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span>
      {<span class="ruby-string">'Accession'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">x</span>[<span class="ruby-value">0</span>],
       <span class="ruby-string">'Version'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">x</span>[<span class="ruby-value">1</span>],
       <span class="ruby-string">' '</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">x</span>[<span class="ruby-value">2</span>],
       <span class="ruby-string">'Molecular Type'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">x</span>[<span class="ruby-value">3</span>]}
    }
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- dr-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="SPTR.html#method-i-embl_dr">embl_dr</a>
          </div>
          

          
        </div><!-- dr-method -->

      
        <div id="dt-method" class="method-detail ">
          <a name="method-i-dt"></a>

          
          <div class="method-heading">
            <span class="method-name">dt</span><span
              class="method-args">(key = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a <a href="../Hash.html">Hash</a> of information in the DT lines.</p>

<pre>hash keys: 
  ['created', 'sequence', 'annotation']</pre>

<p>Since UniProtKB release 7.0 of 07-Feb-2006, the DT line format is changed,
and the word annotation is no longer used in DT lines. Despite the
change, the word annotation is still used for keeping compatibility.</p>

<p>returns a <a href="../String.html">String</a> of information in the DT
lines by a given key.</p>

<h3>DT Line; date (3/entry)</h3>

<pre>DT DD-MMM-YYY (integrated into UniProtKB/XXXXX.)
DT DD-MMM-YYY (sequence version NN)
DT DD-MMM-YYY (entry version NN)</pre>

<p>The format have been changed in UniProtKB release 7.0 of 07-Feb-2006. Below
is the older format.</p>

<h3>Old format of DT Line; date (3/entry)</h3>

<pre>DT DD-MMM-YYY (rel. NN, Created)
DT DD-MMM-YYY (rel. NN, Last sequence update)
DT DD-MMM-YYY (rel. NN, Last annotation update)</pre>
            

            
            <div class="method-source-code" id="dt-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 158</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">dt</span>(<span class="ruby-identifier">key</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">dt</span>[<span class="ruby-identifier">key</span>] <span class="ruby-keyword">if</span> <span class="ruby-identifier">key</span>
  <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DT'</span>] <span class="ruby-keyword">if</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DT'</span>]

  <span class="ruby-identifier">part</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">get</span>(<span class="ruby-string">'DT'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/\n/</span>)
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DT'</span>] = {
    <span class="ruby-string">'created'</span>    =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">part</span>[<span class="ruby-value">0</span>].<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\w{2}   /</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">strip</span>,
    <span class="ruby-string">'sequence'</span>   =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">part</span>[<span class="ruby-value">1</span>].<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\w{2}   /</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">strip</span>,
    <span class="ruby-string">'annotation'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">part</span>[<span class="ruby-value">2</span>].<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\w{2}   /</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">strip</span>
  }
<span class="ruby-keyword">end</span></pre>
            </div><!-- dt-source -->
            
          </div>

          

          
        </div><!-- dt-method -->

      
        <div id="embl_dr-method" class="method-detail method-alias">
          <a name="method-i-embl_dr"></a>

          
          <div class="method-heading">
            <span class="method-name">embl_dr</span><span
              class="method-args">(key = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>Backup Bio::EMBLDB#dr as <a href="SPTR.html#method-i-embl_dr">embl_dr</a></p>
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="SPTR.html#method-i-dr">dr</a>
          </div>
          
        </div><!-- embl_dr-method -->

      
        <div id="entry-method" class="method-detail method-alias">
          <a name="method-i-entry"></a>

          
          <div class="method-heading">
            <span class="method-name">entry</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="SPTR.html#method-i-entry_id">entry_id</a>
          </div>
          
        </div><!-- entry-method -->

      
        <div id="entry_id-method" class="method-detail ">
          <a name="method-i-entry_id"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_id</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a ENTRY_NAME in the ID line.</p>
            

            
            <div class="method-source-code" id="entry_id-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 99</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">entry_id</span>
  <span class="ruby-identifier">id_line</span>(<span class="ruby-string">'ENTRY_NAME'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- entry_id-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="SPTR.html#method-i-entry_name">entry_name</a>, <a href="SPTR.html#method-i-entry">entry</a>
          </div>
          

          
        </div><!-- entry_id-method -->

      
        <div id="entry_name-method" class="method-detail method-alias">
          <a name="method-i-entry_name"></a>

          
          <div class="method-heading">
            <span class="method-name">entry_name</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="SPTR.html#method-i-entry_id">entry_id</a>
          </div>
          
        </div><!-- entry_name-method -->

      
        <div id="ft-method" class="method-detail ">
          <a name="method-i-ft"></a>

          
          <div class="method-heading">
            <span class="method-name">ft</span><span
              class="method-args">(feature_key = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns contents in the feature table.</p>

<h2>Examples</h2>

<pre>sp = Bio::SPTR.new(entry)
ft = sp.ft
ft.class #=&gt; Hash
ft.keys.each do |feature_key|
  ft[feature_key].each do |feature|
    feature['From'] #=&gt; '1'
    feature['To']   #=&gt; '21'
    feature['Description'] #=&gt; ''
    feature['FTId'] #=&gt; ''
    feature['diff'] #=&gt; []
    feature['original'] #=&gt; [feature_key, '1', '21', '', '']
  end
end</pre>
<ul><li>
<p><a href="SPTR.html#method-i-ft">Bio::SPTR#ft</a> -&gt; <a
href="../Hash.html">Hash</a></p>

<pre>{FEATURE_KEY =&gt; [{'From' =&gt; int, 'To' =&gt; int, 
                  'Description' =&gt; aStr, 'FTId' =&gt; aStr,
                  'diff' =&gt; [original_residues, changed_residues],
                  'original' =&gt; aAry }],...}</pre>
</li></ul>

<p>returns an Array of the information about the feature_name in the feature
table.</p>
<ul><li>
<p><a href="SPTR.html#method-i-ft">Bio::SPTR#ft(feature_name)</a> -&gt; Array
of <a href="../Hash.html">Hash</a></p>

<pre>[{'From' =&gt; str, 'To' =&gt; str, 'Description' =&gt; str, 'FTId' =&gt; str},...]</pre>
</li></ul>

<h2>FT Line; feature table data (&gt;=0, optional)</h2>

<pre>Col     Data item
-----   -----------------
 1- 2   FT
 6-13   Feature name 
15-20   `FROM' endpoint
22-27   `TO' endpoint
35-75   Description (&gt;=0 per key)
-----   -----------------</pre>

<p>Note: FROM and TO endopoints are allowed to use non-numerial charactors
including &lt;, &gt; or ?. (c.f. &lt;1, ?42)</p>

<p>See also <a
href="http://www.expasy.org/sprot/userman.html#FT_line">www.expasy.org/sprot/userman.html#FT_line</a></p>
            

            
            <div class="method-source-code" id="ft-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 1196</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ft</span>(<span class="ruby-identifier">feature_key</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">ft</span>[<span class="ruby-identifier">feature_key</span>] <span class="ruby-keyword">if</span> <span class="ruby-identifier">feature_key</span>
  <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'FT'</span>] <span class="ruby-keyword">if</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'FT'</span>]

  <span class="ruby-identifier">table</span> = []
  <span class="ruby-keyword">begin</span>
    <span class="ruby-identifier">get</span>(<span class="ruby-string">'FT'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-string">&quot;\n&quot;</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">line</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">line</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/^FT   \w/</span>
        <span class="ruby-identifier">feature</span> = <span class="ruby-identifier">line</span>.<span class="ruby-identifier">chomp</span>.<span class="ruby-identifier">ljust</span>(<span class="ruby-value">74</span>)
        <span class="ruby-identifier">table</span> <span class="ruby-operator">&lt;&lt;</span> [<span class="ruby-identifier">feature</span>[ <span class="ruby-value">5</span><span class="ruby-operator">..</span><span class="ruby-value">12</span>].<span class="ruby-identifier">strip</span>,   <span class="ruby-comment"># Feature Name</span>
                  <span class="ruby-identifier">feature</span>[<span class="ruby-value">14</span><span class="ruby-operator">..</span><span class="ruby-value">19</span>].<span class="ruby-identifier">strip</span>,   <span class="ruby-comment"># From</span>
                  <span class="ruby-identifier">feature</span>[<span class="ruby-value">21</span><span class="ruby-operator">..</span><span class="ruby-value">26</span>].<span class="ruby-identifier">strip</span>,   <span class="ruby-comment"># To</span>
                  <span class="ruby-identifier">feature</span>[<span class="ruby-value">34</span><span class="ruby-operator">..</span><span class="ruby-value">74</span>].<span class="ruby-identifier">strip</span> ]  <span class="ruby-comment"># Description</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">table</span>.<span class="ruby-identifier">last</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">line</span>.<span class="ruby-identifier">chomp</span>.<span class="ruby-identifier">sub!</span>(<span class="ruby-regexp">/^FT +/</span>, <span class="ruby-string">''</span>)
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>

    <span class="ruby-comment"># Joining Description lines</span>
    <span class="ruby-identifier">table</span> = <span class="ruby-identifier">table</span>.<span class="ruby-identifier">map</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">feature</span><span class="ruby-operator">|</span> 
      <span class="ruby-identifier">ftid</span> = <span class="ruby-identifier">feature</span>.<span class="ruby-identifier">pop</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">feature</span>.<span class="ruby-identifier">last</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/FTId=/</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">feature</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">&gt;</span> <span class="ruby-value">4</span>
        <span class="ruby-identifier">feature</span> = [<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>], 
                   <span class="ruby-identifier">feature</span>[<span class="ruby-value">1</span>], 
                   <span class="ruby-identifier">feature</span>[<span class="ruby-value">2</span>], 
                   <span class="ruby-identifier">feature</span>[<span class="ruby-value">3</span>, <span class="ruby-identifier">feature</span>.<span class="ruby-identifier">size</span> <span class="ruby-operator">-</span> <span class="ruby-value">3</span>].<span class="ruby-identifier">join</span>(<span class="ruby-string">&quot; &quot;</span>)]
      <span class="ruby-keyword">end</span>
      <span class="ruby-identifier">feature</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">ftid</span> <span class="ruby-keyword">then</span> <span class="ruby-identifier">ftid</span> <span class="ruby-keyword">else</span> <span class="ruby-string">''</span> <span class="ruby-keyword">end</span>
    }

    <span class="ruby-identifier">hash</span> = {}
    <span class="ruby-identifier">table</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">feature</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]] = [] <span class="ruby-keyword">unless</span> <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]]
      <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]] <span class="ruby-operator">&lt;&lt;</span> {
        <span class="ruby-comment"># Removing '&lt;', '&gt;' or '?' in FROM/TO endopoint.</span>
        <span class="ruby-string">'From'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">feature</span>[<span class="ruby-value">1</span>].<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\D/</span>, <span class="ruby-string">''</span>).<span class="ruby-identifier">to_i</span>,  
        <span class="ruby-string">'To'</span>   =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">feature</span>[<span class="ruby-value">2</span>].<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\D/</span>, <span class="ruby-string">''</span>).<span class="ruby-identifier">to_i</span>, 
        <span class="ruby-string">'Description'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">feature</span>[<span class="ruby-value">3</span>], 
        <span class="ruby-string">'FTId'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">feature</span>[<span class="ruby-value">4</span>].<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\/FTId=/</span>, <span class="ruby-string">''</span>).<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\.$/</span>, <span class="ruby-string">''</span>),
        <span class="ruby-string">'diff'</span> =<span class="ruby-operator">&gt;</span> [],
        <span class="ruby-string">'original'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">feature</span>
      }

      <span class="ruby-keyword">case</span> <span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]
      <span class="ruby-keyword">when</span> <span class="ruby-string">'VARSPLIC'</span>, <span class="ruby-string">'VARIANT'</span>, <span class="ruby-string">'VAR_SEQ'</span>, <span class="ruby-string">'CONFLICT'</span>
        <span class="ruby-keyword">case</span> <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]].<span class="ruby-identifier">last</span>[<span class="ruby-string">'Description'</span>]
        <span class="ruby-keyword">when</span> <span class="ruby-regexp">/(\w[\w ]*\w*) - ?&gt; (\w[\w ]*\w*)/</span>
          <span class="ruby-identifier">original_res</span> = <span class="ruby-node">$1</span>
          <span class="ruby-identifier">changed_res</span> = <span class="ruby-node">$2</span>
          <span class="ruby-identifier">original_res</span> = <span class="ruby-identifier">original_res</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">/ /</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">strip</span>
          <span class="ruby-identifier">chenged_res</span> = <span class="ruby-identifier">changed_res</span>.<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">/ /</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">strip</span>
        <span class="ruby-keyword">when</span> <span class="ruby-regexp">/Missing/</span>
          <span class="ruby-identifier">original_res</span> = <span class="ruby-identifier">seq</span>.<span class="ruby-identifier">subseq</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]].<span class="ruby-identifier">last</span>[<span class="ruby-string">'From'</span>],
                                    <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]].<span class="ruby-identifier">last</span>[<span class="ruby-string">'To'</span>])
          <span class="ruby-identifier">changed_res</span> = <span class="ruby-string">''</span>
        <span class="ruby-keyword">end</span>
        <span class="ruby-identifier">hash</span>[<span class="ruby-identifier">feature</span>[<span class="ruby-value">0</span>]].<span class="ruby-identifier">last</span>[<span class="ruby-string">'diff'</span>] = [<span class="ruby-identifier">original_res</span>, <span class="ruby-identifier">chenged_res</span>]
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">rescue</span>
    <span class="ruby-identifier">raise</span> <span class="ruby-node">&quot;Invalid FT Lines(#{$!}) in #{entry_id}:, \n'#{self.get('FT')}'\n&quot;</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'FT'</span>] = <span class="ruby-identifier">hash</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- ft-source -->
            
          </div>

          

          
        </div><!-- ft-method -->

      
        <div id="gene_name-method" class="method-detail ">
          <a name="method-i-gene_name"></a>

          
          <div class="method-heading">
            <span class="method-name">gene_name</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a <a href="../String.html">String</a> of the first gene name in the
GN line.</p>
            

            
            <div class="method-source-code" id="gene_name-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 438</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gene_name</span>
  <span class="ruby-identifier">gene_names</span>.<span class="ruby-identifier">first</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- gene_name-source -->
            
          </div>

          

          
        </div><!-- gene_name-method -->

      
        <div id="gene_names-method" class="method-detail ">
          <a name="method-i-gene_names"></a>

          
          <div class="method-heading">
            <span class="method-name">gene_names</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a Array of gene names in the GN line.</p>
            

            
            <div class="method-source-code" id="gene_names-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 427</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gene_names</span>
  <span class="ruby-identifier">gn</span> <span class="ruby-comment"># set @data['GN'] if it hasn't been already done</span>
  <span class="ruby-keyword">if</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'GN'</span>].<span class="ruby-identifier">first</span>.<span class="ruby-identifier">class</span> <span class="ruby-operator">==</span> <span class="ruby-constant">Hash</span> <span class="ruby-keyword">then</span>
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'GN'</span>].<span class="ruby-identifier">collect</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">element</span><span class="ruby-operator">|</span> <span class="ruby-identifier">element</span>[<span class="ruby-value">:name</span>] }
  <span class="ruby-keyword">else</span>
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'GN'</span>].<span class="ruby-identifier">first</span>
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- gene_names-source -->
            
          </div>

          

          
        </div><!-- gene_names-method -->

      
        <div id="gn-method" class="method-detail ">
          <a name="method-i-gn"></a>

          
          <div class="method-heading">
            <span class="method-name">gn</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns gene names in the GN line.</p>

<p>New UniProt/SwissProt format:</p>
<ul><li>
<p><a href="SPTR.html#method-i-gn">Bio::SPTR#gn</a> -&gt; [ &lt;gene
record&gt;* ]</p>
</li></ul>

<p>where &lt;gene record&gt; is:</p>

<pre>{ :name =&gt; '...', 
  :synonyms =&gt; [ 's1', 's2', ... ],
  :loci   =&gt; [ 'l1', 'l2', ... ],
  :orfs     =&gt; [ 'o1', 'o2', ... ] 
}</pre>

<p>Old format:</p>
<ul><li>
<p><a href="SPTR.html#method-i-gn">Bio::SPTR#gn</a> -&gt; Array      # AND</p>
</li><li>
<p><a href="SPTR.html#method-i-gn"><a href="http://0">Bio::SPTR#gn</a></a>
-&gt; Array   # OR</p>
</li></ul>

<h3>GN Line: Gene name(s) (&gt;=0, optional)</h3>
            

            
            <div class="method-source-code" id="gn-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 351</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">gn</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'GN'</span>]
    <span class="ruby-keyword">case</span> <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'GN'</span>)
    <span class="ruby-keyword">when</span> <span class="ruby-regexp">/Name=/</span>,<span class="ruby-regexp">/ORFNames=/</span>,<span class="ruby-regexp">/OrderedLocusNames=/</span>,<span class="ruby-regexp">/Synonyms=/</span>
      <span class="ruby-ivar">@data</span>[<span class="ruby-string">'GN'</span>] = <span class="ruby-identifier">gn_uniprot_parser</span>
    <span class="ruby-keyword">else</span>
      <span class="ruby-ivar">@data</span>[<span class="ruby-string">'GN'</span>] = <span class="ruby-identifier">gn_old_parser</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'GN'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- gn-source -->
            
          </div>

          

          
        </div><!-- gn-method -->

      
        <div id="hi-method" class="method-detail ">
          <a name="method-i-hi"></a>

          
          <div class="method-heading">
            <span class="method-name">hi</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <h3>The HI line</h3>

<p><a href="SPTR.html#method-i-hi">Bio::SPTR#hi</a> #=&gt; hash</p>
            

            
            <div class="method-source-code" id="hi-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 691</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">hi</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'HI'</span>]
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'HI'</span>] = []
    <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'HI'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/\. /</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">hlist</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">hash</span> = {<span class="ruby-string">'Category'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>,  <span class="ruby-string">'Keywords'</span> =<span class="ruby-operator">&gt;</span> [], <span class="ruby-string">'Keyword'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>}
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'Category'</span>], <span class="ruby-identifier">hash</span>[<span class="ruby-string">'Keywords'</span>] = <span class="ruby-identifier">hlist</span>.<span class="ruby-identifier">split</span>(<span class="ruby-string">': '</span>)
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'Keywords'</span>] = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'Keywords'</span>].<span class="ruby-identifier">split</span>(<span class="ruby-string">'; '</span>)
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'Keyword'</span>] = <span class="ruby-identifier">hash</span>[<span class="ruby-string">'Keywords'</span>].<span class="ruby-identifier">pop</span>
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'Keyword'</span>].<span class="ruby-identifier">sub!</span>(<span class="ruby-regexp">/\.$/</span>, <span class="ruby-string">''</span>)
      <span class="ruby-ivar">@data</span>[<span class="ruby-string">'HI'</span>] <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">hash</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'HI'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- hi-source -->
            
          </div>

          

          
        </div><!-- hi-method -->

      
        <div id="id_line-method" class="method-detail ">
          <a name="method-i-id_line"></a>

          
          <div class="method-heading">
            <span class="method-name">id_line</span><span
              class="method-args">(key = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a <a href="../Hash.html">Hash</a> of the ID line.</p>

<p>returns a content (Int or <a href="../String.html">String</a>) of the ID
line by a given key. <a href="../Hash.html">Hash</a> keys: [ENTRY_NAME,
DATA_CLASS, MODECULE_TYPE, SEQUENCE_LENGTH]</p>

<h3>ID Line (since UniProtKB release 9.0 of 31-Oct-2006)</h3>

<pre>ID   P53_HUMAN               Reviewed;         393 AA.
#&quot;ID  #{ENTRY_NAME} #{DATA_CLASS}; #{SEQUENCE_LENGTH}.&quot;</pre>

<h3>Examples</h3>

<pre>obj.id_line  #=&gt; {&quot;ENTRY_NAME&quot;=&gt;&quot;P53_HUMAN&quot;, &quot;DATA_CLASS&quot;=&gt;&quot;Reviewed&quot;, 
                  &quot;SEQUENCE_LENGTH&quot;=&gt;393, &quot;MOLECULE_TYPE&quot;=&gt;nil}

obj.id_line('ENTRY_NAME') #=&gt; &quot;P53_HUMAN&quot;</pre>

<h3>ID Line (older style)</h3>

<pre>ID   P53_HUMAN      STANDARD;      PRT;   393 AA.
#&quot;ID  #{ENTRY_NAME} #{DATA_CLASS}; #{MOLECULE_TYPE}; #{SEQUENCE_LENGTH}.&quot;</pre>

<h3>Examples</h3>

<pre>obj.id_line  #=&gt; {&quot;ENTRY_NAME&quot;=&gt;&quot;P53_HUMAN&quot;, &quot;DATA_CLASS&quot;=&gt;&quot;STANDARD&quot;, 
                  &quot;SEQUENCE_LENGTH&quot;=&gt;393, &quot;MOLECULE_TYPE&quot;=&gt;&quot;PRT&quot;}

obj.id_line('ENTRY_NAME') #=&gt; &quot;P53_HUMAN&quot;</pre>
            

            
            <div class="method-source-code" id="id_line-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 74</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">id_line</span>(<span class="ruby-identifier">key</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">id_line</span>[<span class="ruby-identifier">key</span>] <span class="ruby-keyword">if</span> <span class="ruby-identifier">key</span>
  <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'ID'</span>] <span class="ruby-keyword">if</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'ID'</span>]

  <span class="ruby-identifier">part</span> = <span class="ruby-ivar">@orig</span>[<span class="ruby-string">'ID'</span>].<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/ +/</span>)         
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">part</span>[<span class="ruby-value">4</span>].<span class="ruby-identifier">to_s</span>.<span class="ruby-identifier">chomp</span> <span class="ruby-operator">==</span> <span class="ruby-string">'AA.'</span> <span class="ruby-keyword">then</span>
    <span class="ruby-comment"># after UniProtKB release 9.0 of 31-Oct-2006</span>
    <span class="ruby-comment"># (http://www.uniprot.org/docs/sp_news.htm)</span>
    <span class="ruby-identifier">molecule_type</span>   = <span class="ruby-keyword">nil</span>
    <span class="ruby-identifier">sequence_length</span> = <span class="ruby-identifier">part</span>[<span class="ruby-value">3</span>].<span class="ruby-identifier">to_i</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-identifier">molecule_type</span>   = <span class="ruby-identifier">part</span>[<span class="ruby-value">3</span>].<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/;/</span>,<span class="ruby-string">''</span>)
    <span class="ruby-identifier">sequence_length</span> = <span class="ruby-identifier">part</span>[<span class="ruby-value">4</span>].<span class="ruby-identifier">to_i</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'ID'</span>] = {
    <span class="ruby-string">'ENTRY_NAME'</span>      =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">part</span>[<span class="ruby-value">1</span>],
    <span class="ruby-string">'DATA_CLASS'</span>      =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">part</span>[<span class="ruby-value">2</span>].<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/;/</span>,<span class="ruby-string">''</span>),
    <span class="ruby-string">'MOLECULE_TYPE'</span>   =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">molecule_type</span>,
    <span class="ruby-string">'SEQUENCE_LENGTH'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">sequence_length</span>
  }
<span class="ruby-keyword">end</span></pre>
            </div><!-- id_line-source -->
            
          </div>

          

          
        </div><!-- id_line-method -->

      
        <div id="molecule-method" class="method-detail ">
          <a name="method-i-molecule"></a>

          
          <div class="method-heading">
            <span class="method-name">molecule</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a MOLECULE_TYPE in the ID line.</p>

<p>A short-cut for <a
href="SPTR.html#method-i-id_line">Bio::SPTR#id_line</a>(MOLECULE_TYPE).</p>
            

            
            <div class="method-source-code" id="molecule-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 109</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">molecule</span>
  <span class="ruby-identifier">id_line</span>(<span class="ruby-string">'MOLECULE_TYPE'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- molecule-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="SPTR.html#method-i-molecule_type">molecule_type</a>
          </div>
          

          
        </div><!-- molecule-method -->

      
        <div id="molecule_type-method" class="method-detail method-alias">
          <a name="method-i-molecule_type"></a>

          
          <div class="method-heading">
            <span class="method-name">molecule_type</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
          </div>

          

          
          <div class="aliases">
            Alias for: <a href="SPTR.html#method-i-molecule">molecule</a>
          </div>
          
        </div><!-- molecule_type-method -->

      
        <div id="oh-method" class="method-detail ">
          <a name="method-i-oh"></a>

          
          <div class="method-heading">
            <span class="method-name">oh</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <h3>The OH Line;  </h3>

<p>OH   NCBI_TaxID=TaxID; HostName. <a
href="http://br.expasy.org/sprot/userman.html#OH_line">br.expasy.org/sprot/userman.html#OH_line</a></p>
            

            
            <div class="method-source-code" id="oh-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 521</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">oh</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OH'</span>]
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OH'</span>] = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'OH'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-string">&quot;\. &quot;</span>).<span class="ruby-identifier">map</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">x</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">x</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/NCBI_TaxID=(\d+);/</span>
        <span class="ruby-identifier">taxid</span> = <span class="ruby-node">$1</span>
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">raise</span> <span class="ruby-constant">ArgumentError</span>, [<span class="ruby-node">&quot;Error: Invalid OH line format (#{self.entry_id}):&quot;</span>,
                              <span class="ruby-identifier">$!</span>, <span class="ruby-string">&quot;\n&quot;</span>, <span class="ruby-identifier">get</span>(<span class="ruby-string">'OH'</span>), <span class="ruby-string">&quot;\n&quot;</span>].<span class="ruby-identifier">join</span>
        
      <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">x</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/NCBI_TaxID=\d+; (.+)/</span> 
        <span class="ruby-identifier">host_name</span> = <span class="ruby-node">$1</span>
        <span class="ruby-identifier">host_name</span>.<span class="ruby-identifier">sub!</span>(<span class="ruby-regexp">/\.$/</span>, <span class="ruby-string">''</span>)
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">host_name</span> = <span class="ruby-keyword">nil</span>
      <span class="ruby-keyword">end</span>
      {<span class="ruby-string">'NCBI_TaxID'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">taxid</span>, <span class="ruby-string">'HostName'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">host_name</span>}
    }
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OH'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- oh-source -->
            
          </div>

          

          
        </div><!-- oh-method -->

      
        <div id="os-method" class="method-detail ">
          <a name="method-i-os"></a>

          
          <div class="method-heading">
            <span class="method-name">os</span><span
              class="method-args">(num = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a Array of Hashs or a <a href="../String.html">String</a> of the OS
line when a key given.</p>
<ul><li>
<p>Bio::EMBLDB#os  -&gt; Array</p>
</li></ul>

<pre>[{'name' =&gt; '(Human)', 'os' =&gt; 'Homo sapiens'}, 
 {'name' =&gt; '(Rat)', 'os' =&gt; 'Rattus norveticus'}]</pre>
<ul><li>
<p><a href="http://0">Bio::EPTR#os</a> -&gt; <a href="../Hash.html">Hash</a></p>
</li></ul>

<pre>{'name' =&gt; &quot;(Human)&quot;, 'os' =&gt; 'Homo sapiens'}</pre>
<ul><li>
<p><a href="SPTR.html#method-i-os"><a
href="http://0">Bio::SPTR#os</a></a>['name'] -&gt; "(Human)"</p>
</li><li>
<p>Bio::EPTR#os(0) -&gt; Homo sapiens (Human)</p>
</li></ul>

<h3>OS Line; organism species (&gt;=1)</h3>

<pre>OS   Genus species (name).
OS   Genus species (name0) (name1).
OS   Genus species (name0) (name1).
OS   Genus species (name0), G s0 (name0), and G s (name0) (name1).
OS   Homo sapiens (Human), and Rarrus norveticus (Rat)
OS   Hippotis sp. Clark and Watts 825.
OS   unknown cyperaceous sp.</pre>
            

            
            <div class="method-source-code" id="os-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 460</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">os</span>(<span class="ruby-identifier">num</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OS'</span>]
    <span class="ruby-identifier">os</span> = <span class="ruby-constant">Array</span>.<span class="ruby-identifier">new</span>
    <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'OS'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/, and|, /</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">tmp</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">if</span> <span class="ruby-identifier">tmp</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/(\w+ *[\w\d \:\'\+\-\.]+[\w\d\.])/</span>
        <span class="ruby-identifier">org</span> = <span class="ruby-node">$1</span>
        <span class="ruby-identifier">tmp</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/(\(.+\))/</span> 
        <span class="ruby-identifier">os</span>.<span class="ruby-identifier">push</span>({<span class="ruby-string">'name'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-node">$1</span>, <span class="ruby-string">'os'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-identifier">org</span>})
      <span class="ruby-keyword">else</span>
        <span class="ruby-identifier">raise</span> <span class="ruby-node">&quot;Error: OS Line. #{$!}\n#{fetch('OS')}\n&quot;</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OS'</span>] = <span class="ruby-identifier">os</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-keyword">if</span> <span class="ruby-identifier">num</span>
    <span class="ruby-comment"># EX. &quot;Trifolium repens (white clover)&quot;</span>
    <span class="ruby-keyword">return</span> <span class="ruby-node">&quot;#{@data['OS'][num]['os']} #{@data['OS'][num]['name']}&quot;</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OS'</span>]
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- os-source -->
            
          </div>

          

          
        </div><!-- os-method -->

      
        <div id="ox-method" class="method-detail ">
          <a name="method-i-ox"></a>

          
          <div class="method-heading">
            <span class="method-name">ox</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a <a href="../Hash.html">Hash</a> of oraganism taxonomy
cross-references.</p>
<ul><li>
<p><a href="SPTR.html#method-i-ox">Bio::SPTR#ox</a> -&gt; <a
href="../Hash.html">Hash</a></p>

<pre>{'NCBI_TaxID' =&gt; ['1234','2345','3456','4567'], ...}</pre>
</li></ul>

<h3>OX Line; organism taxonomy cross-reference (&gt;=1 per entry)</h3>

<pre>OX   NCBI_TaxID=1234;
OX   NCBI_TaxID=1234, 2345, 3456, 4567;</pre>
            

            
            <div class="method-source-code" id="ox-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 504</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ox</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OX'</span>]
    <span class="ruby-identifier">tmp</span> = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'OX'</span>).<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\.$/</span>,<span class="ruby-string">''</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/;/</span>).<span class="ruby-identifier">map</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">e</span><span class="ruby-operator">|</span> <span class="ruby-identifier">e</span>.<span class="ruby-identifier">strip</span> }
    <span class="ruby-identifier">hsh</span> = <span class="ruby-constant">Hash</span>.<span class="ruby-identifier">new</span>
    <span class="ruby-identifier">tmp</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">e</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">db</span>,<span class="ruby-identifier">refs</span> = <span class="ruby-identifier">e</span>.<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/=/</span>)
      <span class="ruby-identifier">hsh</span>[<span class="ruby-identifier">db</span>] = <span class="ruby-identifier">refs</span>.<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/, */</span>)
    <span class="ruby-keyword">end</span>
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OX'</span>] = <span class="ruby-identifier">hsh</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'OX'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- ox-source -->
            
          </div>

          

          
        </div><!-- ox-method -->

      
        <div id="protein_name-method" class="method-detail ">
          <a name="method-i-protein_name"></a>

          
          <div class="method-heading">
            <span class="method-name">protein_name</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns the proposed official name of the protein. Returns a <a
href="../String.html">String</a>.</p>

<p>Since UniProtKB release 14.0 of 22-Jul-2008, the DE line format have been
changed. The method returns the full name which is taken from RecName:
Full= or SubName: Full= line normally in the beginning of the DE lines. 
Unlike parser for old format, no special treatments for fragment or
precursor.</p>

<p>For old format, the method parses the DE lines and returns the protein name
as a <a href="../String.html">String</a>.</p>

<h3>DE Line; description (&gt;=1)</h3>

<pre>&quot;DE #{OFFICIAL_NAME} (#{SYNONYM})&quot;
&quot;DE #{OFFICIAL_NAME} (#{SYNONYM}) [CONTEINS: #1; #2].&quot;
OFFICIAL_NAME  1/entry
SYNONYM        &gt;=0
CONTEINS       &gt;=0</pre>
            

            
            <div class="method-source-code" id="protein_name-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 251</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">protein_name</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DE'</span>] <span class="ruby-operator">||=</span> <span class="ruby-identifier">parse_DE_line_rel14</span>(<span class="ruby-identifier">get</span>(<span class="ruby-string">'DE'</span>))
  <span class="ruby-identifier">parsed_de_line</span> = <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DE'</span>]
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">parsed_de_line</span> <span class="ruby-keyword">then</span>
    <span class="ruby-comment"># since UniProtKB release 14.0 of 22-Jul-2008</span>
    <span class="ruby-identifier">name</span> = <span class="ruby-keyword">nil</span>
    <span class="ruby-identifier">parsed_de_line</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">a</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">case</span> <span class="ruby-identifier">a</span>[<span class="ruby-value">0</span>]
      <span class="ruby-keyword">when</span> <span class="ruby-string">'RecName'</span>, <span class="ruby-string">'SubName'</span>
        <span class="ruby-keyword">if</span> <span class="ruby-identifier">name_pair</span> = <span class="ruby-identifier">a</span>[<span class="ruby-value">1</span><span class="ruby-operator">..</span><span class="ruby-value">-1</span>].<span class="ruby-identifier">find</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">b</span><span class="ruby-operator">|</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">0</span>] <span class="ruby-operator">==</span> <span class="ruby-string">'Full'</span> } <span class="ruby-keyword">then</span>
          <span class="ruby-identifier">name</span> = <span class="ruby-identifier">name_pair</span>[<span class="ruby-value">1</span>]
          <span class="ruby-keyword">break</span>
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
    <span class="ruby-identifier">name</span> = <span class="ruby-identifier">name</span>.<span class="ruby-identifier">to_s</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-comment"># old format (before Rel. 13.x)</span>
    <span class="ruby-identifier">name</span> = <span class="ruby-string">&quot;&quot;</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">de_line</span> = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'DE'</span>) <span class="ruby-keyword">then</span>
      <span class="ruby-identifier">str</span> = <span class="ruby-identifier">de_line</span>[<span class="ruby-regexp">/^[^\[]*/</span>] <span class="ruby-comment"># everything preceding the first [ (the &quot;contains&quot; part)</span>
      <span class="ruby-identifier">name</span> = <span class="ruby-identifier">str</span>[<span class="ruby-regexp">/^[^(]*/</span>].<span class="ruby-identifier">strip</span>
      <span class="ruby-identifier">name</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-string">' (Fragment)'</span> <span class="ruby-keyword">if</span> <span class="ruby-identifier">str</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/fragment/</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">name</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- protein_name-source -->
            
          </div>

          

          
        </div><!-- protein_name-method -->

      
        <div id="ref-method" class="method-detail ">
          <a name="method-i-ref"></a>

          
          <div class="method-heading">
            <span class="method-name">ref</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns contents in the R lines.</p>
<ul><li>
<p><a href="EMBLDB/Common.html#method-i-ref">Bio::EMBLDB::Common#ref</a> -&gt;
[ &lt;refernece information Hash&gt;* ]</p>
</li></ul>

<p>where &lt;reference information Hash&gt; is:</p>

<pre>{'RN' =&gt; '', 'RC' =&gt; '', 'RP' =&gt; '', 'RX' =&gt; '', 
 'RA' =&gt; '', 'RT' =&gt; '', 'RL' =&gt; '', 'RG' =&gt; ''}</pre>

<p>R Lines</p>
<ul><li>
<p>RN RC RP RX RA RT RL RG</p>
</li></ul>
            

            
            <div class="method-source-code" id="ref-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 557</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">ref</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'R'</span>]
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'R'</span>] = [<span class="ruby-identifier">get</span>(<span class="ruby-string">'R'</span>).<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/\nRN   /</span>)].<span class="ruby-identifier">flatten</span>.<span class="ruby-identifier">map</span> { <span class="ruby-operator">|</span><span class="ruby-identifier">str</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">hash</span> = {<span class="ruby-string">'RN'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>, <span class="ruby-string">'RC'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>, <span class="ruby-string">'RP'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>, <span class="ruby-string">'RX'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>, 
             <span class="ruby-string">'RA'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>, <span class="ruby-string">'RT'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>, <span class="ruby-string">'RL'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>, <span class="ruby-string">'RG'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-string">''</span>}
      <span class="ruby-identifier">str</span> = <span class="ruby-string">'RN   '</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">str</span> <span class="ruby-keyword">unless</span> <span class="ruby-regexp">/^RN   /</span> <span class="ruby-operator">=~</span> <span class="ruby-identifier">str</span>

      <span class="ruby-identifier">str</span>.<span class="ruby-identifier">split</span>(<span class="ruby-string">&quot;\n&quot;</span>).<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">line</span><span class="ruby-operator">|</span>
        <span class="ruby-keyword">if</span> <span class="ruby-regexp">/^(R[NPXARLCTG])   (.+)/</span> <span class="ruby-operator">=~</span> <span class="ruby-identifier">line</span>
          <span class="ruby-identifier">hash</span>[<span class="ruby-node">$1</span>] <span class="ruby-operator">+=</span> <span class="ruby-node">$2</span> <span class="ruby-operator">+</span> <span class="ruby-string">' '</span>
        <span class="ruby-keyword">else</span>
          <span class="ruby-identifier">raise</span> <span class="ruby-node">&quot;Invalid format in R lines, \n[#{line}]\n&quot;</span>
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span>

      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RN'</span>] = <span class="ruby-identifier">set_RN</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RN'</span>])
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RC'</span>] = <span class="ruby-identifier">set_RC</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RC'</span>])
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RP'</span>] = <span class="ruby-identifier">set_RP</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RP'</span>])
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RX'</span>] = <span class="ruby-identifier">set_RX</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RX'</span>])
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RA'</span>] = <span class="ruby-identifier">set_RA</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RA'</span>])
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RT'</span>] = <span class="ruby-identifier">set_RT</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RT'</span>])
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RL'</span>] = <span class="ruby-identifier">set_RL</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RL'</span>])
      <span class="ruby-identifier">hash</span>[<span class="ruby-string">'RG'</span>] = <span class="ruby-identifier">set_RG</span>(<span class="ruby-identifier">hash</span>[<span class="ruby-string">'RG'</span>])

      <span class="ruby-identifier">hash</span>
    }

  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'R'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- ref-source -->
            
          </div>

          

          
        </div><!-- ref-method -->

      
        <div id="references-method" class="method-detail ">
          <a name="method-i-references"></a>

          
          <div class="method-heading">
            <span class="method-name">references</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns <a href="Reference.html">Bio::Reference</a> object from <a
href="EMBLDB/Common.html#method-i-ref">Bio::EMBLDB::Common#ref</a>.</p>
<ul><li>
<p><a href="EMBLDB/Common.html#method-i-ref">Bio::EMBLDB::Common#ref</a> -&gt;
<a href="References.html">Bio::References</a></p>
</li></ul>
            

            
            <div class="method-source-code" id="references-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 651</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">references</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'references'</span>]
    <span class="ruby-identifier">ary</span> = <span class="ruby-keyword">self</span>.<span class="ruby-identifier">ref</span>.<span class="ruby-identifier">map</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">ent</span><span class="ruby-operator">|</span>
      <span class="ruby-identifier">hash</span> = <span class="ruby-constant">Hash</span>.<span class="ruby-identifier">new</span>(<span class="ruby-string">''</span>)
      <span class="ruby-identifier">ent</span>.<span class="ruby-identifier">each</span> {<span class="ruby-operator">|</span><span class="ruby-identifier">key</span>, <span class="ruby-identifier">value</span><span class="ruby-operator">|</span>
        <span class="ruby-keyword">case</span> <span class="ruby-identifier">key</span>
        <span class="ruby-keyword">when</span> <span class="ruby-string">'RA'</span>
          <span class="ruby-identifier">hash</span>[<span class="ruby-string">'authors'</span>] = <span class="ruby-identifier">value</span>.<span class="ruby-identifier">split</span>(<span class="ruby-regexp">/, /</span>)
        <span class="ruby-keyword">when</span> <span class="ruby-string">'RT'</span>
          <span class="ruby-identifier">hash</span>[<span class="ruby-string">'title'</span>] = <span class="ruby-identifier">value</span>
        <span class="ruby-keyword">when</span> <span class="ruby-string">'RL'</span>
          <span class="ruby-keyword">if</span> <span class="ruby-identifier">value</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/(.*) (\d+) \((\d+)\), (\d+-\d+) \((\d+)\)$/</span>
            <span class="ruby-identifier">hash</span>[<span class="ruby-string">'journal'</span>] = <span class="ruby-node">$1</span>
            <span class="ruby-identifier">hash</span>[<span class="ruby-string">'volume'</span>]  = <span class="ruby-node">$2</span>
            <span class="ruby-identifier">hash</span>[<span class="ruby-string">'issue'</span>]   = <span class="ruby-node">$3</span>
            <span class="ruby-identifier">hash</span>[<span class="ruby-string">'pages'</span>]   = <span class="ruby-node">$4</span>
            <span class="ruby-identifier">hash</span>[<span class="ruby-string">'year'</span>]    = <span class="ruby-node">$5</span>
          <span class="ruby-keyword">else</span>
            <span class="ruby-identifier">hash</span>[<span class="ruby-string">'journal'</span>] = <span class="ruby-identifier">value</span>
          <span class="ruby-keyword">end</span>
        <span class="ruby-keyword">when</span> <span class="ruby-string">'RX'</span>  <span class="ruby-comment"># PUBMED, MEDLINE, DOI</span>
          <span class="ruby-identifier">value</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">tag</span>, <span class="ruby-identifier">xref</span><span class="ruby-operator">|</span>
            <span class="ruby-identifier">hash</span>[ <span class="ruby-identifier">tag</span>.<span class="ruby-identifier">downcase</span> ]  = <span class="ruby-identifier">xref</span>
          <span class="ruby-keyword">end</span>
        <span class="ruby-keyword">end</span>
      }
      <span class="ruby-constant">Reference</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">hash</span>)
    }
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'references'</span>] = <span class="ruby-constant">References</span>.<span class="ruby-identifier">new</span>(<span class="ruby-identifier">ary</span>)
  <span class="ruby-keyword">end</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'references'</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- references-source -->
            
          </div>

          

          
        </div><!-- references-method -->

      
        <div id="seq-method" class="method-detail ">
          <a name="method-i-seq"></a>

          
          <div class="method-heading">
            <span class="method-name">seq</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a <a href="Sequence/AA.html">Bio::Sequence::AA</a> of the amino
acid sequence.</p>
<ul><li>
<p><a href="SPTR.html#method-i-seq">Bio::SPTR#seq</a> -&gt; <a
href="Sequence/AA.html">Bio::Sequence::AA</a></p>
</li></ul>

<p>blank Line; sequence data (&gt;=1)</p>
            

            
            <div class="method-source-code" id="seq-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 1306</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">seq</span>
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">''</span>]
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">''</span>] = <span class="ruby-constant">Sequence</span><span class="ruby-operator">::</span><span class="ruby-constant">AA</span>.<span class="ruby-identifier">new</span>( <span class="ruby-identifier">fetch</span>(<span class="ruby-string">''</span>).<span class="ruby-identifier">gsub</span>(<span class="ruby-regexp">/ |\d+/</span>,<span class="ruby-string">''</span>) )
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">''</span>]
<span class="ruby-keyword">end</span></pre>
            </div><!-- seq-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="SPTR.html#method-i-aaseq">aaseq</a>
          </div>
          

          
        </div><!-- seq-method -->

      
        <div id="sequence_length-method" class="method-detail ">
          <a name="method-i-sequence_length"></a>

          
          <div class="method-heading">
            <span class="method-name">sequence_length</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a SEQUENCE_LENGTH in the ID line.</p>

<p>A short-cut for <a
href="SPTR.html#method-i-id_line">Bio::SPTR#id_line</a>(SEQUENCE_LENGHT).</p>
            

            
            <div class="method-source-code" id="sequence_length-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 118</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">sequence_length</span>
  <span class="ruby-identifier">id_line</span>(<span class="ruby-string">'SEQUENCE_LENGTH'</span>)
<span class="ruby-keyword">end</span></pre>
            </div><!-- sequence_length-source -->
            
          </div>

          
          <div class="aliases">
            Also aliased as: <a href="SPTR.html#method-i-aalen">aalen</a>
          </div>
          

          
        </div><!-- sequence_length-method -->

      
        <div id="set_RN-method" class="method-detail ">
          <a name="method-i-set_RN"></a>

          
          <div class="method-heading">
            <span class="method-name">set_RN</span><span
              class="method-args">(data)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            
            

            
            <div class="method-source-code" id="set_RN-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 588</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">set_RN</span>(<span class="ruby-identifier">data</span>)
  <span class="ruby-identifier">data</span>.<span class="ruby-identifier">strip</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- set_RN-source -->
            
          </div>

          

          
        </div><!-- set_RN-method -->

      
        <div id="sq-method" class="method-detail ">
          <a name="method-i-sq"></a>

          
          <div class="method-heading">
            <span class="method-name">sq</span><span
              class="method-args">(key = nil)</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns a <a href="../Hash.html">Hash</a> of conteins in the SQ lines.</p>
<ul><li>
<p>Bio::SPTRL#sq  -&gt; hsh</p>
</li></ul>

<p>returns a value of a key given in the SQ lines.</p>
<ul><li>
<p>Bio::SPTRL#sq(key)  -&gt; int or str</p>
</li><li>
<p>Keys: [MW, mw, molecular, weight, aalen, len, length,</p>

<pre>'CRC64']</pre>
</li></ul>

<h3>SQ Line; sequence header (1/entry)</h3>

<pre>SQ   SEQUENCE   233 AA;  25630 MW;  146A1B48A1475C86 CRC64;
SQ   SEQUENCE  \d+ AA; \d+ MW;  [0-9A-Z]+ CRC64;</pre>

<p>MW, Dalton unit. CRC64 (64-bit Cyclic Redundancy Check, ISO 3309).</p>
            

            
            <div class="method-source-code" id="sq-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 1278</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">sq</span>(<span class="ruby-identifier">key</span> = <span class="ruby-keyword">nil</span>)
  <span class="ruby-keyword">unless</span> <span class="ruby-ivar">@data</span>[<span class="ruby-string">'SQ'</span>]
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'SQ'</span>) <span class="ruby-operator">=~</span> <span class="ruby-regexp">/(\d+) AA\; (\d+) MW; (.+) CRC64;/</span>
      <span class="ruby-ivar">@data</span>[<span class="ruby-string">'SQ'</span>] = { <span class="ruby-string">'aalen'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-node">$1</span>.<span class="ruby-identifier">to_i</span>, <span class="ruby-string">'MW'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-node">$2</span>.<span class="ruby-identifier">to_i</span>, <span class="ruby-string">'CRC64'</span> =<span class="ruby-operator">&gt;</span> <span class="ruby-node">$3</span> }
    <span class="ruby-keyword">else</span>
      <span class="ruby-identifier">raise</span> <span class="ruby-node">&quot;Invalid SQ Line: \n'#{fetch('SQ')}'&quot;</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>

  <span class="ruby-keyword">if</span> <span class="ruby-identifier">key</span>
    <span class="ruby-keyword">case</span> <span class="ruby-identifier">key</span>
    <span class="ruby-keyword">when</span> <span class="ruby-regexp">/mw/</span>, <span class="ruby-regexp">/molecular/</span>, <span class="ruby-regexp">/weight/</span>
      <span class="ruby-ivar">@data</span>[<span class="ruby-string">'SQ'</span>][<span class="ruby-string">'MW'</span>]
    <span class="ruby-keyword">when</span> <span class="ruby-regexp">/len/</span>, <span class="ruby-regexp">/length/</span>, <span class="ruby-regexp">/AA/</span>
      <span class="ruby-ivar">@data</span>[<span class="ruby-string">'SQ'</span>][<span class="ruby-string">'aalen'</span>]
    <span class="ruby-keyword">else</span>
      <span class="ruby-ivar">@data</span>[<span class="ruby-string">'SQ'</span>][<span class="ruby-identifier">key</span>]
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">else</span> 
    <span class="ruby-ivar">@data</span>[<span class="ruby-string">'SQ'</span>]
  <span class="ruby-keyword">end</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- sq-source -->
            
          </div>

          

          
        </div><!-- sq-method -->

      
        <div id="synonyms-method" class="method-detail ">
          <a name="method-i-synonyms"></a>

          
          <div class="method-heading">
            <span class="method-name">synonyms</span><span
              class="method-args">()</span>
            <span class="method-click-advice">click to toggle source</span>
          </div>
          

          <div class="method-description">
            
            <p>returns synonyms (unofficial and/or alternative names). Returns an Array
containing <a href="../String.html">String</a> objects.</p>

<p>Since UniProtKB release 14.0 of 22-Jul-2008, the DE line format have been
changed. The method returns the full or short names which are taken from
RecName: Short=, RecName: EC=, and AltName lines, except after
Contains: or Includes:. For keeping compatibility with old format
parser, RecName: EC=N.N.N.N is reported as EC N.N.N.N. In addition, to
prevent confusion, Allergen= and CD_antigen=  prefixes are added for
the corresponding fields.</p>

<p>For old format, the method parses the DE lines and returns synonyms.
synonyms are each placed in () following the official name on the DE line.</p>
            

            
            <div class="method-source-code" id="synonyms-source">
<pre>
<span class="ruby-comment"># File lib/bio/db/embl/sptr.rb, line 294</span>
<span class="ruby-keyword">def</span> <span class="ruby-identifier">synonyms</span>
  <span class="ruby-identifier">ary</span> = <span class="ruby-constant">Array</span>.<span class="ruby-identifier">new</span>
  <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DE'</span>] <span class="ruby-operator">||=</span> <span class="ruby-identifier">parse_DE_line_rel14</span>(<span class="ruby-identifier">get</span>(<span class="ruby-string">'DE'</span>))
  <span class="ruby-identifier">parsed_de_line</span> = <span class="ruby-ivar">@data</span>[<span class="ruby-string">'DE'</span>]
  <span class="ruby-keyword">if</span> <span class="ruby-identifier">parsed_de_line</span> <span class="ruby-keyword">then</span>
    <span class="ruby-comment"># since UniProtKB release 14.0 of 22-Jul-2008</span>
    <span class="ruby-identifier">parsed_de_line</span>.<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">a</span><span class="ruby-operator">|</span>
      <span class="ruby-keyword">case</span> <span class="ruby-identifier">a</span>[<span class="ruby-value">0</span>]
      <span class="ruby-keyword">when</span> <span class="ruby-string">'Includes'</span>, <span class="ruby-string">'Contains'</span>
        <span class="ruby-keyword">break</span> <span class="ruby-comment">#the each loop</span>
      <span class="ruby-keyword">when</span> <span class="ruby-string">'RecName'</span>, <span class="ruby-string">'SubName'</span>, <span class="ruby-string">'AltName'</span>
        <span class="ruby-identifier">a</span>[<span class="ruby-value">1</span><span class="ruby-operator">..</span><span class="ruby-value">-1</span>].<span class="ruby-identifier">each</span> <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">b</span><span class="ruby-operator">|</span>
          <span class="ruby-keyword">if</span> <span class="ruby-identifier">name</span> = <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>] <span class="ruby-keyword">and</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>] <span class="ruby-operator">!=</span> <span class="ruby-keyword">self</span>.<span class="ruby-identifier">protein_name</span> <span class="ruby-keyword">then</span>
            <span class="ruby-keyword">case</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">0</span>]
            <span class="ruby-keyword">when</span> <span class="ruby-string">'EC'</span>
              <span class="ruby-identifier">name</span> = <span class="ruby-string">&quot;EC &quot;</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>]
            <span class="ruby-keyword">when</span> <span class="ruby-string">'Allergen'</span>, <span class="ruby-string">'CD_antigen'</span>
              <span class="ruby-identifier">name</span> = <span class="ruby-identifier">b</span>[<span class="ruby-value">0</span>] <span class="ruby-operator">+</span> <span class="ruby-string">'='</span> <span class="ruby-operator">+</span> <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>]
            <span class="ruby-keyword">else</span>
              <span class="ruby-identifier">name</span> = <span class="ruby-identifier">b</span>[<span class="ruby-value">1</span>]
            <span class="ruby-keyword">end</span>
            <span class="ruby-identifier">ary</span>.<span class="ruby-identifier">push</span> <span class="ruby-identifier">name</span>
          <span class="ruby-keyword">end</span>
        <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span> <span class="ruby-comment">#case a[0]</span>
    <span class="ruby-keyword">end</span> <span class="ruby-comment">#parsed_de_line.each</span>
  <span class="ruby-keyword">else</span>
    <span class="ruby-comment"># old format (before Rel. 13.x)</span>
    <span class="ruby-keyword">if</span> <span class="ruby-identifier">de_line</span> = <span class="ruby-identifier">fetch</span>(<span class="ruby-string">'DE'</span>) <span class="ruby-keyword">then</span>
      <span class="ruby-identifier">line</span> = <span class="ruby-identifier">de_line</span>.<span class="ruby-identifier">sub</span>(<span class="ruby-regexp">/\[.*\]/</span>,<span class="ruby-string">''</span>) <span class="ruby-comment"># ignore stuff between [ and ].  That's the &quot;contains&quot; part</span>
    <span class="ruby-identifier">line</span>.<span class="ruby-identifier">scan</span>(<span class="ruby-regexp">/\([^)]+/</span>) <span class="ruby-keyword">do</span> <span class="ruby-operator">|</span><span class="ruby-identifier">synonym</span><span class="ruby-operator">|</span> 
      <span class="ruby-keyword">unless</span> <span class="ruby-identifier">synonym</span> <span class="ruby-operator">=~</span> <span class="ruby-regexp">/fragment/</span> <span class="ruby-keyword">then</span> 
        <span class="ruby-identifier">ary</span> <span class="ruby-operator">&lt;&lt;</span> <span class="ruby-identifier">synonym</span>[<span class="ruby-value">1</span><span class="ruby-operator">..</span><span class="ruby-value">-1</span>].<span class="ruby-identifier">strip</span> <span class="ruby-comment"># index to remove the leading (  </span>
      <span class="ruby-keyword">end</span>
      <span class="ruby-keyword">end</span>
    <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">end</span>
  <span class="ruby-keyword">return</span> <span class="ruby-identifier">ary</span>
<span class="ruby-keyword">end</span></pre>
            </div><!-- synonyms-source -->
            
          </div>

          

          
        </div><!-- synonyms-method -->

      
      </div><!-- public-instance-method-details -->
    
    </div><!-- 5Buntitled-5D -->
  

  </div><!-- documentation -->

  <div id="validator-badges">
    <p><small><a href="http://validator.w3.org/check/referer">[Validate]</a></small></p>
    <p><small>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish
      Rdoc Generator</a> 2</small>.</p>
  </div>

</body>
</html>

